<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Step 3: Aligning reads | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Step 3: Aligning reads | The WHG training resources"><meta data-rh="true" name="description" content="Most analyses of short-read sequence data are based on the basic paradigm made up of two steps:"><meta data-rh="true" property="og:description" content="Most analyses of short-read sequence data are based on the basic paradigm made up of two steps:"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites/">Getting the prerequisites</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/">Practical outline</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs/">Step 1: What&#x27;s in the fastq files?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">Step 2: Inspecting sequence data quality</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">Step 3: Aligning reads</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/">Step 4: Viewing alignments</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/">Challenge questions</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/">Bonus tutorial: how accurate is Illumina sequencing?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">Appendix 1: Paired-end sequencing in theory and practice</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/">Appendix 2: understanding sequence duplication levels</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">Appendix 3: read trimming</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq/">Appendix 4: RNA-seq links</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links/">Useful links</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"><span itemprop="name">Analysing paired-end sequencing data</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Step 3: Aligning reads</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Step 3: Aligning reads</h1><p>Most analyses of short-read sequence data are based on the basic paradigm made up of two steps:</p><ul><li>first, <em>align reads to an established reference genome assembly</em>.</li><li>then <em>use those alignments to identify and call genetic variation for analysis</em>.</li></ul><p>The first step in this paradigm is to get reads aligned. On this page we&#x27;ll implement the basic
read alignment pipeline that starts with the fastqs and ends with alignments for downstream
analysis. </p><h2 class="anchor anchorWithStickyNavbar_mojV" id="the-basic-pipeline">The basic pipeline<a class="hash-link" href="#the-basic-pipeline" title="Direct link to heading">​</a></h2><p>The basic pipeline is:</p><ol><li>align reads to the reference genome assembly</li><li>sort them so they are in genome position order</li><li>&#x27;mark&#x27; reads that look like duplicates (since these are <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">likely to be artifacts</a>.)</li><li>Finally we&#x27;ll gather some statistics about the alignments.</li></ol><p>In theory that&#x27;s it - in practice it gets a bit more fiddly because of the need for various format
conversion and indexing steps. (Some data might also need a <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">read trimming step</a>
but we&#x27;re skipping that.)</p><p>Without further ado let&#x27;s implement these steps using a popular aligner <code>bwa mem</code> and the swiss
army knife-like <code>samtools</code> to do the work. As before I&#x27;ll work with the data for the <code>QG0033-C</code>
malaria sample (which was ascertained in Congo in 2012), but if you know what you are doing you are
free to work with any of the other samples in the folder - or the chromosome 19 human data in the
<code>human/</code> folder if you prefer. (Just remember to use the human reference sequence <code>GRCh38_chr19.fa</code>
in that case.)</p><p>To get started, move to a terminal window and make sure you are in the
<code>sequence_data_analysis/malaria</code> folder:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">cd ~/sequence_data_analysis/malaria</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The pipeline can be run like this:</p><div class="language-sh codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-sh codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain"># Make a temp dir to hold intermediate files</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">mkdir -p tmp</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Create a FM index for the reference</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">bwa index Pf3D7_v3.fa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># align the reads (using 2 threads - don&#x27;t use more please!)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">bwa mem -t 4 -o tmp/QG0033-C-aligned.sam Pf3D7_v3.fa QG0033-C_Illumina-HiSeq_read1.fastq.gz QG0033-C_Illumina-HiSeq_read2.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># convert to BAM</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools view -b -o tmp/QG0033-C-aligned.bam tmp/QG0033-C-aligned.sam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># fix mate-pair information and convert to BAM</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools fixmate -m tmp/QG0033-C-aligned.bam tmp/QG0033-C-fixmate.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># sort reads by genomic position</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools sort -T tmp -o tmp/QG0033-C-sorted.bam tmp/QG0033-C-fixmate.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Identify probable duplicate reads</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools markdup -s tmp/QG0033-C-sorted.bam tmp/QG0033-C-markdup.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># Rename to this folder, and index</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">mv tmp/QG0033-C-markdup.bam QG0033-C.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools index QG0033-C.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>(<strong>Note.</strong> You can see what this has created by running <code>ls</code>.)</p><p>If you run all that together it might take around 10 minutes to complete (the longest steps being the <code>bwa mem</code>
indexing and alignment steps.) It&#x27;s worth running the steps one by one to make sure they complete without errors, and
watching the output to see what is being done. While you&#x27;re waiting for the alignment to complete, consider reading the
information below about what&#x27;s in the aligned data, <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">reading more about paired-end sequencing theory</a>,
or more about the SAM/BAM format and using samtools.</p><p>For example here is a <a href="https://davetang.org/wiki/tiki-index.php?page=SAM" target="_blank" rel="noopener noreferrer">useful overview of SAM format</a>, or a page
with <a href="https://github.com/davetang/learning_bam_file" target="_blank" rel="noopener noreferrer">more tips on using samtools</a>.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="what-did-the-pipeline-generate">What did the pipeline generate?<a class="hash-link" href="#what-did-the-pipeline-generate" title="Direct link to heading">​</a></h3><p>If you followed the above you&#x27;ll now have:</p><ul><li><p>a file <code>tmp/QG0033-C-aligned.sam</code> which contains the alignments in the <a href="https://en.wikipedia.org/wiki/SAM_(file_format)" target="_blank" rel="noopener noreferrer">SAM text file format</a>.</p></li><li><p>a binary version <code>tmp/QG0033-C-aligned.bam</code> of the same thing, in <a href="https://en.wikipedia.org/wiki/Binary_Alignment_Map" target="_blank" rel="noopener noreferrer">BAM format</a>.</p></li><li><p>a coordinate-sorted version of that, in <code>tmp/QG0033-C-sorted.bam</code></p></li><li><p>a final version of the alignments with duplicate reads &#x27;marked&#x27; - called <code>QG0033-C.bam</code></p></li><li><p>and a corresponding index file <code>QG0033-C.bam.bai</code>.</p></li></ul><p><strong>Congratulations!</strong> You should now have an aligned set of reads ready for analysis. </p><p>The rest of this page gives details about what is in the alignment files you&#x27;ve just created. When
you&#x27;re satisfied you know enough, <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/#the-practical-in-a-nutshell">go back to the practical</a>.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="inspecting-the-alignment-output">Inspecting the alignment output<a class="hash-link" href="#inspecting-the-alignment-output" title="Direct link to heading">​</a></h2><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-contents-of-a-sam-file">The contents of a SAM file<a class="hash-link" href="#the-contents-of-a-sam-file" title="Direct link to heading">​</a></h3><p>If you look at the unsorted SAM file (e.g. by typing <code>less -S tmp/QG0033-C-aligned.sam</code> - press <code>q</code> when you want to quit) you&#x27;ll see something like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_08_v3  LN:1472805</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf_M76611    LN:5967</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_01_v3  LN:640851</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_09_v3  LN:1541735</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_03_v3  LN:1067971</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_05_v3  LN:1343557</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_02_v3  LN:947102</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_14_v3  LN:3291936</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_11_v3  LN:2038340</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_10_v3  LN:1687656</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_04_v3  LN:1200490</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_12_v3  LN:2271494</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_13_v3  LN:2925236</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_06_v3  LN:1418242</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:Pf3D7_07_v3  LN:1445207</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@SQ     SN:PF_apicoplast_genome_1       LN:29430</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">@PG     ID:bwa  PN:bwa  VN:0.7.17-r1188 CL:bwa mem -t 2 -o QG0033-C-aligned.sam Pf3D7_v3.fa QG0033-C_Illumina-HiSeq_read1.fast</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ERR377582.7615542       99      Pf3D7_09_v3     37181   21      100M    =       37244   163     AAAAATCCATTTATATCTTTTATGGTTAGTATTATTTATACCTTCCTGTTTTATTTATCCACGTTTTATAAAATTGACGTATTGTTAATAAGTAGTTGTA    B@DECEFEEGFEEHFGGEFFFFFEFFEEFEGHHGGGFFFAEFFFGGEFFHFGEEFFEDEGDDEF@GFGEGEEGHEEDGGGEFEEFFEEEDEFDGDECEEF    NM:i:1  MD:Z:58T41      MC:Z:100M       AS:i:95 XS:i:91</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ERR377582.7615542       147     Pf3D7_09_v3     37244   21      100M    =       37181   -163    TTTTATAAAATTGACGTATTGTTAATAAGTAGTTGTATATTATAAATTATAAGGTAATATAATTGCTATAATTATAATAATCATTATCATAATTATTATC    DHEDGBHEDGFEDAAGEDEDGFHGGGDEFCFGFGFFEEEIBCFCIFDEEDFFGGBEFDGEEGEFFEFHIFDEFEHHEFHEDDFFEEFDFEEFDEDECA@9    NM:i:2  MD:Z:43G49A6    MC:Z:100M       AS:i:90 XS:i:90</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ERR377582.19970658      77      *       0       0       *       *       0       0       TGATCGGTGCGGGCCGGACTGAATTGCTACGCCTGATTTTCGGTGCCGACCTGGCCGACAGTGGCACGGTGGCCTTGGGGTCGCCAGCGCAGGTGGTGAG    *:?D&lt;CF3-D@EE@EDDGFF?DDEFACEFEBHCGD/EFEE:?E;F?GE=2FE&gt;FFC:EHD??H@DFFB89CD@D-ECC8;EF.A:;6EEDDFA,D&gt;&#x27;D@F    AS:i:0  XS:i:0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">...</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The file consists of some <em>metadata</em> (lines starting with <code>@</code>, and recording information about the sequences in the reference
assembly and the program that was run to generate the alignments) followed by the alignments themselves.</p><p>The alignment columns rows consist of</p><ul><li>The read ID (straight from the fastq file)</li><li>Some <em>flags</em> encoded as an integer.  The best way to understand these is through the <a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="noopener noreferrer">&quot;Explain SAM flags&quot; webpage</a>.  We&#x27;ll go through some examples below.</li><li>The chromosome and position at which the read aligns</li><li>The <em>mapping quality</em>, detailed more below.</li><li>The <em>CIGAR</em> string which gives some detail about how the read aligns</li><li>The chromosome and position of the other read in the pair <em>if it&#x27;s aligned</em>. (The
chromosome is often <code>=</code>, which means the same chromosome as this read, which you&#x27;d expect since they both came from the same fragment. A <code>*</code> means the other read was not aligned).</li><li>The &#x27;template length&#x27; (how long was the span of the read pair on the reference contig?)</li><li>The read bases and mapping qualities.  These are straight from the fastq but <em>reverse-complemented if necessary</em> so they are in the same order as the reference bases.</li><li>Finally there are some <strong>optional fields</strong>.  Among these are <strong>NM</strong> (number of mismatches between the read and the reference), and <strong>MD</strong> (a kind of counterpoint to the CIGAR string in column 6).</li></ul><p>If you look at the read IDs and flags above (via
<a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="noopener noreferrer">this page</a>), you&#x27;ll see that the read pairs
are represented as pairs of alignments with the same identifiers but with one flagged as &#x27;first in
pair&#x27; and the other as &#x27;second in pair&#x27;.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="how-many-alignments-are-there">How many alignments are there?<a class="hash-link" href="#how-many-alignments-are-there" title="Direct link to heading">​</a></h3><p>We can use the UNIX <code>wc</code> command to count the alignments:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">samtools view tmp/QG0033-C-aligned.sam | wc -l</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note.</strong> The BAM file is exactly the same, it&#x27;s just encoded in a binary format.  So this also works and is slightly quicker:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">samtools view tmp/QG0033-C-aligned.bam | wc -l</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you run this with the above data it says there are 4,057,972 alignments. But wait - there were only 2 million read pairs!</p><p>This illustrates an important point: <strong>there are more alignments than reads in the data</strong>.  How can this be?</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="primary-and-supplementary-alignments">Primary and supplementary alignments<a class="hash-link" href="#primary-and-supplementary-alignments" title="Direct link to heading">​</a></h3><p>To figure out why, let&#x27;s look at an example: the read with ID <code>ERR377582.20226793</code>. You can extract
this using <code>grep</code> like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">grep &#x27;ERR377582.20226793&#x27; tmp/QG0033-C-aligned.sam</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you look at the alignment flags (second columns) via <a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="noopener noreferrer">this page</a>, you&#x27;ll see that:</p><ul><li>The first alignment (flag=81) represents the first read in the pair.  It is aligned on the reverse strand of the reference.</li><li>The second alignment (flag=2113) <em>also</em> represents the first read in the pair.  It is flagged as a <strong>supplementary alignment</strong>.</li><li>The third alignment (flag=161) represents the second read in the pair.</li></ul><p><strong>Terminology</strong>: A <em>supplementary alignment</em> occurs when a read aligns in more than one part. (The
additional alignments aren&#x27;t adjacent to the primary one, otherwise they would be part of it.)</p><p>You can see how this works for this read by looking at the <em>CIGAR string</em>. For the first alignment
the CIGAR is <code>60M40S</code>. This means that the first 60 bases of the 100bp read were aligned, and the
following 40 were &#x27;soft clipped&#x27; - not part of the alignment. For the supplementary alignment, the
CIGAR is <code>54M46H</code>. If you look at the read bases you will see they are the reverse complement of
the bases from the original read (which are fully represented in the first alignment). The upshot
is that <code>bwa</code> thinks the first ~60 bases align to chromosome <code>Pf3D7_10_v3</code>, but the last ~54 bases
align in reverse orientation to chromosome <code>Pf3D7_09_v3</code>.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="interpreting-mapping-qualities">Interpreting mapping qualities<a class="hash-link" href="#interpreting-mapping-qualities" title="Direct link to heading">​</a></h3><p>Taken at face value you might think that this means this sample has a translocation between
chromosome 9 and 10. But hold on. If you look at the <em>mapping quality</em> column (column 5) you&#x27;ll see
that all these alignments actually have <strong>zero mapping quality</strong>. This means that confidence in
these alignments is very low!</p><p>Like base qualities, mapping qualities are expressed on the PHRED scale. They are the aligner&#x27;s
estimate of the quantity: the <em>probability that the read actually aligns elsewhere</em>. On the PHRED
scale this is expressed as:</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mtext>mapping quality</mtext><mo>=</mo><mo>−</mo><mn>10</mn><msub><mrow><mi>log</mi><mo>⁡</mo></mrow><mn>10</mn></msub><mrow><mo fence="true">(</mo><mi>P</mi><mo stretchy="false">(</mo><mtext>the read actually maps elsewhere</mtext><mo stretchy="false">)</mo><mo fence="true">)</mo></mrow></mrow><annotation encoding="application/x-tex">\text{mapping quality} = -10 \log_{10} \left( P( \text{the read actually maps elsewhere}) \right)</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8889em;vertical-align:-0.1944em"></span><span class="mord text"><span class="mord">mapping quality</span></span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-0.25em"></span><span class="mord">−</span><span class="mord">10</span><span class="mspace" style="margin-right:0.1667em"></span><span class="mop"><span class="mop">lo<span style="margin-right:0.01389em">g</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.207em"><span style="top:-2.4559em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">10</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2441em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em"></span><span class="minner"><span class="mopen delimcenter" style="top:0em">(</span><span class="mord mathnormal" style="margin-right:0.13889em">P</span><span class="mopen">(</span><span class="mord text"><span class="mord">the read actually maps elsewhere</span></span><span class="mclose">)</span><span class="mclose delimcenter" style="top:0em">)</span></span></span></span></span></span></div><p>which gives this relationship:</p><table><thead><tr><th>mapping quality</th><th>Estimated probability alignment is wrong</th></tr></thead><tbody><tr><td>0</td><td>100%</td></tr><tr><td>10</td><td>10%</td></tr><tr><td>20</td><td>1%</td></tr><tr><td>30</td><td>0.1%</td></tr><tr><td>40</td><td>0.001%</td></tr></tbody></table><p>A mapping quality of zero translates to a 100% probability that the read aligns elsewhere, that is,
that <code>bwa</code> is not confident at all in this alignment. </p><p><strong>Note.</strong> Also like base qualities, this is only an estimate of the probability.  It may not be well calibrated.</p><p>(What&#x27;s really going on is that these reads original from the telomeres. Telomeres in malaria, as
in humans and other organisms, are highly repetitive and are very hard to analyse using short-read
data.)</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="what-types-of-alignment-are-there">What types of alignment are there?<a class="hash-link" href="#what-types-of-alignment-are-there" title="Direct link to heading">​</a></h3><p>To get a better sense of the types of alignment let&#x27;s use <code>samtools flagstat</code>.<br>
<!-- -->This counts reads according to the <code>flags</code> column. Run it like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">samtools flagstat tmp/QG0033-C-aligned.sam </span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>From the output you should see: </p><ul><li>Of 4 million reads (&quot;4000000 paired in sequencing&quot;), only 3,804,730 were actually aligned by <code>bwa</code>.  That&#x27;s nearly 200,000 that weren&#x27;t aligned at all.</li><li>Most of these (3731534) were were aligned in pairs.</li><li>Most of <em>these</em> (3537816) were &#x27;properly paired&#x27; - that is, both reads mapped to the same chromosome, in the right orientation (i.e. <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">facing each other</a>) and about the right distance apart.</li><li>But a small subset (264863 in total) had unmapped mate, mate on a different chromosome, or not close or in the wrong orientation to the original read.</li></ul><p><strong>Note.</strong> A much more detailed view of reads can be generated by running <code>samtools stats</code>. See more
on the <a href="http://www.htslib.org/doc/samtools-stats.html" target="_blank" rel="noopener noreferrer"><code>samtools stats</code> documentation page</a>.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="what-did-markdup-do">What did <code>markdup</code> do?<a class="hash-link" href="#what-did-markdup-do" title="Direct link to heading">​</a></h3><p><strong>Question</strong> what <strong>did</strong> the <code>samtools markdup</code> step do?</p><p><strong>Hint 1</strong>. Try viewing the first reads in the coordinate-sorted and final (post-markdup BAM).  E.g.:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">samtools view tmp/QG0033-C-sorted.bam  | head</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools view QG0033-C.bam  | head</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Can you spot the difference</p><p><strong>Hint</strong>. Try running <code>samtools flagstat</code> on these two files.  What&#x27;s different?</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="important--cleaning-up">Important!  Cleaning up<a class="hash-link" href="#important--cleaning-up" title="Direct link to heading">​</a></h3><p>Our pipeline has left behind several copies of the original data that we don&#x27;t need.  See them by looking in the temp directory:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">ls -lh tmp</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">du -ch tmp</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>That&#x27;s 2.4Gb of space essentially wasted!  Now is a good time to get rid of these:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">rm tmp/*</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">rmdir tmp</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note.</strong> <strong>Don&#x27;t</strong> delete the main output file <code>QG0033-C.bam</code> or its index - we&#x27;ll look at that in a moment.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="next-steps">Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">​</a></h2><p>Now that we have an aligned set of reads, move on to <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/">viewing your alignments</a>.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Step 2: Inspecting sequence data quality</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Step 4: Viewing alignments</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#the-basic-pipeline" class="table-of-contents__link toc-highlight">The basic pipeline</a><ul><li><a href="#what-did-the-pipeline-generate" class="table-of-contents__link toc-highlight">What did the pipeline generate?</a></li></ul></li><li><a href="#inspecting-the-alignment-output" class="table-of-contents__link toc-highlight">Inspecting the alignment output</a><ul><li><a href="#the-contents-of-a-sam-file" class="table-of-contents__link toc-highlight">The contents of a SAM file</a></li><li><a href="#how-many-alignments-are-there" class="table-of-contents__link toc-highlight">How many alignments are there?</a></li><li><a href="#primary-and-supplementary-alignments" class="table-of-contents__link toc-highlight">Primary and supplementary alignments</a></li><li><a href="#interpreting-mapping-qualities" class="table-of-contents__link toc-highlight">Interpreting mapping qualities</a></li><li><a href="#what-types-of-alignment-are-there" class="table-of-contents__link toc-highlight">What types of alignment are there?</a></li><li><a href="#what-did-markdup-do" class="table-of-contents__link toc-highlight">What did <code>markdup</code> do?</a></li><li><a href="#important--cleaning-up" class="table-of-contents__link toc-highlight">Important!  Cleaning up</a></li></ul></li><li><a href="#next-steps" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>