"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[1688],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>h});var a=n(7294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function r(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},o=Object.keys(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var l=a.createContext({}),p=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):r(r({},t),e)),n},c=function(e){var t=p(e.components);return a.createElement(l.Provider,{value:t},e.children)},m="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},u=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,o=e.originalType,l=e.parentName,c=s(e,["components","mdxType","originalType","parentName"]),m=p(n),u=i,h=m["".concat(l,".").concat(u)]||m[u]||d[u]||o;return n?a.createElement(h,r(r({ref:t},c),{},{components:n})):a.createElement(h,r({ref:t},c))}));function h(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var o=n.length,r=new Array(o);r[0]=u;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[m]="string"==typeof e?e:i,r[1]=s;for(var p=2;p<o;p++)r[p]=n[p];return a.createElement.apply(null,r)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},2595:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>r,default:()=>d,frontMatter:()=>o,metadata:()=>s,toc:()=>p});var a=n(7462),i=(n(7294),n(3905));const o={sidebar_position:1.25},r="Plink warm-up",s={unversionedId:"genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies",id:"genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies",title:"Plink warm-up",description:"Plink is a tool for analysis of genome-wide genotype data that supports many functions including those related to data",source:"@site/docs/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies.md",sourceDirName:"genome_wide_association_studies/genome_wide_association_analysis",slug:"/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies.md",tags:[],version:"current",sidebarPosition:1.25,frontMatter:{sidebar_position:1.25},sidebar:"sidebar8",previous:{title:"Getting setup",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/getting_setup"},next:{title:"A toy GWAS",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink"}},l={},p=[{value:"Computing allele frequencies",id:"computing-allele-frequencies",level:2}],c={toc:p},m="wrapper";function d(e){let{components:t,...n}=e;return(0,i.kt)(m,(0,a.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"plink-warm-up"},"Plink warm-up"),(0,i.kt)("p",null,"Plink is a tool for analysis of genome-wide genotype data that supports many functions including those related to data\norganization, formatting, quality control, association testing, population stratification and much more. In this\npractical we'll use it to quality control (QC) some data and run a real (simulated) GWAS study.  Will we find any\nassociations?"),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"You can find more information about plink at the plink website:"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink2"},"https://www.cog-genomics.org/plink2")),(0,i.kt)("p",{parentName:"div"},"There is detailed documentation about all the options available, file formats and examples of commands.   (The version\ncurrently in greatest use is called '1.9 beta'.)"),(0,i.kt)("p",{parentName:"div"},"Another detailed tutorial (similar to work we have done here) is available at:"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("a",{parentName:"p",href:"http://pngu.mgh.harvard.edu/~purcell/plink/tutorial.shtml"},"http://pngu.mgh.harvard.edu/~purcell/plink/tutorial.shtml")))),(0,i.kt)("h2",{id:"computing-allele-frequencies"},"Computing allele frequencies"),(0,i.kt)("p",null,"A basic first task is to compute allele frequencies. Let's try that out now by running plink like this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --freq  --vcf snp-example.vcf\n")),(0,i.kt)("p",null,"The ",(0,i.kt)("inlineCode",{parentName:"p"},"--freq")," command asks PLINK to calculate allele frequencies. ",(0,i.kt)("inlineCode",{parentName:"p"},"--vcf snp-example.vcf")," tells\nPLINK to read in data from the VCF file ",(0,i.kt)("inlineCode",{parentName:"p"},"snp-example.vcf"),"."),(0,i.kt)("p",null,"If you type ",(0,i.kt)("inlineCode",{parentName:"p"},"ls")," you'll see that PLINK has written two output files in the directory we are working in: 1. a log file called\n",(0,i.kt)("inlineCode",{parentName:"p"},"plink.log")," that summarizes what it has done (this is the same as the output that appeared in the terminal window), and 2.\n",(0,i.kt)("inlineCode",{parentName:"p"},"plink.frq"),", which contains the allele frequency information. "),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Use a text editor, ",(0,i.kt)("inlineCode",{parentName:"p"},"less"),", or load them into R to look at all the input and output files."),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Q"),". How many SNPs were in this dataset? What are their names? What are their allele frequencies?"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Q"),". How many samples are included in total?"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Advanced Question"),". The column names can be frankly somewhat cryptic. Can you find out what\n",(0,i.kt)("inlineCode",{parentName:"p"},"NCHROBS")," means from the documentation? (You may find you need to look at the documentation for the\noriginal version of plink (",(0,i.kt)("a",{parentName:"p",href:"http://zzz.bwh.harvard.edu/plink/summary.shtml"},"http://zzz.bwh.harvard.edu/plink/summary.shtml"),"). On the other hand,\nlooking at the documentation on ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/"},"https://www.cog-genomics.org/plink/1.9/"),", are there other or better\n",(0,i.kt)("inlineCode",{parentName:"p"},"--freq"),"-like commands you can run? What do they output? Which version do you like?"))))}d.isMDXComponent=!0}}]);