"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[978],{3905:(e,t,a)=>{a.d(t,{Zo:()=>d,kt:()=>g});var n=a(7294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function s(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function o(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var l=n.createContext({}),u=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):s(s({},t),e)),a},d=function(e){var t=u(e.components);return n.createElement(l.Provider,{value:t},e.children)},c="mdxType",p={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},m=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,d=o(e,["components","mdxType","originalType","parentName"]),c=u(a),m=i,g=c["".concat(l,".").concat(m)]||c[m]||p[m]||r;return a?n.createElement(g,s(s({ref:t},d),{},{components:a})):n.createElement(g,s({ref:t},d))}));function g(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=a.length,s=new Array(r);s[0]=m;var o={};for(var l in t)hasOwnProperty.call(t,l)&&(o[l]=t[l]);o.originalType=e,o[c]="string"==typeof e?e:i,s[1]=o;for(var u=2;u<r;u++)s[u]=a[u];return n.createElement.apply(null,s)}return n.createElement.apply(null,a)}m.displayName="MDXCreateElement"},1724:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>s,default:()=>p,frontMatter:()=>r,metadata:()=>o,toc:()=>u});var n=a(7462),i=(a(7294),a(3905));const r={sidebar_position:1,author:"Eddie Cano-Gamez",date:"2022-10-04"},s="Visualise RNA-seq data in R",o={unversionedId:"data_visualisation/visualising_rna-seq_data/README",id:"data_visualisation/visualising_rna-seq_data/README",title:"Visualise RNA-seq data in R",description:"Author: Eddie Cano-Gamez.",source:"@site/docs/data_visualisation/visualising_rna-seq_data/README.md",sourceDirName:"data_visualisation/visualising_rna-seq_data",slug:"/data_visualisation/visualising_rna-seq_data/",permalink:"/whg-training-resources/data_visualisation/visualising_rna-seq_data/",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/data_visualisation/visualising_rna-seq_data/README.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1,author:"Eddie Cano-Gamez",date:"2022-10-04"},sidebar:"sidebar3",previous:{title:"Data visualisation tutorials",permalink:"/whg-training-resources/data_visualisation/"},next:{title:"Getting setup",permalink:"/whg-training-resources/data_visualisation/visualising_rna-seq_data/getting_setup"}},l={},u=[{value:"The GTEx RNA-seq dataset",id:"the-gtex-rna-seq-dataset",level:2}],d={toc:u},c="wrapper";function p(e){let{components:t,...r}=e;return(0,i.kt)(c,(0,n.Z)({},d,r,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"visualise-rna-seq-data-in-r"},"Visualise RNA-seq data in R"),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Author"),": Eddie Cano-Gamez."),(0,i.kt)("p",null,"Welcome!  This tutorial will show you some of the elementary skills you need to begin visualising RNA-seq\ndata (specifically, abundance estimates for gene RNA transcripts) in R. To do so, we will use a\npublicly available data set generated by the Genotype-Tissue Expression (GTEx) consortium."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"If you prefer, there is also an R markdown version of this tutorial at ",(0,i.kt)("a",{target:"_blank",href:a(9222).Z},"this link")," (suitable for use in RStudio)."))),(0,i.kt)("h2",{id:"the-gtex-rna-seq-dataset"},"The GTEx RNA-seq dataset"),(0,i.kt)("p",null,"The GTEx consortium is an ongoing project which has collected tissue material from over 1,000\nindividuals. This material has been used to generate RNA-sequencing data and study patterns of gene\nactivity across different organs in the human body. For more detailed information on how the GTEx\ndata was generated, you can refer to their ",(0,i.kt)("a",{parentName:"p",href:"https://gtexportal.org/home/"},"online portal"),"."),(0,i.kt)("p",null,"For the purposes of this tutorial, you will be working with a previously processed and cleaned file\ncontaining gene expression measurements for different tissues. The measurements in this file\nrepresent an average profile, generated by computing the mean expression of each gene across all\nindividuals in the cohort. Furthermore, the course organisers cleaned this data set by removing any\nunannotated genes or any genes which were too lowly expressed to be of interest for this tutorial.\nMeasurements were normalised to account for systematic differences between batches, and expression\nvalues are presented as Transcripts per Million (TPM). For more information on why normalisation is\nneeded and how it is performed, you can refer to the accompanying slides."),(0,i.kt)("p",null,"To get started, go ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/data_visualisation/visualising_rna-seq_data/getting_setup"},"here"),"."))}p.isMDXComponent=!0},9222:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/files/Data-visualisation-practical_GTEx-6857bb2ec3b1d9079f256ea252f34510.Rmd"}}]);