"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[4097],{3905:(e,t,a)=>{a.d(t,{Zo:()=>p,kt:()=>u});var n=a(7294);function i(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function o(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function s(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?o(Object(a),!0).forEach((function(t){i(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):o(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,i=function(e,t){if(null==e)return{};var a,n,i={},o=Object.keys(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||(i[a]=e[a]);return i}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(n=0;n<o.length;n++)a=o[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(i[a]=e[a])}return i}var r=n.createContext({}),m=function(e){var t=n.useContext(r),a=t;return e&&(a="function"==typeof e?e(t):s(s({},t),e)),a},p=function(e){var t=m(e.components);return n.createElement(r.Provider,{value:t},e.children)},c="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},h=n.forwardRef((function(e,t){var a=e.components,i=e.mdxType,o=e.originalType,r=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),c=m(a),h=i,u=c["".concat(r,".").concat(h)]||c[h]||d[h]||o;return a?n.createElement(u,s(s({ref:t},p),{},{components:a})):n.createElement(u,s({ref:t},p))}));function u(e,t){var a=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var o=a.length,s=new Array(o);s[0]=h;var l={};for(var r in t)hasOwnProperty.call(t,r)&&(l[r]=t[r]);l.originalType=e,l[c]="string"==typeof e?e:i,s[1]=l;for(var m=2;m<o;m++)s[m]=a[m];return n.createElement.apply(null,s)}return n.createElement.apply(null,a)}h.displayName="MDXCreateElement"},2067:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>r,contentTitle:()=>s,default:()=>d,frontMatter:()=>o,metadata:()=>l,toc:()=>m});var n=a(7462),i=(a(7294),a(3905));const o={sidebar_position:2},s="Analysing a GWAS",l={unversionedId:"genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control",id:"genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control",title:"Analysing a GWAS",description:"We will now work with a set of data files containing many SNPs from chromosome 19 genotyped on controls and cases. Data from a",source:"@site/docs/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control.md",sourceDirName:"genome_wide_association_studies/genome_wide_association_analysis",slug:"/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{sidebar_position:2},sidebar:"sidebar8",previous:{title:"A toy GWAS",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink"},next:{title:"Association Practical",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_association"}},r={},m=[{value:"Basic quality control using plink",id:"basic-quality-control-using-plink",level:2},{value:"Converting data to binary format",id:"converting-data-to-binary-format",level:3},{value:"Computing and analysing QC metrics",id:"computing-and-analysing-qc-metrics",level:3},{value:"Creating an analysis-ready dataset",id:"creating-an-analysis-ready-dataset",level:3},{value:"Next steps",id:"next-steps",level:3}],p={toc:m},c="wrapper";function d(e){let{components:t,...a}=e;return(0,i.kt)(c,(0,n.Z)({},p,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"analysing-a-gwas"},"Analysing a GWAS"),(0,i.kt)("p",null,"We will now work with a set of data files containing many SNPs from chromosome 19 genotyped on controls and cases. Data from a\nGWAS would contain SNPs at this density across the entire genome, but we will focus on just one chromosome to make the\nexercises more tractable."),(0,i.kt)("p",null,"Our first task will be to ",(0,i.kt)("strong",{parentName:"p"},"quality control")," the dataset to remove unusual SNPs and samples."),(0,i.kt)("h2",{id:"basic-quality-control-using-plink"},"Basic quality control using plink"),(0,i.kt)("h3",{id:"converting-data-to-binary-format"},"Converting data to binary format"),(0,i.kt)("p",null,"The key files are ",(0,i.kt)("inlineCode",{parentName:"p"},"chr19-example.vcf.gz")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"chr19-example.samples"),". If you followed the ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/getting_setup"},"getting started\nsection"),", you should already have these in your folder."),(0,i.kt)("p",null,"Before getting started, let's convert these files from VCF format to a 'binary PED' format that PLINK can read more\neasily:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --make-bed \\\n--vcf chr19-example.vcf.gz \\\n--pheno chr19-example.samples \\\n--update-sex chr19-example.samples 2 \\\n--out chr19-example \\\n--keep-allele-order\n")),(0,i.kt)("p",null,"This might take a moment to run.  You should see plink has created three new files:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"chr19-example.bed\nchr19-example.bim\nchr19-example.fam\n")),(0,i.kt)("p",null,"These files contain information about the samples and SNPs, as well as the genotypes for each of the samples at each of the\nSNPs. You can look at the ",(0,i.kt)("inlineCode",{parentName:"p"},".bim")," and ",(0,i.kt)("inlineCode",{parentName:"p"},".fam")," files using ",(0,i.kt)("inlineCode",{parentName:"p"},"less"),", but the ",(0,i.kt)("inlineCode",{parentName:"p"},".bed")," file is in a special binary format that only\nplink and other software tools can understand."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"We added the ",(0,i.kt)("inlineCode",{parentName:"p"},"--keep-allele-order")," option here to the above command to make sure plink preserves the two alleles at each\nSNP in the same order as in the VCF file. This is ",(0,i.kt)("strong",{parentName:"p"},"important")," because in a GWAS analysis you will want to be able to\nkeep track of the direction of estimated effects."))),(0,i.kt)("p",null,"We can tell ",(0,i.kt)("inlineCode",{parentName:"p"},"plink")," to load data in the binary plink format using the ",(0,i.kt)("inlineCode",{parentName:"p"},"--bfile")," option. For instance, to calculate\nallele frequencies we now use:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --bfile chr19-example --freq --out chr19-example\n")),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"How many SNPs and samples are in this dataset?"))),(0,i.kt)("div",{className:"admonition admonition-note alert alert--secondary"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Many other formats are in use for genetic data, including 'GEN' format, BCF (binary VCF) format, and ",(0,i.kt)("a",{parentName:"p",href:"http://www.bgenformat.org"},"'BGEN'\nformat"),". Learning how to deal with these is part of the job description."))),(0,i.kt)("h3",{id:"computing-and-analysing-qc-metrics"},"Computing and analysing QC metrics"),(0,i.kt)("p",null,"There are many useful QC metrics that we can calculate for our dataset. These metrics can tell us about the quality of\nloci (i.e. SNPs), and of samples. For instance, we can calculation information about missingness:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --bfile chr19-example --missing --out miss-info\n")),(0,i.kt)("p",null,"This will produce a ",(0,i.kt)("inlineCode",{parentName:"p"},".imiss")," file with information about individuals and .lmiss with information about loci (SNPs)."),(0,i.kt)("p",null,"Another QC metric is sample heterozygosity:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --bfile chr19-example --het --out het-info\n")),(0,i.kt)("p",null,"Let's use R to visualize these results. "),(0,i.kt)("p",null,"You can load the output into R to look at them in more detail - we'll use ",(0,i.kt)("inlineCode",{parentName:"p"},"tidyverse")," in some of the examples, so load that now as well:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'# In R:\nlibrary( tidyverse )\nMissData = read_table( "miss-info.imiss", col_types = "ccciid" )\nHetData <- read_table( "het-info.het", col_types = "ccdddd" )\n')),(0,i.kt)("div",{className:"admonition admonition-note alert alert--secondary"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"If you prefer, you can also look at the data using a spreadsheet like Excel. Open a new spreadsheet and then use the\n",(0,i.kt)("inlineCode",{parentName:"p"},"File->Import")," menu to load the data. The files are space-delimited text files."))),(0,i.kt)("p",null,"Load the data now and take a look."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Quick question")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"What columns do these files have? (",(0,i.kt)("strong",{parentName:"p"},"Note:")," the column names can be pretty cryptic, but they're described in the ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/formats#imiss"},"plink\ndocumentation"),".)"),(0,i.kt)("p",{parentName:"div"},"Which SNP has the highest missing rate?"))),(0,i.kt)("p",null,"To make sense of this, let's plot the metrics now - we'll start with the heterozygosity. Curiously enough, even though the\ncommand is called ",(0,i.kt)("inlineCode",{parentName:"p"},"--het"),", plink does not actually output the heterzygosity rates in this file. Let's compute it now from the\n",(0,i.kt)("em",{parentName:"p"},"number of observed homozygotes")," and plot:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"# in R\nHetData$heterozygosity = ( HetData[['N(NM)']] - HetData[['O(HOM)']] ) / HetData[['N(NM)']]\nprint(\n    ggplot(\n        data = HetData,\n        aes( x = heterozygosity )\n    )\n    + geom_histogram( bins = 100, col = 'black', fill = 'grey' )\n    + theme_minimal()\n    + xlab( \"Sample heterozygosity\" )\n    \n)\n")),(0,i.kt)("p",null,"The graph you see shows the distribution of ",(0,i.kt)("em",{parentName:"p"},"heterozygosity")," across samples. QC plots often look like this; there is a large peak of samples that lie within a range of normal values (the normal samples), and then a small number of outlier samples that are usually poor quality samples. "),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"The idea here is that in any reasonable homogenous population, the level of heterozygosity - that is, the chance that\nthe two chromosomes in an individual differ at a randomly chosen SNP - will be roughtly similar across individuals.\nOr, if the population is structured, we might see subsets of the population with different values."),(0,i.kt)("p",{parentName:"div"},"If single individuals appear to have drastically different heterozygosity, this could be for example because"),(0,i.kt)("ul",{parentName:"div"},(0,i.kt)("li",{parentName:"ul"},"the samples was contaminated (it contains more than one person's DNA); or"),(0,i.kt)("li",{parentName:"ul"},"the person was somewhat inbred (contains more homozygosity than expected); or"),(0,i.kt)("li",{parentName:"ul"},"the DNA sample was poor quality (so that genotype calling was poor).")))),(0,i.kt)("p",null,"To improve the plot without losing any data, let's ",(0,i.kt)("em",{parentName:"p"},"clamp")," the values into a smaller x axis range:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"clamp <- function( x, lower = -Inf, upper = Inf ) {\n    pmax( pmin( x, upper ), lower )\n}\np = (\n    ggplot(\n        data = HetData,\n        aes( x = clamp( heterozygosity, 0.1, 0.3 ))\n    )\n    + geom_histogram( bins = 100, col = 'black', fill = 'grey' )\n    + theme_minimal()\n    + xlab( \"Sample heterozygosity, clamped to [0.1 - 0.3]\" )\n)\nprint(p)\n")),(0,i.kt)("p",null,"You can see that there is a large peak in heterozygosity around 0.2, with a number of outliers below 0.15 or above 0.25."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Saving images")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"In Rstudio you can save any figure for later reference - go to the 'Export' button above the plot and choose a destination\nfile.  Alternatively you can save it directly from ggplot using the ",(0,i.kt)("inlineCode",{parentName:"p"},"png()")," command like this:"),(0,i.kt)("pre",{parentName:"div"},(0,i.kt)("code",{parentName:"pre",className:"language-R"},'# In R:\nggsave( p, file = "HetHist.pdf" )\n')))),(0,i.kt)("p",null,"Next, lets plot the missingness values."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-R"},"# In R:\nprint(\n    ggplot(\n        data = MissData,\n        aes( x = F_MISS )\n    )\n    + geom_histogram( bins = 100, col = 'black', fill = 'grey' )\n    + theme_minimal()\n    + xlab( \"Sample missing data rate\" )\n)\n")),(0,i.kt)("p",null,"That's again not very informative because of the scale... let's zoom in to the region near zero on the x axis by clamping\nvalues to 0.2:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"print(\n    ggplot(\n        data = MissData,\n        aes( x = clamp( F_MISS, 0, 0.1 ) )\n    )\n    + geom_histogram( bins = 100, col = 'black', fill = 'grey' )\n    + theme_minimal()\n    + xlab( \"Sample missing data rate, clamped to [0 - 0.1]\" )\n)\n")),(0,i.kt)("p",null,"Again, most of the samples have low missingness (close to zero), with a number of outliers above 0.025. You can see some\nother features there as well such as the two bumps in the distribution which looks a bit suspicious - in a real study\nyou would want to look into this. For now, let's combine the two QC metrics (missingness and heterozygosity) to select\noutlying samples:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-R"},'# In R:\nqcFails <- MissData[\n    MissData$F_MISS > 0.02\n    | HetData$heterozygosity < 0.18\n    | HetData$heterozygosity > 0.22,\n    c(1:2)                   # Just capture the sample identifier fields\n]\nwrite_delim( qcFails, file = "qcFails.txt" )\n\n')),(0,i.kt)("p",null,"Here's a plot showing the thresholds we've chosen:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'print(\n    ggplot(\n        data = cbind( MissData, HetData[,3:7] ),\n        aes(\n            x = clamp( F_MISS, 0, 0.1 ),\n            y = clamp( heterozygosity, 0.1, 0.3 )\n        )\n    )\n    + geom_point()\n    + theme_minimal()\n    + geom_vline( xintercept = c( 0.02 ), col = \'red\' )\n    + geom_hline( yintercept = c( 0.18, 0.22 ))\n    + xlab( "Sample missing data rate, clamped to [0 - 0.1]" )\n    + ylab( "Sample heterozygosity, clamped to [0.1 - 0.3]" )\n)\n')),(0,i.kt)("p",null,"We should definitely do something about those two clusters!"),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge question")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Explore the dataset. Is there anything different about the samples in the two clusters? (Information on the samples is contained\nin the",(0,i.kt)("inlineCode",{parentName:"p"},"chr19-example.fam")," file (format ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/formats#fam"},"here"),") or the\n",(0,i.kt)("inlineCode",{parentName:"p"},"chr19-example.samples")," file."))),(0,i.kt)("h3",{id:"creating-an-analysis-ready-dataset"},"Creating an analysis-ready dataset"),(0,i.kt)("p",null,"Let's apply these sample filters and some basic SNP quality filters, to try to create a 'clean' dataset for analysis. We will:"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"filter samples out that have outlying missingness or heterozygosity, according to the list generated above;")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("p",{parentName:"li"},"and filter out SNPs based on SNP QC metrics using the ",(0,i.kt)("inlineCode",{parentName:"p"},"--geno"),", ",(0,i.kt)("inlineCode",{parentName:"p"},"--hwe"),", and ",(0,i.kt)("inlineCode",{parentName:"p"},"--maf")," options."))),(0,i.kt)("p",null,"Here is a plink command that does this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink \\\n--bfile chr19-example \\\n--remove qcFails.txt \\\n--hwe 1e-4 include-nonctrl \\\n--geno 0.01 \\\n--maf 0.01 \\\n--make-bed \\\n--out chr19-clean \\\n--keep-allele-order\n")),(0,i.kt)("div",{className:"admonition admonition-warning alert alert--danger"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"The backslash characters above are 'line continuation characters' - they're just there to make the command work as if we typed\nit all one one line."))),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"What SNP QC filters have we applied here? (To find out, see the\n",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/filter#hwe"},"documentation on --hwe"),",\non ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/filter#missing"},"--geno"),",\nand on ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/filter#maf"},"--maf"),")."),(0,i.kt)("p",{parentName:"div"},"Read the output of the command carefully. How many samples did the command retain? Check in R this is the expected number. And\nhow many SNPs did we keep - does this number look sensible?"))),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Important question")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Re-compute the sample missingness and heterozygosity metrics and re-create the above QC plot using the clean data. Have the SNP\nQC filters dealt with the issue that showed two clusters of samples?"),(0,i.kt)("p",{parentName:"div"},"(And why are the new missingness values much smaller than the old ones?)"))),(0,i.kt)("h3",{id:"next-steps"},"Next steps"),(0,i.kt)("p",null,"Congratulations!  You now have a carefully QC'd dataset to work with.  Now go on to the ",(0,i.kt)("a",{parentName:"p",href:"./gwas_association/"},"association practical"),"."))}d.isMDXComponent=!0}}]);