"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[5274],{3905:(e,t,a)=>{a.d(t,{Zo:()=>c,kt:()=>h});var n=a(7294);function s(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){s(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function i(e,t){if(null==e)return{};var a,n,s=function(e,t){if(null==e)return{};var a,n,s={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(s[a]=e[a]);return s}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(s[a]=e[a])}return s}var l=n.createContext({}),m=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},c=function(e){var t=m(e.components);return n.createElement(l.Provider,{value:t},e.children)},p="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},g=n.forwardRef((function(e,t){var a=e.components,s=e.mdxType,r=e.originalType,l=e.parentName,c=i(e,["components","mdxType","originalType","parentName"]),p=m(a),g=s,h=p["".concat(l,".").concat(g)]||p[g]||d[g]||r;return a?n.createElement(h,o(o({ref:t},c),{},{components:a})):n.createElement(h,o({ref:t},c))}));function h(e,t){var a=arguments,s=t&&t.mdxType;if("string"==typeof e||s){var r=a.length,o=new Array(r);o[0]=g;var i={};for(var l in t)hasOwnProperty.call(t,l)&&(i[l]=t[l]);i.originalType=e,i[p]="string"==typeof e?e:s,o[1]=i;for(var m=2;m<r;m++)o[m]=a[m];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}g.displayName="MDXCreateElement"},2360:(e,t,a)=>{a.d(t,{Z:()=>o});var n=a(7294),s=a(6010);const r={tabItem:"tabItem_OmH5"};function o(e){let{children:t,hidden:a,className:o}=e;return n.createElement("div",{role:"tabpanel",className:(0,s.Z)(r.tabItem,o),hidden:a},t)}},9877:(e,t,a)=>{a.d(t,{Z:()=>d});var n=a(7462),s=a(7294),r=a(2389),o=a(7392),i=a(7094),l=a(2466),m=a(6010);const c={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function p(e){const{lazy:t,block:a,defaultValue:r,values:p,groupId:d,className:g}=e,h=s.Children.map(e.children,(e=>{if((0,s.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),u=p??h.map((e=>{let{props:{value:t,label:a,attributes:n}}=e;return{value:t,label:a,attributes:n}})),k=(0,o.l)(u,((e,t)=>e.value===t.value));if(k.length>0)throw new Error(`Docusaurus error: Duplicate values "${k.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const v=null===r?r:r??h.find((e=>e.props.default))?.props.value??h[0]?.props.value;if(null!==v&&!u.some((e=>e.value===v)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${v}" but none of its children has the corresponding value. Available values are: ${u.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:N,setTabGroupChoices:b}=(0,i.U)(),[f,y]=(0,s.useState)(v),w=[],{blockElementScrollPositionUntilNextRender:_}=(0,l.o5)();if(null!=d){const e=N[d];null!=e&&e!==f&&u.some((t=>t.value===e))&&y(e)}const x=e=>{const t=e.currentTarget,a=w.indexOf(t),n=u[a].value;n!==f&&(_(t),y(n),null!=d&&b(d,n))},T=e=>{let t=null;switch(e.key){case"ArrowRight":{const a=w.indexOf(e.currentTarget)+1;t=w[a]||w[0];break}case"ArrowLeft":{const a=w.indexOf(e.currentTarget)-1;t=w[a]||w[w.length-1];break}}t?.focus()};return s.createElement("div",{className:(0,m.Z)("tabs-container",c.tabList)},s.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,m.Z)("tabs",{"tabs--block":a},g)},u.map((e=>{let{value:t,label:a,attributes:r}=e;return s.createElement("li",(0,n.Z)({role:"tab",tabIndex:f===t?0:-1,"aria-selected":f===t,key:t,ref:e=>w.push(e),onKeyDown:T,onFocus:x,onClick:x},r,{className:(0,m.Z)("tabs__item",c.tabItem,r?.className,{"tabs__item--active":f===t})}),a??t)}))),t?(0,s.cloneElement)(h.filter((e=>e.props.value===f))[0],{className:"margin-top--md"}):s.createElement("div",{className:"margin-top--md"},h.map(((e,t)=>(0,s.cloneElement)(e,{key:t,hidden:e.props.value!==f})))))}function d(e){const t=(0,r.Z)();return s.createElement(p,(0,n.Z)({key:String(t)},e))}},4556:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>c,contentTitle:()=>l,default:()=>h,frontMatter:()=>i,metadata:()=>m,toc:()=>p});var n=a(7462),s=(a(7294),a(3905)),r=a(9877),o=a(2360);const i={sidebar_position:2},l="Computing coding length",m={unversionedId:"programming/analysing_genes_across_species/extreme_genes/long_genes_2",id:"programming/analysing_genes_across_species/extreme_genes/long_genes_2",title:"Computing coding length",description:"If you followed so far you'll have looked at some very long genes - some of them are pretty interesting.",source:"@site/docs/programming/analysing_genes_across_species/extreme_genes/002_long_genes_2.md",sourceDirName:"programming/analysing_genes_across_species/extreme_genes",slug:"/programming/analysing_genes_across_species/extreme_genes/long_genes_2",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/long_genes_2",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/programming/analysing_genes_across_species/extreme_genes/002_long_genes_2.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{sidebar_position:2},sidebar:"sidebar5",previous:{title:"Gene length",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/long_genes_1"},next:{title:"Picking a 'canonical' transcript",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/canonical_transcripts"}},c={},p=[{value:"Computing coding sequence length",id:"computing-coding-sequence-length",level:2}],d={toc:p},g="wrapper";function h(e){let{components:t,...a}=e;return(0,s.kt)(g,(0,n.Z)({},d,a,{components:t,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"computing-coding-length"},"Computing coding length"),(0,s.kt)("p",null,"If you followed so far you'll have looked at some ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/long_genes_1"},"very long genes")," - some of them are pretty interesting."),(0,s.kt)("p",null,"But wait, that only measured gene 'length' as it is on the chromosome.  Surely a better measure would be the coding sequence length.  Let's try to compute that now."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"The coding sequence length, in bases, is of course related to the length of the encoded protein.  What's the\nrelationship?"))),(0,s.kt)("h2",{id:"computing-coding-sequence-length"},"Computing coding sequence length"),(0,s.kt)("p",null,"To compute 'the coding sequence length of a gene' we run into two problems."),(0,s.kt)("p",null,"Here we run into two problems.  The first problem is that we somehow have to compute the coding length of each transcript, which isn't listed in the dataframe at the moment.  The second is that the coding length depends on the transcript, and there are multiple transcripts per gene - we'll somehow need to pick one."),(0,s.kt)("p",null,"Let's solve the first problem now by computing the coding length of each transcript.  (We'll then solve the second\nproblem by ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/canonical_transcripts"},"computing a 'canonical' representative transcript of each gene"),")."),(0,s.kt)("p",null,"Make sure you've followed ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/analysing_genes_across_species/filter_join_merge"},"this page")," first so you will know how to do this.  You should\nalso have the ",(0,s.kt)("inlineCode",{parentName:"p"},"genes")," and ",(0,s.kt)("inlineCode",{parentName:"p"},"transcripts")," tables from that page loaded.  "),(0,s.kt)("p",null,"You can compute the coding length like this:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("strong",{parentName:"li"},"First"),", load the coding sequence records.  We will be 'joining' the result to the transcripts, so makre sure to rename the ",(0,s.kt)("inlineCode",{parentName:"li"},"Parent")," column as ",(0,s.kt)("inlineCode",{parentName:"li"},"transcript_id"),", and add a ",(0,s.kt)("inlineCode",{parentName:"li"},"cds_")," prefix to other column names to distringuish them from the transcript table.")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Solution")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)(r.Z,{mdxType:"Tabs"},(0,s.kt)(o.Z,{value:"hint",label:"Don't show me!",mdxType:"TabItem"},(0,s.kt)("p",{parentName:"div"},"Please try this yourself!"),(0,s.kt)("p",{parentName:"div"},"If you get stuck, you can click the tabs above to see solutions.")),(0,s.kt)(o.Z,{value:"R",label:"R code",mdxType:"TabItem"},(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"cds = dbGetQuery(\n    db,\n    \"SELECT dataset, ID AS cds_id, Parent AS transcript_id,\n    seqid AS cds_seqid, start AS cds_start, end AS cds_end\n    FROM gff\n    WHERE type == 'CDS'\"\n)\n"))),(0,s.kt)(o.Z,{value:"dbplyr",label:"dbplyr code",mdxType:"TabItem"},(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"cds = (\n    db\n    %>% tbl( 'gff' )\n    %>% filter( type == \"CDS\" )\n    %>% select(\n        dataset,\n        cds_id = ID,\n        transcript_id = Parent,\n        cds_seqid = seqid,\n        cds_start = start,\n        cds_end = end\n    )\n    %>% collect()\n)\n")))))),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Secondly"),", use a join to link the CDS records to the transcripts:"),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Solution")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)(r.Z,{mdxType:"Tabs"},(0,s.kt)(o.Z,{value:"hint",label:"Don't show me!",mdxType:"TabItem"},(0,s.kt)("p",{parentName:"div"},"Please try this yourself!  You can click the tabs above to see solutions.")),(0,s.kt)(o.Z,{value:"R",label:"R code",mdxType:"TabItem"},(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},'joined = left_join(\n    transcripts,\n    cds,\n    by = c( "dataset", "transcript_id" )\n)\n')),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note")," you should always sanity check to see things are correct.\nAre there any transcripts without CDS records?  Any CDS without transcripts?  Remembering that we're focussing on proteing-coding genes, satisfy yourself that the join has done the right thing."))))),(0,s.kt)("p",null,"And ",(0,s.kt)("strong",{parentName:"p"},"lastly"),", use the ",(0,s.kt)("inlineCode",{parentName:"p"},"group_by()")," and ",(0,s.kt)("inlineCode",{parentName:"p"},"summarise()")," operations to compute the coding length of each transcript."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Solution")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)(r.Z,{mdxType:"Tabs"},(0,s.kt)(o.Z,{value:"hint",label:"Don't show me!",mdxType:"TabItem"},(0,s.kt)("p",{parentName:"div"},"Please try this yourself!  You can click the tab above to see solutions.")),(0,s.kt)(o.Z,{value:"R",label:"R code",mdxType:"TabItem"},(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"transcripts_and_cds = (\n    joined\n    %>% group_by( dataset, gene_id, transcript_id )\n    %>% summarise(\n        number_of_cds = n(),\n        cds_length = sum( cds_end - cds_start + 1 ),\n        cds_start = min( cds_start ),\n        cds_end = max( cds_end )\n    )\n)\n")))))),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"As always"),", you should sanity check the results - for example add up the coding length by hand for one gene to see that the above is right.  Don't forget that the length of each record is ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mo",{parentName:"mrow",stretchy:"false"},"("),(0,s.kt)("mtext",{parentName:"mrow"},"end"),(0,s.kt)("mo",{parentName:"mrow"},"\u2212"),(0,s.kt)("mtext",{parentName:"mrow"},"start"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mn",{parentName:"mrow"},"1"),(0,s.kt)("mo",{parentName:"mrow",stretchy:"false"},")")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"(\\text{end} - \\text{start} + 1)")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1em",verticalAlign:"-0.25em"}}),(0,s.kt)("span",{parentName:"span",className:"mopen"},"("),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"end")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\u2212"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6984em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"start")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1em",verticalAlign:"-0.25em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1"),(0,s.kt)("span",{parentName:"span",className:"mclose"},")"))))),"."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"There is of course another good sanity check.  Proteins are encoded in groups of three nucleotides, so if we've got this right then all the coding lengths should be multiples of three."),(0,s.kt)("p",{parentName:"div"},"In R you can check this by using the 'modulo operator' ",(0,s.kt)("inlineCode",{parentName:"p"},"%%")," - it computes the remainder of any number after dividing by three.  Let's try:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-r"},"transcripts_and_cds$length_mod_3 = transcripts_and_cds$cds_length %% 3\ntable( transcripts_and_cds$length_mod_3 )\n")),(0,s.kt)("p",{parentName:"div"},"When I did this, it worked fine for almost all transcripts, but..."),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"     0      1      2 \n311392  14769  14877\n")),(0,s.kt)("p",{parentName:"div"},"Investigate a few of those transcripts with the 'wrong' lengths now - for example by looking at them in the ",(0,s.kt)("a",{parentName:"p",href:"https://www.ensembl.org"},"Ensembl browser")," or inspecting the data in the 'attributes' column.  Is this something we've done wrong, or something in the data?"),(0,s.kt)("p",{parentName:"div"},"The majority of genes ",(0,s.kt)("em",{parentName:"p"},"do")," have coding lengths a multiple of three, so (unless you've found a bug in your code) I suggest putting up with this issue for now.  (Another option would be to filter these records out.)"))))}h.isMDXComponent=!0}}]);