"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[5730],{3905:(e,a,t)=>{t.d(a,{Zo:()=>m,kt:()=>u});var n=t(7294);function s(e,a,t){return a in e?Object.defineProperty(e,a,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[a]=t,e}function i(e,a){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);a&&(n=n.filter((function(a){return Object.getOwnPropertyDescriptor(e,a).enumerable}))),t.push.apply(t,n)}return t}function o(e){for(var a=1;a<arguments.length;a++){var t=null!=arguments[a]?arguments[a]:{};a%2?i(Object(t),!0).forEach((function(a){s(e,a,t[a])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):i(Object(t)).forEach((function(a){Object.defineProperty(e,a,Object.getOwnPropertyDescriptor(t,a))}))}return e}function r(e,a){if(null==e)return{};var t,n,s=function(e,a){if(null==e)return{};var t,n,s={},i=Object.keys(e);for(n=0;n<i.length;n++)t=i[n],a.indexOf(t)>=0||(s[t]=e[t]);return s}(e,a);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)t=i[n],a.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(s[t]=e[t])}return s}var p=n.createContext({}),l=function(e){var a=n.useContext(p),t=a;return e&&(t="function"==typeof e?e(a):o(o({},a),e)),t},m=function(e){var a=l(e.components);return n.createElement(p.Provider,{value:a},e.children)},c="mdxType",h={inlineCode:"code",wrapper:function(e){var a=e.children;return n.createElement(n.Fragment,{},a)}},d=n.forwardRef((function(e,a){var t=e.components,s=e.mdxType,i=e.originalType,p=e.parentName,m=r(e,["components","mdxType","originalType","parentName"]),c=l(t),d=s,u=c["".concat(p,".").concat(d)]||c[d]||h[d]||i;return t?n.createElement(u,o(o({ref:a},m),{},{components:t})):n.createElement(u,o({ref:a},m))}));function u(e,a){var t=arguments,s=a&&a.mdxType;if("string"==typeof e||s){var i=t.length,o=new Array(i);o[0]=d;var r={};for(var p in a)hasOwnProperty.call(a,p)&&(r[p]=a[p]);r.originalType=e,r[c]="string"==typeof e?e:s,o[1]=r;for(var l=2;l<i;l++)o[l]=t[l];return n.createElement.apply(null,o)}return n.createElement.apply(null,t)}d.displayName="MDXCreateElement"},5006:(e,a,t)=>{t.r(a),t.d(a,{assets:()=>p,contentTitle:()=>o,default:()=>h,frontMatter:()=>i,metadata:()=>r,toc:()=>l});var n=t(7462),s=(t(7294),t(3905));const i={sidebar_position:6},o="Making a regional association plot",r={unversionedId:"genome_wide_association_studies/genome_wide_association_analysis/hitplots",id:"genome_wide_association_studies/genome_wide_association_analysis/hitplots",title:"Making a regional association plot",description:"In this part of the practical, we will hone our R plotting skills to make beautiful regional association plots (\u2018hitplots\u2019) of",source:"@site/docs/genome_wide_association_studies/genome_wide_association_analysis/hitplots.md",sourceDirName:"genome_wide_association_studies/genome_wide_association_analysis",slug:"/genome_wide_association_studies/genome_wide_association_analysis/hitplots",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/hitplots",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/genome_wide_association_analysis/hitplots.md",tags:[],version:"current",sidebarPosition:6,frontMatter:{sidebar_position:6},sidebar:"sidebar8",previous:{title:"Controlling for population structure",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/population_structure"},next:{title:"Genome-wide association analysis",permalink:"/whg-training-resources/genome_wide_association_studies/"}},p={},l=[{value:"Plotting associated regions",id:"plotting-associated-regions",level:2},{value:"Anatomy of a hitplot",id:"anatomy-of-a-hitplot",level:3},{value:"Computing linkage disequilibrium values",id:"computing-linkage-disequilibrium-values",level:3},{value:"Loading gene annotations",id:"loading-gene-annotations",level:3},{value:"Loading the recombination rate",id:"loading-the-recombination-rate",level:3},{value:"Finding a focal region",id:"finding-a-focal-region",level:3},{value:"Making an initial hitplot",id:"making-an-initial-hitplot",level:3},{value:"Refining the hitplot",id:"refining-the-hitplot",level:3},{value:"Next steps",id:"next-steps",level:3}],m={toc:l},c="wrapper";function h(e){let{components:a,...t}=e;return(0,s.kt)(c,(0,n.Z)({},m,t,{components:a,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"making-a-regional-association-plot"},"Making a regional association plot"),(0,s.kt)("p",null,"In this part of the practical, we will hone our R plotting skills to make beautiful regional association plots (\u2018hitplots\u2019) of\nthe most associated regions in our scan."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"This section of the practical is optional - it is about making publication-ready plots of the results. You should only carry\nit out if you\u2019ve successfully completed the other parts and have time to go on."))),(0,s.kt)("h2",{id:"plotting-associated-regions"},"Plotting associated regions"),(0,s.kt)("h3",{id:"anatomy-of-a-hitplot"},"Anatomy of a hitplot"),(0,s.kt)("p",null,"A good regional association plot - or 'hitplot' for short - shows several things:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},"The evidence for association across the region"),(0,s.kt)("li",{parentName:"ul"},"Information about the recombination landscape and linkage disequilibrium"),(0,s.kt)("li",{parentName:"ul"},"Regional genes and other potentially interesting annotations."),(0,s.kt)("li",{parentName:"ul"},"Information on the variants being tested \u2013 such as whether they are directly typed or imputed (in this practical, we assume all variants are directly typed) or whether they have predicted function.")),(0,s.kt)("p",null,"To make a hitplot we need to load data on all of these things. To begin, make sure you have loaded the association scan from\nthe last part of the practical:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'# In R:\ndata <- read.table( "pccorrected-test.assoc.logistic", header = TRUE )\ndata <- data[data$TEST == "ADD",]\n')),(0,s.kt)("h3",{id:"computing-linkage-disequilibrium-values"},"Computing linkage disequilibrium values"),(0,s.kt)("p",null,"We\u2019ll start by getting plink to compute LD between the most-associated SNP in each of the strongest three association peaks\nand the other SNPs in the dataset. First let\u2019s identify what the top SNPs in each region are (we'll take those with the lowest\nP-value here):"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"# In R:\ndata <- data[order( data$P ), ]\ndata[ which( data$BP > 2.5E7 & data$BP < 3.5E7 )[1],]\ndata[ which( data$BP < 1.5E7 )[1],]\ndata[ which( data$BP > 4.5E7 )[1],]\n")),(0,s.kt)("p",null,"Note down the rsids (in the \u2018SNP\u2019 column) of the three SNPs. As a sanity check, all three should have a P-value less than\n",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"1"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mn",{parentName:"mrow"},"1"),(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mn",{parentName:"msup"},"0"),(0,s.kt)("mrow",{parentName:"msup"},(0,s.kt)("mo",{parentName:"mrow"},"\u2212"),(0,s.kt)("mn",{parentName:"mrow"},"6")))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"1\\times 10^{-6}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7278em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1"),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord"},"0"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"\u2212"),(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"6"))))))))))))),". In this practical we\u2019ll concentrate on the third SNP, which has the lowest P-value, but a question will ask\nyou to repeat this for the other two SNPs. Now in the terminal let\u2019s use plink to compute LD:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"./plink \\\n--bfile chr19-clean \\\n--keep resources/controls.txt \\\n--r2 dprime \\\n--ld-window 10000 \\\n--ld-window-kb 1000000 \\\n--ld-window-r2 0.05 \\\n--ld-snps rs57558361 rs376047 rs112820994 \\\n--threads 1\n")),(0,s.kt)("p",null,"(We've added ",(0,s.kt)("inlineCode",{parentName:"p"},"--threads 1")," to this command to make sure plink doesn't use lots of threads.)"),(0,s.kt)("p",null,"This command says to compute pairwise ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"r"),(0,s.kt)("mn",{parentName:"msup"},"2"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"r^{2}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"r"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2")))))))))))))," and ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"D"),(0,s.kt)("mtext",{parentName:"mrow"},"\u2019")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"D\u2019")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"D"),(0,s.kt)("span",{parentName:"span",className:"mord"},"\u2019")))))," statistics (which are measures of LD) between the three lead SNPs and\nall other SNPs within a window around them. Only SNPs with ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"r"),(0,s.kt)("mn",{parentName:"msup"},"2")),(0,s.kt)("mo",{parentName:"mrow"},">"),(0,s.kt)("mn",{parentName:"mrow"},"0.05")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"r^2 > 0.05")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8532em",verticalAlign:"-0.0391em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"r"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2")))))))),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},">"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.05")))))," will appear in the output file. (We\u2019ve restricted the\ncomputation to control samples. If these are population controls then we are therefore estimating LD in the population). Let\u2019s\nload the result into our R session:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'# In R:\nld <- read.table( "plink.ld", hea=T, as.is=T )\nhead(ld)\n')),(0,s.kt)("p",null,"Have a look at the results. The last two columns contain estimates of ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"r"),(0,s.kt)("mn",{parentName:"msup"},"2"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"r^2")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"r"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2"))))))))))))," and ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"D"),(0,s.kt)("mo",{parentName:"msup",mathvariant:"normal",lspace:"0em",rspace:"0em"},"\u2032"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"D'")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7519em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"D"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.7519em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"\u2032")))))))))))))," between the SNPs listed in SNP_A and\nSNP_B columns. For plotting, it's simplest to bin those values:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"# In R:\nld$R2_bin <- cut( ld$R2, breaks = seq( 0, 1, by = 0.1 ))\nhead(ld)\n")),(0,s.kt)("h3",{id:"loading-gene-annotations"},"Loading gene annotations"),(0,s.kt)("p",null,"Now let's load the other data we need. First, we'll load information on genes from the file ",(0,s.kt)("inlineCode",{parentName:"p"},"resources/refGene_chr19.txt"),"\n(which originates from the UCSC Genome browser):"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'genes <- read.table( "resources/refGene_chr19.txt.gz", header = TRUE, as.is = TRUE )\nhead( genes )\n')),(0,s.kt)("p",null,"This file is a bit like the ",(0,s.kt)("inlineCode",{parentName:"p"},".gff")," format gene annotation files you may have seen elsewhere, but in a slightly different\nformat. It contains gene names (in the ",(0,s.kt)("inlineCode",{parentName:"p"},"name2")," column), transcription start and end points (",(0,s.kt)("inlineCode",{parentName:"p"},"txStart"),", ",(0,s.kt)("inlineCode",{parentName:"p"},"txEnd"),"), and the\npositions of exons (",(0,s.kt)("inlineCode",{parentName:"p"},"exonStarts"),", ",(0,s.kt)("inlineCode",{parentName:"p"},"exonEnds"),")."),(0,s.kt)("p",null,"Note that many genes in the file (such as ",(0,s.kt)("em",{parentName:"p"},"FUT2"),") have several transcripts. That's useful information but is a bit too much detail for\nour plot, so for our purposes let's just keep the longest transcript of each gene:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"select_longest_gene = function( genes ) {\n    result = genes[ order( genes$txEnd - genes$txStart, genes$exonCount, decreasing = TRUE ), ]\n    result = result[ which( !duplicated( result$name2 )), ]\n    return( result )\n}\ngenes = select_longest_gene( genes )\n")),(0,s.kt)("p",null,"We'll also load some helper code that will help us plot these genes:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"source( 'scripts/plot.genes.R' )\n")),(0,s.kt)("h3",{id:"loading-the-recombination-rate"},"Loading the recombination rate"),(0,s.kt)("p",null,"Next we'll load recombination rate estimates from the HapMap project:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'genetic_map <- read.table( "resources/genetic_map_chr19_combined_b37.txt.gz", hea=T, as.is=T )\nhead( genetic_map )\n')),(0,s.kt)("p",null,"The recombination rate is expressed in ",(0,s.kt)("em",{parentName:"p"},"centimorgans per megabase")," and provides values along the genome."),(0,s.kt)("h3",{id:"finding-a-focal-region"},"Finding a focal region"),(0,s.kt)("p",null,"We're nearly ready to plot! We'll make a hit plot near the most-associated SNP (rs112820994). Let's choose a sensible region\naround it:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"focus.snp = 'rs112820994'\nfocus.pos = data$BP[ data$SNP == focus.snp ]\nregion = c( focus.pos - 250000, focus.pos + 250000 )\n")),(0,s.kt)("h3",{id:"making-an-initial-hitplot"},"Making an initial hitplot"),(0,s.kt)("p",null,"We'll make a plot divided into sections, something like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"panel 1: Main hitplot\npanel 2: Genes\npanel 3: Recombination rate  \n")),(0,s.kt)("p",null,"To do this, we're going to use 'base R' graphics which gives us great flexibility (although it can be a bit cryptic). The\n",(0,s.kt)("inlineCode",{parentName:"p"},"layout()")," command can be used to split the plotting area into sections with different heights."),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"layout.matrix = matrix( c(1,2,3), ncol = 1 )\nlayout(\n    layout.matrix,\n    heights = c(1,0.5,0.5)\n)\n")),(0,s.kt)("p",null,"First let's plot the association statistics. Since we're plotting other stuff underneath, we'll suppress the X axis label. We\nalso use par(mar = \u2026) to reduce the space around the plot:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"par( mar = c(1,4,2,2))\nplot(\n    data$BP,\n    -log10( data$P ),\n    xlim = region,\n    xlab = '',\n    ylab = \"-log10( P-value )\"\n)\n")),(0,s.kt)("p",null,"Now let's use the ",(0,s.kt)("inlineCode",{parentName:"p"},"plot.genes")," function loaded above to plot genes in the second panel, and plot the recombination rate"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"plot.genes( genes, region )\nplot(\n    genetic_map$position,\n    genetic_map$ COMBINED_rate.cM.Mb.,\n    type = 'l',\n    xlim = region,\n    ylab = \"cM/Mb\"\n)\n")),(0,s.kt)("p",null,"Beautiful!  But not beautiful enough.  Let's refine it."),(0,s.kt)("h3",{id:"refining-the-hitplot"},"Refining the hitplot"),(0,s.kt)("p",null,"Now we'll make several modifications to make the plot really publication-ready. First we'll colour points using the binned ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"r"),(0,s.kt)("mn",{parentName:"msup"},"2"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"r^2")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02778em"}},"r"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2")))))))))))),"\nvalues we loaded earlier. (To do this, we'll use the ",(0,s.kt)("inlineCode",{parentName:"p"},"heat.colors()")," function to generate a nice palette for this):"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'palette = heat.colors(10)\nregion.ld = ld[ ld$SNP_A == focus.snp & ld$BP_B >= region[1] & ld$BP_B <= region[2] & !is.na( ld$R2_bin ), ]\ndata$colour = "black"\ndata$colour[ match( region.ld$SNP_B, data$SNP ) ] = palette[ as.integer( region.ld$R2_bin ) ]\n')),(0,s.kt)("p",null,"We can plot it again using these colours like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"par( mar = c( 1, 4, 2, 2 ))\nplot(\n    data$BP, -log10( data$P ),\n    col = data$colour,\n    pch = 20,\n    xlim = region,\n    xlab = '',\n    ylab = \"-log10( P-value )\"\n)\n")),(0,s.kt)("p",null,"(In the above we used ",(0,s.kt)("inlineCode",{parentName:"p"},"pch=20")," to get filled circles instead of open ones.)"),(0,s.kt)("p",null,"It's also helpful to provide a legend. The ",(0,s.kt)("inlineCode",{parentName:"p"},"legend()")," command can be used to add one to the plot. A legend with helpful labels\nis:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},'legend(\n    "topright",\n    col=c("black",heat.colors(10)),\n    pch = 20,\n    legend = c(\n        "r\xb2<0.1",\n        sprintf( "r\xb2\u2265%.1f",seq(0.1,0.9,by=0.1 ))\n    )\n)\n')),(0,s.kt)("p",null,"We can also add a grid to let viewers locate points better:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"grid()\n")),(0,s.kt)("p",null,"The plot is... getting there."),(0,s.kt)("h3",{id:""}),(0,s.kt)("p",null,"To production-ise this, we have wrapped up the above code - with a few more cosmetic tweaks - into a function ",(0,s.kt)("inlineCode",{parentName:"p"},"hitplot()"),",\nstored in the file ",(0,s.kt)("inlineCode",{parentName:"p"},"scripts/hitplot.R"),". Load this into Rstudio and run it. You can use it to produce the finished plot like\nthis:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"source( 'scripts/hitplot.R' )\nhitplot( 'rs112820994', data, genes, ld, margin = 200000 )\n")),(0,s.kt)("p",null,"Or save to a pdf like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"pdf( file = \"rs112820994_hitplot.pdf\", width = 8, height = 6 )\nhitplot( 'rs112820994', data, genes, ld, margin = 200000 )\ndev.off()\n")),(0,s.kt)("p",null,"Now use your plot to answer these questions:"),(0,s.kt)("div",{className:"admonition admonition-note alert alert--secondary"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"Questions")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("ul",{parentName:"div"},(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"What gene is rs112820994 in? Is it in an exon? Are other top SNPs in the region also in genes? (You may need to play around\nwith the ",(0,s.kt)("inlineCode",{parentName:"p"},"margin")," argument to zoom in or out of the plot. You can cross-check results using genome browsers such as the\n",(0,s.kt)("a",{parentName:"p",href:"https://genome.ucsc.edu/cgi-bin/hgGateway"},"UCSC Genome Browser")," - note that we are using GRCh37/hg19 coordinates).")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"Does the extent of association look sensible with respect to LD? Are there any obvious features near the boundaries of the\nregion of associated SNPs that could explain why the association signal dies off there?")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"Make a similar plot for each of the other regions we computed LD for above. Do they look similar? Is there anything odd\nabout them?")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"Having looked at the variants in the genome browser \u2013 what other genomic features would you add to the hitplot if you had\ntime? How would you add these?"))))),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"This one is a real coding challenge - only try this if you've got time. In the sample folder there's a file called\n\"resources/snp138_chr19.txt.gz\" containing information from dbSNP on variants in each region.  You can load this in like this:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},'functional_data = read.table( "resources/snp138_chr19.txt.gz", sep = "\\t", header = TRUE )\n')),(0,s.kt)("p",{parentName:"div"},"The column called ",(0,s.kt)("inlineCode",{parentName:"p"},"func")," lists putative functional consequences of each variant."),(0,s.kt)("p",{parentName:"div"},"The challenge is to create a version of the hit plot that overplots different shapes to show the functional annotation."),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Hints:")),(0,s.kt)("ol",{parentName:"div"},(0,s.kt)("li",{parentName:"ol"},"The 'func' column in snp138 contains many categories and you'll first need to simplify it for plotting. One way is to take\nthe most 'severe' category for each SNP. A rough list of useful categories, in descending order of imporance, might be:\nmissense, nonsense, stop-gain, stop-loss, splice-3, splice-5, untranslated-3, untranslated-5, intron. E.g. the following code\nshows how to make a simpler variable func2:")),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-R"},"snp138$func2 = NA\nsnp138$func2[grep('intron',snp138$func)] = 'intron'\nsnp138$func2[grep('untranslated-5',snp138$func)] = 'untranslated-5'\nsnp138$func2[grep('untranslated-3',snp138$func)] = 'untranslated-3'\n# \u2026etc.\n")),(0,s.kt)("p",{parentName:"div"},"(To cut down copy-pasting, you may want to turn the above into a loop - make sure to go through them in least-to-most important order.)"),(0,s.kt)("ol",{parentName:"div",start:2},(0,s.kt)("li",{parentName:"ol"},(0,s.kt)("p",{parentName:"li"},"Use this to add a 'functional_annotation' column to the association test data frame, data. You should be able to match snp138 to data using\nthe name column. (This depends on the SNPs being named correctly though! Are there any SNPs that match by position but not\nname?)")),(0,s.kt)("li",{parentName:"ol"},(0,s.kt)("p",{parentName:"li"},"The ",(0,s.kt)("inlineCode",{parentName:"p"},"points()")," command can be used to overplot points on an existing plot. Use the pch argument to specify shape. (Only plot\npoints that have functional_annotation != 'none').")),(0,s.kt)("li",{parentName:"ol"},(0,s.kt)("p",{parentName:"li"},"pch takes an argument that is a positive integer in the range 1-20, say (google for 'R shapes' to see a table of shapes).\nOne way to turn categories into integers for pch is to first convert them to a factor, e.g."))),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre",className:"language-R"},"data$functional_annotation = factor(\n    data$functional_annotation,\n    levels = c(\n        'missense', 'nonsense', 'stop-gain', 'stop-loss', 'splice-3', 'splice-5', 'untranslated-3', 'untranslated-5', 'intron'\n    )\ntable( as.integer(data$functional_annotation) )\n")),(0,s.kt)("ol",{parentName:"div",start:5},(0,s.kt)("li",{parentName:"ol"},"You also need a legend!")))),(0,s.kt)("h3",{id:"next-steps"},"Next steps"),(0,s.kt)("p",null,"Congratulations! You have now conducted a complete GWAS study and started on the next steps: trying to figure out what the\nassociated SNPs do.  This marks the end of this practical."))}h.isMDXComponent=!0}}]);