"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[783],{3905:(a,e,t)=>{t.d(e,{Zo:()=>o,kt:()=>k});var s=t(7294);function n(a,e,t){return e in a?Object.defineProperty(a,e,{value:t,enumerable:!0,configurable:!0,writable:!0}):a[e]=t,a}function m(a,e){var t=Object.keys(a);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(a);e&&(s=s.filter((function(e){return Object.getOwnPropertyDescriptor(a,e).enumerable}))),t.push.apply(t,s)}return t}function p(a){for(var e=1;e<arguments.length;e++){var t=null!=arguments[e]?arguments[e]:{};e%2?m(Object(t),!0).forEach((function(e){n(a,e,t[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(a,Object.getOwnPropertyDescriptors(t)):m(Object(t)).forEach((function(e){Object.defineProperty(a,e,Object.getOwnPropertyDescriptor(t,e))}))}return a}function i(a,e){if(null==a)return{};var t,s,n=function(a,e){if(null==a)return{};var t,s,n={},m=Object.keys(a);for(s=0;s<m.length;s++)t=m[s],e.indexOf(t)>=0||(n[t]=a[t]);return n}(a,e);if(Object.getOwnPropertySymbols){var m=Object.getOwnPropertySymbols(a);for(s=0;s<m.length;s++)t=m[s],e.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(a,t)&&(n[t]=a[t])}return n}var r=s.createContext({}),l=function(a){var e=s.useContext(r),t=e;return a&&(t="function"==typeof a?a(e):p(p({},e),a)),t},o=function(a){var e=l(a.components);return s.createElement(r.Provider,{value:e},a.children)},N="mdxType",c={inlineCode:"code",wrapper:function(a){var e=a.children;return s.createElement(s.Fragment,{},e)}},h=s.forwardRef((function(a,e){var t=a.components,n=a.mdxType,m=a.originalType,r=a.parentName,o=i(a,["components","mdxType","originalType","parentName"]),N=l(t),h=n,k=N["".concat(r,".").concat(h)]||N[h]||c[h]||m;return t?s.createElement(k,p(p({ref:e},o),{},{components:t})):s.createElement(k,p({ref:e},o))}));function k(a,e){var t=arguments,n=e&&e.mdxType;if("string"==typeof a||n){var m=t.length,p=new Array(m);p[0]=h;var i={};for(var r in e)hasOwnProperty.call(e,r)&&(i[r]=e[r]);i.originalType=a,i[N]="string"==typeof a?a:n,p[1]=i;for(var l=2;l<m;l++)p[l]=t[l];return s.createElement.apply(null,p)}return s.createElement.apply(null,t)}h.displayName="MDXCreateElement"},6833:(a,e,t)=>{t.r(e),t.d(e,{assets:()=>r,contentTitle:()=>p,default:()=>c,frontMatter:()=>m,metadata:()=>i,toc:()=>l});var s=t(7462),n=(t(7294),t(3905));const m={sidebar_position:2},p="Meta-analysing two studies.",i={unversionedId:"genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis",id:"genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis",title:"Meta-analysing two studies.",description:"Up to the table of contents / Back to the Introduction / Forward to the fine-mapping section",source:"@site/docs/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis.md",sourceDirName:"genome_wide_association_studies/meta-analysis_and_fine-mapping",slug:"/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis",permalink:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{sidebar_position:2},sidebar:"sidebar8",previous:{title:"Meta-analysis and fine-mapping practical",permalink:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction"},next:{title:"Using FINEMAP to fine-map associations",permalink:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping"}},r={},l=[{value:"Conducting fixed-effect meta-analysis",id:"conducting-fixed-effect-meta-analysis",level:3},{value:"Running the meta-analsis",id:"running-the-meta-analsis",level:3},{value:"Making a forest plot",id:"making-a-forest-plot",level:3},{value:"Other meta-analysis models",id:"other-meta-analysis-models",level:3},{value:"Fine-mapping the association",id:"fine-mapping-the-association",level:3}],o={toc:l},N="wrapper";function c(a){let{components:e,...m}=a;return(0,n.kt)(N,(0,s.Z)({},o,m,{components:e,mdxType:"MDXLayout"}),(0,n.kt)("h1",{id:"meta-analysing-two-studies"},"Meta-analysing two studies."),(0,n.kt)("p",null,(0,n.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"},"Up to the table of contents")," / ",(0,n.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction"},"Back to the Introduction")," / ",(0,n.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping"},"Forward to the fine-mapping section")),(0,n.kt)("h3",{id:"conducting-fixed-effect-meta-analysis"},"Conducting fixed-effect meta-analysis"),(0,n.kt)("p",null,"The simplest way to perform meta-analysis is as follows.  We assume:"),(0,n.kt)("ul",null,(0,n.kt)("li",{parentName:"ul"},"that the ",(0,n.kt)("em",{parentName:"li"},"true effect")," is the same in both studies."),(0,n.kt)("li",{parentName:"ul"},"that the ",(0,n.kt)("em",{parentName:"li"},"observed effect")," in each study is equal to the ",(0,n.kt)("em",{parentName:"li"},"true effect")," plus noise.  (The noise is given by the study standard error, and is assumed to be gaussian in the usual way.)")),(0,n.kt)("p",null,"The full name of this is ",(0,n.kt)("em",{parentName:"p"},"inverse variance weighted fixed-effect meta-analysis"),".  "),(0,n.kt)("p",null,"Here's how it works: form a weighted average of the effect estimates, weighted by the inverse of the variances:"),(0,n.kt)("div",{className:"math math-display"},(0,n.kt)("span",{parentName:"div",className:"katex-display"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("mi",{parentName:"mrow"},"b"),(0,n.kt)("mo",{parentName:"mrow"},"="),(0,n.kt)("munder",{parentName:"mrow"},(0,n.kt)("mo",{parentName:"munder"},"\u2211"),(0,n.kt)("mi",{parentName:"munder"},"i")),(0,n.kt)("mfrac",{parentName:"mrow"},(0,n.kt)("msub",{parentName:"mfrac"},(0,n.kt)("mi",{parentName:"msub"},"\u03b2"),(0,n.kt)("mi",{parentName:"msub"},"i")),(0,n.kt)("msubsup",{parentName:"mfrac"},(0,n.kt)("mtext",{parentName:"msubsup"},"se"),(0,n.kt)("mi",{parentName:"msubsup"},"i"),(0,n.kt)("mn",{parentName:"msubsup"},"2")))),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"b = \\sum_i \\frac{\\beta_i}{\\text{se}_i^2}")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6944em"}}),(0,n.kt)("span",{parentName:"span",className:"mord mathnormal"},"b"),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,n.kt)("span",{parentName:"span",className:"mrel"},"="),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"2.6491em",verticalAlign:"-1.2777em"}}),(0,n.kt)("span",{parentName:"span",className:"mop op-limits"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.05em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-1.8723em",marginLeft:"0em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3.05em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i"))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3.05em"}}),(0,n.kt)("span",{parentName:"span"},(0,n.kt)("span",{parentName:"span",className:"mop op-symbol large-op"},"\u2211")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.2777em"}},(0,n.kt)("span",{parentName:"span"}))))),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.1667em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,n.kt)("span",{parentName:"span",className:"mfrac"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.3714em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.314em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord text"},(0,n.kt)("span",{parentName:"span",className:"mord"},"se")),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.7959em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.4231em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i"))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.0448em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},"2")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.2769em"}},(0,n.kt)("span",{parentName:"span"})))))))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,n.kt)("span",{parentName:"span",style:{top:"-3.677em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3117em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"-0.0528em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,n.kt)("span",{parentName:"span"}))))))))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.9629em"}},(0,n.kt)("span",{parentName:"span"}))))),(0,n.kt)("span",{parentName:"span",className:"mclose nulldelimiter"}))))))),(0,n.kt)("p",null,"Compute also the sum of weights:"),(0,n.kt)("div",{className:"math math-display"},(0,n.kt)("span",{parentName:"div",className:"katex-display"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("mi",{parentName:"mrow"},"w"),(0,n.kt)("mo",{parentName:"mrow"},"="),(0,n.kt)("munder",{parentName:"mrow"},(0,n.kt)("mo",{parentName:"munder"},"\u2211"),(0,n.kt)("mi",{parentName:"munder"},"i")),(0,n.kt)("mfrac",{parentName:"mrow"},(0,n.kt)("mn",{parentName:"mfrac"},"1"),(0,n.kt)("msubsup",{parentName:"mfrac"},(0,n.kt)("mtext",{parentName:"msubsup"},"se"),(0,n.kt)("mi",{parentName:"msubsup"},"i"),(0,n.kt)("mn",{parentName:"msubsup"},"2")))),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"w = \\sum_i \\frac{1}{\\text{se}_i^2}")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.4306em"}}),(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02691em"}},"w"),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,n.kt)("span",{parentName:"span",className:"mrel"},"="),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"2.5991em",verticalAlign:"-1.2777em"}}),(0,n.kt)("span",{parentName:"span",className:"mop op-limits"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.05em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-1.8723em",marginLeft:"0em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3.05em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i"))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3.05em"}}),(0,n.kt)("span",{parentName:"span"},(0,n.kt)("span",{parentName:"span",className:"mop op-symbol large-op"},"\u2211")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.2777em"}},(0,n.kt)("span",{parentName:"span"}))))),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.1667em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,n.kt)("span",{parentName:"span",className:"mfrac"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.3214em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.314em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord text"},(0,n.kt)("span",{parentName:"span",className:"mord"},"se")),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.7959em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.4231em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i"))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.0448em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},"2")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.2769em"}},(0,n.kt)("span",{parentName:"span"})))))))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,n.kt)("span",{parentName:"span",style:{top:"-3.677em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord"},"1")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.9629em"}},(0,n.kt)("span",{parentName:"span"}))))),(0,n.kt)("span",{parentName:"span",className:"mclose nulldelimiter"}))))))),(0,n.kt)("p",null,"Then the ",(0,n.kt)("em",{parentName:"p"},"meta-analysis estimate")," is"),(0,n.kt)("div",{className:"math math-display"},(0,n.kt)("span",{parentName:"div",className:"katex-display"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("msub",{parentName:"mrow"},(0,n.kt)("mi",{parentName:"msub"},"\u03b2"),(0,n.kt)("mtext",{parentName:"msub"},"meta")),(0,n.kt)("mo",{parentName:"mrow"},"="),(0,n.kt)("mi",{parentName:"mrow"},"b"),(0,n.kt)("mo",{parentName:"mrow"},"\u2217"),(0,n.kt)("mi",{parentName:"mrow"},"w")),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta_{\\text{meta}} = b * w")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.2806em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"-0.0528em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},(0,n.kt)("span",{parentName:"span",className:"mord text mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},"meta")))))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,n.kt)("span",{parentName:"span"})))))),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,n.kt)("span",{parentName:"span",className:"mrel"},"="),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6944em"}}),(0,n.kt)("span",{parentName:"span",className:"mord mathnormal"},"b"),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,n.kt)("span",{parentName:"span",className:"mbin"},"\u2217"),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.4306em"}}),(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02691em"}},"w")))))),(0,n.kt)("p",null,"and the ",(0,n.kt)("em",{parentName:"p"},"meta-analysis standard error")," is"),(0,n.kt)("div",{className:"math math-display"},(0,n.kt)("span",{parentName:"div",className:"katex-display"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("msub",{parentName:"mrow"},(0,n.kt)("mi",{parentName:"msub"},"\u03b2"),(0,n.kt)("mtext",{parentName:"msub"},"meta")),(0,n.kt)("mo",{parentName:"mrow"},"="),(0,n.kt)("mfrac",{parentName:"mrow"},(0,n.kt)("mn",{parentName:"mfrac"},"1"),(0,n.kt)("msqrt",{parentName:"mfrac"},(0,n.kt)("mi",{parentName:"msqrt"},"w")))),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta_{\\text{meta}} = \\frac{1}{\\sqrt{w}}")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.2806em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"-0.0528em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},(0,n.kt)("span",{parentName:"span",className:"mord text mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mtight"},"meta")))))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,n.kt)("span",{parentName:"span"})))))),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,n.kt)("span",{parentName:"span",className:"mrel"},"="),(0,n.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"2.2514em",verticalAlign:"-0.93em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,n.kt)("span",{parentName:"span",className:"mfrac"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"1.3214em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.3097em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord sqrt"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8003em"}},(0,n.kt)("span",{parentName:"span",className:"svg-align",style:{top:"-3em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord",style:{paddingLeft:"0.833em"}},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.02691em"}},"w"))),(0,n.kt)("span",{parentName:"span",style:{top:"-2.7603em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"hide-tail",style:{minWidth:"0.853em",height:"1.08em"}},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"400em",height:"1.08em",viewBox:"0 0 400000 1080",preserveAspectRatio:"xMinYMin slice"},(0,n.kt)("path",{parentName:"svg",d:"M95,702\nc-2.7,0,-7.17,-2.7,-13.5,-8c-5.8,-5.3,-9.5,-10,-9.5,-14\nc0,-2,0.3,-3.3,1,-4c1.3,-2.7,23.83,-20.7,67.5,-54\nc44.2,-33.3,65.8,-50.3,66.5,-51c1.3,-1.3,3,-2,5,-2c4.7,0,8.7,3.3,12,10\ns173,378,173,378c0.7,0,35.3,-71,104,-213c68.7,-142,137.5,-285,206.5,-429\nc69,-144,104.5,-217.7,106.5,-221\nl0 -0\nc5.3,-9.3,12,-14,20,-14\nH400000v40H845.2724\ns-225.272,467,-225.272,467s-235,486,-235,486c-2.7,4.7,-9,7,-19,7\nc-6,0,-10,-1,-12,-3s-194,-422,-194,-422s-65,47,-65,47z\nM834 80h400000v40h-400000z"}))))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.2397em"}},(0,n.kt)("span",{parentName:"span"}))))))),(0,n.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,n.kt)("span",{parentName:"span",style:{top:"-3.677em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord"},"1")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.93em"}},(0,n.kt)("span",{parentName:"span"}))))),(0,n.kt)("span",{parentName:"span",className:"mclose nulldelimiter"}))))))),(0,n.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,n.kt)("div",{parentName:"div",className:"admonition-heading"},(0,n.kt)("h5",{parentName:"div"},(0,n.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,n.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,n.kt)("div",{parentName:"div",className:"admonition-content"},(0,n.kt)("p",{parentName:"div"},"There are two good reasons to compute the estimate this way. Firstly, it makes sense: the meta-analysis estimate is a\nweighted average of the per-study estimates, and they are weighted by the variance. So studies with lots of uncertainty (large\nvariance) get weighted down while studies with little uncertainty (small variance) get higher weight."),(0,n.kt)("p",{parentName:"div"},"The second reason is that, as it turns out, this meta-analysis estimate is precisely ",(0,n.kt)("strong",{parentName:"p"},"what you would have computed")," if you\nhad estimated using all the data together** (at least approximately). The maths behind this is based on the remarkable fact\nthat products of normal distributions are also normal distributions. If we approximate the likelihood function from each study\nby a normal - that is, by its estimate ",(0,n.kt)("span",{parentName:"p",className:"math math-inline"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("msub",{parentName:"mrow"},(0,n.kt)("mi",{parentName:"msub"},"\u03b2"),(0,n.kt)("mi",{parentName:"msub"},"i"))),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta_i")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3117em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.55em",marginLeft:"-0.0528em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,n.kt)("span",{parentName:"span"}))))))))))," and standard deviation ",(0,n.kt)("span",{parentName:"p",className:"math math-inline"},(0,n.kt)("span",{parentName:"span",className:"katex"},(0,n.kt)("span",{parentName:"span",className:"katex-mathml"},(0,n.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,n.kt)("semantics",{parentName:"math"},(0,n.kt)("mrow",{parentName:"semantics"},(0,n.kt)("msub",{parentName:"mrow"},(0,n.kt)("mtext",{parentName:"msub"},"se"),(0,n.kt)("mi",{parentName:"msub"},"i"))),(0,n.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\text{se}_i")))),(0,n.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,n.kt)("span",{parentName:"span",className:"base"},(0,n.kt)("span",{parentName:"span",className:"strut",style:{height:"0.5806em",verticalAlign:"-0.15em"}}),(0,n.kt)("span",{parentName:"span",className:"mord"},(0,n.kt)("span",{parentName:"span",className:"mord text"},(0,n.kt)("span",{parentName:"span",className:"mord"},"se")),(0,n.kt)("span",{parentName:"span",className:"msupsub"},(0,n.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.3117em"}},(0,n.kt)("span",{parentName:"span",style:{top:"-2.55em",marginRight:"0.05em"}},(0,n.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,n.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,n.kt)("span",{parentName:"span",className:"mord mathnormal mtight"},"i")))),(0,n.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,n.kt)("span",{parentName:"span",className:"vlist-r"},(0,n.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.15em"}},(0,n.kt)("span",{parentName:"span"}))))))))))," - then it turns out that this product is\na new normal density, and it has exactly the meta-analysis estimates as mean and standard deviation."))),(0,n.kt)("h3",{id:"running-the-meta-analsis"},"Running the meta-analsis"),(0,n.kt)("p",null,"Here is a function to run the meta-analysis from our two studies:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"meta.analyse <- function( beta1, se1, beta2, se2 ) {\n  inverse.variances = c( 1/se1^2, 1/se2^2 )\n  b = c( beta1, beta2 ) * inverse.variances\n  meta.inverse.variance = sum( inverse.variances )\n  meta.beta = sum( b ) / meta.inverse.variance\n  meta.se = sqrt( 1 / meta.inverse.variance )\n  \n  # This formulation computes a two-tailed P-value\n  # as discussed in lectures\n  meta.P = pnorm( -abs( meta.beta ), sd = meta.se ) * 2\n  \n  # This formulation computes a Bayes factor under a N(0,0.2^2) prior\n  meta.BF = (\n    pnorm( meta.beta, mean = 0, sd = sqrt( meta.se^2 + 0.2 ) ) /\n    pnorm( meta.beta, mean = 0, sd = meta.se )\n  )\n\n  return( list(\n    meta.beta = meta.beta,\n    meta.se = meta.se,\n    meta.P = meta.P,\n    log10_meta.BF = log10( meta.BF )\n  ))\n}\n")),(0,n.kt)("p",null,"Try it on some fake data, for example:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"> meta.analyse( beta1 = 1, se1 = 0.1, beta2 = 1, se2 = 0.1 )\n")),(0,n.kt)("p",null,"Try a few different input values.  You should see:"),(0,n.kt)("ul",null,(0,n.kt)("li",{parentName:"ul"},"The ",(0,n.kt)("em",{parentName:"li"},"meta-analysis estimate")," should be somewhere between the two input estimates."),(0,n.kt)("li",{parentName:"ul"},"The ",(0,n.kt)("em",{parentName:"li"},"meta-analysis standard error")," should be smaller than either of the two study standard errors."),(0,n.kt)("li",{parentName:"ul"},"The ",(0,n.kt)("em",{parentName:"li"},"meta-analysis P-value")," depends on how well the two estimates 'stack up' on each other.  If they are both in the same direction and of roughly comparable size, the P-value will be lower than either of the study P-values.")),(0,n.kt)("p",null,"Here is a way to run the above function across all the data in our study."),(0,n.kt)("p",null,(0,n.kt)("strong",{parentName:"p"},"Note.")," We will use the ",(0,n.kt)("a",{parentName:"p",href:"https://purrr.tidyverse.org"},(0,n.kt)("inlineCode",{parentName:"a"},"purrr"))," function ",(0,n.kt)("inlineCode",{parentName:"p"},"map_dfr()")," to run the\nabove function across all rows and return a data frame. If you don't have ",(0,n.kt)("inlineCode",{parentName:"p"},"purrr"),", you could use a\nbase R function like ",(0,n.kt)("inlineCode",{parentName:"p"},"lapply()")," instead."),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"meta_analysis = map_dfr(\n  1:nrow(study1),\n  function( i ) {\n    c(\n      rsid = study1$rsid[i],\n      chromosome = study1$chromosome[i],\n      position = study1$position[i],\n      allele1 = study1$allele1[i],\n      allele2 = study1$allele2[i],\n      study1.beta = study1$beta[i],\n      study1.se = study1$se[i],\n      study1.P = study1$P[i],\n      log10_study1.BF = study1$log10_BF[i],\n      study2.beta = study2$beta[i],\n      study2.se = study2$se[i],\n      study2.P = study2$P[i],\n      log10_study2.BF = study2$log10_BF[i],\n      meta.analyse(\n        study1$beta[i], study1$se[i],\n        study2$beta[i], study2$se[i]\n      )\n    )\n  }\n)\n")),(0,n.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,n.kt)("div",{parentName:"div",className:"admonition-heading"},(0,n.kt)("h5",{parentName:"div"},(0,n.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,n.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,n.kt)("div",{parentName:"div",className:"admonition-content"},(0,n.kt)("p",{parentName:"div"},"What does the meta-analysis look like for the 'top' SNPs (e.g. those with the lowest\nP-values or highest Bayes factors) in study1 or 2? Is the evidence stronger or weaker across\nstudies? Has the meta-analysis changed the SNP or SNPs with the strongest evidence? Why?"),(0,n.kt)("p",{parentName:"div"},(0,n.kt)("strong",{parentName:"p"},"Hint.")," One way to pull out the 'top' SNps is to use the ",(0,n.kt)("inlineCode",{parentName:"p"},"order()")," function. For example to pull\nout the results for the ten SNPs with the lowest P-values in study 1, in order:"),(0,n.kt)("pre",{parentName:"div"},(0,n.kt)("code",{parentName:"pre"},"meta_analysis[ order( meta_analysis$study1.P )[1:10], ]\n")))),(0,n.kt)("h3",{id:"making-a-forest-plot"},"Making a forest plot"),(0,n.kt)("p",null,"To make best sense of this the tool for the job is a ",(0,n.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Forest_plot"},(0,n.kt)("em",{parentName:"a"},"forest plot")),".  We plot the study estimates and the meta-analysis estimate, along with their confidence/credible intervals, on seperate lines.  Let's do that now.  First here's our utility function to make a blank canvas:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"blank.plot <- function( xlim = c( 0, 1 ), ylim = c( 0, 1 ), xlab = \"\", ylab = \"\" ) {\n  # this function plots a blank canvas\n  plot(\n    0, 0,\n    col = 'white', # draw points white\n    bty = 'n',     # no border\n    xaxt = 'n',    # no x axis\n    yaxt = 'n',    # no y axis\n    xlab = xlab,   # no x axis label\n    ylab = ylab,   # no x axis label\n    xlim = xlim,\n    ylim = ylim\n  )\n}\n")),(0,n.kt)("p",null,"Now let's make a general forest plot function:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"\ndraw.forest.plot <- function(\n  betas,\n  ses,\n  names\n) {\n  # y axis locations for the lines.\n  # We assume the meta-analysis will go on the last line, so we separate it slightly by putting it at 1/2\n  y = c( length(betas):2, 0.5 )\n  \n  # learn a good x axis range by going out 3 standard errors from each estimate:\n  xlim = c( min( betas - 3 * ses ), max( betas + 3 * ses ))\n  # Also let's make sure to include zero in our range\n  xlim[1] = min( xlim[1], 0 )\n  xlim[2] = max( xlim[2], 0 )\n\n  # expand the range slightly\n  xcentre = mean(xlim)\n  xlim[1] = xcentre + (xlim[1] - xcentre) * 1.1\n  xlim[2] = xcentre + (xlim[2] - xcentre) * 1.1\n\n  # Give ourselves a big left margin for the row labels\n  par( mar = c( 4.1, 6.1, 2.1, 2.1 ))\n  blank.plot(\n    xlim = xlim,\n    ylim = c( range(y) + c( -0.5, 0.5 )),\n    xlab = \"Estimate and 95% CI\"\n  )\n  \n  # Draw the intervals first so they don't draw over the points\n  segments(\n    x0 = betas - 1.96 * ses, x1 = betas + 1.96 * ses,\n    y0 = y, y1 = y,\n    col = 'grey'\n  )\n\n  # Now plot the estimates\n  points(\n    x = betas,\n    y = y,\n    col = 'black',\n    pch = 19\n  )\n\n  # ... and add labels.  We put them 10% further left than the leftmost point\n  # and we right-align them\n  text.x = xcentre + (xlim[1] - xcentre) * 1.1\n  text(\n    x = text.x,\n    y = y,\n    labels = names,\n    adj = 1, # right-adjust\n    xpd = NA # this means \"Allow text outside the plot area\"\n  )\n  \n  # Add an x axis\n  axis( 1 )\n  \n  # add grid lines\n  grid()\n  \n  # add a solid line at 0\n  abline( v = 0, col = rgb( 0, 0, 0, 0.2 ), lwd = 2 )\n}\n")),(0,n.kt)("p",null,"For example, we could now make a forest plot for the first SNP:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},'  forest.plot <- function( row ) {\n    betas = c( row$study1.beta, row$study2.beta, row$meta.beta )\n    ses = c( row$study1.se, row$study2.se, row$meta.se )\n    names = c( "Study 1", "Study 2", "Meta-analysis" )\n    draw.forest.plot( betas, ses, names )\n  }\n  \n  forest.plot( meta_analysis[1,] )\n')),(0,n.kt)("p",null,(0,n.kt)("img",{alt:"img",src:t(6051).Z,width:"400",height:"300"})),(0,n.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,n.kt)("div",{parentName:"div",className:"admonition-heading"},(0,n.kt)("h5",{parentName:"div"},(0,n.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,n.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,n.kt)("div",{parentName:"div",className:"admonition-content"},(0,n.kt)("p",{parentName:"div"},"What does the forest plot look like for the 'top' SNPs, i.e. for those with the lowest P-values (or highest Bayes factors)?  Plot a few of them and look at them.  Make sure you understand how the data in the meta-analysis file corresponds to the data on the plot."))),(0,n.kt)("p",null,(0,n.kt)("strong",{parentName:"p"},"Note.")," Forest plots are deceptively simple - they don't look like there's much going on, but whenever I try to draw one I\nfind it takes quite a bit of code. (Like the ",(0,n.kt)("inlineCode",{parentName:"p"},"draw.forest.plot()")," function above, which already has ~40 lines of code). This\nis typical of visualisation in general - the code always gets lengthy - and I think it is not really surprising. It is because\nin a visualistion you are trying to convey a great deal of information clearly in a small space: this often takes great deal\nof careful tweaking to get right. Don't skimp on this!"),(0,n.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,n.kt)("div",{parentName:"div",className:"admonition-heading"},(0,n.kt)("h5",{parentName:"div"},(0,n.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,n.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,n.kt)("div",{parentName:"div",className:"admonition-content"},(0,n.kt)("p",{parentName:"div"},' For a "working" plot, one thing it might be good to add to the plot would be more text listing the numerical\nvalues (i.e. the estimate and 95% confidence intervals) E.g. this code:'),(0,n.kt)("pre",{parentName:"div"},(0,n.kt)("code",{parentName:"pre"},'labels = sprintf(\n  "%.2f (%.2f - %.2f)",\n  betas,\n  betas - 1.96 * ses,\n  betas + 1.96 * ses\n)\n')),(0,n.kt)("p",{parentName:"div"},"produces the right kind of text.  Can you add this to the plot in the right place?"))),(0,n.kt)("h3",{id:"other-meta-analysis-models"},"Other meta-analysis models"),(0,n.kt)("p",null,"The above is a simple meta-analysis that assumes that the true effect is the same in both studies. However, as we saw in the\nlecture, more complex models are also possible, including a 'random' effects analysis (that assumes that true effects are\nnormally distributed around some mean) and include a heterogeneity test. Here is a way to do this."),(0,n.kt)("p",null,"First, you will need to install the ",(0,n.kt)("inlineCode",{parentName:"p"},"meta")," library: run"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},'install.packages("meta")\n')),(0,n.kt)("p",null,"or use the ",(0,n.kt)("inlineCode",{parentName:"p"},"Tools")," -> ",(0,n.kt)("inlineCode",{parentName:"p"},"Install Packages")," menu in Rstudio."),(0,n.kt)("p",null,"We can then run a random-effect meta-analysis like this:"),(0,n.kt)("pre",null,(0,n.kt)("code",{parentName:"pre"},"library(meta)\nforest.meta(\n  metagen(\n    c( study1$beta[1], study2$beta[1] ),\n    c( study1$se[1],study2$se[1] )\n  )\n)\n")),(0,n.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,n.kt)("div",{parentName:"div",className:"admonition-heading"},(0,n.kt)("h5",{parentName:"div"},(0,n.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,n.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,n.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,n.kt)("div",{parentName:"div",className:"admonition-content"},(0,n.kt)("p",{parentName:"div"},"What does this look like for the most-associated SNPs? Is there any evidence of differences in effect between studies?"))),(0,n.kt)("h3",{id:"fine-mapping-the-association"},"Fine-mapping the association"),(0,n.kt)("p",null,"By now you should have a good sense of which SNPs have lots of evidence for association in the two studies - and maybe these\nare the 'causal' SNPs. However, an obvious possibility (if the gene is relevant for the disease) is that there could be\nmultiple causal SNPs. In the ",(0,n.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping"},"next section")," we will see one way to try to discover how many causal SNPs there\nare - and what they are."))}c.isMDXComponent=!0},6051:(a,e,t)=>{t.d(e,{Z:()=>s});const s=t.p+"assets/images/forest_plot_1st_SNP-cac3aaf427b2a427fb43946ff64bb08b.png"}}]);