"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[8777],{3905:(e,a,t)=>{t.d(a,{Zo:()=>o,kt:()=>N});var n=t(7294);function s(e,a,t){return a in e?Object.defineProperty(e,a,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[a]=t,e}function m(e,a){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);a&&(n=n.filter((function(a){return Object.getOwnPropertyDescriptor(e,a).enumerable}))),t.push.apply(t,n)}return t}function i(e){for(var a=1;a<arguments.length;a++){var t=null!=arguments[a]?arguments[a]:{};a%2?m(Object(t),!0).forEach((function(a){s(e,a,t[a])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):m(Object(t)).forEach((function(a){Object.defineProperty(e,a,Object.getOwnPropertyDescriptor(t,a))}))}return e}function r(e,a){if(null==e)return{};var t,n,s=function(e,a){if(null==e)return{};var t,n,s={},m=Object.keys(e);for(n=0;n<m.length;n++)t=m[n],a.indexOf(t)>=0||(s[t]=e[t]);return s}(e,a);if(Object.getOwnPropertySymbols){var m=Object.getOwnPropertySymbols(e);for(n=0;n<m.length;n++)t=m[n],a.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(s[t]=e[t])}return s}var p=n.createContext({}),l=function(e){var a=n.useContext(p),t=a;return e&&(t="function"==typeof e?e(a):i(i({},a),e)),t},o=function(e){var a=l(e.components);return n.createElement(p.Provider,{value:a},e.children)},c="mdxType",h={inlineCode:"code",wrapper:function(e){var a=e.children;return n.createElement(n.Fragment,{},a)}},d=n.forwardRef((function(e,a){var t=e.components,s=e.mdxType,m=e.originalType,p=e.parentName,o=r(e,["components","mdxType","originalType","parentName"]),c=l(t),d=s,N=c["".concat(p,".").concat(d)]||c[d]||h[d]||m;return t?n.createElement(N,i(i({ref:a},o),{},{components:t})):n.createElement(N,i({ref:a},o))}));function N(e,a){var t=arguments,s=a&&a.mdxType;if("string"==typeof e||s){var m=t.length,i=new Array(m);i[0]=d;var r={};for(var p in a)hasOwnProperty.call(a,p)&&(r[p]=a[p]);r.originalType=e,r[c]="string"==typeof e?e:s,i[1]=r;for(var l=2;l<m;l++)i[l]=t[l];return n.createElement.apply(null,i)}return n.createElement.apply(null,t)}d.displayName="MDXCreateElement"},2360:(e,a,t)=>{t.d(a,{Z:()=>i});var n=t(7294),s=t(6010);const m={tabItem:"tabItem_OmH5"};function i(e){let{children:a,hidden:t,className:i}=e;return n.createElement("div",{role:"tabpanel",className:(0,s.Z)(m.tabItem,i),hidden:t},a)}},9877:(e,a,t)=>{t.d(a,{Z:()=>h});var n=t(7462),s=t(7294),m=t(2389),i=t(7392),r=t(7094),p=t(2466),l=t(6010);const o={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function c(e){const{lazy:a,block:t,defaultValue:m,values:c,groupId:h,className:d}=e,N=s.Children.map(e.children,(e=>{if((0,s.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),k=c??N.map((e=>{let{props:{value:a,label:t,attributes:n}}=e;return{value:a,label:t,attributes:n}})),g=(0,i.l)(k,((e,a)=>e.value===a.value));if(g.length>0)throw new Error(`Docusaurus error: Duplicate values "${g.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const u=null===m?m:m??N.find((e=>e.props.default))?.props.value??N[0]?.props.value;if(null!==u&&!k.some((e=>e.value===u)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${u}" but none of its children has the corresponding value. Available values are: ${k.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:v,setTabGroupChoices:w}=(0,r.U)(),[y,f]=(0,s.useState)(u),b=[],{blockElementScrollPositionUntilNextRender:x}=(0,p.o5)();if(null!=h){const e=v[h];null!=e&&e!==y&&k.some((a=>a.value===e))&&f(e)}const _=e=>{const a=e.currentTarget,t=b.indexOf(a),n=k[t].value;n!==y&&(x(a),f(n),null!=h&&w(h,n))},M=e=>{let a=null;switch(e.key){case"ArrowRight":{const t=b.indexOf(e.currentTarget)+1;a=b[t]||b[0];break}case"ArrowLeft":{const t=b.indexOf(e.currentTarget)-1;a=b[t]||b[b.length-1];break}}a?.focus()};return s.createElement("div",{className:(0,l.Z)("tabs-container",o.tabList)},s.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,l.Z)("tabs",{"tabs--block":t},d)},k.map((e=>{let{value:a,label:t,attributes:m}=e;return s.createElement("li",(0,n.Z)({role:"tab",tabIndex:y===a?0:-1,"aria-selected":y===a,key:a,ref:e=>b.push(e),onKeyDown:M,onFocus:_,onClick:_},m,{className:(0,l.Z)("tabs__item",o.tabItem,m?.className,{"tabs__item--active":y===a})}),t??a)}))),a?(0,s.cloneElement)(N.filter((e=>e.props.value===y))[0],{className:"margin-top--md"}):s.createElement("div",{className:"margin-top--md"},N.map(((e,a)=>(0,s.cloneElement)(e,{key:a,hidden:e.props.value!==y})))))}function h(e){const a=(0,m.Z)();return s.createElement(c,(0,n.Z)({key:String(a)},e))}},9183:(e,a,t)=>{t.r(a),t.d(a,{assets:()=>o,contentTitle:()=>p,default:()=>N,frontMatter:()=>r,metadata:()=>l,toc:()=>c});var n=t(7462),s=(t(7294),t(3905)),m=t(9877),i=t(2360);const r={sidebar_position:1},p="Getting started: linear regression",l={unversionedId:"statistical_modelling/regression_modelling/linear_regression_1",id:"statistical_modelling/regression_modelling/linear_regression_1",title:"Getting started: linear regression",description:"Prerequisites",source:"@site/docs/statistical_modelling/regression_modelling/linear_regression_1.md",sourceDirName:"statistical_modelling/regression_modelling",slug:"/statistical_modelling/regression_modelling/linear_regression_1",permalink:"/whg-training-resources/statistical_modelling/regression_modelling/linear_regression_1",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/statistical_modelling/regression_modelling/linear_regression_1.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"sidebar6",previous:{title:"Statistical modelling I - regression",permalink:"/whg-training-resources/statistical_modelling/regression_modelling/"},next:{title:"Logistic regression",permalink:"/whg-training-resources/statistical_modelling/regression_modelling/logistic_regression_1"}},o={},c=[{value:"Prerequisites",id:"prerequisites",level:3},{value:"Getting the data",id:"getting-the-data",level:3},{value:"What&#39;s in the data?",id:"whats-in-the-data",level:3},{value:"Quick look at the genotypes:",id:"quick-look-at-the-genotypes",level:3},{value:"Is rs1419114 associated with <em>ATP2B4</em> expression?",id:"is-rs1419114-associated-with-atp2b4-expression",level:2},{value:"Plotting the relationship",id:"plotting-the-relationship",level:3},{value:"Normalising read counts",id:"normalising-read-counts",level:3},{value:"Running a regression",id:"running-a-regression",level:3},{value:"Interpreting the regression",id:"interpreting-the-regression",level:3},{value:"P-values and all that",id:"p-values-and-all-that",level:3},{value:"Including covariates",id:"including-covariates",level:3},{value:"How good is the fit?",id:"how-good-is-the-fit",level:3},{value:"Next steps",id:"next-steps",level:2}],h={toc:c},d="wrapper";function N(e){let{components:a,...r}=e;return(0,s.kt)(d,(0,n.Z)({},h,r,{components:a,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"getting-started-linear-regression"},"Getting started: linear regression"),(0,s.kt)("h3",{id:"prerequisites"},"Prerequisites"),(0,s.kt)("p",null,"For this tutorial you will need to have ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/prerequisites/R"},"installed R and Rstudio"),". We will use R to analyse some expression\ndata.  I'll also be using the ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/prerequisites/tidyverse"},"tidyverse")," in examples."),(0,s.kt)("h3",{id:"getting-the-data"},"Getting the data"),(0,s.kt)("p",null,"In this part of the practical we will use linear regression to estimate the possible effect of a genotype on gene expression\nvalues. To get started, create a new empty directory and change into it:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"mkdir regression_tutorial\ncd regression_tutorial\n")),(0,s.kt)("p",null,"The data can be found in ",(0,s.kt)("a",{parentName:"p",href:"https://github.com/whg-training/whg-training-resources/tree/main/docs/statistical_modelling/regression_modelling/data/"},"this\nfolder")," -\ndownload the file ",(0,s.kt)("inlineCode",{parentName:"p"},"atp2b4_per_gene_data.tsv")," and place it in your folder:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"curl -O https://raw.githubusercontent.com/whg-training/whg-training-resources/main/docs/statistical_modelling/regression_modelling/data/atp2b4_per_gene_data.tsv\n")),(0,s.kt)("p",null,"(Or use the ",(0,s.kt)("a",{target:"_blank",href:t(5101).Z},"direct link"),".)"),(0,s.kt)("p",null,"Start an R session and make sure you are working in this directory (e.g. using ",(0,s.kt)("inlineCode",{parentName:"p"},"setwd()")," or ",(0,s.kt)("inlineCode",{parentName:"p"},"Session")," -> ",(0,s.kt)("inlineCode",{parentName:"p"},"Set Working Directory"),"\nin RStudio).  Let's load the data now and take a look:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'library( tidyverse )\ndata = read_delim( "atp2b4_per_gene_data.tsv", delim = "\\t" )\n')),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"I am using the ",(0,s.kt)("a",{parentName:"p",href:"https://www.tidyverse.org"},"tidyverse")," for these examples. If you don't\nhave it installed, either install it or use base R (e.g. ",(0,s.kt)("inlineCode",{parentName:"p"},"read.delim()"),") instead."))),(0,s.kt)("h3",{id:"whats-in-the-data"},"What's in the data?"),(0,s.kt)("p",null,"The data represents gene expression values (expressed as the mean number of RNA-seq reads per base, averaged across the gene\nexons) for the gene ",(0,s.kt)("em",{parentName:"p"},"ATP2B4"),". The data comes from ",(0,s.kt)("a",{parentName:"p",href:"https://doi.org/10.1172/JCI94378"},"this paper")," which reported that local\ngenotypes - including at the SNP rs1419114 - are associated with expression changes. Let's see if we can reconstruct this."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Gene quantification is usually measured in more sophisticated units - 'transcripts per million' (TPM) or 'fragments per\nkilobase per million'`(FPKM). They involve scaling by the computed value across all genes, which helps to normalise for\nvariation in expression of other genes.   We'll skip that here but you should consider it if doing a real transcriptomic\nstudy."))),(0,s.kt)("h3",{id:"quick-look-at-the-genotypes"},"Quick look at the genotypes:"),(0,s.kt)("p",null,"First let's rename the genotype column - too long!"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'colnames(data)[4] = "genotype"\n')),(0,s.kt)("p",null,"Let's look at this column now:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"table(data$genotype)\n")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"How many samples are in the data?  How many of each genotype?"))),(0,s.kt)("p",null,"There's only one A/A genotype in the data. This might look a bit odd at first, but if you look up rs1419114 on\n",(0,s.kt)("a",{parentName:"p",href:"https://ensembl.org"},"https://ensembl.org")," you'll see why. The frequency of the ",(0,s.kt)("inlineCode",{parentName:"p"},"A")," allele in non-African populations is around ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mfrac",{parentName:"mrow"},(0,s.kt)("mn",{parentName:"mfrac"},"1"),(0,s.kt)("mn",{parentName:"mfrac"},"10"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\frac{1}{10}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1.1901em",verticalAlign:"-0.345em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,s.kt)("span",{parentName:"span",className:"mfrac"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8451em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.655em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"10")))),(0,s.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,s.kt)("span",{parentName:"span",style:{top:"-3.394em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"1"))))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.345em"}},(0,s.kt)("span",{parentName:"span"}))))),(0,s.kt)("span",{parentName:"span",className:"mclose nulldelimiter"}))))))," or\nless. So assuming the two chromosomes are drawn independently we'd only expect about ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"24"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mfrac",{parentName:"mrow"},(0,s.kt)("mn",{parentName:"mfrac"},"1"),(0,s.kt)("mn",{parentName:"mfrac"},"100"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"24 \\times \\frac{1}{100}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7278em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"24"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1.1901em",verticalAlign:"-0.345em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,s.kt)("span",{parentName:"span",className:"mfrac"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8451em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.655em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"100")))),(0,s.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,s.kt)("span",{parentName:"span",style:{top:"-3.394em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"1"))))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.345em"}},(0,s.kt)("span",{parentName:"span"}))))),(0,s.kt)("span",{parentName:"span",className:"mclose nulldelimiter"})))))),", so not so\nsurprising we got only one."),(0,s.kt)("h2",{id:"is-rs1419114-associated-with-atp2b4-expression"},"Is rs1419114 associated with ",(0,s.kt)("em",{parentName:"h2"},"ATP2B4")," expression?"),(0,s.kt)("h3",{id:"plotting-the-relationship"},"Plotting the relationship"),(0,s.kt)("p",null,"The best way to start figuring this out is by plotting:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'library( ggplot2 )\n(\n    ggplot( data = data, aes( x = genotype, y = mean_reads_per_base ))\n    + geom_violin()\n    + geom_point()\n    + xlab( "rs1419114 genotype" )\n    + ylab( "Reads per base")\n)\n')),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:t(5009).Z,width:"600",height:"400"})),(0,s.kt)("p",null,"Hmm.  There looks visually like a correspondence.  "),(0,s.kt)("p",null,"But wait!  Maybe that's just to do with how many reads were sequenced?"),(0,s.kt)("h3",{id:"normalising-read-counts"},"Normalising read counts"),(0,s.kt)("p",null,"Different runs of the sequencer produce different numbers of reads due to prosaic features to do with library preparation and\nreagents and so on - producing variation in the counts that has nothing to do with biology.  So let's normalise by this number now:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"data$reads_per_base_per_million = 1E6 * data$mean_reads_per_base / data$total_reads\n")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"I included added a factor of 1 million here just to bring the values back into roughly the right level. The values are the\n",(0,s.kt)("em",{parentName:"p"},"number of reads per base of the gene per million sequenced reads"),". (For most work I suggest computing ",(0,s.kt)("strong",{parentName:"p"},"TPM")," instead.)"))),(0,s.kt)("p",null,"Let's plot again:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'p = (\n    ggplot( data = data, aes( x = genotype, y = reads_per_base_per_million ))\n    + geom_violin()\n    + geom_point()\n    + xlab( "rs1419114 genotype" )\n    + ylab( "Reads per base per million sequenced reads")\n)\n')),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:t(8170).Z,width:"600",height:"400"})),(0,s.kt)("p",null,"It certainly looks visually like there's a relationship."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Expression values are often explored after log-transforming them.  What difference does that make here?"))),(0,s.kt)("p",null,"In what follows I will use 'expression' to mean these normalised expression values - just remember that these are a transformed\nmeasurement of the actual expression of the gene."),(0,s.kt)("h3",{id:"running-a-regression"},"Running a regression"),(0,s.kt)("p",null,"Let's estimate the relationship between genotype and mRNA expression by fitting a ",(0,s.kt)("strong",{parentName:"p"},"linear regression model"),"."),(0,s.kt)("p",null,"To start off let's recode our genotype by the number of ",(0,s.kt)("strong",{parentName:"p"},"G")," alleles:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"data$dosage = 0\ndata$dosage[ data$genotype == 'A/G' ] = 1\ndata$dosage[ data$genotype == 'G/G' ] = 2\ndata$dosage = as.integer( data$dosage )\ntable( data$dosage )\n")),(0,s.kt)("p",null,"In R it's easy to fit a linear regression model: use ",(0,s.kt)("inlineCode",{parentName:"p"},"lm()"),":"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"fit = lm(\n    reads_per_base_per_million ~ dosage,\n    data = data\n)\n")),(0,s.kt)("p",null,"The best way to look at the result of this is to use the ",(0,s.kt)("inlineCode",{parentName:"p"},"summary()")," command to pull out the estimated coefficients and standard\nerrors. You can do that like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"summary(fit)$coefficients\n")),(0,s.kt)("p",null,"This will probably produce something like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"              Estimate Std. Error  t value  Pr(>|t|)\n(Intercept) 0.04981156 0.02097038 2.375329 0.0266714\ndosage      0.02090930 0.01245827 1.678347 0.1074296\n")),(0,s.kt)("p",null,"To make sense of this, you have to understand something about the model that is being fit. In math syntax it can be written as\nfollows:"),(0,s.kt)("div",{className:"math math-display"},(0,s.kt)("span",{parentName:"div",className:"katex-display"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mtext",{parentName:"mrow"},"expression\xa0level"),(0,s.kt)("mo",{parentName:"mrow"},"="),(0,s.kt)("mi",{parentName:"mrow"},"\u03bc"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mi",{parentName:"mrow"},"\u03b2"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mtext",{parentName:"mrow"},"genotype"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mrow",{parentName:"mrow"},(0,s.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"gaussian"),(0,s.kt)("mtext",{parentName:"mrow"},"\xa0"),(0,s.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"noise"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\text{expression level} = \\mu + \\beta \\times \\text{genotype} + \\text{\\textit{gaussian noise}}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"expression\xa0level")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"="),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7778em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"\u03bc"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8095em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"genotype")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8623em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord textit"},"gaussian\xa0noise")))))))),(0,s.kt)("p",null,"Here ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03bc")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\mu")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.625em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"\u03bc")))))," and ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," are the two parameters (output in the two rows above) - ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03bc")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\mu")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.625em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"\u03bc")))))," is the ",(0,s.kt)("em",{parentName:"p"},"intercept"),", which captures the\naverage expression values; and ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," is the ",(0,s.kt)("em",{parentName:"p"},"regression effect size parameter"),". "),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," should therefore be interpreted as: the estimated increase in expression associated with each additional copy of the 'G'\nallele. It is often called the ",(0,s.kt)("em",{parentName:"p"},"association effect size"),"."))),(0,s.kt)("p",null,"All ",(0,s.kt)("inlineCode",{parentName:"p"},"lm")," does is find the values of ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03bc")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\mu")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.625em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"\u03bc")))))," and ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," which best fit the data."),(0,s.kt)("h3",{id:"interpreting-the-regression"},"Interpreting the regression"),(0,s.kt)("p",null,"So what does this output say?  It says that:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},"the estimated mean expression level is ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0498")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0498")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0498"))))),"."),(0,s.kt)("li",{parentName:"ul"},"the ",(0,s.kt)("em",{parentName:"li"},"standard error")," of this estimate is ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0210")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0210")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0210"))))),"."),(0,s.kt)("li",{parentName:"ul"},"the estimated ",(0,s.kt)("em",{parentName:"li"},"association effect size")," is ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0209")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0209")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0209"))))),"."),(0,s.kt)("li",{parentName:"ul"},"the standard error of this estimate is ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0125")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0125")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0125"))))),".")),(0,s.kt)("p",null,"Let's plot this on the above graph:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'p = (\n    ggplot( data = data, aes( x = dosage, y = reads_per_base_per_million ) )\n    + geom_point()\n    + xlab( "rs1419114 genotype" )\n    + ylab( "Reads per base per million reads")\n    + geom_abline( intercept = 0.0498, slope = 0.0209, col = \'red\' )\n)\nprint(p)\n')),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Important"),". The regression estimate of ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," is the ",(0,s.kt)("em",{parentName:"p"},"slope")," of fitted line."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Which of these statements do you agree with?"),(0,s.kt)("ul",{parentName:"div"},(0,s.kt)("li",{parentName:"ul"},"The rs1419114 'G' allele is associated with increased expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),"."),(0,s.kt)("li",{parentName:"ul"},"The rs1419114 'G' allele is not associated with increased expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),"."),(0,s.kt)("li",{parentName:"ul"},"We can't tell from these data if rs1419114 is associated with expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),".")))),(0,s.kt)("div",{className:"admonition admonition-note alert alert--secondary"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"14",height:"16",viewBox:"0 0 14 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"}))),"Visualising the fit")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"If you did the ",(0,s.kt)("a",{parentName:"p",href:"/data_visualisation/data_visualisation_with_ggplot2/"},"visualisation tutorial")," you will have seen a method to get\nggplot2 to add a regression line and confidence interval (using ",(0,s.kt)("inlineCode",{parentName:"p"},"geom_smooth()"),"). How is that added?"),(0,s.kt)("p",{parentName:"div"},"The calculation can be done using ",(0,s.kt)("inlineCode",{parentName:"p"},"predict"),". Specifically let's get it to predict the values and upper and lower confidence\nintervals at a range of values from 0 to 2:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"x = seq( from = 0, to = 2, by = 0.01 )\ninterval = predict(\n    fit2,\n    newdata = data.frame( dosage = x ),\n    interval = \"confidence\",\n    level = 0.95\n)\n\ninterval_data = tibble(\n    x = x,\n    lower = interval[,'lwr'],\n    upper = interval[,'upr']\n)\nprint(\n    p\n    + geom_line( data = interval_data, aes( x = x, y = lower ))\n    + geom_line( data = interval_data, aes( x = x, y = upper ))\n)\n")),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("img",{alt:"img",src:t(6267).Z,width:"400",height:"300"})),(0,s.kt)("p",{parentName:"div"},"Compare this with ",(0,s.kt)("inlineCode",{parentName:"p"},"geom_smooth"),":"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},'p + geom_smooth( method = "lm" )\n')),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("img",{alt:"img",src:t(1451).Z,width:"400",height:"300"})),(0,s.kt)("p",{parentName:"div"},"They do... exactly the same thing!"),(0,s.kt)("p",{parentName:"div"},"Aside: If you want to get the cool shading from the ",(0,s.kt)("inlineCode",{parentName:"p"},"predict")," method, that's doable too:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"transparent_grey = rgb( 0, 0, 0, 0.2 )\nprint(\n    p\n    + geom_ribbon(\n        data = interval_data,\n        # need to set y here even though not used\n        aes( x = x, y = 0, ymin = lower, ymax = upper ),\n        fill = transparent_grey\n    )\n)\n")),(0,s.kt)("p",{parentName:"div"},"Yet another way to do this is to ",(0,s.kt)("strong",{parentName:"p"},"sample some regression lines")," from the model fit. This is easy with a bit of an approximation.\nFirst get the variance-covariance matrix of the parameter estimates:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"V = vcov(fit)\n")),(0,s.kt)("p",{parentName:"div"},"Now sample some regression lines from the fitted parameter distribution. To do this let's approximate it by a multivariate\ngaussian. The sampling can be done using the ",(0,s.kt)("inlineCode",{parentName:"p"},"mvtnorm")," package:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"library( mvtnorm )\nsampled_lines = rmvnorm( 100, sigma = V, mean = c( 0.0498, 0.0209 ) )\n")),(0,s.kt)("p",{parentName:"div"},"Now let's add quantiles from those sampled lines to the plot:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"sampled_interval = tibble(\n    dosage = x,\n    lower = sapply( x, function(p) { quantile( sampled_lines[,1] + p * sampled_lines[,2], 0.025 ) } ),\n    upper = sapply( x, function(p) { quantile( sampled_lines[,1] + p * sampled_lines[,2], 0.975 ) } )\n)\nprint(\n    p\n    + geom_line( data = sampled_interval, aes( x = dosage, y = lower ))\n    + geom_line( data = sampled_interval, aes( x = dosage, y = upper ))\n)\n")),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("img",{alt:"img",src:t(6324).Z,width:"400",height:"300"})),(0,s.kt)("p",{parentName:"div"},"Ok that looks a bit wobbly! But if you turn up the number of samples to (say) 10,000 you'll see this is the same thing again."),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note.")," This last way of plotting the regression confidence interval is helpful because it reveals how we think of the regression\nfit: namely it determines a ",(0,s.kt)("em",{parentName:"p"},"joint distribution over the parameters"),"."))),(0,s.kt)("h3",{id:"p-values-and-all-that"},"P-values and all that"),(0,s.kt)("p",null,"In the above, the estimate was ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0209")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0209")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0209")))))," with a standard error of ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0125")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0125")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0125"))))),".  What does that standard error mean?"),(0,s.kt)("p",null,"Here are two ways of thinking about it.  First, in the context of bayes' theorem, this is really expressing the\nvariation in the posterior distribution.  Roughtly speaking, it is saying that the posterior is (approximately) a\nGaussian distribution with the given estimate as mode and standard deviation as the standard error.  (For linear\nregression it's actually a T distribution not a gaussian, but that doesn't matter very much.)  In other words this is\ndescribing what we should believe about the 'true' parameter estimate given the data we've seen.  This is a ",(0,s.kt)("strong",{parentName:"p"},"good way"),"\nof thinking about it."),(0,s.kt)("p",null,"Another useful way of thinking about the standard error is that it tells us the ",(0,s.kt)("strong",{parentName:"p"},"sampling error")," of the estimate.  To\nunderstand this, you have to think not just of the dataset we have loaded, but also imagine what would happen if we\ngenerated and estimated in other datasets generated the same way.  Each dataset would generate its own estimate, which\nwould differ a bit from our estimate due to all the random noise and so on.  The ",(0,s.kt)("strong",{parentName:"p"},"standard error")," tells us how much\nvariability we should expect in that estimate."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"In this context, an odd feature of linear regression is that it treats the predictors (i.e. the genotypes) as fixed - they are treated as\nknown beforehand rather than as part of the 'observed data' for statistical purposes. So this is really about imagining\npossible expression values for samples with the given genotypes."))),(0,s.kt)("p",null,"To explore this ",(0,s.kt)("strong",{parentName:"p"},"sampling distribution")," interopretation, suppose there's actually no association and suppose -\nhypothetically - we were to repeat our estimate on a million different sets of 24 samples (with the same genotypes). The\n",(0,s.kt)("em",{parentName:"p"},"sampling distribution")," of the estimate we would obtain would look something like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'sampling_distribution = tibble(\n    x = seq( from = -0.1, to = 0.1, by = 0.001 )\n)\nsampling_distribution$density = dnorm( x, mean = 0, sd = 0.01245827 )\n\np2 = (\n    ggplot( data = sampling_distribution )\n    + geom_line( aes( x = x, y = density ) )\n    + geom_vline( xintercept = 0.0209, linetype = 2 )\n    + geom_text( x = 0.0209, y = 30, label = "observed estimate", hjust = -0.1 )\n    + xlab( "estimate" )\n    + ylab( "density" )\n)\nprint(p2)\n')),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:t(4648).Z,width:"400",height:"300"})),(0,s.kt)("p",null,"What this is saying is: our estimate is fairly well within the distribution that would be expected if there is no true association.\nWhichever way you look at this it's hard to get excited about the association."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"The P-value given is a two-tailed p-value, and it is just computed as the ",(0,s.kt)("strong",{parentName:"p"},"mass of our estimate under the two tails above"),". That\nlooks like this:"),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("img",{alt:"img",src:t(1406).Z,width:"400",height:"300"})),(0,s.kt)("p",{parentName:"div"},"To compute it, just compute this mass:"),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"print(\n    pnorm( -0.0209, mean = 0, sd = 0.0125 ) # mass under left tail\n    * 2                                     # get right tail too\n)\n")),(0,s.kt)("pre",{parentName:"div"},(0,s.kt)("code",{parentName:"pre"},"[1] 0.09452432\n")),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note.")," The p-value computed above is ",(0,s.kt)("em",{parentName:"p"},"almost")," but not quite the same as the one listed in the regression output. That's because,\nfor linear regression, the distribution is actually T not gaussian. However, this doesn't matter much and gets less important as\nthe sample size grows, so I'm going to ignore it."),(0,s.kt)("p",{parentName:"div"},"Becuase of this mass-under-the-tails behaviour, the p-value is interpreted as 'the probability of seeing an effect as\nlarge as ",(0,s.kt)("em",{parentName:"p"},"or larger than")," the one we actually saw, if there wasn't any true effect.  "),(0,s.kt)("p",{parentName:"div"},"However, the real point I am making here is that ",(0,s.kt)("strong",{parentName:"p"},"the P-value gives no new information relative to the estimate and its\nstandard error"),". If you had to choose what to report, it should be the estimate of ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," and its standard error, and\nyou can also think of that as summarising the posterior."))),(0,s.kt)("p",null,"Whichever way you dice it, this association does not look very exciting, right?"),(0,s.kt)("h3",{id:"including-covariates"},"Including covariates"),(0,s.kt)("p",null,"But wait! There's another variable in the data. Specifically, the samples were either from bone marrow from adults or fetal liver.\nMaybe we should be including that in the regression?"),(0,s.kt)("p",null,"Including covariates is easy in linear regression - this is one of its great strengths.  Try it now:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-R"},"fit2 = lm(\n    reads_per_base_per_million ~ dosage + stage,\n    data = data\n)\nsummary(fit2)$coefficients\n")),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"What do these 'fetal' and 'adult' monikers mean? Well, mature red blood cells (erthrocytes) are ",(0,s.kt)("em",{parentName:"p"},"enucleated")," - they don't have\nnuclei. They are basically bags of haemoglobin, and although there might be a few fragments of RNA floating around, it will be\npretty degraded.  They are not much good for an expression study."),(0,s.kt)("p",{parentName:"div"},"For this study the authors therefore instead worked with haematopoetic stem and progenitor cells (HSPCs) from fetal liver and adult\nbone marrow, and experimentally differentiated them into ",(0,s.kt)("em",{parentName:"p"},"proerythroblasts"),". They then measured expression in these still-nucleated\ncells."),(0,s.kt)("p",{parentName:"div"},"(This issue of needing to differentiate is why red cells are not found in many expression databases - such as\n",(0,s.kt)("a",{parentName:"p",href:"http://gtexportal.org/home/index.html"},"GTEx"),".)"))),(0,s.kt)("p",null,"Your fit might look something like this:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"               Estimate Std. Error   t value     Pr(>|t|)\n(Intercept)  0.04473606 0.01650016  2.711250 0.0130793191\ndosage       0.03918108 0.01086677  3.605588 0.0016614274\nstagefetal  -0.04770967 0.01241647 -3.842451 0.0009462768\n")),(0,s.kt)("p",null,"The model is now:"),(0,s.kt)("div",{className:"math math-display"},(0,s.kt)("span",{parentName:"div",className:"katex-display"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mtext",{parentName:"mrow"},"expression"),(0,s.kt)("mo",{parentName:"mrow"},"="),(0,s.kt)("mi",{parentName:"mrow"},"\u03bc"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mi",{parentName:"mrow"},"\u03b2"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mtext",{parentName:"mrow"},"genotype"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mi",{parentName:"mrow"},"\u03b3"),(0,s.kt)("mo",{parentName:"mrow"},"\xd7"),(0,s.kt)("mtext",{parentName:"mrow"},"stage"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mrow",{parentName:"mrow"},(0,s.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"gaussian"),(0,s.kt)("mtext",{parentName:"mrow"},"\xa0"),(0,s.kt)("mtext",{parentName:"mrow",mathvariant:"italic"},"noise"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\text{expression} = \\mu + \\beta\\times\\text{genotype} + \\gamma \\times \\text{stage} + \\text{\\textit{gaussian noise}}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8623em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"expression")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}}),(0,s.kt)("span",{parentName:"span",className:"mrel"},"="),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2778em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7778em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal"},"\u03bc"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8095em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"genotype")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7778em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05556em"}},"\u03b3"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\xd7"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8095em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"stage")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8623em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord textit"},"gaussian\xa0noise")))))))),(0,s.kt)("p",null,"and we have three parameter estimates.  "),(0,s.kt)("p",null,"What has happened?"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},"The intercept has got slightly lower"),(0,s.kt)("li",{parentName:"ul"},"The estimate of ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," is now about twice as large"),(0,s.kt)("li",{parentName:"ul"},"The standard error of ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b2")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\beta")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8889em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05278em"}},"\u03b2")))))," is smaller"),(0,s.kt)("li",{parentName:"ul"},"And the new parameter for fetal/adult status, ",(0,s.kt)("span",{parentName:"li",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"\u03b3")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\gamma")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.625em",verticalAlign:"-0.1944em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.05556em"}},"\u03b3"))))),", is negative.")),(0,s.kt)("p",null,"The parameter estimate for the genotype ",(0,s.kt)("inlineCode",{parentName:"p"},"dosage")," is now ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mn",{parentName:"mrow"},"0.0392")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"0.0392")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"0.0392")))))," - and that's now ",(0,s.kt)("em",{parentName:"p"},"four times the standard error"),". As the P-value\nshows that's way out in the tail of the sampling distribution."),(0,s.kt)(m.Z,{mdxType:"Tabs"},(0,s.kt)(i.Z,{value:"code",label:"Plot code",mdxType:"TabItem"},(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'p = (\n    ggplot( data = data, aes( x = dosage, y = reads_per_base_per_million ) )\n    + geom_point()\n    + xlab( "rs1419114 genotype" )\n    + ylab( "Reads per base per million reads")\n    + geom_abline( intercept = 0.04473606, slope = 0.03918108, col = \'red\' )\n)\n\nsampled_lines = rmvnorm( 10000, sigma = vcov( fit2 ), mean = c( 0.04473606, 0.03918108, -0.04770967 ) )\nsampled_interval = tibble(\n    dosage = x,\n    lower = sapply( x, function(p) { quantile( sampled_lines[,1] + p * sampled_lines[,2], 0.025 ) } ),\n    upper = sapply( x, function(p) { quantile( sampled_lines[,1] + p * sampled_lines[,2], 0.975 ) } )\n)\nprint(\n    p\n    + geom_ribbon(\n        data = sampled_interval,\n        aes( x = dosage, y = 0, ymin = lower, ymax = upper ),\n        fill = transparent_grey\n    )\n)\n\n# Plot the tail distribution\nsampling_distribution = tibble(\n    x = seq( from = -0.05, to = 0.1, by = 0.001 )\n)\nsampling_distribution$density = dnorm( sampling_distribution$x, mean = 0, sd = 0.01086677 )\n\np2 = (\n    ggplot( data = sampling_distribution )\n    + geom_line( aes( x = x, y = density ) )\n    + geom_vline( xintercept = 0.03922, linetype = 2 )\n    + geom_text( x = 0.03922, y = 35, label = "estimate", hjust = -0.1 )\n    + xlab( "estimate" )\n    + ylab( "density" )\n)\n\nlower_tail = which( sampling_distribution$x <= -0.03922 )\nupper_tail = which( sampling_distribution$x >= 0.03922 )\n\nprint(\n    p2\n    + geom_area( data = sampling_distribution[ lower_tail, ], aes( x = x, y = density ), fill = transparent_grey )\n    + geom_area( data = sampling_distribution[ upper_tail, ], aes( x = x, y = density ), fill = transparent_grey )\n)\n\n'))),(0,s.kt)(i.Z,{value:"image",label:"Updated plot",default:!0,mdxType:"TabItem"},(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:t(3664).Z,width:"400",height:"300"}),"\n",(0,s.kt)("img",{alt:"img",src:t(3842).Z,width:"400",height:"300"})))),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Which of these statements do you agree with now?"),(0,s.kt)("ul",{parentName:"div"},(0,s.kt)("li",{parentName:"ul"},"The rs1419114 'G' allele is associated with increased expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),"."),(0,s.kt)("li",{parentName:"ul"},"The rs1419114 'G' allele is not associated with increased expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),"."),(0,s.kt)("li",{parentName:"ul"},"We can't tell from these data if rs1419114 is associated with expression of ",(0,s.kt)("em",{parentName:"li"},"ATP2B4"),".")),(0,s.kt)("p",{parentName:"div"},"Should we include the stage as a covariate, or shouldn't we?"))),(0,s.kt)("h3",{id:"how-good-is-the-fit"},"How good is the fit?"),(0,s.kt)("p",null,"Before moving on we ought to check that the model is at all sensible for the data."),(0,s.kt)("p",null,"One way to do this is the ",(0,s.kt)("strong",{parentName:"p"},"quantile-quantile")," or *qq plot**. The qqplot is used to compare an expected distribution (which, in\nthis case, can be the fitted distribution of the expression values) to the actual observed distribution."),(0,s.kt)("p",null,"To do this let's examine the ",(0,s.kt)("strong",{parentName:"p"},"residuals")," - that is, the difference between the observed values and the fitted straight-line\nmodel. If the model is accurate, these residuals should look like 'gaussian noise' - i.e. be as if drawn from a Gaussian\ndistribution.  We can check this by comparing them to quantiles of a gaussian now:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},'qq_data = tibble(\n    observed = sort( fit2$residuals ),\n    expected = qnorm(\n        1:nrow(data)/(nrow(data)+1),\n        sd = sigma( fit2 )              # sigma() is the fitted residual standard error\n    )\n)\nprint(\n    ggplot(\n        data = qq_data,\n        aes( x = observed, y = expected )\n    )\n    + geom_point()\n    + geom_abline( intercept = 0, slope = 1, colour = "red" )\n)\n')),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:t(3374).Z,width:"400",height:"300"})),(0,s.kt)("p",null,"This plot says the fit is pretty good."),(0,s.kt)("p",null,"If you want to go further than this, one way is to compute a P-value comparing the distribution of residuals against the fitted\nmodel."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge Question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Compute a P-value for the residuals against the fitted model."),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Hint")," if the residuals are gaussian with variance ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"\u03c3"),(0,s.kt)("mn",{parentName:"msup"},"2"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\sigma^2")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.03588em"}},"\u03c3"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2")))))))))))),", then their sum is gaussian with variance equal to ",(0,s.kt)("span",{parentName:"p",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mi",{parentName:"mrow"},"N"),(0,s.kt)("msup",{parentName:"mrow"},(0,s.kt)("mi",{parentName:"msup"},"\u03c3"),(0,s.kt)("mn",{parentName:"msup"},"2"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"N\\sigma^2")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.8141em"}}),(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.10903em"}},"N"),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mord mathnormal",style:{marginRight:"0.03588em"}},"\u03c3"),(0,s.kt)("span",{parentName:"span",className:"msupsub"},(0,s.kt)("span",{parentName:"span",className:"vlist-t"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8141em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-3.063em",marginRight:"0.05em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"2.7em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"2")))))))))))),".\nHow far out in the tail is the sum of observed residuals?"))),(0,s.kt)("h2",{id:"next-steps"},"Next steps"),(0,s.kt)("p",null,"Now let's look at a slightly different example using ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/statistical_modelling/regression_modelling/logistic_regression_1"},"logistic regression"),"."))}N.isMDXComponent=!0},5101:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/files/atp2b4_per_gene_data-df61916f2fd59d7401e9ce17d5c9e162.tsv"},6267:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_confidence_interval_1-dc0ac489136133bd58e195ffea767cc1.png"},1451:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_confidence_interval_2-a2966da70d7daeb4e95352d1c5162662.png"},6324:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_confidence_interval_3-cd7320089298a37afec6cc66087f7733.png"},3374:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_qqplot-e15055e83ed82976edec7257378ce974.png"},3664:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_refit_model_1-4a49e616c16ad61729637d1d5df6d0ed.png"},3842:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/lm_refit_tail-1418940e584b1e92d48dd465b50e2663.png"},5009:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/mean_per_genotype-adae91726137cbf2f4ee56a9d90bff81.png"},8170:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/normalised_per_genotype-aa52aa743b9b904fad26e6221d83948f.png"},4648:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/sampling_distribution_1-0cd85a000b0321c1c9ebb99eadb26729.png"},1406:(e,a,t)=>{t.d(a,{Z:()=>n});const n=t.p+"assets/images/sampling_distribution_shaded-0162baf05d477a64d12471f82f9cf169.png"}}]);