"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[9905],{3905:(e,n,t)=>{t.d(n,{Zo:()=>m,kt:()=>g});var r=t(7294);function i(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function a(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);n&&(r=r.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,r)}return t}function o(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?a(Object(t),!0).forEach((function(n){i(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):a(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,r,i=function(e,n){if(null==e)return{};var t,r,i={},a=Object.keys(e);for(r=0;r<a.length;r++)t=a[r],n.indexOf(t)>=0||(i[t]=e[t]);return i}(e,n);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(r=0;r<a.length;r++)t=a[r],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(i[t]=e[t])}return i}var p=r.createContext({}),c=function(e){var n=r.useContext(p),t=n;return e&&(t="function"==typeof e?e(n):o(o({},n),e)),t},m=function(e){var n=c(e.components);return r.createElement(p.Provider,{value:n},e.children)},u="mdxType",l={inlineCode:"code",wrapper:function(e){var n=e.children;return r.createElement(r.Fragment,{},n)}},f=r.forwardRef((function(e,n){var t=e.components,i=e.mdxType,a=e.originalType,p=e.parentName,m=s(e,["components","mdxType","originalType","parentName"]),u=c(t),f=i,g=u["".concat(p,".").concat(f)]||u[f]||l[f]||a;return t?r.createElement(g,o(o({ref:n},m),{},{components:t})):r.createElement(g,o({ref:n},m))}));function g(e,n){var t=arguments,i=n&&n.mdxType;if("string"==typeof e||i){var a=t.length,o=new Array(a);o[0]=f;var s={};for(var p in n)hasOwnProperty.call(n,p)&&(s[p]=n[p]);s.originalType=e,s[u]="string"==typeof e?e:i,o[1]=s;for(var c=2;c<a;c++)o[c]=t[c];return r.createElement.apply(null,o)}return r.createElement.apply(null,t)}f.displayName="MDXCreateElement"},2708:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>p,contentTitle:()=>o,default:()=>l,frontMatter:()=>a,metadata:()=>s,toc:()=>c});var r=t(7462),i=(t(7294),t(3905));const a={sidebar_position:2},o="Using snakemake",s={unversionedId:"bioinformatics/pipelining_with_snakemake/using_snakemake",id:"bioinformatics/pipelining_with_snakemake/using_snakemake",title:"Using snakemake",description:"(This page is under construction).",source:"@site/docs/bioinformatics/pipelining_with_snakemake/02_using_snakemake.md",sourceDirName:"bioinformatics/pipelining_with_snakemake",slug:"/bioinformatics/pipelining_with_snakemake/using_snakemake",permalink:"/whg-training-resources/bioinformatics/pipelining_with_snakemake/using_snakemake",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/bioinformatics/pipelining_with_snakemake/02_using_snakemake.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{sidebar_position:2},sidebar:"sidebar2",previous:{title:"Getting set up",permalink:"/whg-training-resources/bioinformatics/pipelining_with_snakemake/prerequisites"}},p={},c=[{value:"The <code>summarise_ensembl_gff.R</code> program",id:"the-summarise_ensembl_gffr-program",level:2}],m={toc:c},u="wrapper";function l(e){let{components:n,...t}=e;return(0,i.kt)(u,(0,r.Z)({},m,t,{components:n,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"using-snakemake"},"Using snakemake"),(0,i.kt)("p",null,"(This page is under construction)."),(0,i.kt)("h2",{id:"the-summarise_ensembl_gffr-program"},"The ",(0,i.kt)("inlineCode",{parentName:"h2"},"summarise_ensembl_gff.R")," program"),(0,i.kt)("p",null,"For reference, here is how you could run the ",(0,i.kt)("inlineCode",{parentName:"p"},"summarise_ensembl_gff.R")," program using a GFF file downloaded from the\n",(0,i.kt)("a",{parentName:"p",href:"https://ftp.ensembl.org/pub/current_gff3/"},"Ensembl FTP site"),";"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-r"},"Rscript --vanilla summarise_ensembl_gff.R --input /path/to/gff_file.gff --output summary.tsv\n")),(0,i.kt)("p",null,"It reads in the gff file and outputs a summary of the number and median size of each record in the file, split by the\ntype (and biotype attribute) of the records."))}l.isMDXComponent=!0}}]);