"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[606],{3905:(e,t,a)=>{a.d(t,{Zo:()=>m,kt:()=>h});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function s(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var l=n.createContext({}),p=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},m=function(e){var t=p(e.components);return n.createElement(l.Provider,{value:t},e.children)},c="mdxType",u={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,l=e.parentName,m=s(e,["components","mdxType","originalType","parentName"]),c=p(a),d=r,h=c["".concat(l,".").concat(d)]||c[d]||u[d]||i;return a?n.createElement(h,o(o({ref:t},m),{},{components:a})):n.createElement(h,o({ref:t},m))}));function h(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,o=new Array(i);o[0]=d;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[c]="string"==typeof e?e:r,o[1]=s;for(var p=2;p<i;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}d.displayName="MDXCreateElement"},2360:(e,t,a)=>{a.d(t,{Z:()=>o});var n=a(7294),r=a(6010);const i={tabItem:"tabItem_OmH5"};function o(e){let{children:t,hidden:a,className:o}=e;return n.createElement("div",{role:"tabpanel",className:(0,r.Z)(i.tabItem,o),hidden:a},t)}},9877:(e,t,a)=>{a.d(t,{Z:()=>u});var n=a(7462),r=a(7294),i=a(2389),o=a(7392),s=a(7094),l=a(2466),p=a(6010);const m={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function c(e){const{lazy:t,block:a,defaultValue:i,values:c,groupId:u,className:d}=e,h=r.Children.map(e.children,(e=>{if((0,r.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),g=c??h.map((e=>{let{props:{value:t,label:a,attributes:n}}=e;return{value:t,label:a,attributes:n}})),f=(0,o.l)(g,((e,t)=>e.value===t.value));if(f.length>0)throw new Error(`Docusaurus error: Duplicate values "${f.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const v=null===i?i:i??h.find((e=>e.props.default))?.props.value??h[0]?.props.value;if(null!==v&&!g.some((e=>e.value===v)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${v}" but none of its children has the corresponding value. Available values are: ${g.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:k,setTabGroupChoices:N}=(0,s.U)(),[b,w]=(0,r.useState)(v),y=[],{blockElementScrollPositionUntilNextRender:_}=(0,l.o5)();if(null!=u){const e=k[u];null!=e&&e!==b&&g.some((t=>t.value===e))&&w(e)}const C=e=>{const t=e.currentTarget,a=y.indexOf(t),n=g[a].value;n!==b&&(_(t),w(n),null!=u&&N(u,n))},x=e=>{let t=null;switch(e.key){case"ArrowRight":{const a=y.indexOf(e.currentTarget)+1;t=y[a]||y[0];break}case"ArrowLeft":{const a=y.indexOf(e.currentTarget)-1;t=y[a]||y[y.length-1];break}}t?.focus()};return r.createElement("div",{className:(0,p.Z)("tabs-container",m.tabList)},r.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,p.Z)("tabs",{"tabs--block":a},d)},g.map((e=>{let{value:t,label:a,attributes:i}=e;return r.createElement("li",(0,n.Z)({role:"tab",tabIndex:b===t?0:-1,"aria-selected":b===t,key:t,ref:e=>y.push(e),onKeyDown:x,onFocus:C,onClick:C},i,{className:(0,p.Z)("tabs__item",m.tabItem,i?.className,{"tabs__item--active":b===t})}),a??t)}))),t?(0,r.cloneElement)(h.filter((e=>e.props.value===b))[0],{className:"margin-top--md"}):r.createElement("div",{className:"margin-top--md"},h.map(((e,t)=>(0,r.cloneElement)(e,{key:t,hidden:e.props.value!==b})))))}function u(e){const t=(0,i.Z)();return r.createElement(c,(0,n.Z)({key:String(t)},e))}},2306:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>u,frontMatter:()=>i,metadata:()=>s,toc:()=>p});var n=a(7462),r=(a(7294),a(3905));a(9877),a(2360);const i={sidebar_position:9},o="Challenge questions",s={unversionedId:"programming/programming_with_gene_annotations3/challenge_questions",id:"programming/programming_with_gene_annotations3/challenge_questions",title:"Challenge questions",description:"If you reach this point, you have:",source:"@site/docs/programming/programming_with_gene_annotations3/009_challenge_questions.md",sourceDirName:"programming/programming_with_gene_annotations3",slug:"/programming/programming_with_gene_annotations3/challenge_questions",permalink:"/whg-training-resources/programming/programming_with_gene_annotations3/challenge_questions",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/programming/programming_with_gene_annotations3/009_challenge_questions.md",tags:[],version:"current",sidebarPosition:9,frontMatter:{sidebar_position:9},sidebar:"sidebar5",previous:{title:"Writing a useful conversion program",permalink:"/whg-training-resources/programming/programming_with_gene_annotations3/Converting_gff_to_sqlite"},next:{title:"Reading results from a sqlite file in any language",permalink:"/whg-training-resources/programming/programming_with_gene_annotations3/appendices/sqlite_access"}},l={},p=[{value:"Challenge 1: extract more attributes",id:"challenge-1-extract-more-attributes",level:2},{value:"Challenge 2: shrink <code>attributes</code>",id:"challenge-2-shrink-attributes",level:2},{value:"Challenge 3: split the output",id:"challenge-3-split-the-output",level:2},{value:"Challenge 4: the refactor",id:"challenge-4-the-refactor",level:2},{value:"The finished product",id:"the-finished-product",level:2}],m={toc:p},c="wrapper";function u(e){let{components:t,...a}=e;return(0,r.kt)(c,(0,n.Z)({},m,a,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"challenge-questions"},"Challenge questions"),(0,r.kt)("p",null,"If you reach this point, you have:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"written a function",(0,r.kt)("inlineCode",{parentName:"li"},"parse_gff3_to_dataframe()")," that can read GFF data into R (or python)"),(0,r.kt)("li",{parentName:"ul"},"written an R package, or a python module, to contain that function"),(0,r.kt)("li",{parentName:"ul"},"written a command-line program ",(0,r.kt)("inlineCode",{parentName:"li"},"gff_to_sqlite.R")," (or ",(0,r.kt)("inlineCode",{parentName:"li"},"gff_to_sqlite.py"),") that uses that function to convert GFF file(s) to a database format.")),(0,r.kt)("p",null,"That's pretty good - but it could be better.  Here are some challenges to try. "),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"For these challenges we suggest to focus on the GFF files ",(0,r.kt)("a",{parentName:"p",href:"http://ftp.ensembl.org/pub/current_gff3/homo_sapiens/"},"from\nEnsembl"),", at least at first.  Files from other sources such as\n",(0,r.kt)("a",{parentName:"p",href:"https://www.gencodegenes.org/human/"},"gencode")," have minor formatting differences that will make this task harder."))),(0,r.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Wartning")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Some of these challenges involve changes to your ",(0,r.kt)("inlineCode",{parentName:"p"},"parse_gff3_to_dataframe()")," function. Make sure and test it to ensure\nit still works!"),(0,r.kt)("p",{parentName:"div"},"Also, if you like your edits and put them in your ",(0,r.kt)("strong",{parentName:"p"},"gmsgff")," package, remember to run ",(0,r.kt)("inlineCode",{parentName:"p"},"R CMD INSTALL gmsgff")," again to\ninstall the new version of the package."))),(0,r.kt)("h2",{id:"challenge-1-extract-more-attributes"},"Challenge 1: extract more attributes"),(0,r.kt)("p",null,"Wouldn't it be nice to have other attributes extracted as columns?  For example in the Ensembl files the ",(0,r.kt)("inlineCode",{parentName:"p"},"Name")," and\n",(0,r.kt)("inlineCode",{parentName:"p"},"biotype")," are ",(0,r.kt)("a",{parentName:"p",href:"https://ftp.ensembl.org/pub/current_gff3/homo_sapiens/README"},"useful attributes")," of gene records."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Give ",(0,r.kt)("inlineCode",{parentName:"p"},"parse_gff3_to_dataframe()")," a second argument called 'attributes', which should be a list (or in R,\na vector) of attribute names. For each attribute in the list, you should extract it and add it into a seperate column of\nthe result dataframe.  (This should be in addition to the ",(0,r.kt)("inlineCode",{parentName:"p"},"ID")," and ",(0,r.kt)("inlineCode",{parentName:"p"},"Parent")," attributes, which you should always\nextract.)"),(0,r.kt)("p",{parentName:"div"},"For bonus marks, alter ",(0,r.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite.R")," or ",(0,r.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite.py")," so you can also specify these attributes on the\ncommand-line. ",(0,r.kt)("strong",{parentName:"p"},"Hint")," You will need to use the ",(0,r.kt)("inlineCode",{parentName:"p"},'nargs = "+"')," option to the argument parser ",(0,r.kt)("inlineCode",{parentName:"p"},"add_argument()")," call, so\nthat the option can take several values.  The argument can then be used as a list (in python) or vector (in R)."))),(0,r.kt)("h2",{id:"challenge-2-shrink-attributes"},"Challenge 2: shrink ",(0,r.kt)("inlineCode",{parentName:"h2"},"attributes")),(0,r.kt)("p",null,"Currently, ",(0,r.kt)("inlineCode",{parentName:"p"},"parse_gff3_to_dataframe()")," extracts ",(0,r.kt)("inlineCode",{parentName:"p"},"ID")," and ",(0,r.kt)("inlineCode",{parentName:"p"},"Parent")," (and, if you do the challenge above, other attributes too).\nBut it also leaves these fields in the ",(0,r.kt)("inlineCode",{parentName:"p"},"attributes")," column.  Since the files are so bit, this can waste a lot of space."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Find a way to remove the extracted fields from ",(0,r.kt)("inlineCode",{parentName:"p"},"attributes")," when you extract them.\nMake sure to remove the semicolon too, if it's there!"))),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Hint 1")," The ",(0,r.kt)("a",{parentName:"p",href:"https://stringr.tidyverse.org/reference/str_remove.html"},(0,r.kt)("inlineCode",{parentName:"a"},"str_remove()")," function")," can help with this -\nyou will need to use the right regular expression."),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Hint 2")," The regular expression syntax '(;|$)' will match ",(0,r.kt)("em",{parentName:"p"},"either")," a semicolon ",(0,r.kt)("em",{parentName:"p"},"or")," the end of the string - may be\nuseful."),(0,r.kt)("h2",{id:"challenge-3-split-the-output"},"Challenge 3: split the output"),(0,r.kt)("p",null,"Currently, our ",(0,r.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite")," program outputs all fields into a single table (called ",(0,r.kt)("inlineCode",{parentName:"p"},"gff"),").\nHowever, wouldn't it be better to split out the different types into different tables?\nFor example:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"chromosomes"),(0,r.kt)("li",{parentName:"ul"},"genes"),(0,r.kt)("li",{parentName:"ul"},"transcripts"),(0,r.kt)("li",{parentName:"ul"},"exons"),(0,r.kt)("li",{parentName:"ul"},"CDS"),(0,r.kt)("li",{parentName:"ul"},"utr")),(0,r.kt)("p",null,"may all make sense as seperate tables."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Create a version of ",(0,r.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite")," (say, ",(0,r.kt)("inlineCode",{parentName:"p"},"gff_to_sqlite_split"),") that outputs different gff record types into different\ntables."),(0,r.kt)("p",{parentName:"div"},(0,r.kt)("strong",{parentName:"p"},"Note")," This is easiest if you focus on GFF files from one source (say Ensembl) - since the GFFs from other sources\n(such as gencode) have different types in them."))),(0,r.kt)("p",null,"In fact the best version of this would extract different attributes for each table as well.  For example, the gene name or symbol\nis only needed for gene records."),(0,r.kt)("h2",{id:"challenge-4-the-refactor"},"Challenge 4: the refactor"),(0,r.kt)("p",null,"See ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/programming_with_gene_annotations3/appendices/refactoring"},"the appendix")," for this challenge."),(0,r.kt)("h2",{id:"the-finished-product"},"The finished product"),(0,r.kt)("p",null,"If you get through all (or any) of that - congratulations!"),(0,r.kt)("p",null,"For a tutorial that uses this code - see the follow-on ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/analysing_genes_across_species/"},"Analysing genes across species\ntutorial"),"."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"If instead you'd like to see my version of the finished product - the R package and the command-line program with the\nimprovements listed here - there's also a link on the ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/programming/analysing_genes_across_species/"},"Analysing genes across species\ntutorial")," page."))))}u.isMDXComponent=!0}}]);