"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[2242],{3905:(e,n,t)=>{t.d(n,{Zo:()=>m,kt:()=>h});var a=t(7294);function o(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function r(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);n&&(a=a.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,a)}return t}function i(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?r(Object(t),!0).forEach((function(n){o(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):r(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,a,o=function(e,n){if(null==e)return{};var t,a,o={},r=Object.keys(e);for(a=0;a<r.length;a++)t=r[a],n.indexOf(t)>=0||(o[t]=e[t]);return o}(e,n);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)t=r[a],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(o[t]=e[t])}return o}var l=a.createContext({}),c=function(e){var n=a.useContext(l),t=n;return e&&(t="function"==typeof e?e(n):i(i({},n),e)),t},m=function(e){var n=c(e.components);return a.createElement(l.Provider,{value:n},e.children)},p="mdxType",u={inlineCode:"code",wrapper:function(e){var n=e.children;return a.createElement(a.Fragment,{},n)}},d=a.forwardRef((function(e,n){var t=e.components,o=e.mdxType,r=e.originalType,l=e.parentName,m=s(e,["components","mdxType","originalType","parentName"]),p=c(t),d=o,h=p["".concat(l,".").concat(d)]||p[d]||u[d]||r;return t?a.createElement(h,i(i({ref:n},m),{},{components:t})):a.createElement(h,i({ref:n},m))}));function h(e,n){var t=arguments,o=n&&n.mdxType;if("string"==typeof e||o){var r=t.length,i=new Array(r);i[0]=d;var s={};for(var l in n)hasOwnProperty.call(n,l)&&(s[l]=n[l]);s.originalType=e,s[p]="string"==typeof e?e:o,i[1]=s;for(var c=2;c<r;c++)i[c]=t[c];return a.createElement.apply(null,i)}return a.createElement.apply(null,t)}d.displayName="MDXCreateElement"},2360:(e,n,t)=>{t.d(n,{Z:()=>i});var a=t(7294),o=t(6010);const r={tabItem:"tabItem_OmH5"};function i(e){let{children:n,hidden:t,className:i}=e;return a.createElement("div",{role:"tabpanel",className:(0,o.Z)(r.tabItem,i),hidden:t},n)}},9877:(e,n,t)=>{t.d(n,{Z:()=>u});var a=t(7462),o=t(7294),r=t(2389),i=t(7392),s=t(7094),l=t(2466),c=t(6010);const m={tabList:"tabList_uSqn",tabItem:"tabItem_LplD"};function p(e){const{lazy:n,block:t,defaultValue:r,values:p,groupId:u,className:d}=e,h=o.Children.map(e.children,(e=>{if((0,o.isValidElement)(e)&&void 0!==e.props.value)return e;throw new Error(`Docusaurus error: Bad <Tabs> child <${"string"==typeof e.type?e.type:e.type.name}>: all children of the <Tabs> component should be <TabItem>, and every <TabItem> should have a unique "value" prop.`)})),g=p??h.map((e=>{let{props:{value:n,label:t,attributes:a}}=e;return{value:n,label:t,attributes:a}})),v=(0,i.l)(g,((e,n)=>e.value===n.value));if(v.length>0)throw new Error(`Docusaurus error: Duplicate values "${v.map((e=>e.value)).join(", ")}" found in <Tabs>. Every value needs to be unique.`);const _=null===r?r:r??h.find((e=>e.props.default))?.props.value??h[0]?.props.value;if(null!==_&&!g.some((e=>e.value===_)))throw new Error(`Docusaurus error: The <Tabs> has a defaultValue "${_}" but none of its children has the corresponding value. Available values are: ${g.map((e=>e.value)).join(", ")}. If you intend to show no default tab, use defaultValue={null} instead.`);const{tabGroupChoices:f,setTabGroupChoices:b}=(0,s.U)(),[w,k]=(0,o.useState)(_),y=[],{blockElementScrollPositionUntilNextRender:N}=(0,l.o5)();if(null!=u){const e=f[u];null!=e&&e!==w&&g.some((n=>n.value===e))&&k(e)}const x=e=>{const n=e.currentTarget,t=y.indexOf(n),a=g[t].value;a!==w&&(N(n),k(a),null!=u&&b(u,a))},C=e=>{let n=null;switch(e.key){case"ArrowRight":{const t=y.indexOf(e.currentTarget)+1;n=y[t]||y[0];break}case"ArrowLeft":{const t=y.indexOf(e.currentTarget)-1;n=y[t]||y[y.length-1];break}}n?.focus()};return o.createElement("div",{className:(0,c.Z)("tabs-container",m.tabList)},o.createElement("ul",{role:"tablist","aria-orientation":"horizontal",className:(0,c.Z)("tabs",{"tabs--block":t},d)},g.map((e=>{let{value:n,label:t,attributes:r}=e;return o.createElement("li",(0,a.Z)({role:"tab",tabIndex:w===n?0:-1,"aria-selected":w===n,key:n,ref:e=>y.push(e),onKeyDown:C,onFocus:x,onClick:x},r,{className:(0,c.Z)("tabs__item",m.tabItem,r?.className,{"tabs__item--active":w===n})}),t??n)}))),n?(0,o.cloneElement)(h.filter((e=>e.props.value===w))[0],{className:"margin-top--md"}):o.createElement("div",{className:"margin-top--md"},h.map(((e,n)=>(0,o.cloneElement)(e,{key:n,hidden:e.props.value!==w})))))}function u(e){const n=(0,r.Z)();return o.createElement(p,(0,a.Z)({key:String(n)},e))}},9574:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>l,contentTitle:()=>i,default:()=>u,frontMatter:()=>r,metadata:()=>s,toc:()=>c});var a=t(7462),o=(t(7294),t(3905));t(9877),t(2360);const r={sidebar_position:4},i="How much of the genome is in exons?",s={unversionedId:"programming/analysing_genes_across_species/how_much_in_genes/how_much_in_exons",id:"programming/analysing_genes_across_species/how_much_in_genes/how_much_in_exons",title:"How much of the genome is in exons?",description:"You might be a bit underwhelmed by the difference between our 'naive' and 'correct' calculations for genes. However the",source:"@site/docs/programming/analysing_genes_across_species/how_much_in_genes/004_how_much_in_exons.md",sourceDirName:"programming/analysing_genes_across_species/how_much_in_genes",slug:"/programming/analysing_genes_across_species/how_much_in_genes/how_much_in_exons",permalink:"/whg-training-resources/programming/analysing_genes_across_species/how_much_in_genes/how_much_in_exons",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/programming/analysing_genes_across_species/how_much_in_genes/004_how_much_in_exons.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{sidebar_position:4},sidebar:"sidebar5",previous:{title:"How much of the genome is *really* in genes?",permalink:"/whg-training-resources/programming/analysing_genes_across_species/how_much_in_genes/how_much_II"},next:{title:"Conclusions and challenges",permalink:"/whg-training-resources/programming/analysing_genes_across_species/challenges"}},l={},c=[],m={toc:c},p="wrapper";function u(e){let{components:n,...r}=e;return(0,o.kt)(p,(0,a.Z)({},m,r,{components:n,mdxType:"MDXLayout"}),(0,o.kt)("h1",{id:"how-much-of-the-genome-is-in-exons"},"How much of the genome is in exons?"),(0,o.kt)("p",null,"You might be a bit underwhelmed by the difference between our 'naive' and 'correct' calculations for genes. However the\npower of our functions really comes into play if we do the same thing for exons, or coding sequence.  "),(0,o.kt)("p",null,"To do this, we just have to put back the ",(0,o.kt)("inlineCode",{parentName:"p"},"start")," and ",(0,o.kt)("inlineCode",{parentName:"p"},"end")," columns, since we renamed them earlier:"),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-r"},"covered_by_exons = compute_lengths_per_chromosome(\n    exons %>% mutate( start = exon_start, end = exon_end )\n)\n\nexon_proportions = compute_proportion_of_genome_covered( covered_by_exons )\n\n")),(0,o.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,o.kt)("div",{parentName:"div",className:"admonition-heading"},(0,o.kt)("h5",{parentName:"div"},(0,o.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,o.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,o.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Warning")),(0,o.kt)("div",{parentName:"div",className:"admonition-content"},(0,o.kt)("p",{parentName:"div"},"This might several minutes to run!  (There are a lot of exons and coding sequence regions.)"))),(0,o.kt)("p",null,"The result will look like this:"),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre"},'> print( exon_proportions[, c( "dataset", "total_genome_length", "proportion_covered" )] )\n# A tibble: 11 \xd7 3\n   dataset                                           total_genome_length proportion_covered\n   <chr>                                                           <dbl>              <dbl>\n 1 Acanthochromis_polyacanthus.ASM210954v1.110                 831322490            0.0776 \n 2 Asparagus_officinalis.Aspof.V1.57                          1015711224            0.0240 \n 3 Bufo_bufo-GCA_905171765.1-2022_05-genes                    5004510881            0.00917\n 4 Camelus_dromedarius.CamDro2.110.chr                        2052758708            0.0225 \n 5 Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.110.chr          1041139641            0.0933 \n 6 Homo_sapiens.GRCh38.110.chr                                3088286401            0.0507 \n 7 Mus_musculus.GRCm39.110.chr                                2723431143            0.0483 \n 8 Pan_troglodytes.Pan_tro_3.0.110.chr                        2967125077            0.0220 \n 9 Plasmodium_falciparum.ASM276v2.57                            23292622            0.538  \n10 Plasmodium_knowlesi.ASM635v1.57                              23462187            0.476  \n11 Plasmodium_vivax.ASM241v2.57                                 23771395            0.529 \n')),(0,o.kt)("p",null,"Or you can plot it:"),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-r"},'p = (\n   ggplot( data = exon_proportions )\n   + geom_col(\n      aes(\n         x = proportion_covered,\n         y = dataset\n      )\n   )\n   + ylab( "Species" )\n   + xlab( "Proportion of genome in exons" )\n   + xlim( 0, 1 )\n   + theme_minimal(14)\n)\nprint(p)\n')),(0,o.kt)("p",null,(0,o.kt)("img",{alt:"img",src:t(3463).Z,width:"2184",height:"1088"})),(0,o.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,o.kt)("div",{parentName:"div",className:"admonition-heading"},(0,o.kt)("h5",{parentName:"div"},(0,o.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,o.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,o.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,o.kt)("div",{parentName:"div",className:"admonition-content"},(0,o.kt)("p",{parentName:"div"},"Does this agree with what you understand about how the human or other genomes are organised?\nTry looking at a gene in the UCSC genome browser, does it roughly tie in with the above?"))),(0,o.kt)("p",null,"##\xa0Conclusion and challenge"),(0,o.kt)("p",null,"I think it is pretty incredible that only a tiny fraction of many genomes actually codes for mRNA. For example, in the\nhuman genome (which must be the best-annotated genomes available), only about 5% is annotated as protein coding."),(0,o.kt)("p",null,"On the other hand some genomes have much higher proportions.  The genomes of malaria species, in particular, seem to be\npacked full of genes."),(0,o.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,o.kt)("div",{parentName:"div",className:"admonition-heading"},(0,o.kt)("h5",{parentName:"div"},(0,o.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,o.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,o.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,o.kt)("div",{parentName:"div",className:"admonition-content"},(0,o.kt)("p",{parentName:"div"},"Now do the same thing for coding sequence.  (Make sure and fix the ",(0,o.kt)("inlineCode",{parentName:"p"},"seqid"),", ",(0,o.kt)("inlineCode",{parentName:"p"},"start")," and ",(0,o.kt)("inlineCode",{parentName:"p"},"end")," columns, since we renamed them\nin ",(0,o.kt)("inlineCode",{parentName:"p"},"cds"),"). You might see something like this:"),(0,o.kt)("p",{parentName:"div"},(0,o.kt)("img",{alt:"img",src:t(6878).Z,width:"2172",height:"1076"})),(0,o.kt)("p",{parentName:"div"},"Are you surprised by these numbers?  Which species have the least proportion of genome in genes?  Which have the most?"))),(0,o.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,o.kt)("div",{parentName:"div",className:"admonition-heading"},(0,o.kt)("h5",{parentName:"div"},(0,o.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,o.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,o.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,o.kt)("div",{parentName:"div",className:"admonition-content"},(0,o.kt)("p",{parentName:"div"},"How much of the genome is in untranslated regions?  What about:"),(0,o.kt)("ul",{parentName:"div"},(0,o.kt)("li",{parentName:"ul"},"in ",(0,o.kt)("inlineCode",{parentName:"li"},"tRNA")," fields (which encode transfer RNAs?)"),(0,o.kt)("li",{parentName:"ul"},"in non-coding RNAs?"),(0,o.kt)("li",{parentName:"ul"},"in pseudogenes?")))))}u.isMDXComponent=!0},6878:(e,n,t)=>{t.d(n,{Z:()=>a});const a=t.p+"assets/images/propn_in_cds-c81086f9dc3c163219e401fa22735ec9.png"},3463:(e,n,t)=>{t.d(n,{Z:()=>a});const a=t.p+"assets/images/propn_in_exons-125527a57444f021c236c366f06c099a.png"}}]);