"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[9841],{3905:(e,t,n)=>{n.d(t,{Zo:()=>g,kt:()=>d});var a=n(7294);function r(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function s(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?s(Object(n),!0).forEach((function(t){r(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):s(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function o(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},s=Object.keys(e);for(a=0;a<s.length;a++)n=s[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(e);for(a=0;a<s.length;a++)n=s[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=a.createContext({}),p=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},g=function(e){var t=p(e.components);return a.createElement(l.Provider,{value:t},e.children)},m="mdxType",h={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},c=a.forwardRef((function(e,t){var n=e.components,r=e.mdxType,s=e.originalType,l=e.parentName,g=o(e,["components","mdxType","originalType","parentName"]),m=p(n),c=r,d=m["".concat(l,".").concat(c)]||m[c]||h[c]||s;return n?a.createElement(d,i(i({ref:t},g),{},{components:n})):a.createElement(d,i({ref:t},g))}));function d(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var s=n.length,i=new Array(s);i[0]=c;var o={};for(var l in t)hasOwnProperty.call(t,l)&&(o[l]=t[l]);o.originalType=e,o[m]="string"==typeof e?e:r,i[1]=o;for(var p=2;p<s;p++)i[p]=n[p];return a.createElement.apply(null,i)}return a.createElement.apply(null,n)}c.displayName="MDXCreateElement"},1454:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>i,default:()=>h,frontMatter:()=>s,metadata:()=>o,toc:()=>p});var a=n(7462),r=(n(7294),n(3905));const s={sidebar_position:5},i="Extreme genes revisited",o={unversionedId:"programming/analysing_genes_across_species/extreme_genes/extreme_genes_revisited",id:"programming/analysing_genes_across_species/extreme_genes/extreme_genes_revisited",title:"Extreme genes revisited",description:"Phew!  If you reached this point you have computed the coding sequence length of each transcript, and have produced a table of genes linked to a 'canonical' transcript.  You may also have computed the number of exons in each gene.",source:"@site/docs/programming/analysing_genes_across_species/extreme_genes/005_extreme_genes_revisited.md",sourceDirName:"programming/analysing_genes_across_species/extreme_genes",slug:"/programming/analysing_genes_across_species/extreme_genes/extreme_genes_revisited",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/extreme_genes_revisited",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/programming/analysing_genes_across_species/extreme_genes/005_extreme_genes_revisited.md",tags:[],version:"current",sidebarPosition:5,frontMatter:{sidebar_position:5},sidebar:"sidebar5",previous:{title:"The number of exons",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/how_many_exons"},next:{title:"Too long, didn't read",permalink:"/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/tldr"}},l={},p=[],g={toc:p},m="wrapper";function h(e){let{components:t,...s}=e;return(0,r.kt)(m,(0,a.Z)({},g,s,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"extreme-genes-revisited"},"Extreme genes revisited"),(0,r.kt)("p",null,"Phew!  If you reached this point you have ",(0,r.kt)("a",{parentName:"p",href:"./006_long_genes_2.md"},"computed the coding sequence length")," of each transcript, and have produced a ",(0,r.kt)("a",{parentName:"p",href:"./007_canonical_transcripts.md"},"table of genes linked to a 'canonical' transcript"),".  You may also have computed the ",(0,r.kt)("a",{parentName:"p",href:"./007_how_many_exons.md"},"number of exons in each gene"),"."),(0,r.kt)("p",null,"You're in good shape to find 'extreme' genes!"),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Investigate 'extreme' genes now."),(0,r.kt)("p",{parentName:"div"},"For example what are the biggest genes?  The genes with the most exons?  Are they similar across species?  Do they have related biological function?  What about the smallest genes?  What's an average gene size in each species?  What's an average number of exons?  What proportion of genes have just one exon?"),(0,r.kt)("p",{parentName:"div"},"Use your R skills, along with genome browsers and/or literature search to investigate - and have fun!"),(0,r.kt)("p",{parentName:"div"},(0,r.kt)("strong",{parentName:"p"},"Note"),". You should follow whatever interests you most here - you won't be able to look at everything."))),(0,r.kt)("p",null,"As a starting point I suggest making histograms of these variables and plot them against each other.  For example, you could histogram the coding sequence length:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-r"},'(\n    ggplot( data = genes )\n    + geom_histogram(\n        aes(\n            x = cds_length\n        ),\n        bins = 100\n    )\n    + xlab( "Coding sequence length of chosen transcript" )\n    + scale_x_log10()\n    + facet_wrap( ~ dataset )\n)\n')),(0,r.kt)("div",{className:"admonition admonition-caution alert alert--warning"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"16",height:"16",viewBox:"0 0 16 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"}))),"Note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"This histogram above has weird-looking artifacts around length 1,000 in some species.  What are those?"))),(0,r.kt)("p",null,"Or you could plot it against the genome length of each gene:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-r"},'genes$length = genes$end - genes$start + 1\n(\n    ggplot( data = genes )\n    + geom_point( aes(\n        x = length,\n        y = cds_length,\n        col = dataset\n    ))\n    + scale_x_log10()\n    + scale_y_log10()\n    + xlab( "Genome length of gene" )\n    + ylab( "Coding sequence length" )\n)\n')),(0,r.kt)("p",null,(0,r.kt)("img",{alt:"img",src:n(4401).Z,width:"3329",height:"1624"})),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"I used log scales above - what does the plot look like without that?  What if you facet over the datasets instead of\ncolouring them?  As always, experiment until you like the plot."),(0,r.kt)("p",{parentName:"div"},"In this plot, all the points lie under the diagonal - why?"))),(0,r.kt)("p",null,"You can then use filtering and sorting (",(0,r.kt)("inlineCode",{parentName:"p"},"arrange()"),") to identify the outlying genes in question, and investigate."),(0,r.kt)("p",null,"Good luck!  For example, here are some things I noticed:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"In all the data I looked at, there's only one gene with > 100 transcripts. It is the human gene ",(0,r.kt)("a",{parentName:"p",href:"https://www.uniprot.org/uniprot/P53779"},(0,r.kt)("em",{parentName:"a"},"MAPK10"),',\n"Mitogen-activated protein kinase 10"')," and it has >151 transcripts. (It has so\nmany that it takes a while to load in\n",(0,r.kt)("a",{parentName:"p",href:"http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000109339;r=4:85990007-86594625"},"Ensembl")," or the\n",(0,r.kt)("a",{parentName:"p",href:"https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr4%3A86016491%2D86594110&hgsid=275896231_HieWdPQTMOsgYQUFAnwTALgAECs0"},"UCSC Genome\nbrowser"),".)\nIf you explore these genome browsers you will find a wealth of information on this gene including its expression\npatterns and function. Is there much published literature about it? What does it look like in other species?")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Only one gene has a coding sequence length over 100kb - the phenomenal ",(0,r.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Titin"},(0,r.kt)("em",{parentName:"a"},"Titin")),".\nIt is also by far the longest gene in chimpanzees, camels and mice.  It ",(0,r.kt)("em",{parentName:"p"},"also")," stands out in terms of number of exons"))),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"having a slightly incredible 184 exons in Chimpanzees (",(0,r.kt)("em",{parentName:"li"},"Pan troglodytes"),") and ~113 in humans.")),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Another gene with lots of exons is ",(0,r.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Nebulin"},(0,r.kt)("em",{parentName:"a"},"Nebulin")),". These genes seem especially huge in Chimpanzees. Are\nthey big in all species? In all great apes? Is there literature that might explain the results for these genes, or other\ngenes with lots of exons?\n")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"The genes with lots of exons aren't the largest though (by genome length). In humans and chimpanzees, that accolade goes\nto ",(0,r.kt)("a",{parentName:"p",href:"http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000078328;r=16:5239802-7713340"},(0,r.kt)("inlineCode",{parentName:"a"},"RBFOX1")),"\nwith a whopping length of around 2 and a half megabases, or 0.08% of the genome. What does this look\nlike across species? Where is it expressed? (Interestingly, this gene has been previously associated ",(0,r.kt)("a",{parentName:"p",href:"https://www.nature.com/articles/s41380-018-0068-7"},"with aggressive\nbehaviour"),").")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"There are also some pretty tiny genes around. In humans one of the smallest seems to be\n",(0,r.kt)("a",{parentName:"p",href:"http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000288608;r=7:151061928-151061978;t=ENST00000674552"},(0,r.kt)("em",{parentName:"a"},"ENSG00000288608")),",\nwhich is only 50 base pairs long. (However, it seems to be contained in an exon of the much\nlonger ",(0,r.kt)("a",{parentName:"p",href:"https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr7%3A151061877%2D151062029&hgsid=275886241_mxgWOGDr4elcf2SdW0zGz6ukwVJW"},(0,r.kt)("em",{parentName:"a"},"SLC4A2")," Anion exhange protein 2"),".\nI'm not quite sure what this means...)"))),(0,r.kt)("p",null,"Essentially everything you look at is interesting here..."))}h.isMDXComponent=!0},4401:(e,t,n)=>{n.d(t,{Z:()=>a});const a=n.p+"assets/images/coding_length_vs_gene_length-42359d384bd6850124074453b424ef24.png"}}]);