"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[9147],{3905:(e,t,a)=>{a.d(t,{Zo:()=>c,kt:()=>m});var n=a(7294);function s(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function r(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?r(Object(a),!0).forEach((function(t){s(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):r(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function i(e,t){if(null==e)return{};var a,n,s=function(e,t){if(null==e)return{};var a,n,s={},r=Object.keys(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||(s[a]=e[a]);return s}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)a=r[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(s[a]=e[a])}return s}var l=n.createContext({}),u=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},c=function(e){var t=u(e.components);return n.createElement(l.Provider,{value:t},e.children)},p="mdxType",h={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var a=e.components,s=e.mdxType,r=e.originalType,l=e.parentName,c=i(e,["components","mdxType","originalType","parentName"]),p=u(a),d=s,m=p["".concat(l,".").concat(d)]||p[d]||h[d]||r;return a?n.createElement(m,o(o({ref:t},c),{},{components:a})):n.createElement(m,o({ref:t},c))}));function m(e,t){var a=arguments,s=t&&t.mdxType;if("string"==typeof e||s){var r=a.length,o=new Array(r);o[0]=d;var i={};for(var l in t)hasOwnProperty.call(t,l)&&(i[l]=t[l]);i.originalType=e,i[p]="string"==typeof e?e:s,o[1]=i;for(var u=2;u<r;u++)o[u]=a[u];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}d.displayName="MDXCreateElement"},5165:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>h,frontMatter:()=>r,metadata:()=>i,toc:()=>u});var n=a(7462),s=(a(7294),a(3905));const r={sidebar_position:3},o="Step 2: Inspecting sequence data quality",i={unversionedId:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control",id:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control",title:"Step 2: Inspecting sequence data quality",description:"There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality",source:"@site/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control.md",sourceDirName:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis",slug:"/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{sidebar_position:3},sidebar:"sidebar9",previous:{title:"Step 1: What's in the fastq files?",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs"},next:{title:"Step 3: Aligning reads",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads"}},l={},u=[{value:"Running <code>fastqc</code>",id:"running-fastqc",level:2},{value:"A survey of sequencing performance",id:"a-survey-of-sequencing-performance",level:2},{value:"Next steps",id:"next-steps",level:3},{value:"Interpreting fastqc output",id:"interpreting-fastqc-output",level:2},{value:"Basic statistics",id:"basic-statistics",level:3},{value:"Per base sequence quality",id:"per-base-sequence-quality",level:3},{value:"Per tile sequence quality",id:"per-tile-sequence-quality",level:3},{value:"Per-base sequence content",id:"per-base-sequence-content",level:3},{value:"Per-sequence GC content",id:"per-sequence-gc-content",level:3},{value:"Sequence duplication levels, adapter content and over-represented sequences",id:"sequence-duplication-levels-adapter-content-and-over-represented-sequences",level:3},{value:"Adapter content and over-represented sequences",id:"adapter-content-and-over-represented-sequences",level:3},{value:"Next steps",id:"next-steps-1",level:3}],c={toc:u},p="wrapper";function h(e){let{components:t,...r}=e;return(0,s.kt)(p,(0,n.Z)({},c,r,{components:t,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"step-2-inspecting-sequence-data-quality"},"Step 2: Inspecting sequence data quality"),(0,s.kt)("p",null,"There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality\nof sequencing. These include looking for poor quality data, looking for read duplication, assessing\nGC content, looking for over-represented sequences and the presence of sequence adapters in reads.\nLuckily there are also well-established tools that make running these analyses easy. The one we'll\ntry is called ",(0,s.kt)("inlineCode",{parentName:"p"},"fastqc")," and it can be run as follows."),(0,s.kt)("h2",{id:"running-fastqc"},"Running ",(0,s.kt)("inlineCode",{parentName:"h2"},"fastqc")),(0,s.kt)("p",null,"First make a directory for the output and run fastqc:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"mkdir fastqc_output\nfastqc -o fastqc_output malaria/*.fastq.gz\n")),(0,s.kt)("p",null,"This will process both reads and will take a couple of minutes to run."),(0,s.kt)("p",null,"If you look in the folder now you'll see there are some new files:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"$ ls fastq_output\nQG0033-C_Illumina-HiSeq_read1.fastq.zip\nQG0033-C_Illumina-HiSeq_read1.fastq.html\nQG0033-C_Illumina-HiSeq_read2.fastq.zip\nQG0033-C_Illumina-HiSeq_read2.fastq.html\n...\n")),(0,s.kt)("p",null,(0,s.kt)("inlineCode",{parentName:"p"},"fastqc")," has created some ",(0,s.kt)("inlineCode",{parentName:"p"},".html")," files, as well as some ",(0,s.kt)("inlineCode",{parentName:"p"},".zip")," files. Look at these by opening the\n",(0,s.kt)("inlineCode",{parentName:"p"},".html")," files in a web browser."),(0,s.kt)("h2",{id:"a-survey-of-sequencing-performance"},"A survey of sequencing performance"),(0,s.kt)("p",null,"In the rest of the page below I'm going to look at the ",(0,s.kt)("inlineCode",{parentName:"p"},"QG0033-C")," dataset, which you will have\ngenerated if you followed the instructions above. However, it's useful to see what the output looks\nlike across datasets - so I collected some.  See:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},(0,s.kt)("strong",{parentName:"p"},"a malaria sample")," sequenced on a HiSeq with an unusual GC bump.  (That's this dataset, described further below). ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read1_fastqc.html#M5"},"-> link")," ")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"a ",(0,s.kt)("strong",{parentName:"p"},"second malaria sample")," sequenced on the Illumina XTEN for which the reads have PCR primer contamination from the library prep step. ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read1_fastqc.html#M9"},"-> read 1 link"),", ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read2_fastqc.html#M9"},"read 2 link"),".")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},(0,s.kt)("strong",{parentName:"p"},"a human sample")," sequenced on a Novaseq at the ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk"},"WHG"),", for which read 2 is enriched for poly-G sequences. ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-illumina_novaseq_2_fastqc.html#M9"},"-> link"),".")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},(0,s.kt)("strong",{parentName:"p"},"another human sample"),", sequenced during a fragmentation testing process, that has the same problem but much worse (nearly 3% of reads are like this.) ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/NA12878-illumina_novaseq-NE755566_R2_fastqc.html#M5"},"-> link"),".  ")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"The ",(0,s.kt)("strong",{parentName:"p"},"same human sample")," sequenced ",(0,s.kt)("a",{parentName:"p",href:"https://www.ebi.ac.uk/ena/browser/view/ERR3239334?show=reads"},"at the NYGC"),", where\nthe base qualities have been compressed to a different set of levels.  ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/ERR3239334-Illumina_NovaSeq_6000_read2_fastqc.html#M1"},"->\nlink"),"."))),(0,s.kt)("p",null,"Have a look at these. All except the second-last dataset would probably be considered reasonable quality data, but they\nall have at least some level of artifacts."),(0,s.kt)("p",null,"(More examples can be found ",(0,s.kt)("a",{parentName:"p",href:"https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"},"on the fastq documentation\npage"),".)"),(0,s.kt)("p",null,"Finally here is an example of ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-PacBio_CCS_fastqc.html"},"a long-read human dataset")," (PacBio Hifi) - this looks quite different."),(0,s.kt)("h3",{id:"next-steps"},"Next steps"),(0,s.kt)("p",null,"The rest of this page contains information on what's in this fastqc output. There's lots in here - feel free to read.\nWhen you're ready move on to ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads"},"align the reads to the reference genome"),"."),(0,s.kt)("h2",{id:"interpreting-fastqc-output"},"Interpreting fastqc output"),(0,s.kt)("p",null,"For ",(0,s.kt)("inlineCode",{parentName:"p"},"QG0033-C")," fastqc has provided us a detailed overview of information it can gain purely from\nthe fastq files. Let's take a look."),(0,s.kt)("h3",{id:"basic-statistics"},"Basic statistics"),(0,s.kt)("p",null,"The 'Basic statistics' section tells you some stuff you already know - the number of reads, read\nlength and so on: "),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:a(9710).Z,width:"675",height:"307"})),(0,s.kt)("p",null,"It also tells you the average GC content (the proportion of read bases that are\nGs or Cs). This is around 23%, which is about right because the ",(0,s.kt)("em",{parentName:"p"},"P.falciparum")," genome ",(0,s.kt)("a",{parentName:"p",href:"https://doi.org/10.1038/nature01097"},"has such\nhigh A/T content"),"."),(0,s.kt)("h3",{id:"per-base-sequence-quality"},"Per base sequence quality"),(0,s.kt)("p",null,"The 'Per base sequence quality' section is interesting: it suggests that the ",(0,s.kt)("strong",{parentName:"p"},"quality of\nsequencing is lower at the ends of the read"),"."),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:a(1189).Z,width:"910",height:"676"})),(0,s.kt)("p",null,"Here, this is particularly seen at the start of the read, which is slightly unusual. If you look at\ne.g. the results from the data run on XTEN you'll see that it really tails off toward the end of\nthe read - but that's not so noticeable here."),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Note.")," It's important to realise this is the sequencer's estimate of base quality - it's generally\nthought pretty good, but is not necessarily fully calibrated."),(0,s.kt)("h3",{id:"per-tile-sequence-quality"},"Per tile sequence quality"),(0,s.kt)("p",null,"This is a map of the 'tiles' on the flowcell with the average base quality for each tile on the\nIllumina sequencer. (Here, a 'tile' is a physical region of a sequencer's flowcell lane that is\nimaged to detect bases. Inside each tile will be multiple 'clusters' which, if the chemistry works\nout, are each amplified from a single DNA fragment and correspond to one read pair. See the ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory"},"short\nread theory page")," for more on this)."),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Question.")," How does fastqc know which tile the reads are from?"),(0,s.kt)("p",null,"In this output you can see some tiles that have performed badly (red indicating below average base\nquality) but most look pretty good."),(0,s.kt)("h3",{id:"per-base-sequence-content"},"Per-base sequence content"),(0,s.kt)("p",null,"This shows the average proportion of each base in each location in the reads:"),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:a(9239).Z,width:"924",height:"676"})),(0,s.kt)("p",null,"Here you see a fairly typical picture, as follows:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},"In general ",(0,s.kt)("em",{parentName:"li"},"there should be about the same number of G as C bases, and the same number of A as T\nbases"),". This is because either strand of DNA ought to be equally likely to be sequenced. In the\nabove just under 40% of bases are G or C, and just over 10% are A or T, and it is pretty flat\nacross the read. So this looks about right for a malaria genome with ~23% A/T content."),(0,s.kt)("li",{parentName:"ul"},"There is nevertheless a skew of base content toward the start of the read. This seems to be\ntypical for Illumina sequencing and it could be caused by a number of things. One possibility is\nthat there are ",(0,s.kt)("a",{parentName:"li",href:"https://doi.org/10.1186/s12859-016-0976-y"},"biases in the sequence content due to DNA\nfragmentation"),"; it's also true that base quality was\nlower at these bases so maybe there is another aspect of the chemistry causing this.")),(0,s.kt)("h3",{id:"per-sequence-gc-content"},"Per-sequence GC content"),(0,s.kt)("p",null,"This plot measures the distribution of G and C bases ('GC' content) in all reads - and for this\nsample it looks kind of interesting:"),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:a(2152).Z,width:"902",height:"656"})),(0,s.kt)("p",null,"The bump on the left is expected, since ",(0,s.kt)("em",{parentName:"p"},"P.falciparum")," has around 23% GC content on average. This\nvaries a bit around the genome so we should expect to see some spread around 23%."),(0,s.kt)("p",null,"But this sample also has a second bump of reads at around 60% GC. What could this be? Here are some\ndata points:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"If you look at the ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read2_fastqc.html#M5"},"read 2 fastqc output")," file you'll see it there as well.")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"It's not seen in other samples (such as those linked above) - so it's something about this sample.")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"Human DNA has about 41% GC content, but this is up at 60%... so it's not obviously whole-genome human DNA contamination."))),(0,s.kt)("p",null,"Something fishy is going on... what could it be?"),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Note.")," This will actually be one of our ",(0,s.kt)("strong",{parentName:"p"},(0,s.kt)("a",{parentName:"strong",href:"challenges.md"},"challenge questions")),". Leave it for now as\nwe'll need to align reads to make progress on it."),(0,s.kt)("h3",{id:"sequence-duplication-levels-adapter-content-and-over-represented-sequences"},"Sequence duplication levels, adapter content and over-represented sequences"),(0,s.kt)("p",null,"Near the end are a set of plots that can provide evidence of contamination or chemistry issues -\nsuch as high levels of duplicaets which could indicate a 'low-diversity' (low DNA quantity input)\nlibrary, presence of adapter sequences which could be due to too short fragments, or other\nartifacts."),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Note.")," The list of possible artifacts is long - see the ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory"},"paired-end sequencing\ntheory")," for a description of some of this. (The\n",(0,s.kt)("a",{parentName:"p",href:"https://sequencing.qcfail.com"},"sequencing.qcfail.com")," site is also a good read if you want a\nmore in-depth look at problems.) First of these analyses is the sequence duplication plot:"),(0,s.kt)("p",null,(0,s.kt)("img",{alt:"img",src:a(8155).Z,width:"1754",height:"1314"})),(0,s.kt)("p",null,"For this sample about 3% of sequences seem to be duplicated.  To interpret this, you should note that"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"fastqc is only looking at one of the reads here (not both pairs, which would be a better analysis\nto look for artifacts, because most duplication arising from amplification will be of the whole\nfragment not just one of the ends).  So it might actually be over-estimating.")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"on the other hand sequencing errors might make duplicate fragments appear not duplicated, leading to underestimation\nof the number of duplicates. For this reason fastqc focusses on the first 50bp of each read, but ",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read1_fastqc.html#M1"},"as we saw above\nthere can be low-quality bases at the start of the read as\nwell"),".")),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("p",{parentName:"li"},"Some level of 'genuine' duplication is expected in high-coverage data , of course. This is both\nbecause we are sequencing randomly-placed fragments around the genome that occasionally overlap\nby chance, and because genomes contain repetitive sequence. For more on this see the\n",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates"},"understanding sequence duplication levels")," page."))),(0,s.kt)("p",null,"The upshot of this particular plot is that there aren't worrying levels of duplicate reads in this\ndataset, although because of the low qualities at the start of the read, you might guess this is an\nunderestimate."),(0,s.kt)("p",null,(0,s.kt)("strong",{parentName:"p"},"Note.")," It is common to see much higher levels of duplication - for example this\n",(0,s.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/ERR3239334-Illumina_NovaSeq_6000_read2_fastqc.html#M8"},"sequencing of human sample NA12878"),"\nshows around a 20% duplication rate even though it used a PCR-free library preparation.\nPCR-based library preparation typically needs to still higher duplication rates."),(0,s.kt)("h3",{id:"adapter-content-and-over-represented-sequences"},"Adapter content and over-represented sequences"),(0,s.kt)("p",null,"Finally fastqc has a couple of analyses that search reads for the presence of known adapter and\nprimer sequences, or other sequences that are unusually over-represented and suggest contamination\nor chemistry issues. In brief, adapter sequences can appear in the reads ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory"},"if the fragments are too\nshort")," so that the sequencer 'reads through' into the adapters; PCR primers\ncould appear for example due to primer self-amplification during PCR; other sequences (such as\npoly-G sequences) can arise when chemistry for one of the primers fails."),(0,s.kt)("p",null,"This sample doesn't seem to have any problems here.  But see:"),(0,s.kt)("ul",null,(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("a",{parentName:"li",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read2_fastqc.html#M9"},"this malaria dataset")," with PCR primer contamination;"),(0,s.kt)("li",{parentName:"ul"},(0,s.kt)("a",{parentName:"li",href:"https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-illumina_novaseq_2_fastqc.html#M9"},"this human dataset")," which has poly-G sequences in the second read.")),(0,s.kt)("p",null,"If you see high levels of artifacts of this type, you may want to try ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming"},"read trimming")," before analysis."),(0,s.kt)("h3",{id:"next-steps-1"},"Next steps"),(0,s.kt)("p",null,"When you've read enough, move on to the section on ",(0,s.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads"},"aligning reads"),"."))}h.isMDXComponent=!0},9710:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/fastqc_basic_stats-936918dadde9e9f6d2ab81a3e230f17b.png"},8155:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/fastqc_duplication-57a97a49454b52b7b91dd6f76ae0941c.png"},2152:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/fastqc_gc_content-484e6b7afd5a3c86f25b9ebbcc832ed3.png"},9239:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/fastqc_per_base_content-55a12a5f28059ec9bdcaf46c3ca33bd6.png"},1189:(e,t,a)=>{a.d(t,{Z:()=>n});const n=a.p+"assets/images/fastqc_per_base_quality-4afa967904b1a3a534da7489ed88eb0e.png"}}]);