"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[4128],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>g});var a=n(7294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function s(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?s(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):s(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function r(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},s=Object.keys(e);for(a=0;a<s.length;a++)n=s[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var s=Object.getOwnPropertySymbols(e);for(a=0;a<s.length;a++)n=s[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var l=a.createContext({}),p=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},c=function(e){var t=p(e.components);return a.createElement(l.Provider,{value:t},e.children)},d="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},u=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,s=e.originalType,l=e.parentName,c=r(e,["components","mdxType","originalType","parentName"]),d=p(n),u=i,g=d["".concat(l,".").concat(u)]||d[u]||m[u]||s;return n?a.createElement(g,o(o({ref:t},c),{},{components:n})):a.createElement(g,o({ref:t},c))}));function g(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var s=n.length,o=new Array(s);o[0]=u;var r={};for(var l in t)hasOwnProperty.call(t,l)&&(r[l]=t[l]);r.originalType=e,r[d]="string"==typeof e?e:i,o[1]=r;for(var p=2;p<s;p++)o[p]=n[p];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},315:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>m,frontMatter:()=>s,metadata:()=>r,toc:()=>p});var a=n(7462),i=(n(7294),n(3905));const s={sidebar_position:1.5},o="A toy GWAS",r={unversionedId:"genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink",id:"genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink",title:"A toy GWAS",description:"Before we get started properly let's run a little toy example.",source:"@site/docs/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink.md",sourceDirName:"genome_wide_association_studies/genome_wide_association_analysis",slug:"/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/genome_wide_association_analysis/gwas_analysis_plink.md",tags:[],version:"current",sidebarPosition:1.5,frontMatter:{sidebar_position:1.5},sidebar:"sidebar8",previous:{title:"Plink warm-up",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/allele_frequencies"},next:{title:"Analysing a GWAS",permalink:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control"}},l={},p=[{value:"Testing for association",id:"testing-for-association",level:2},{value:"Forest plotting",id:"forest-plotting",level:2},{value:"Renaming output files",id:"renaming-output-files",level:2}],c={toc:p},d="wrapper";function m(e){let{components:t,...n}=e;return(0,i.kt)(d,(0,a.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"a-toy-gwas"},"A toy GWAS"),(0,i.kt)("p",null,"Before we get started properly let's run a little toy example."),(0,i.kt)("h2",{id:"testing-for-association"},"Testing for association"),(0,i.kt)("p",null,"A basic GWAS study begins by running regression (usually linear regression for quantitative traits,\nor logistic regression for case/control traits like the one we are using) between each genetic\nvariant in the genome and the phenotype of interest. We can run a first tiny GWAS by running:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --vcf snp-example.vcf --logistic --pheno snp-example.samples --allow-no-sex\n")),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"The ",(0,i.kt)("inlineCode",{parentName:"p"},"--allow-no-sex")," option is needed when working with VCF files in plink. This works around plink's default behaviour which\nremoves samples without a sex assignment - VCF files don't record sample sex, so this is necessary."))),(0,i.kt)("p",null,"PLINK has again generated two files in the directory we are working in: ",(0,i.kt)("inlineCode",{parentName:"p"},"plink.log")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"plink.assoc.logistic"),". The log file\nsimply captures the status information that PLINK reports with each run. The other gives the logistic regression output."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"How many cases and controls are included in this data?"))),(0,i.kt)("p",null,"If you look in the ",(0,i.kt)("inlineCode",{parentName:"p"},"plink.assoc.logistic")," file we will see that this SNP, rs8135996, is associated with our phenotype."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Q"),". What is the p-value of association?  What is the odds ratio?"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Q"),". And what does the odds ratio mean anyway? If this is a disease trait, is it good news to\nhave the 'G' allele? Or to have the 'A' allele? And how bad is it to have the risk allele?"),(0,i.kt)("p",{parentName:"div"},(0,i.kt)("strong",{parentName:"p"},"Advanced Question"),". What does ",(0,i.kt)("inlineCode",{parentName:"p"},"ADD")," mean in the output? Can you figure out how to run a\nnon-additive test? Which mode of inheritance has the greatest evidence?"))),(0,i.kt)("h2",{id:"forest-plotting"},"Forest plotting"),(0,i.kt)("p",null,"If you followed other parts of our training on statistical modelling, we argued that what you want is to summarise the\nlikelihood by its ",(0,i.kt)("em",{parentName:"p"},"maximum likelihood estimate")," and its ",(0,i.kt)("em",{parentName:"p"},"standard error")," - thus approximating the likelihood function by a\ngaussian function. For logistic regression the appropriate values are the ",(0,i.kt)("em",{parentName:"p"},"log odds ratio")," and its standard error."),(0,i.kt)("p",null,"By default plink hasn't given us the standard error so let's compute it now.\nFirst let's get plink to give us the logistic regression estimate - the ",(0,i.kt)("em",{parentName:"p"},"log odds ratio")," instead of the odds ratio.\nReading the ",(0,i.kt)("a",{parentName:"p",href:"https://www.cog-genomics.org/plink/1.9/assoc#linear"},"documentation for\n--logistic"),' you\'ll see there is an option called "beta" to the --logistic\ncommand that does this.  '),(0,i.kt)("blockquote",null,(0,i.kt)("p",{parentName:"blockquote"},"\"For logistic regressions, the 'beta' modifier causes regression coefficients instead of odds ratios to be reported.\"")),(0,i.kt)("p",null,"Finally there is also a ",(0,i.kt)("inlineCode",{parentName:"p"},"--ci")," option that computes a confidence interval.  Let's apply those now:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink --vcf snp-example.vcf --logistic beta --ci 0.95 --pheno snp-example.samples --allow-no-sex\n")),(0,i.kt)("p",null,"Load this output file into R now:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-R"},'# in R\nX = read_table( "plink.assoc.logistic" )\n')),(0,i.kt)("p",null,"If you look at the result, you'll see it has output a ",(0,i.kt)("inlineCode",{parentName:"p"},"BETA")," column instad of the original ",(0,i.kt)("inlineCode",{parentName:"p"},"OR"),", and also has ",(0,i.kt)("inlineCode",{parentName:"p"},"SE")," (standard\nerror) and confidence interval columns.  Let's plot that now:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-r"},'(\n    ggplot( data = X )\n    + geom_point( aes( x = BETA, y = SNP ))\n    + geom_segment( aes( x = BETA - 1.96 * SE, xend = BETA + 1.96 * SE, y = SNP, yend = SNP ))\n    + geom_vline( xintercept = 0, linetype = 2 )\n    + xlab( "Log odds-ratio" )\n    + ylab( "SNP" )\n    + theme_minimal( 16 )\n    + theme( axis.title.y = element_text( angle = 0, vjust = 0.5 ))\n)\n')),(0,i.kt)("p",null,"or in base R:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'plot.betas <- function( betas, ses ) {\n    plot(\n        betas, 1:length(betas),\n        pch = 19, # make filled dots\n        main = "rs8135996 forest plot",\n        xlab = "log odds ratio",\n        ylab = "",\n        xlim = c( -0.5, 0 ),\n        yaxt = \'n\' # Turn off y axis ticks\n    )\n    segments(\n        x0 = betas - 1.96 * ses,\n        x1 = betas + 1.96 * ses,\n        y0 = 1, y1 = 1,\n        col = \'grey\'\n    )\n    abline( v = 0 ) # draw a solid black line at 0\n    grid()\n}\n\nplot.betas( X$BETA, X$SE )\n')),(0,i.kt)("p",null,"Congratulations! This is your first GWAS forest plot (albeit with only one row)."),(0,i.kt)("h2",{id:"renaming-output-files"},"Renaming output files"),(0,i.kt)("p",null,"Running ",(0,i.kt)("inlineCode",{parentName:"p"},"plink --logistic")," is all very well, but it's a bit annoying to have files named simply ",(0,i.kt)("inlineCode",{parentName:"p"},"plink.logistic.assoc"),"\nand so on.  Instead lets re-run plink telling it to rename its output files (important since we\u2019ll be generating lots of\nthem in the practicals).  The ",(0,i.kt)("inlineCode",{parentName:"p"},"--out")," option can be used to do this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"./plink \\\n--vcf snp-example.vcf \\\n--logistic beta \\\n--beta \\\n--ci 0.95 \\\n--pheno snp-example.samples \\\n--allow-no-sex \\\n--out getting-started\n")),(0,i.kt)("p",null,"Now the output files will be named ",(0,i.kt)("inlineCode",{parentName:"p"},"getting-started.log")," and ",(0,i.kt)("inlineCode",{parentName:"p"},"getting-started.assoc.logistic"),". This is the basic pattern\nto working with PLINK: specifying input files and analyses, along with an output name to save results."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"The backslashes in the above command are shell line continuation characters - they let us put the command on multiple lines.\nSome of these commands will get pretty long, so it's useful to put them on multiple lines like this!"))),(0,i.kt)("p",null,"With this background in using plink - it's time to go on to ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/gwas_quality_control"},"analysing a simulated GWAS study"),"."))}m.isMDXComponent=!0}}]);