"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[8214],{3905:(e,t,a)=>{a.d(t,{Zo:()=>l,kt:()=>d});var n=a(7294);function s(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function r(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){s(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function o(e,t){if(null==e)return{};var a,n,s=function(e,t){if(null==e)return{};var a,n,s={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(s[a]=e[a]);return s}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(s[a]=e[a])}return s}var p=n.createContext({}),m=function(e){var t=n.useContext(p),a=t;return e&&(a="function"==typeof e?e(t):r(r({},t),e)),a},l=function(e){var t=m(e.components);return n.createElement(p.Provider,{value:t},e.children)},c="mdxType",g={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},h=n.forwardRef((function(e,t){var a=e.components,s=e.mdxType,i=e.originalType,p=e.parentName,l=o(e,["components","mdxType","originalType","parentName"]),c=m(a),h=s,d=c["".concat(p,".").concat(h)]||c[h]||g[h]||i;return a?n.createElement(d,r(r({ref:t},l),{},{components:a})):n.createElement(d,r({ref:t},l))}));function d(e,t){var a=arguments,s=t&&t.mdxType;if("string"==typeof e||s){var i=a.length,r=new Array(i);r[0]=h;var o={};for(var p in t)hasOwnProperty.call(t,p)&&(o[p]=t[p]);o.originalType=e,o[c]="string"==typeof e?e:s,r[1]=o;for(var m=2;m<i;m++)r[m]=a[m];return n.createElement.apply(null,r)}return n.createElement.apply(null,a)}h.displayName="MDXCreateElement"},3152:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>p,contentTitle:()=>r,default:()=>g,frontMatter:()=>i,metadata:()=>o,toc:()=>m});var n=a(7462),s=(a(7294),a(3905));const i={sidebar_position:8},r="Investigating specific genes",o={unversionedId:"bioinformatics/exploring_gene_annotations_in_bash/investigating",id:"bioinformatics/exploring_gene_annotations_in_bash/investigating",title:"Investigating specific genes",description:"Let's switch track and try to dig out info about a specific gene -",source:"@site/docs/bioinformatics/exploring_gene_annotations_in_bash/investigating.md",sourceDirName:"bioinformatics/exploring_gene_annotations_in_bash",slug:"/bioinformatics/exploring_gene_annotations_in_bash/investigating",permalink:"/whg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/investigating",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/bioinformatics/exploring_gene_annotations_in_bash/investigating.md",tags:[],version:"current",sidebarPosition:8,frontMatter:{sidebar_position:8},sidebar:"sidebar2",previous:{title:"Counting genes again",permalink:"/whg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/counting2"},next:{title:"Browsing the genome",permalink:"/whg-training-resources/bioinformatics/exploring_gene_annotations_in_bash/browsing"}},p={},m=[{value:"Finding the gene",id:"finding-the-gene",level:2},{value:"Finding transcripts",id:"finding-transcripts",level:2},{value:"Finding exons",id:"finding-exons",level:2}],l={toc:m},c="wrapper";function g(e){let{components:t,...a}=e;return(0,s.kt)(c,(0,n.Z)({},l,a,{components:t,mdxType:"MDXLayout"}),(0,s.kt)("h1",{id:"investigating-specific-genes"},"Investigating specific genes"),(0,s.kt)("p",null,"Let's switch track and try to dig out info about a specific gene -\n",(0,s.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/FUT2"},(0,s.kt)("em",{parentName:"a"},"FUT2")),". That's an interesting gene because it encodes a\nfucosyltransferase which is involved the synthesis pathway for 'soluble' A and B antigens - that is\nA and B antigens found in bodily fluids. Quoth\n",(0,s.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Secretor_status"},"wikipedia"),":"),(0,s.kt)("p",null,(0,s.kt)("em",{parentName:"p"},'"Secretor status refers to the presence or absence of water-soluble ABO blood group antigens in a person\'s bodily fluids, such as saliva, tears, breast milk, urine, and semen. People who secrete these antigens in their bodily fluids are referred to as secretors, while people who do not are termed non-secretors. Secretor status is controlled by the FUT2 gene (also called the Se gene), and the secretor phenotype is inherited in an autosomal dominant manner."')," - Wikipedia"),(0,s.kt)("p",null,"A specific mutation (rs601338 G>A, at position 48703417 on GRCh38 assembly) in ",(0,s.kt)("em",{parentName:"p"},"FUT2")," is the major determinant of\nsecretor status. Because pathogens including norovirus bind to these secreted antigens, these mutations can confer\nprotection against norovirus."),(0,s.kt)("p",null,"Let's look at ",(0,s.kt)("em",{parentName:"p"},"FUT2")," in our annotation file now."),(0,s.kt)("h2",{id:"finding-the-gene"},"Finding the gene"),(0,s.kt)("p",null,"A simple way to look this up is to just to grep (i.e. conduct a text search) for ",(0,s.kt)("inlineCode",{parentName:"p"},"FUT2"),":"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"grep FUT2 gencode.v41.annotation.gff3 | less -S\n")),(0,s.kt)("p",null,"Unfortunately that returns a lot of rows - let's just get genes:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"grep FUT2 gencode.v41.annotation.gff3 | awk '$3 == \"gene\"' | less -S\n")),(0,s.kt)("p",null,"Ok, this returns two records. If you look at the ",(0,s.kt)("inlineCode",{parentName:"p"},"gene_name")," attribute you'll see one, on chromosome 19, is\n",(0,s.kt)("em",{parentName:"p"},"FUT2"),", while the other is a different gene called ",(0,s.kt)("inlineCode",{parentName:"p"},"POFUT2"),". Let's use that to do a bit better:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"grep 'gene_name=FUT2' gencode.v41.annotation.gff3 | awk '$3 == \"gene\"' | less -S\n")),(0,s.kt)("p",null,"We got it!  Copy its ID to the clipboard - in my file it is ",(0,s.kt)("inlineCode",{parentName:"p"},"ENSG00000176920.13"),"."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Questions")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("ul",{parentName:"div"},(0,s.kt)("li",{parentName:"ul"},"How long is FUT2 on the chromosome?")),(0,s.kt)("p",{parentName:"div"},(0,s.kt)("strong",{parentName:"p"},"Note.")," to get the answer 100% right, you actually have to take the formula"),(0,s.kt)("div",{parentName:"div",className:"math math-display"},(0,s.kt)("span",{parentName:"div",className:"katex-display"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML",display:"block"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mtext",{parentName:"mrow"},"end\xa0coordinate"),(0,s.kt)("mo",{parentName:"mrow"},"\u2212"),(0,s.kt)("mtext",{parentName:"mrow"},"start\xa0coordinate"),(0,s.kt)("mo",{parentName:"mrow"},"+"),(0,s.kt)("mn",{parentName:"mrow"},"1")),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\text{end coordinate} - \\text{start coordinate} + 1")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7778em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"end\xa0coordinate")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"\u2212"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.7778em",verticalAlign:"-0.0833em"}}),(0,s.kt)("span",{parentName:"span",className:"mord text"},(0,s.kt)("span",{parentName:"span",className:"mord"},"start\xa0coordinate")),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}}),(0,s.kt)("span",{parentName:"span",className:"mbin"},"+"),(0,s.kt)("span",{parentName:"span",className:"mspace",style:{marginRight:"0.2222em"}})),(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"0.6444em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},"1")))))),(0,s.kt)("p",{parentName:"div"},"This is because both start and end are expressed in a ",(0,s.kt)("em",{parentName:"p"},"1-based, closed coordinate system")," i.e. they both point\nat bases included in the gene.  (Think of a gene with only two bases in it to see why this is.)"))),(0,s.kt)("h2",{id:"finding-transcripts"},"Finding transcripts"),(0,s.kt)("p",null,"So how many transcripts does ",(0,s.kt)("em",{parentName:"p"},"FUT2")," have? Well we know how to do this - look for ",(0,s.kt)("em",{parentName:"p"},"transcript")," records with the\n",(0,s.kt)("em",{parentName:"p"},"FUT2")," gene as parent:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"grep 'Parent=ENSG00000176920.13' gencode.v41.annotation.gff3 | awk '$3 == \"transcript\"' | less -S\n")),(0,s.kt)("p",null,"So it has 4 transcripts - that is, the file suggests the gene may be transcribed to mRNA in 4 different ways.\nScroll around a bit to look at the attributes of these transcripts.  If you look closely you'll see there is some more information in there.\nFor example a ",(0,s.kt)("strong",{parentName:"p"},"transcript support\nlevel")," which reflects how confident GENCODE is about the transcript. See\n",(0,s.kt)("a",{parentName:"p",href:"https://www.ensembl.org/info/genome/genebuild/transcript_quality_tags.html"},"the Ensembl page"),"\nfor a description of these."),(0,s.kt)("p",null,'One of these transcripts (ENST00000425340.3) is also marked as \u2018Ensemble canonical\u2019 which means\n"',(0,s.kt)("a",{parentName:"p",href:"https://www.ensembl.org/info/genome/genebuild/canonical.html"},"a single, representative transcript identified at every locus"),"\".\nSo let's focus on that transcript and dig a bit deeper"),(0,s.kt)("h2",{id:"finding-exons"},"Finding exons"),(0,s.kt)("p",null,"This is easy now:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre",className:"language-sh"},"grep 'Parent=ENST00000425340.3' gencode.v41.annotation.gff3 | awk '$3 == \"exon\"' | less -S\n")),(0,s.kt)("p",null,"Aha, it has two exons."),(0,s.kt)("p",null,"So, how long are these exons?  To make that easier let's use ",(0,s.kt)("inlineCode",{parentName:"p"},"cut")," to get rid of the noise:"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"grep 'Parent=ENST00000425340.3' gencode.v41.annotation.gff3 | awk '$3 == \"exon\"' | cut -f1,3-5\n")),(0,s.kt)("p",null,"Adding that up, the two exons have length 119 and 2,997 - so only about 30% of the gene is actually transcribed into RNA!"),(0,s.kt)("p",null,"What about the bit that codes for protein? We can find that by looking for the ",(0,s.kt)("em",{parentName:"p"},"coding sequence")," records - they\nhave ",(0,s.kt)("inlineCode",{parentName:"p"},"type=CDS"),":"),(0,s.kt)("pre",null,(0,s.kt)("code",{parentName:"pre"},"grep 'Parent=ENST00000425340.3' gencode.v41.annotation.gff3 | awk '$3 == \"CDS\"' | cut -f1,3-5\n")),(0,s.kt)("p",null,"If you look at this you'll see the gene has one annotated coding sequence, and it lives entirely inside the\nsecond exon. Its length is 1032 base pairs. So ",(0,s.kt)("strong",{parentName:"p"},"only about ",(0,s.kt)("span",{parentName:"strong",className:"math math-inline"},(0,s.kt)("span",{parentName:"span",className:"katex"},(0,s.kt)("span",{parentName:"span",className:"katex-mathml"},(0,s.kt)("math",{parentName:"span",xmlns:"http://www.w3.org/1998/Math/MathML"},(0,s.kt)("semantics",{parentName:"math"},(0,s.kt)("mrow",{parentName:"semantics"},(0,s.kt)("mfrac",{parentName:"mrow"},(0,s.kt)("mn",{parentName:"mfrac"},"1"),(0,s.kt)("mn",{parentName:"mfrac"},"10"))),(0,s.kt)("annotation",{parentName:"semantics",encoding:"application/x-tex"},"\\tfrac{1}{10}")))),(0,s.kt)("span",{parentName:"span",className:"katex-html","aria-hidden":"true"},(0,s.kt)("span",{parentName:"span",className:"base"},(0,s.kt)("span",{parentName:"span",className:"strut",style:{height:"1.1901em",verticalAlign:"-0.345em"}}),(0,s.kt)("span",{parentName:"span",className:"mord"},(0,s.kt)("span",{parentName:"span",className:"mopen nulldelimiter"}),(0,s.kt)("span",{parentName:"span",className:"mfrac"},(0,s.kt)("span",{parentName:"span",className:"vlist-t vlist-t2"},(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.8451em"}},(0,s.kt)("span",{parentName:"span",style:{top:"-2.655em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"10")))),(0,s.kt)("span",{parentName:"span",style:{top:"-3.23em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"frac-line",style:{borderBottomWidth:"0.04em"}})),(0,s.kt)("span",{parentName:"span",style:{top:"-3.394em"}},(0,s.kt)("span",{parentName:"span",className:"pstrut",style:{height:"3em"}}),(0,s.kt)("span",{parentName:"span",className:"sizing reset-size6 size3 mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},(0,s.kt)("span",{parentName:"span",className:"mord mtight"},"1"))))),(0,s.kt)("span",{parentName:"span",className:"vlist-s"},"\u200b")),(0,s.kt)("span",{parentName:"span",className:"vlist-r"},(0,s.kt)("span",{parentName:"span",className:"vlist",style:{height:"0.345em"}},(0,s.kt)("span",{parentName:"span"}))))),(0,s.kt)("span",{parentName:"span",className:"mclose nulldelimiter"})))))),"th of the gene codes for protein"),"."),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"If we've got this right then the nucleotide length of the coding sequence should be a multiple of something - what?\nIs 1032 an appropriate multiple?"))),(0,s.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,s.kt)("div",{parentName:"div",className:"admonition-heading"},(0,s.kt)("h5",{parentName:"div"},(0,s.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,s.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,s.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Challenge question")),(0,s.kt)("div",{parentName:"div",className:"admonition-content"},(0,s.kt)("p",{parentName:"div"},"Now repeat the above process for another gene and see if things look similar. For example, try the genes that encode\n",(0,s.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Hemoglobin_subunit_alpha"},"alpha globin"),", named ",(0,s.kt)("inlineCode",{parentName:"p"},"HBA1")," and ",(0,s.kt)("inlineCode",{parentName:"p"},"HBA2"),"."))))}g.isMDXComponent=!0}}]);