"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[3292],{3905:(e,t,a)=>{a.d(t,{Zo:()=>d,kt:()=>h});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function s(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var l=n.createContext({}),p=function(e){var t=n.useContext(l),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},d=function(e){var t=p(e.components);return n.createElement(l.Provider,{value:t},e.children)},u="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},m=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,l=e.parentName,d=s(e,["components","mdxType","originalType","parentName"]),u=p(a),m=r,h=u["".concat(l,".").concat(m)]||u[m]||c[m]||i;return a?n.createElement(h,o(o({ref:t},d),{},{components:a})):n.createElement(h,o({ref:t},d))}));function h(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,o=new Array(i);o[0]=m;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[u]="string"==typeof e?e:r,o[1]=s;for(var p=2;p<i;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}m.displayName="MDXCreateElement"},7137:(e,t,a)=>{a.r(t),a.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>c,frontMatter:()=>i,metadata:()=>s,toc:()=>p});var n=a(7462),r=(a(7294),a(3905));const i={sidebar_position:3},o="Exploring Reads in IGV",s={unversionedId:"sequence_data_analysis/IGV/README",id:"sequence_data_analysis/IGV/README",title:"Exploring Reads in IGV",description:"Contributors:  Gavin Band, Annie Forster",source:"@site/docs/sequence_data_analysis/IGV/README.md",sourceDirName:"sequence_data_analysis/IGV",slug:"/sequence_data_analysis/IGV/",permalink:"/whg-training-resources/sequence_data_analysis/IGV/",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/IGV/README.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{sidebar_position:3},sidebar:"sidebar9",previous:{title:"Useful links",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links"},next:{title:"Viewing reads",permalink:"/whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV"}},l={},p=[{value:"Getting the data",id:"getting-the-data",level:2},{value:"Loading data",id:"loading-data",level:2}],d={toc:p},u="wrapper";function c(e){let{components:t,...a}=e;return(0,r.kt)(u,(0,n.Z)({},d,a,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"exploring-reads-in-igv"},"Exploring Reads in IGV"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Contributors"),":  Gavin Band, Annie Forster"),(0,r.kt)("h2",{id:"getting-the-data"},"Getting the data"),(0,r.kt)("p",null,"In this tutorial you are going to be inspecting read alignments in a program called ",(0,r.kt)("a",{parentName:"p",href:"https://software.broadinstitute.org/software/igv"},"Integrative Genomics\nViewer"),", or IGV for short. IGV is a very handy piece of software which allows you to visually inspect alignments across the genome. "),(0,r.kt)("p",null,"First you will need to download IGV. You can find the software\n",(0,r.kt)("a",{parentName:"p",href:"https://software.broadinstitute.org/software/igv/download"},"here")," - please download and install it now."),(0,r.kt)("p",null,"For this tutorial we will use sequencing data from ",(0,r.kt)("em",{parentName:"p"},"P.falciparum")," malaria parasites. If you completed the earlier\ntutorial ",(0,r.kt)("a",{parentName:"p",href:"../introduction_to_next_generation_sequencing_data_analysis/"},"on paired-end sequencing"),", or the ",(0,r.kt)("a",{parentName:"p",href:"../building_an_ngs_pipeline/"},"pipelining\ntutorial")," you will already have some aligned BAM files which you can use for this.  For this tutorial you should load (at least) these files:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("inlineCode",{parentName:"li"},"QG0033-C.bam")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("inlineCode",{parentName:"li"},"QG0041-C.bam"))),(0,r.kt)("p",null,"You are also welcome to load and look at others as well."),(0,r.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,r.kt)("div",{parentName:"div",className:"admonition-heading"},(0,r.kt)("h5",{parentName:"div"},(0,r.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,r.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,r.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Note")),(0,r.kt)("div",{parentName:"div",className:"admonition-content"},(0,r.kt)("p",{parentName:"div"},"Alternatively, my versions of the aligned files can be found ",(0,r.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/whg-training-resources/data/sequence_data_analysis/building_an_ngs_pipeline/results/aligned"},"in this\nfolder"),"."),(0,r.kt)("p",{parentName:"div"},"An easy way to load these into IGV is to use the ",(0,r.kt)("inlineCode",{parentName:"p"},"File -> Load from URL")," option.  Then paste in the link to the BAM file."),(0,r.kt)("p",{parentName:"div"},(0,r.kt)("strong",{parentName:"p"},"Alternatively")," you can download the ",(0,r.kt)("inlineCode",{parentName:"p"},".bam")," and ",(0,r.kt)("inlineCode",{parentName:"p"},".bai")," files to your computer first."))),(0,r.kt)("p",null,"If you followed the ",(0,r.kt)("a",{parentName:"p",href:"../building_an_ngs_pipeline"},"snakemake tutorial"),", you should also be able to look at:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Your coverage (",(0,r.kt)("inlineCode",{parentName:"p"},".bedgraph"),") files for the above samples. (Although this isn't very important because IGV shows you coverage anyway.)  You can find them in ",(0,r.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/whg-training-resources/data/sequence_data_analysis/building_an_ngs_pipeline/results/coverage"},"this folder")," if you don't have them already.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Your ",(0,r.kt)("inlineCode",{parentName:"p"},"variants.vcf.gz")," file containing the variant calls from Octopus. (If yours doesn't work, this may be due to an Octopus issue that has now been fixed. I filtered out the problem variants in ",(0,r.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/whg-training-resources/data/sequence_data_analysis/building_an_ngs_pipeline/results/variants"},"my\nversion")," which you can try instead.)"))),(0,r.kt)("h2",{id:"loading-data"},"Loading data"),(0,r.kt)("p",null,"To load these files into IGV, choose ",(0,r.kt)("inlineCode",{parentName:"p"},"File -> Load from File...")," from the menu. You should be prompted to download the appropriate reference; \u201cPfalciparum3D7\u201d should already be available from the provided selection.  (If you're not prompted, make sure you are using the ",(0,r.kt)("inlineCode",{parentName:"p"},"Pfalciparum3D7 (v9.0)")," genome. This should appear in the drop-down box at the top left of the window. If not, click on the dropdown and choose ",(0,r.kt)("inlineCode",{parentName:"p"},"...")," and then select ",(0,r.kt)("inlineCode",{parentName:"p"},"P. falciparum 3D7 (V9.0)")," from the list)."),(0,r.kt)("p",null,"Load in the BAM files for ",(0,r.kt)("inlineCode",{parentName:"p"},"QG0033-C.bam")," and ",(0,r.kt)("inlineCode",{parentName:"p"},"QG0041-C.bam")," now, as well as the ",(0,r.kt)("inlineCode",{parentName:"p"},"variants.vcf.gz")," file if you have it.  Once the data is loaded you're ready to go and ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV"},"view the reads"),"."))}c.isMDXComponent=!0}}]);