"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[9418],{3905:(e,a,t)=>{t.d(a,{Zo:()=>p,kt:()=>g});var n=t(7294);function i(e,a,t){return a in e?Object.defineProperty(e,a,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[a]=t,e}function r(e,a){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);a&&(n=n.filter((function(a){return Object.getOwnPropertyDescriptor(e,a).enumerable}))),t.push.apply(t,n)}return t}function s(e){for(var a=1;a<arguments.length;a++){var t=null!=arguments[a]?arguments[a]:{};a%2?r(Object(t),!0).forEach((function(a){i(e,a,t[a])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):r(Object(t)).forEach((function(a){Object.defineProperty(e,a,Object.getOwnPropertyDescriptor(t,a))}))}return e}function o(e,a){if(null==e)return{};var t,n,i=function(e,a){if(null==e)return{};var t,n,i={},r=Object.keys(e);for(n=0;n<r.length;n++)t=r[n],a.indexOf(t)>=0||(i[t]=e[t]);return i}(e,a);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(n=0;n<r.length;n++)t=r[n],a.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(i[t]=e[t])}return i}var l=n.createContext({}),c=function(e){var a=n.useContext(l),t=a;return e&&(t="function"==typeof e?e(a):s(s({},a),e)),t},p=function(e){var a=c(e.components);return n.createElement(l.Provider,{value:a},e.children)},u="mdxType",m={inlineCode:"code",wrapper:function(e){var a=e.children;return n.createElement(n.Fragment,{},a)}},d=n.forwardRef((function(e,a){var t=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,p=o(e,["components","mdxType","originalType","parentName"]),u=c(t),d=i,g=u["".concat(l,".").concat(d)]||u[d]||m[d]||r;return t?n.createElement(g,s(s({ref:a},p),{},{components:t})):n.createElement(g,s({ref:a},p))}));function g(e,a){var t=arguments,i=a&&a.mdxType;if("string"==typeof e||i){var r=t.length,s=new Array(r);s[0]=d;var o={};for(var l in a)hasOwnProperty.call(a,l)&&(o[l]=a[l]);o.originalType=e,o[u]="string"==typeof e?e:i,s[1]=o;for(var c=2;c<r;c++)s[c]=t[c];return n.createElement.apply(null,s)}return n.createElement.apply(null,t)}d.displayName="MDXCreateElement"},4314:(e,a,t)=>{t.r(a),t.d(a,{assets:()=>l,contentTitle:()=>s,default:()=>m,frontMatter:()=>r,metadata:()=>o,toc:()=>c});var n=t(7462),i=(t(7294),t(3905));const r={sidebar_position:6},s="Variant calling, phasing and imputation",o={unversionedId:"sequence_data_analysis/variant_calling_and_imputation/README",id:"sequence_data_analysis/variant_calling_and_imputation/README",title:"Variant calling, phasing and imputation",description:"This morning we focussed on quality control, aligning, and inspecting some sequence data for a single sample.",source:"@site/docs/sequence_data_analysis/variant_calling_and_imputation/README.md",sourceDirName:"sequence_data_analysis/variant_calling_and_imputation",slug:"/sequence_data_analysis/variant_calling_and_imputation/",permalink:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/variant_calling_and_imputation/README.md",tags:[],version:"current",sidebarPosition:6,frontMatter:{sidebar_position:6},sidebar:"sidebar9",previous:{title:"Tips and tricks",permalink:"/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/tips_and_tricks"},next:{title:"Getting the data",permalink:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Prerequisites"}},l={},c=[{value:"Steps in the practical",id:"steps-in-the-practical",level:2}],p={toc:c},u="wrapper";function m(e){let{components:a,...t}=e;return(0,i.kt)(u,(0,n.Z)({},p,t,{components:a,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"variant-calling-phasing-and-imputation"},"Variant calling, phasing and imputation"),(0,i.kt)("p",null,"This morning we focussed on quality control, aligning, and inspecting some sequence data for a single sample."),(0,i.kt)("p",null,"In this afternoon's session we're going to focus on using a set of sequenced samples to do a few things:"),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},"identify genetic variants in a region (and work out the sample genotypes at those variants)"),(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("em",{parentName:"li"},"phase")," the genotypes so we can see how they align along haplotypes"),(0,i.kt)("li",{parentName:"ol"},"and use this to impute variants from a second dataset (for which we only have microarray data).")),(0,i.kt)("p",null,"Not every possible genetic variant is a real one and to make this work well we will need to do some\nmore ",(0,i.kt)("strong",{parentName:"p"},"quality control"),". This time we'll take care to filter the set of variants based on sensible\nmetrics before we use the data."),(0,i.kt)("p",null,"All the data in this practical comes from the ",(0,i.kt)("a",{parentName:"p",href:"https://www.internationalgenome.org"},"IGSR"),", which\nlists a huge number of open-access datasets that you can use in your analysis (including the 1000\nGenomes Project data)."),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Note.")," Before doing anything else, please make sure you have ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Prerequisites"},"downloaded the data"),".\nThen come back here."),(0,i.kt)("h2",{id:"steps-in-the-practical"},"Steps in the practical"),(0,i.kt)("p",null,"The practical has four main steps:"),(0,i.kt)("ol",null,(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("p",{parentName:"li"},"The first step is to ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Variant_calling"},"generate a VCF file of variant calls")," from some 1000 Genomes Project samples.")),(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("p",{parentName:"li"},"We'll then perform ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Variant_quality_control"},"quality control on the initial variant calls"),".")),(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("p",{parentName:"li"},"Next we will ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Phasing"},"phase the calls")," so we know how they stack up on haplotypes.")),(0,i.kt)("li",{parentName:"ol"},(0,i.kt)("p",{parentName:"li"},"And then we'll ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Imputation"},"use these haplotypes to impute some microarray data")))),(0,i.kt)("p",null,"Finally we have - you guessed it - some ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Challenge_questions"},"challenge question"),".  Good luck!"))}m.isMDXComponent=!0}}]);