"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[1097],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>d});var a=n(7294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},r=Object.keys(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var l=a.createContext({}),p=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},c=function(e){var t=p(e.components);return a.createElement(l.Provider,{value:t},e.children)},u="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},h=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,c=s(e,["components","mdxType","originalType","parentName"]),u=p(n),h=i,d=u["".concat(l,".").concat(h)]||u[h]||m[h]||r;return n?a.createElement(d,o(o({ref:t},c),{},{components:n})):a.createElement(d,o({ref:t},c))}));function d(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=n.length,o=new Array(r);o[0]=h;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[u]="string"==typeof e?e:i,o[1]=s;for(var p=2;p<r;p++)o[p]=n[p];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}h.displayName="MDXCreateElement"},3157:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>m,frontMatter:()=>r,metadata:()=>s,toc:()=>p});var a=n(7462),i=(n(7294),n(3905));const r={sidebar_position:6},o="Imputing a set of microarray data",s={unversionedId:"sequence_data_analysis/variant_calling_and_imputation/Imputation",id:"sequence_data_analysis/variant_calling_and_imputation/Imputation",title:"Imputing a set of microarray data",description:"Ok, so we've called genetic variants, we've [QCd",source:"@site/docs/sequence_data_analysis/variant_calling_and_imputation/Imputation.md",sourceDirName:"sequence_data_analysis/variant_calling_and_imputation",slug:"/sequence_data_analysis/variant_calling_and_imputation/Imputation",permalink:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Imputation",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/variant_calling_and_imputation/Imputation.md",tags:[],version:"current",sidebarPosition:6,frontMatter:{sidebar_position:6},sidebar:"sidebar9",previous:{title:"Phasing the variant calls",permalink:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Phasing"},next:{title:"Challenge questions on imputation",permalink:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Challenge_questions"}},l={},p=[{value:"What did imputation do?",id:"what-did-imputation-do",level:3},{value:"Inspecting the secretor status",id:"inspecting-the-secretor-status",level:3},{value:"Next steps",id:"next-steps",level:3}],c={toc:p},u="wrapper";function m(e){let{components:t,...n}=e;return(0,i.kt)(u,(0,a.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"imputing-a-set-of-microarray-data"},"Imputing a set of microarray data"),(0,i.kt)("p",null,"Ok, so we've ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Variant_calling"},"called genetic variants"),", we've ",(0,i.kt)("a",{parentName:"p",href:"Variant_quality_controls.md"},"QCd\nthem"),", and we've ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/Phasing"},"phased them into haplotypes"),". What can\nwe do with that?"),(0,i.kt)("p",null,"One of the important ways that genetic variation reference panels is used is to ",(0,i.kt)("strong",{parentName:"p"},"impute"),"\ngenotypes in other sample sets that (for reasons of cost or practicality) haven't been sequenced.\nThis is the basic paradigm, for example, for most analyses of the\n",(0,i.kt)("a",{parentName:"p",href:"https://www.nature.com/articles/s41586-018-0579-z"},"genetic resource in the UK Biobank"),"."),(0,i.kt)("p",null,"Imputation will be covered in more detail in another session, but the basic idea is that - even if\nwe haven't genotyped a particular variant in our sample set - we have genotyped variants on the\nsame haplotype. So by statistically matching haplotypes between our genotyped set and the reference\npanel we've created using sequencing we might be able to infer their genotypes, even at untyped\nmarkers. (This process relies on the interesting structure that human haplotypes have, in which\nvariants along the genome become correlated due to genetic drift, but these patterns are broken\ndown by recombination.)"),(0,i.kt)("p",null,"Imputation from very large reference panels can now be done easily online using for example the\n",(0,i.kt)("a",{parentName:"p",href:"https://imputationserver.sph.umich.edu/index.html"},"Michigan Imputation Server"),"\nor the ",(0,i.kt)("a",{parentName:"p",href:"https://imputation.sanger.ac.uk"},"Sanger imputation service"),".  But here we want to demonstrate how it's done."),(0,i.kt)("p",null,"To this end we've placed a dataset from microarray-based genotyping of a second set of Gambian\nsamples in the file: ",(0,i.kt)("inlineCode",{parentName:"p"},"GGVP/omni2.5M/GGVP-illumina_omni2.5M.phased.vcf.gz"),". (They come\nfrom the\n",(0,i.kt)("a",{parentName:"p",href:"https://www.internationalgenome.org/gambian-genome-variation-project/"},"Gambian Genome Variation Project"),"\nwhich is why we've called them ",(0,i.kt)("inlineCode",{parentName:"p"},"GGVP"),".)"),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Note.")," We have ",(0,i.kt)("strong",{parentName:"p"},"already QCd and phased these genotypes")," for you - in a real analysis you might\nhave to do that yourself."),(0,i.kt)("p",null,"Let's see if we can impute the secretor status SNP.  You can check that it is not genotype in the input data:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"bcftools view -H -i 'POS=48703417' GGVP/omni2.5M/GGVP-illumina_omni2.5M.phased.vcf.gz | wc -l\n")),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Note.")," recall that ",(0,i.kt)("inlineCode",{parentName:"p"},"wc -l")," counts the number of lines in its output."),(0,i.kt)("p",null,"To use imputation we will use the program ",(0,i.kt)("inlineCode",{parentName:"p"},"MINIMAC"),". The first step is to convert our reference\npanel:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'Minimac3 \\\n  --refHaps GWD_30x_calls.phased.vcf.gz \\\n  --processReference \\\n  --prefix GWD_30x_calls.phased \\\n  --chr "chr19"\n')),(0,i.kt)("p",null,"This should output a transformed file ",(0,i.kt)("inlineCode",{parentName:"p"},"GWD_30x_calls.phased.m3vcf.gz"),"."),(0,i.kt)("p",null,"Now use it to impute (note we'll use minimac version 4 for this):"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},'minimac4 \\\n  --refHaps "GWD_30x_calls.phased.m3vcf.gz" \\\n  --mapFile genetic_map/genetic_map_hg38_withX.txt.gz \\\n  --haps GGVP/omni2.5M/GGVP-illumina_omni2.5M.phased.vcf.gz \\\n  --ignoreDuplicates \\\n  --format GT \\\n  --prefix GGVP-illumina_omni2.5M.imputed\n')),(0,i.kt)("h3",{id:"what-did-imputation-do"},"What did imputation do?"),(0,i.kt)("p",null,"What's happened there?  Well let's look at how many variants were in the files."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"#Number of variants in the microarray data:\nbcftools view -H GGVP/omni2.5M/GGVP-illumina_omni2.5M.phased.vcf.gz | wc -l\n\n# Number of variants in the reference panel\nbcftools view -H GWD_30x_calls.phased.vcf.gz | wc -l\n\n# Number of variants in the imputed data\nbcftools view -H GGVP-illumina_omni2.5M.imputed.dose.vcf.gz | wc -l\n")),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Question.")," What do these numbers mean?"),(0,i.kt)("h3",{id:"inspecting-the-secretor-status"},"Inspecting the secretor status"),(0,i.kt)("p",null,"Let's have a look at the secretor status SNP in the output:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"bcftools view -H -i 'POS=48703417' GGVP-illumina_omni2.5M.imputed.dose.vcf.gz\n")),(0,i.kt)("p",null,"You should see something like this:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre"},"chr19   48703417    chr19:48703417:G:A  G   A   .   PASS    AF=0.53395;MAF=0.46605;R2=0.99822;IMPUTED   GT  0|0 1|0 1|0 1|1 0|1 1|0 0|0 1|1 0|0 0|1 0|1 1|0 0|1 0|1 0|0 1|1 1|1 1|0 0|1 1|1 0|0 0|0 1|0 1|1 0|0 0|1 0|0 1|1 0|0 1|0 0|0 1|1 0|1 0|1 0|0 1|1 1|0 1|1 1|1 0|0 1|0 0|0 0|1 1|0 1|0 1|1 0|0 0|0 1|0 0|1 0|0 0|0 0|1 0|0 0|1 0|1 1|0 1|1 0|1 1|1 1|1 1|0 1|1 0|0 0|1 1|0 0|0 1|0 1|1 1|0 0|0 1|0 0|1 0|1 1|1 1|0 1|1 0|0 0|1 0|0 0|1 0|1 1|0 1|1 0|0 1|0 0|1 0|1 1|0 1|1 1|0 0|0 1|0 1|0 0|0 1|1 1|1 1|0 1|0 1|0 0|1 1|1 1|1 1|0 0|0 0|1 0|1 0|0 1|1 0|0 1|1 1|1 0|0 1|0 1|0 0|1 1|0 1|1 1|1 0|1 1|0 0|1 1|1 1|1 1|1 1|1 1|0 0|0 1|0 0|1 1|0 1|0 1|0 0|1 1|1 0|1 1|1 0|1 0|1 0|0 0|1 1|0 0|0 1|0 1|1 0|0 1|1 0|0 0|1 1|1 0|1 0|1 0|1 1|1 0|1 0|0 0|1 1|0 0|1 0|0 1|1 1|0 0|0 1|0 1|0 1|0 1|1 1|1 1|0 1|1 1|0 1|1 1|1 1|0 1|1 1|0 1|0 1|0 1|1 1|1 0|0 0|1 0|1 0|1 1|1 0|1 1|1 1|1 0|1 0|0 0|0 0|0 1|1 0|1 0|0 1|1 1|1 1|0 1|0 1|1 0|1 0|1 1|1 1|0 1|0 1|1 1|1 1|0 0|0 0|0 1|0 1|0 0|1 1|1 0|1 0|1 1|1 1|0 1|1 0|0 1|1 0|0 1|1 1|1 1|0 0|0 1|1 1|0 1|1 0|0 1|1 0|1 0|1 1|0 1|0 0|0 0|1 0|0 1|0 0|0 0|0 1|1 0|1 1|1 0|1 0|1 1|0 0|1 0|1 1|0 0|1 1|0 0|0 0|1 0|0 1|1 1|0 1|0 1|1 0|0 1|1 0|1 1|0 0|0 1|1 0|1 1|0 1|0 1|0 0|1 0|0 0|1 0|1 1|1 1|1 0|1 0|0 1|0 1|1 1|0 1|1 1|0 1|0 1|0 0|0 1|0 1|1 1|1 1|0 0|0 1|1 0|0 1|0 1|1 1|0 1|1 1|0 0|1 0|0 1|0 0|1 0|1 0|1 0|0 1|1 1|1 1|1 1|0 0|0 1|1 1|1 1|0 1|0 1|0 0|1 1|1 1|1 1|0 0|0 1|1 1|0 0|1 0|1 0|1 1|1 1|0 1|0 1|0 0|1 0|1 1|0 0|1 1|0 0|1 1|0 1|1 1|0 1|0 0|1 1|1 1|0 0|1 1|0 0|1 1|1 1|0 1|1 0|0 0|1 1|1 0|0 0|0 1|1 1|0 0|0 0|1 0|1 0|0 1|0 1|1 0|0 0|1 0|0 0|1 1|0 0|1 1|0 0|1 1|0 0|0 1|0 0|0 1|1 0|1 1|1 0|1 1|0 0|0 1|1 0|1 1|1 0|0 1|1 1|0 1|1 0|0 1|0 1|0 1|0 0|1 1|0 1|1 1|0 0|0 1|1 1|0 0|1 0|0 0|1 1|1 0|1 0|0 1|1 0|0 1|0 1|1 0|1\n")),(0,i.kt)("p",null,"It worked!  Imputation has generated best-guess genotypes for ",(0,i.kt)("inlineCode",{parentName:"p"},"rs601338")," (at ",(0,i.kt)("inlineCode",{parentName:"p"},"chr19:48703417"),") for us."),(0,i.kt)("h3",{id:"next-steps"},"Next steps"),(0,i.kt)("p",null,"You have successfully used a set of sequence data to identify genetic variants,\nquality control and phase them. And you have used that to impute the important secretor status (and\nother variants) into another dataset.  Congratulations!"),(0,i.kt)("p",null,"To finish the practical, go back and try the ",(0,i.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/#steps-in-the-practical"},"challenge questions"),"."))}m.isMDXComponent=!0}}]);