"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[6599],{3905:(e,t,n)=>{n.d(t,{Zo:()=>c,kt:()=>h});var a=n(7294);function r(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function i(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?i(Object(n),!0).forEach((function(t){r(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):i(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},i=Object.keys(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=a.createContext({}),u=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},c=function(e){var t=u(e.components);return a.createElement(l.Provider,{value:t},e.children)},p="mdxType",d={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},g=a.forwardRef((function(e,t){var n=e.components,r=e.mdxType,i=e.originalType,l=e.parentName,c=s(e,["components","mdxType","originalType","parentName"]),p=u(n),g=r,h=p["".concat(l,".").concat(g)]||p[g]||d[g]||i;return n?a.createElement(h,o(o({ref:t},c),{},{components:n})):a.createElement(h,o({ref:t},c))}));function h(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=n.length,o=new Array(i);o[0]=g;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s[p]="string"==typeof e?e:r,o[1]=s;for(var u=2;u<i;u++)o[u]=n[u];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}g.displayName="MDXCreateElement"},2393:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>o,default:()=>d,frontMatter:()=>i,metadata:()=>s,toc:()=>u});var a=n(7462),r=(n(7294),n(3905));const i={sidebar_position:1},o="Getting the prerequisites",s={unversionedId:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites",id:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites",title:"Getting the prerequisites",description:"To run this tutorial you will need two things: some software and some data.  This page will get you set up.",source:"@site/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites.md",sourceDirName:"sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis",slug:"/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"sidebar9",previous:{title:"Analysing paired-end sequencing data",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"},next:{title:"Practical outline",permalink:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline"}},l={},u=[{value:"Getting the software",id:"getting-the-software",level:2},{value:"Getting the data",id:"getting-the-data",level:2}],c={toc:u},p="wrapper";function d(e){let{components:t,...n}=e;return(0,r.kt)(p,(0,a.Z)({},c,n,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"getting-the-prerequisites"},"Getting the prerequisites"),(0,r.kt)("p",null,"To run this tutorial you will need two things: some software and some data.  This page will get you set up."),(0,r.kt)("h2",{id:"getting-the-software"},"Getting the software"),(0,r.kt)("p",null,"You need a version of the following tools:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("a",{parentName:"li",href:"http://samtools.github.io"},"samtools"),", a program for manipulating next-generation sequencing data reads"),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("a",{parentName:"li",href:"https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"},"fastqc"),", a program for performing quality control fo sequence data"),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("a",{parentName:"li",href:"https://github.com/lh3/bwa"},"bwa"),", a program for aligning reads to a reference sequence"),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("a",{parentName:"li",href:"https://github.com/zippav/Jellyfish-2"},"jellyfish 2"),", a program for counting k-mers (short sequences of fixed length k) in sequence data reads."),(0,r.kt)("li",{parentName:"ul"},"And it is also useful to install the ",(0,r.kt)("a",{parentName:"li",href:"https://igv.org"},"Integrated Genomics Viewer"),".")),(0,r.kt)("p",null,"You can get this software in a few different ways:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},"If you are working on a JupyterHub instance we have setup, these tools should be installed\nalready."),(0,r.kt)("li",{parentName:"ul"},"If you have set up a ",(0,r.kt)("a",{parentName:"li",href:"prerequiesites/README.md"},"conda installation as described here")," (including the bioconda and conda-forge channels),\nyou should be able to install everything except IGV with the command")),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-sh"},"conda install samtools fastqc bwa jellyfish\n")),(0,r.kt)("p",null,"or if mamba isn't working,"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-sh"},"mamba install samtools fastqc bwa jellyfish\n")),(0,r.kt)("p",null,"run in your ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/prerequisites/UNIX"},"UNIX terminal"),"."),(0,r.kt)("p",null,"The IGV desktop application has to be downloaded and installed seperately - see ",(0,r.kt)("a",{parentName:"p",href:"https://igv.org"},"the IGV website"),"."),(0,r.kt)("p",null,"If neither of these work, you may be able to install using your system's package manager (such as ",(0,r.kt)("inlineCode",{parentName:"p"},"apt")," on Ubuntu) or by\nbuilding from source. Doing this is out of scope of this tutorial but if you know what you're doing, feel free to try\nthis."),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Testing it.")," To check that you have the right software, open a bash terminal and try getting\nthem to print their version information.  You should see something like:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"$ jellyfish --version\njellyfish 2.3.0\n\n$ fastqc --version\nFastQC v0.11.9\n\n$ samtools --version\nsamtools 1.8\nUsing htslib 1.8\nCopyright (C) 2018 Genome Research Ltd.\n\n$ bwa\n\nProgram: bwa (alignment via Burrows-Wheeler transformation)\nVersion: 0.7.17-r1188\nContact: Heng Li <lh3@sanger.ac.uk>\n\nUsage:   bwa <command> [options]\n\n[...]\n")),(0,r.kt)("p",null,"(Don't type the dollar signs, those are just there to indicate the command prompt.)"),(0,r.kt)("h2",{id:"getting-the-data"},"Getting the data"),(0,r.kt)("p",null,"To get the data for this tutorial, download the tarball from ",(0,r.kt)("a",{parentName:"p",href:"https://www.well.ox.ac.uk/~gav/projects/whg-training-resources/data/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"},"this folder")," - for example using curl:"),(0,r.kt)("p",null,"For example using ",(0,r.kt)("inlineCode",{parentName:"p"},"curl"),":"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-sh"},"curl -O https://www.well.ox.ac.uk/~gav/projects/whg-training-resources/data/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/sequence_data_analysis.tgz\n")),(0,r.kt)("p",null,"This will take a minute or so to download.  Extract it :"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"tar -xzf sequence_data_analysis.tgz\n")),(0,r.kt)("p",null,"You should now have a folder called ",(0,r.kt)("inlineCode",{parentName:"p"},"sequence_data_analysis"),". For the tutorial, delete the tarball\nand then change into that directory:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"rm sequence_data_analysis.tgz\ncd sequence_data_analysis\n")),(0,r.kt)("p",null,"Now you're ",(0,r.kt)("a",{parentName:"p",href:"/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline"},"ready to start"),"."))}d.isMDXComponent=!0}}]);