"use strict";(self.webpackChunkwhg_training_resources=self.webpackChunkwhg_training_resources||[]).push([[7917],{3905:(e,n,t)=>{t.d(n,{Zo:()=>c,kt:()=>g});var r=t(7294);function a(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function i(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);n&&(r=r.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,r)}return t}function o(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?i(Object(t),!0).forEach((function(n){a(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):i(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function l(e,n){if(null==e)return{};var t,r,a=function(e,n){if(null==e)return{};var t,r,a={},i=Object.keys(e);for(r=0;r<i.length;r++)t=i[r],n.indexOf(t)>=0||(a[t]=e[t]);return a}(e,n);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(r=0;r<i.length;r++)t=i[r],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(a[t]=e[t])}return a}var s=r.createContext({}),p=function(e){var n=r.useContext(s),t=n;return e&&(t="function"==typeof e?e(n):o(o({},n),e)),t},c=function(e){var n=p(e.components);return r.createElement(s.Provider,{value:n},e.children)},u="mdxType",m={inlineCode:"code",wrapper:function(e){var n=e.children;return r.createElement(r.Fragment,{},n)}},d=r.forwardRef((function(e,n){var t=e.components,a=e.mdxType,i=e.originalType,s=e.parentName,c=l(e,["components","mdxType","originalType","parentName"]),u=p(t),d=a,g=u["".concat(s,".").concat(d)]||u[d]||m[d]||i;return t?r.createElement(g,o(o({ref:n},c),{},{components:t})):r.createElement(g,o({ref:n},c))}));function g(e,n){var t=arguments,a=n&&n.mdxType;if("string"==typeof e||a){var i=t.length,o=new Array(i);o[0]=d;var l={};for(var s in n)hasOwnProperty.call(n,s)&&(l[s]=n[s]);l.originalType=e,l[u]="string"==typeof e?e:a,o[1]=l;for(var p=2;p<i;p++)o[p]=t[p];return r.createElement.apply(null,o)}return r.createElement.apply(null,t)}d.displayName="MDXCreateElement"},3137:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>s,contentTitle:()=>o,default:()=>m,frontMatter:()=>i,metadata:()=>l,toc:()=>p});var r=t(7462),a=(t(7294),t(3905));const i={sidebar_position:4},o="Removing highly-correlated SNPs",l={unversionedId:"population_genetics/principal_components_analysis/ld_pruning",id:"population_genetics/principal_components_analysis/ld_pruning",title:"Removing highly-correlated SNPs",description:"If you followed the population genetics simulation tutorial, you'll know that",source:"@site/docs/population_genetics/principal_components_analysis/ld_pruning.md",sourceDirName:"population_genetics/principal_components_analysis",slug:"/population_genetics/principal_components_analysis/ld_pruning",permalink:"/whg-training-resources/population_genetics/principal_components_analysis/ld_pruning",draft:!1,editUrl:"https://github.com/whg-training/whg-training-resources/edit/main/docs/population_genetics/principal_components_analysis/ld_pruning.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{sidebar_position:4},sidebar:"sidebar7",previous:{title:"Overview of the practical",permalink:"/whg-training-resources/population_genetics/principal_components_analysis/overview"},next:{title:"Removing closely-related samples",permalink:"/whg-training-resources/population_genetics/principal_components_analysis/relatedness_pruning"}},s={},p=[{value:"Relatedness pruning",id:"relatedness-pruning",level:2}],c={toc:p},u="wrapper";function m(e){let{components:n,...t}=e;return(0,a.kt)(u,(0,r.Z)({},c,t,{components:n,mdxType:"MDXLayout"}),(0,a.kt)("h1",{id:"removing-highly-correlated-snps"},"Removing highly-correlated SNPs"),(0,a.kt)("p",null,"If you followed the ",(0,a.kt)("a",{parentName:"p",href:"/population_genetics/simulation/"},"population genetics simulation tutorial"),", you'll know that\ncorrelation between nearby SNPs arises naturally as a result of genetic drift (or selection). This can lead to patterns\nof local variation that dominate principal components. "),(0,a.kt)("p",null,"Because for our purposes we want to capture 'genome-wide' patterns\nof relationships, we will first get rid of any too-correlated groups of SNPs."),(0,a.kt)("p",null,"LD pruning removes correlated pairs of SNPs so that the remaining SNPs are roughly independent. (It also helps\nto make subsequent computations quicker.) Run the following command in your terminal to prune the dataset:"),(0,a.kt)("pre",null,(0,a.kt)("code",{parentName:"pre"},"plink --vcf chr19-clean.vcf.gz --maf 0.01 --indep-pairwise 50 5 0.2 --out chr19-clean \n")),(0,a.kt)("p",null,"The above command tells plink to load the file ",(0,a.kt)("inlineCode",{parentName:"p"},"chr19-clean.vcf.gz")," and to prune SNPs, leaving SNPs\nwith minor allele frequency (MAF) at least 1%, and with no pairs remaining with pairwise\nr",(0,a.kt)("sup",null,"2"),">0.2. (The other parameters, here 50 and 5, affect how the computation works in\nwindows across the genome. You can read about the behaviour here:\n",(0,a.kt)("a",{parentName:"p",href:"http://www.cog-genomics.org/plink2/ld"},"http://www.cog-genomics.org/plink2/ld"),")."),(0,a.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,a.kt)("div",{parentName:"div",className:"admonition-heading"},(0,a.kt)("h5",{parentName:"div"},(0,a.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,a.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,a.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Question")),(0,a.kt)("div",{parentName:"div",className:"admonition-content"},(0,a.kt)("p",{parentName:"div"},"Look at the screen output from the above plink command.  "),(0,a.kt)("ul",{parentName:"div"},(0,a.kt)("li",{parentName:"ul"},"How many variants were in the original dataset?"),(0,a.kt)("li",{parentName:"ul"},"How many were removed because their frequency was below 1%?"),(0,a.kt)("li",{parentName:"ul"},"How many variants were removed due to LD pruning?"),(0,a.kt)("li",{parentName:"ul"},"How many variants remain?")))),(0,a.kt)("p",null,"Type ",(0,a.kt)("inlineCode",{parentName:"p"},"ls")," or use the file manager to view the directory.  The command above produced a number of files that all begin with the ",(0,a.kt)("inlineCode",{parentName:"p"},"chr19-clean")," prefix.  For our purposes, the most important one is ",(0,a.kt)("inlineCode",{parentName:"p"},"chr19-clean.prune.in"),", as this lists the SNPs that remain after pruning.  Feel free to look at all these files using less or a text editor."),(0,a.kt)("h2",{id:"relatedness-pruning"},"Relatedness pruning"),(0,a.kt)("p",null,"When you're ready, ",(0,a.kt)("a",{parentName:"p",href:"/whg-training-resources/population_genetics/principal_components_analysis/relatedness_pruning"},"go to the next page to identify and remove close relationships"),"."))}m.isMDXComponent=!0}}]);