<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Warmup: examining read coverage in a region | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Warmup: examining read coverage in a region | The WHG training resources"><meta data-rh="true" name="description" content="Welcome! The practical studies a region of the human genome on chromosome 4 that is known to"><meta data-rh="true" property="og:description" content="Welcome! The practical studies a region of the human genome on chromosome 4 that is known to"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/statistical_modelling/introduction/">Introduction to statistics for genomics</a><button aria-label="Toggle the collapsible sidebar category &#x27;Introduction to statistics for genomics&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/statistical_modelling/probability_cheatsheet/">Probability cheatsheet</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/statistical_modelling/regression_modelling/">Statistical modelling I - regression</a><button aria-label="Toggle the collapsible sidebar category &#x27;Statistical modelling I - regression&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/statistical_modelling/">Statistical modelling</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/">Glycophorin CNV calling tutorial</a><button aria-label="Toggle the collapsible sidebar category &#x27;Glycophorin CNV calling tutorial&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup/">Warmup: examining read coverage in a region</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm/">Modelling CNVs using a Hidden Markov Model</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--sublist-caret" aria-expanded="false" href="/whg-training-resources/statistical_modelling/notes/computing_pvalues/">notes</a></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/"><span itemprop="name">Glycophorin CNV calling tutorial</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Warmup: examining read coverage in a region</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Warmup: examining read coverage in a region</h1><p>Welcome! The practical studies a region of the human genome on chromosome 4 that is known to
contain large <em>copy number variants</em>. These are genetic variants that occur due to mistakes in DNA
replication and lead to deletions, duplications, or other rearrangements of large tracts of DNA. In
general copy number variants (and other structural variants that don&#x27;t change copy number, such as
inversions) are thought to be extremely important determinants of human disease - the CNVs in
this practical delete or duplicate whole genes, and some of them are
<a href="https://dx.doi.org/10.1126/science.aam6393" target="_blank" rel="noopener noreferrer">associatied with malaria susceptibility</a>.</p><p>On the other hand CNVs and other structural variation are not that well studied, particularly when they
occur in regions of genome duplication or paralogy.</p><p>In this practical the plan is to try to genotype CNVs in one such region - the region containing
<a href="https://en.wikipedia.org/wiki/Glycophorin_A" target="_blank" rel="noopener noreferrer"><em>GYPA</em></a>. <em>GYPA</em> encodes Glycophorin A, which is one
of the most abundant red cell surface proteins. Malaria parasites are known to interact with
Glycophorin A during invasion, and we got interested in this because it turns out that these CNVs
can provide protection against malaria. The practical is based on data from this paper
<a href="https://dx.doi.org/10.1126/science.aam6393" target="_blank" rel="noopener noreferrer">https://dx.doi.org/10.1126/science.aam6393</a> where we investigated that protection.</p><p>To call CNVs we will look at <em>sequence coverage data</em> (i.e. how many reads aligned to each genomic
location from short-read sequence data) and look for variation in coverage that might indicate loss
or gain of DNA copies. To make this simple, we have grouped the genome into consecutve 1600bp bins
and we work with <em>mean coverage in each bin</em>.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="first-look-at-the-data">First look at the data<a class="hash-link" href="#first-look-at-the-data" title="Direct link to heading">​</a></h2><p>First of all let&#x27;s load and look at the data - we&#x27;ll use R for this practical (but if you are expert you are welcome to
explore other methods).</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">data = read.delim( &quot;glycophorin_binned_coverage.tsv.gz&quot;, header = T, as.is = T, sep = &quot;\t&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">View(data)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The data has position information in the first few columns, and samples in columns 4 onwards.
Let&#x27;s split these things out for easier handling.  We&#x27;ll call the actual data <code>X</code>:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">sites = data[,1:3]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">X = as.matrix( data[,4:ncol(data)] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">rownames(X) = sites$position</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samples = colnames( X )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>How could we plot this data?  One way is just to make a heatmap:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">image( X, x = 1:nrow(sites), y = 1:length(samples) )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This is not all that edifying but a few features are evident. Clearly samples vary in coverage (some rows are more red
than others). Also, some sites seem to have more coverage than others (some columns are more red). Also some sites have
missing data! (White columns). The reason for this is that the region contains paralogous gene copies which make read
mapping difficult - we have excluded bins where mappability was poor.</p><p>If you stare hard between bins 300-400, you may start to see some samples seem to have something going on (long
horizontal bands of more yellow or more red). This is the signal we want to extract.  There are a few ways we could try this - for example, a dimension reduction method, like PCA, might work.  Feel free to try that!  Here we are going to explore a modelling approach.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="using-an-empirical-model-to-handle-variation-in-the-data">Using an empirical model to handle variation in the data<a class="hash-link" href="#using-an-empirical-model-to-handle-variation-in-the-data" title="Direct link to heading">​</a></h2><p>Let&#x27;s look at how coverage actually looks across sites, for the first few samples:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">par( mar = c( 2, 2, 2, 1 ) )                                    # this line adjust margins to get more on plot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">layout( matrix( 1:25, byrow = T, ncol = 5 ))                    # put multiple panes on one plot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">for( i in 1:25 ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    h = hist(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        X[,i],                      # coverage values for individual i</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        breaks = 25,                # Number of bars to plot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        freq = FALSE,               # This scales the y axis so density sums to 1, i.e. empirical distribution</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        main = samples[i]           # This is the plot title</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The data for each sample looks kind of uni-modal and sort of symmetrical-ish in general, with a few bumps. Of course
any CNVs might affect that, and that could explain some of the bumps - that&#x27;s what we want to find out.</p><p>For a practical approach we will assume that this binned coverage follows a Gaussian distribution. And for a first
guess, we will fit the gaussian using all the data in the 1st 100 bins (which from the plot above don&#x27;t obviously seem
to contain many CNVs). So let&#x27;s go ahead and compute the parameters of these gaussians now:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">coverage.parameters = data.frame(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    sample = samples,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    mean = sapply( 1:ncol( X ), function(i) { mean( X[1:100,i], na.rm = T ) }),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    variance = sapply( 1:ncol( X ), function(i) { var( X[1:100,i], na.rm = T ) })</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">View( coverage.parameters )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>We&#x27;re going to use a Gaussian likelihood function to model sequence coverage data.  The <code>dnorm()</code> function in R can be used to implement this. In the spirit of making code readable, let&#x27;s use this to write a &#x27;gaussian.ll()` function that we will use in our code:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">gaussian.ll &lt;- function(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    data,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    params = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        mean = 0,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        variance = 1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    return(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        dnorm( data, mean = params$mean, sd = sqrt( params$variance ), log = TRUE )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Let&#x27;s plot data again and see how the fit looks:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">par( mar = c( 2, 2, 2, 1 ) )                                    # this line adjust margins to get more on plot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">layout( matrix( 1:25, byrow = T, ncol = 5 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">x = seq( from = 0, to = 20, by = 0.01 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">for( i in 1:25 ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    h = hist(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        X[,i],                      # coverage values for individual i</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        breaks = 25,                # Number of bars to plot</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        freq = FALSE,               # This scales the y axis so density sums to 1, i.e. empirical distribution</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        main = samples[i]           # This is the plot title</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    points(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        x,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        exp(gaussian.ll(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            data = x,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            params = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                mean = coverage.parameters$mean[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                variance = coverage.parameters$variance[i]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        )),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        type = &#x27;l&#x27;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        col = &#x27;red&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The fit looks ... sort of ok.  Not perfect, but not bad, depending on the sample.</p><p><strong>Exercise</strong> feel free to see how this looks for other samples.  You could also make a qq-plot for each sample to see how well the model fits.</p><p>Clearly this model is not a perfect model of the data, but it might be enough to work with - for the purposes of this
practical we&#x27;ll go with it.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="modelling-copy-number-variation">Modelling copy number variation<a class="hash-link" href="#modelling-copy-number-variation" title="Direct link to heading">​</a></h2><p>Our general model of the effect of CNVs on coverage is that coverage of a site with copy number <code>c</code> should be <code>c</code> times
as large as at a site with copy number 1 - plus noise.  (Humans are diploid so the copy number at most sites is 2). If we think of sequence reads as being generated from each
copy independently, a bit of thought shows that the same relationship happens for the variance as well. So our model
for copy number c would be that</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">     coverage ~ N( c * mean, c * variance )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This is good because it provides a direct link from copy number to coverage (accounting for noise in the coverage
values), that might allow us to infer copy numbers.</p><p>Let&#x27;s try that now. For simplicity:</p><ul><li>We will assume only copy numbers of 0 to 5 are possible (i.e. from a homozygous deletion up to a possible three extra copies).  Feel free to increase this if you want to!</li><li>We&#x27;ll put the results in a giant multidimensional array of <code>sites</code> x <code>samples</code> x <code>copy number states</code>.</li></ul><p>We start by computing the per-site likelihood of each data point for each possible copy number. Importantly as in the
stats modelling session <em>we will work throughout in log space</em> to avoid any numerical issues. This makes some
computations more difficult, but is a generally good idea when more than a few data points are involved.  So we&#x27;ll make our array <em>an array of log-likelihoods</em>:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">copy.numbers = 0:5 </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">copy.number.lls = array(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    NA,                                 # fill with missing data to start</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dim = c(                            # dimensions</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        nrow(sites),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        length(samples),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        length( copy.numbers )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dimnames = list(                    # A good feature of R is you can name everything</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        sites$position,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        samples,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        sprintf( &quot;cf=%d&quot;, copy.numbers )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Let&#x27;s compute these lls:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">for( sample in 1:length(samples) ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    for( i in 1:length(copy.numbers) ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number = copy.numbers[i]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        mean.multiplier = copy.number/2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        variance.multiplier = max( copy.number/2, 0.01 ) # this to be error tolerant, as explained below</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number.lls[,sample,i] = gaussian.ll(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            X[,sample],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            params = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                mean = coverage.parameters$mean[sample] * mean.multiplier,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                variance = coverage.parameters$variance[sample] * variance.multiplier</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # Handle the missing data sites.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # We will just treat missing data as having likelihood 1 (loglikelihood zero) here.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        copy.number.lls[ is.na(X[,sample]),sample, ] = 0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Look at the top left of the output:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">View( copy.number.lls[1:10,1:10,] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note</strong> In practice even sites with 0 &quot;real&quot; coverage might get some coverage, e.g. due to spurious read alignment or mis-alignment.  For that reason we used a <code>variance.multiplier</code> variable above.  It equals <code>copy.number / 2</code> except for copy number zero, where it allows a small amount of coverage to exist.</p><p>These copy number log-likelihoods should now capture some of the important signal in the data.  But how to plot it given it&#x27;s 3-dimensional?  Here are two ways we could do it.</p><p>We could take the maximum likelihood copy number call and plot that (for each sample and each site).</p><p>Or, we could recognise that the above is daft because it doesn&#x27;t take into account what we know - that most people will be diploid at most sites.  Instead, we could use a Bayesian approach to build this information in.</p><p>The rest of this practical does both these things.  We&#x27;ll use this function to plot the results:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers &lt;- function(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    copy.number,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    filename = NULL,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    title = NULL,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    palette = c( &quot;darkorange2&quot;, &quot;darkgoldenrod1&quot;, &quot;forestgreen&quot;, &quot;deepskyblue2&quot;, &quot;deepskyblue4&quot;, &quot;blueviolet&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) { </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    if( !is.null( filename )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        pdf( file = filename, width = 6, height = 8 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    } </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    par(mfrow=c(1,1))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    par( mar = c( 4, 4, 2 + 2 *!is.null(title), 2 ) + 0.1 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    image( copy.number, x = 1:nrow(copy.number), y = 1:ncol(copy.number), xlab = &quot;sites&quot;, ylab = &quot;samples&quot;, col = palette, zlim = c( 0, length(palette)-1 ), main = title )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    legend( &quot;topleft&quot;, pch = 22, col = &#x27;black&#x27;, pt.bg = palette, legend = c( &quot;hom deletion&quot;, &quot;het deletion&quot;, &quot;normal&quot;, &quot;1 extra copy&quot;, &quot;2 extra copies&quot;, &quot;3 extra copies&quot; ), bg = &quot;white&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    if( !is.null( filename )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        dev.off()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="copy-number-inference-using-mles">Copy number inference using MLEs<a class="hash-link" href="#copy-number-inference-using-mles" title="Direct link to heading">​</a></h2><p>Let&#x27;s compute the maximum likelihood copy number state for each individual at each site:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">maximum.likelihood.state = array( NA, dim = c( nrow(X), ncol( X ) ), dimnames = list( sites$position, samples ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">for( variant in 1:nrow( X )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    for( sample in 1:ncol(X)) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        w = which.max( copy.number.lls[variant,sample,] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        stopifnot( !is.na(w) )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        maximum.likelihood.state[variant,sample] = copy.numbers[w]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers( maximum.likelihood.state, title = &quot;Maximum likelihood copy number&quot; )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>That plot definitely looks cleaner!  Still lots of noise though, can we do better?</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="bayesian-copy-number-inference">Bayesian copy number inference<a class="hash-link" href="#bayesian-copy-number-inference" title="Direct link to heading">​</a></h2><p>Taking the maximum likelihood approach is daft because we have salient prior information: most people will be diploid
at most sites, with possibly a few CNVs sprinkled in. Let&#x27;s build that knowledge in by applying our bayesian reasoning.
Still working seperately for each sample and each site, we will take a prior distribution that puts most weight on
diploid state, and estiamte copy number state by taking posterior expectations.</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain"># our prior - note this should sum to one</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">prior = c(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=0` = 0.02,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=1` = 0.02,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=2` = 0.9,       # I put 90% weight on diploid state...</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=3` = 0.02,      # ...and 2% weight on everything else - feel free to try different values (make it sum to 1)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=4` = 0.02, </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    `cn=5` = 0.02</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) </span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Here is another big array to put our results in:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">expected.posterior.state = array(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    NA,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dim = c( nrow(sites), length(samples) ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dimnames = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        sites$position,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        samples</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now let&#x27;s compute it.</p><p>Here a small potential technical problem occurs. To compute the normalisation factor in Bayes rule we need to sum over
the possible copy numbers. (Just like with the fair/unfair dice example fom the stats modelling session.) The
probabilities for different copy numbers vary widely in magnitude (from close to -Inf to &gt; 1) and this is a classic
case that causes numerical difficulties. To solve that we are working in log space.</p><p>But to compute expectations we now need to compute a sum over probabilities. A direct approach would use <code>log( sum(
exp( values )))</code> but this runs the risk of the above numerical problems. Instead, a more numerically stable computation
uses the &quot;log-sum-exp&quot; formula, which computes the same value but avoids numerical errors by separating out the largest
value. Here is a fairly robust implementation:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">log.sum.exp &lt;- function(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    x,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    na.rm = FALSE               # This mimics the na.rm argument of base R&#x27;s sum() function</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    result = NA</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    z = max( x )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    if( !is.na( z )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        terms = x-z</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        if( na.rm ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            w = which( !is.na( terms ) )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        } else {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            w = 1:length( terms )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        if( length( w ) &gt; 0 ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">            result = z + log( sum( exp( terms[w] ) ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    return( result )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now let&#x27;s compute the posteriors:</p><div class="language-R codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-R codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">for( variant in 1:nrow( sites )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    for( sample in 1:length(samples)) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # These three lines implement Bayes rule, but working in log space</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        log.unnormalised.posterior = copy.number.lls[variant,sample,] + log(prior)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        log.normalisation.constant = log.sum.exp( log.unnormalised.posterior )      # equivalent to: log( sum( exp( log.unnormalised.posterior )))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        log.posterior = log.unnormalised.posterior - log.normalisation.constant</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        # Now compute expected copy number given the posterior distribution</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        expected.posterior.state[variant,sample] = sum( exp(log.posterior) * copy.numbers )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers( expected.posterior.state, title = &quot;Expected posterior copy number&quot; )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Compare this with the maximum likelihood copy number calls above - the posterior version is much much cleaner.</p><p>It&#x27;s clear now that there is something goin on in that region - a number of samples have clear runs of deleted or
apparently duplicated bins. To draw out this signal we can cluster samples - here using hierarchical clustering:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">clustered.order = hclust(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    # we will cluster based on a distances between expected posterior state vectors</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dist( t(expected.posterior.state) )             </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)$order</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.copy.numbers( expected.posterior.state[,clustered.order], title = &quot;Expected posterior copy number (clustered)&quot; )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note</strong> when you get here, please email me (<a href="mailto:gavin.band@well.ox.ac.uk" target="_blank" rel="noopener noreferrer">gavin.band@well.ox.ac.uk</a>) with the results of your last plot!</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="future-directions">Future directions<a class="hash-link" href="#future-directions" title="Direct link to heading">​</a></h2><p>This is the end of this part of the tutorial.  However, this model still isn&#x27;t good enough because it still only works marginally at each site and sample.  In fact, we know how copy number variants generally arise (unequal crossover leading to long runs of duplicated or deleted DNA) and we&#x27;d like to include that information in the model too, if only we could figure out how to put it in.  In the <a href="/whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm/">next part of the tutorial</a> we will see how that can be done by linking this to a type of model known as a Hidden Markov Model, that models copy number state across the region.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/statistical_modelling/Hidden_markov_models/glycophorin_cnv_warmup.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Glycophorin CNV calling tutorial</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/statistical_modelling/Hidden_markov_models/glycophorin_cnv_hmm/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Modelling CNVs using a Hidden Markov Model</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#first-look-at-the-data" class="table-of-contents__link toc-highlight">First look at the data</a></li><li><a href="#using-an-empirical-model-to-handle-variation-in-the-data" class="table-of-contents__link toc-highlight">Using an empirical model to handle variation in the data</a></li><li><a href="#modelling-copy-number-variation" class="table-of-contents__link toc-highlight">Modelling copy number variation</a></li><li><a href="#copy-number-inference-using-mles" class="table-of-contents__link toc-highlight">Copy number inference using MLEs</a></li><li><a href="#bayesian-copy-number-inference" class="table-of-contents__link toc-highlight">Bayesian copy number inference</a></li><li><a href="#future-directions" class="table-of-contents__link toc-highlight">Future directions</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>