<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/IGV/Exploring_reads_in_IGV">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Viewing reads | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Viewing reads | The WHG training resources"><meta data-rh="true" name="description" content="You should now have load some data into IGV (if not, go and do that now.)"><meta data-rh="true" property="og:description" content="You should now have load some data into IGV (if not, go and do that now.)"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/IGV/Exploring_reads_in_IGV/">Viewing reads</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/IGV/"><span itemprop="name">Exploring Reads in IGV</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Viewing reads</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Viewing reads</h1><p>You should now have load some data into IGV (if not, go and <a href="/whg-training-resources/sequence_data_analysis/IGV/#loading-data">do that now</a>.)
You&#x27;ll probably see something like this:</p><p>To see the reads you will need to zoom into a location on the genome. For example, let&#x27;s zoom into the gene that
harbours the famous chloroquine resistance mutation, <a href="https://plasmodb.org/plasmo/app/record/gene/PF3D7_0709000" target="_blank" rel="noopener noreferrer">Chloroquine resistance transporter
(<em>CRT</em>)</a>. To search for it you will need its ID, which is
<code>PF3D7_0709000</code>. Voila! Some reads.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>IGV hints</h5></div><div class="admonition-content"><p>This is a good point to try a few options to get used to IGV.</p><ul><li><p>For paired-end data, try the &#x27;view as pairs&#x27; option. It can be found in the context menu, obtained by right-clicking
anywhere on the track.</p></li><li><p>If you don&#x27;t like how reads are displayed, try &#x27;collapsed&#x27;, &#x27;expanded&#x27;, or &#x27;squished&#x27; from the context menu.  Which one do you like best?</p></li><li><p>Try clicking on a read.  What is all that info?  Where does it come from?  What does it mean?</p></li><li><p>Try moving around and zooming in/out.  </p></li><li><p>There&#x27;s also a <code>genes</code> track - it can be annoying when it is squished.  Try right-clicking and choose &#x27;expanded&#x27; to see the
genes.</p></li></ul></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="basics">Basics<a class="hash-link" href="#basics" title="Direct link to heading">​</a></h2><p>Scroll around a bit to look at the reads.</p><p>Can you find:</p><ul><li>a sequencing error?</li><li>a SNP?</li><li>an insertion / deletion variant?</li></ul><p>What makes you think so?</p><p>:::</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>By default you are looking at all reads (including those that didn&#x27;t map very well.)  There&#x27;s an option hidden away
under <code>Preferences</code> -&gt; <code>Alignment</code> that lets you set a mapping quality threshold.  A good default value to put here is
20 (but remember that by doing this you are filtering some reads!)</p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="looking-at-snps">Looking at SNPs<a class="hash-link" href="#looking-at-snps" title="Direct link to heading">​</a></h2><p>Let&#x27;s try looking some SNPs</p><p>The mutation that causes <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2954758/" target="_blank" rel="noopener noreferrer">chloroquine resistance</a> is <code>CRT
K76&gt;T</code>, at <code>Pf3D7_07_v3:403,625</code>. (It actually involves a small haplotype including other nearby mutations).</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Questions</h5></div><div class="admonition-content"><ul><li><p>Can you find these mutations?</p></li><li><p>Parasites with the non-reference allele are resistant to chloroquine - are your parasites resistant or susceptible?  (NB. all these samples come from <a href="https://www.google.com/maps/place/Democratic+Republic+of+the+Congo/@-3.9835365,12.6862829,5z/data=!3m1!4b1!4m5!3m4!1s0x1979facf9a7546bd:0x4c63e5eac93f141!8m2!3d-4.038333!4d21.758664" target="_blank" rel="noopener noreferrer">Democratic Republic of The Congo</a>).</p></li><li><p>Zoom into this location to look at the sequence.  Can you figure out the amino acid change(s)?</p></li><li><p>How many reads support this in each sample?  What alleles do they carry?  What are their IDs?  Are they well mapped?
Are they properly paired?  Where is their pair?</p></li></ul><p><strong>Note.</strong> &#x27;properly paired&#x27; means that the two reads in the pair align to the same region in the right orientation, and roughly fall within the distribution of insert sizes inferred from all the read pairs in the data.  The aligner (<code>bwa</code> in this case) sets a flag in the BAM file to reflect this.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="looking-at-insertions-and-deletions">Looking at insertions and deletions<a class="hash-link" href="#looking-at-insertions-and-deletions" title="Direct link to heading">​</a></h2><p><strong>Question.</strong> The variants above are single nucleotide polymorphisms (SNPs) - or rather multiple nucleotide polymorphisms (MNPs) since several come together.  What about other types of variation?  Scroll around a bit and see if you can find a deletion.  (If stuck, try looking around the gene <a href="https://plasmodb.org/plasmo/app/search?q=PF3D7_0220300" target="_blank" rel="noopener noreferrer"><code>PF3D7_0215300</code></a>.  What read evidence supports this deletion?</p><p><strong>Note.</strong> There might also be insertions, which are denoted by a purple bar across a read in IGV.  Can you find any of these?</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="the-trouble-with-mapping-is">The trouble with mapping is...<a class="hash-link" href="#the-trouble-with-mapping-is" title="Direct link to heading">​</a></h2><p><strong>Question.</strong> If you scroll around a bit, after a while you&#x27;ll come across regions where no reads (or very few reads)
seem to align.  Can you find one?  What could cause this? </p></div></div><p>If you can&#x27;t find any, try the ends of chromosomes, or the gene <code>PF3D7_0223500</code> (which encodes one of the highly
polymorphic / duplicated &#x27;var&#x27; genes.)</p><p>:::</p><p><strong>Hint.</strong> You could also try searching for the relevant piece of genome sequence using <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank" rel="noopener noreferrer">NCBI
BLAST</a>.  For example, I tried this for one such location:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">$ samtools faidx data/reference/Pf3D7_v3.fa.gz Pf3D7_07_v3:343,180-343,380</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;Pf3D7_07_v3:343,180-343,380</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">tatatatatatatatatatatatatatatatatatatataaatatatatatatgtatgta</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">tgtatgtaatattttagtgcaaaaaaaaaaaaaaaaaaaaaaaaaaagtatataatgaaa</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">aattattatatatatatatattatatatatatatatatatatatatatatatatatatat</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">atatatatatatatatatata</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>and then pasted the above into the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank" rel="noopener noreferrer">Nucleotide BLAST page</a>.  (Another version of this can be found <a href="https://plasmodb.org/plasmo/app/search/transcript/UnifiedBlast" target="_blank" rel="noopener noreferrer">on PlasmoDB</a>.  <strong>Note.</strong> you may need to turn off the &#x27;low complexity region&#x27; filter to get useful results.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="examining-larger-structural-variants">Examining larger structural variants<a class="hash-link" href="#examining-larger-structural-variants" title="Direct link to heading">​</a></h2><p>Let&#x27;s have a look at a larger structural variant.  Zoom your IGV to the region around the gene <a href="https://plasmodb.org/plasmo/app/record/gene/PF3D7_1127000" target="_blank" rel="noopener noreferrer"><code>Pf3D7_1127000</code></a>.  And zoom out a bit so you can see the whole region including the two flanking genes.</p><p>The samples from this tutorial were chosen because some of them contain a large structural variant in this region. </p><p><strong>Challenge.</strong> Can you find this structural variant in one or more of the samples you&#x27;ve loaded?</p><p><strong>Hints.</strong></p><ul><li><p>One way to find structural variants is by looking at their effects on copy number (i.e. on read coverage.)  Do any samples have unusual copy number profiles?  (NB. this is generally easier in human sequence data because in P.falciparum the coverage varies a lot anyway due to varying AT content.)</p></li><li><p>Your reads are paired-end reads.  If the genome has a different structure to the reference, then these might not align close together in the right orientation or insert size.  Can you find reads like this?  (Hint: the &#x27;view as pairs&#x27;, &#x27;group by&#x27; and/or &#x27;color by&#x27; options are very useful here.)  What is their orientation - are the read pairs the &#x27;right&#x27; way round or the &#x27;wrong&#x27; way round?  What does this mean?  Are these single reads or are they supported by others?</p></li></ul><h2 class="anchor anchorWithStickyNavbar_mojV" id="examining-structural-variant-breakpoints">Examining structural variant breakpoints<a class="hash-link" href="#examining-structural-variant-breakpoints" title="Direct link to heading">​</a></h2><p>By a &#x27;breakpoint&#x27; we mean the points where the segments of the sequenced genome are connected together in a different way to the reference genome.  The long, wrongly oriented / wrong insert size reads above potentially span these breakpoints.</p><p>A good way to look at this is as follows.  First zoom into one of the reads in the pair. Now right-click on a read and choose &#x27;View mate region in split screen&#x27;.  Voila!  You are now looking at both ends of the same read pair.</p><p>To take this further there is an additional option.  If the read in the pair actually crosses a structural variant breakpoint, it could be that the end of the read does not align well.  If so it will typically be &#x27;soft-clipped&#x27; by the aligner.  (The <a href="https://sites.google.com/site/bioinformaticsremarks/bioinfo/sam-bam-format/what-is-a-cigar" target="_blank" rel="noopener noreferrer"><code>CIGAR string</code></a> for the alignment will have a sequence of &#x27;S&#x27;s at the end).  Under <code>Preferences</code> -&gt; <code>Alignments</code> there is an option &#x27;Show soft-clipped bases&#x27; to visualise these.  Go and turn this on.  Does this help figure out the basepair location of the possible breakpoint?</p><p><strong>NB.</strong> In the current data this works best for the reads at <code>Pf3D7_11_v3:1,058,745-1,058,915</code> - if you processed the full coverage data then it might work at all the breakpoints.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="figuring-out-the-structural-variant-structure">Figuring out the structural variant structure<a class="hash-link" href="#figuring-out-the-structural-variant-structure" title="Direct link to heading">​</a></h2><p>::tip Extra hard challenge</p><p>Figure out the structure of the structural variant genomes from these reads.</p><p><strong>Note.</strong> Good luck!  This really is difficult.</p><p>If you want some inspiration, see Figure 6 of <a href="https://doi.org/10.1126/science.aam6393" target="_blank" rel="noopener noreferrer">this paper on the Dantu blood group</a> to see a similar process applied to a human structural variant.</p><p>:::</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="summary">Summary<a class="hash-link" href="#summary" title="Direct link to heading">​</a></h2><p>Congratulations!  You are now an IGV expert!</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/IGV/Exploring_reads_in_IGV.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/IGV/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Exploring Reads in IGV</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">A sequence data pipelining challenge!</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#basics" class="table-of-contents__link toc-highlight">Basics</a></li><li><a href="#looking-at-snps" class="table-of-contents__link toc-highlight">Looking at SNPs</a></li><li><a href="#examining-larger-structural-variants" class="table-of-contents__link toc-highlight">Examining larger structural variants</a></li><li><a href="#examining-structural-variant-breakpoints" class="table-of-contents__link toc-highlight">Examining structural variant breakpoints</a></li><li><a href="#figuring-out-the-structural-variant-structure" class="table-of-contents__link toc-highlight">Figuring out the structural variant structure</a></li><li><a href="#summary" class="table-of-contents__link toc-highlight">Summary</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>