<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-programming/analysing_genes_across_species/Counting_genes_1">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Counting genes | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Counting genes | The WHG training resources"><meta data-rh="true" name="description" content="For simplicity, from here on in we will work with the GFF3 files from the [Ensembl ftp"><meta data-rh="true" property="og:description" content="For simplicity, from here on in we will work with the GFF3 files from the [Ensembl ftp"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/programming/">Programming</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/programming/introduction_to_R/">Introduction to R</a><button aria-label="Toggle the collapsible sidebar category &#x27;Introduction to R&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/programming/programming_with_gene_annotations3/">Learn to program with gene annotations!</a><button aria-label="Toggle the collapsible sidebar category &#x27;Learn to program with gene annotations!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/programming/analysing_genes_across_species/">Analysing genes across species</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing genes across species&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/getting_setup/">Getting setup</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/">Counting genes</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/Counting_genes_2/">What&#x27;s not a gene?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/filter_join_merge/">Filter, group, summarise, join, arrange...</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-2 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/extreme_genes/">Extreme genes!</a><button aria-label="Toggle the collapsible sidebar category &#x27;Extreme genes!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/getting_chromosomes/">Getting chromosome lengths</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-2 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/how_much_in_genes/">How much of the genome is in genes?</a><button aria-label="Toggle the collapsible sidebar category &#x27;How much of the genome is in genes?&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/programming/analysing_genes_across_species/challenges/">Conclusions and challenges</a></li></ul></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/programming/analysing_genes_across_species/"><span itemprop="name">Analysing genes across species</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Counting genes</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Counting genes</h1><p>For simplicity, from here on in we will work with the GFF3 files from the <a href="http://ftp.ensembl.org/pub/current_gff3/" target="_blank" rel="noopener noreferrer">Ensembl ftp
site</a> (vertebrate genomes) or from <a href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/" target="_blank" rel="noopener noreferrer">Ensembl
genomes</a> (non-vertebrate genomes). These use the terminology <code>mRNA</code> for a
gene transcript, and they also have the genome sequence lengths written in the metadata, making life easy for us. Before
starting, download some of these files now. </p><p>We will focus on protein-coding genes, and their transcripts, exons and coding sequence. They have type <code>gene</code>, <code>mRNA</code>,
<code>exon</code> and <code>CDS</code> in the files Ensembl respectively. They come in a basic hierarchy:</p><ul><li><p>Each gene can have multiple transcripts (i.e. multiple expressed forms at mRNA level - e.g. they
might differ in how exons are spliced together, or have different transcription start or end
sites).</p></li><li><p>Each transcript is made up of one or more exons. The introns in between are spliced out. (This
happens during the transcription process.)</p></li><li><p>Of the exon sequence that makes it into a transcript, only a portion is actually translated into
the mature protein during processing by the ribosomes. This is the <em>coding sequence</em> and the rest
is <em>untranslated sequence</em>.</p></li></ul><p>The files we&#x27;re looking are (roughly speaking) humanities&#x27; best guess at what this picture looks like in each organism.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="getting-data">Getting data<a class="hash-link" href="#getting-data" title="Direct link to heading">​</a></h2><p>I&#x27;m going to assume you have found a set of files from the <a href="http://ftp.ensembl.org/pub/current_gff3/" target="_blank" rel="noopener noreferrer">Ensembl ftp site</a> or
elsewhere, and run them through your <code>gff_to_sqlite.R</code> program to get them into a database - say <code>genes.sqlite</code>. If not
<strong>please go and do that now</strong>.  For example, like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">Rscript --vanilla gff_to_sqlite.R --attributes Name biotype --input http://ftp.ensembl.org/pub/current_gff3/homo_sapiens/Homo_sapiens.GRCh38.110.chr.gff3.gz --output genes.sqlite</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Rscript --vanilla gff_to_sqlite.R --attributes Name biotype --input http://ftp.ensembl.org/pub/current_gff3/pan_troglodytes/Pan_troglodytes.Pan_tro_3.0.110.chr.gff3.gz --output genes.sqlite</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Rscript --vanilla gff_to_sqlite.R --attributes Name biotype --input http://ftp.ensembl.org/pub/current_gff3/camelus_dromedarius/Camelus_dromedarius.CamDro2.110.chr.gff3.gz --output genes.sqlite</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># ...and so on</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>I suggest to pick a few species for now - choose ones that interest you.  </p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>There are a lot to choose from.  For example you can get:</p><ul><li>Vertebrate genomes from <a href="http://ftp.ensembl.org/pub/current_gff3/" target="_blank" rel="noopener noreferrer">Ensembl</a></li><li>Protists from <a href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/protists/current/gff3/" target="_blank" rel="noopener noreferrer">Ensembl protists</a></li><li>Fungi from <a href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/fungi/current/gff3/" target="_blank" rel="noopener noreferrer">Ensembl fungi</a></li><li>Plants from <a href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/current/gff3/" target="_blank" rel="noopener noreferrer">Ensembl plants</a></li><li>or Bacteria from <a href="https://ftp.ensemblgenomes.ebi.ac.uk/pub/bacteria/current/gff3/" target="_blank" rel="noopener noreferrer">Ensembl bacteria</a></li></ul><p>Yet another place to look is the <a href="https://portal.darwintreeoflife.org" target="_blank" rel="noopener noreferrer">Darwin Tree of Life Data Portal</a>, which at the
time of writing has nearly 300 genomes in &#x27;Annotation complete&#x27; state.</p></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>What are all those species?</h5></div><div class="admonition-content"><p>If you&#x27;re confused about how all those species relate, it&#x27;s time to look at <a href="http://www.onezoom.org" target="_blank" rel="noopener noreferrer">OneZoom.org</a>:</p><iframe src="http://www.onezoom.org" width="100%" height="800px"></iframe><p>For example, did you know that Dolphins, like Camels, are <a href="http://www.onezoom.org/life/@Cetartiodactyla=622916" target="_blank" rel="noopener noreferrer">cloven-hoofed
ungulates</a>?  Did you know that all species of ape - except humans -
are <a href="http://www.onezoom.org/life/@Catarrhini=842867?img=best_any&amp;anim=flight#x-1130,y379,w6.2977" target="_blank" rel="noopener noreferrer">endangered or critically
endangered</a>?</p><p>Another place to look is  the <a href="https://www.ensembl.org/info/about/species.html" target="_blank" rel="noopener noreferrer">Ensembl species list</a> and the <a href="https://www.ensembl.org/info/about/speciestree.html" target="_blank" rel="noopener noreferrer">Ensembl
species tree</a>.  There are also similar pages for each of the
branches of life on the <a href="https://ensemblgenomes.org" target="_blank" rel="noopener noreferrer">Ensembl Genomes</a> site.</p></div></div><p>Pick some genomes to start with and load them into your database.  Once done, you are actually in a good shape for many
of our scientific questions.  Let&#x27;s start by counting genes:</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="counting-genes-in-sqlite">Counting genes in sqlite<a class="hash-link" href="#counting-genes-in-sqlite" title="Direct link to heading">​</a></h2><p>A simple way to count genes is to use sqlite. For example using the sqlite file you created in the last step, you can count all gene records like this:</p><div class="language-sh codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-sh codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">sqlite&gt; SELECT COUNT(*) FROM gff WHERE type == &#x27;gene&#x27; ;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>To run the above, you have to be working interactively in sqlite3 - you get there by typing <code>sqlite3 genes.sqlite</code> in
the shell. Type <code>.quit</code> again when you want to quit.  </p><p>Alternatively you can run from the shell directly, for example to produce a csv file with column names:</p><div class="language-sh codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-sh codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">% sqlite3 -csv -header genes.sqlite &quot;SELECT COUNT(*) FROM gff WHERE type == &#x27;gene&#x27;&quot;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><p>Of course, that just counts all the genes in the file - we&#x27;d like it split by species.  Simple! If you followed the
suggestions above your file will also have a <code>dataset</code> column that lets you differentiate the records for different
species. So you can make these counts for multiple species:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">sqlite&gt; .mode column</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sqlite&gt; .header on</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">sqlite&gt; SELECT dataset, COUNT(*) FROM gff WHERE type==&#x27;gene&#x27; GROUP BY dataset ;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>In my data, which includes spiny chromis, dromedary camels, red junglefowl, humans, mice, chimpanzees, and
malaria parasites, this gives:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">dataset                                      COUNT(*)  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">-------------------------------------------  ----------</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Acanthochromis_polyacanthus.ASM210954v1.104  24027     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Camelus_dromedarius.CamDro2.104.chr.gff3     18919     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Gallus_gallus.GRCg6a.104                     16666     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Homo_sapiens.GRCh38.104                      21451     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Mus_musculus.GRCm39.104                      25655     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Pan_troglodytes.Pan_tro_3.0.104.chr          22056     </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">PlasmoDB-54_Pfalciparum3D7                   5318      </span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Interestingly, both <a href="https://en.wikipedia.org/wiki/House_mouse" target="_blank" rel="noopener noreferrer">house mice</a> and <a href="https://en.wikipedia.org/wiki/Spiny_chromis" target="_blank" rel="noopener noreferrer">spiny
chromis</a> have more (annotated) genes than humans.
Chimpanzees have a similar number, while <a href="https://en.wikipedia.org/wiki/Red_junglefowl" target="_blank" rel="noopener noreferrer">Red
junglefowl</a> and <a href="https://en.wikipedia.org/wiki/Dromedary" target="_blank" rel="noopener noreferrer">Dromedary
Camels</a> have respectively 17% and 5% fewer (annotated)
protein-coding genes than humans. <em>Plasmodium falciparum</em> has about a quarter of the number of
genes. (But that&#x27;s still pretty impressive because its genome size is less than a hundredth that of
humans.)</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>To make the above query work the <a href="https://plasmodb.org/plasmo/app/downloads/Current_Release/" target="_blank" rel="noopener noreferrer"><em>P.falciparum</em> data from
PlasmoDB</a>, I kludged it by running this sql:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">UPDATE gff</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">SET type = &#x27;gene&#x27;, biotype = &#x27;protein_coding&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">WHERE dataset LIKE &#x27;PlasmoDB-%_Pfalciparum3D7&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">AND type == &#x27;protein_coding_gene&#x27; ;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>I wouldn&#x27;t generally recommend this type of manually-fix-the-data-until-it-works approach
(not least because it would have to be re-done every time we imported new data) but it&#x27;ll do for this tutorial.</p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="counting-genes-in-r-or-python">Counting genes in R or python<a class="hash-link" href="#counting-genes-in-r-or-python" title="Direct link to heading">​</a></h2><p>So how would we do this in R or python?  One of the messages of this tutorial is that the same operations are generally
possible in different languages, with minor differences in syntax.  The tabs below show a few different ways:</p><div class="tabs-container tabList_uSqn"><ul role="tablist" aria-orientation="horizontal" class="tabs"><li role="tab" tabindex="0" aria-selected="true" class="tabs__item tabItem_LplD tabs__item--active">R code</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LplD">R and sql combined</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LplD">dbplyr code</li><li role="tab" tabindex="-1" aria-selected="false" class="tabs__item tabItem_LplD">Python code</li></ul><div class="margin-top--md"><div role="tabpanel" class="tabItem_OmH5"><p>First we will load the whole dataframe into R.  To do this we have to open a &#x27;connection&#x27; to the database and then read
from it.  (We&#x27;ll filter down to genes here as well, so that we don&#x27;t use up too much memory).</p><div class="language-r codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-r codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">library</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> RSQLite </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">library</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> dplyr </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">db </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> DBI</span><span class="token operator" style="color:#393A34">::</span><span class="token plain">dbConnect</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> RSQLite</span><span class="token operator" style="color:#393A34">::</span><span class="token plain">SQLite</span><span class="token punctuation" style="color:#393A34">(</span><span class="token punctuation" style="color:#393A34">)</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;genes.sqlite&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">genes </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> dbGetQuery</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> db</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;SELECT * FROM gff WHERE type == &#x27;gene&#x27;&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now let&#x27;s group by dataset, and summarise just as we did in sql:</p><div class="language-r codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-r codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    genes</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token percent-operator operator" style="color:#393A34">%&gt;%</span><span class="token plain"> group_by</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> dataset </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token percent-operator operator" style="color:#393A34">%&gt;%</span><span class="token plain"> summarise</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> count </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> n</span><span class="token punctuation" style="color:#393A34">(</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You should see the same output as above, but now in R:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain"># A tibble: 7 × 2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  dataset                                                count</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  &lt;chr&gt;                                                  &lt;int&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">1 Acanthochromis_polyacanthus.ASM210954v1.110.gff3       24027</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">2 Camelus_dromedarius.CamDro2.110.chr.gff3               18919</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">3 Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.110.chr.gff3 18121</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">4 Homo_sapiens.GRCh38.110.chr.gff3                       21532</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">5 Mus_musculus.GRCm39.110.chr.gff3                        5946</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">6 Pan_troglodytes.Pan_tro_3.0.110.chr.gff3               22056</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">7 PlasmoDB-65_Pfalciparum3D7                              5318</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div><div role="tabpanel" class="tabItem_OmH5" hidden=""><p>Of course there&#x27;s no particular reason not to combine R and sql in the same code:</p><div class="language-r codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-r codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">dbGetQuery</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> db</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;SELECT dataset, COUNT(*) FROM gff WHERE type==&#x27;gene&#x27; GROUP BY dataset&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">                                                 dataset COUNT(*)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">1       Acanthochromis_polyacanthus.ASM210954v1.110.gff3    24027</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">2               Camelus_dromedarius.CamDro2.110.chr.gff3    18919</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">3 Gallus_gallus.bGalGal1.mat.broiler.GRCg7b.110.chr.gff3    18121</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">4                       Homo_sapiens.GRCh38.110.chr.gff3    21532</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">5                       Mus_musculus.GRCm39.110.chr.gff3     5946</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">6               Pan_troglodytes.Pan_tro_3.0.110.chr.gff3    22056</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">7                             PlasmoDB-65_Pfalciparum3D7     5318</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>(Use <code>as_tibble()</code> if you want this in the form of a tidyverse tibble).</p><p>This has one <strong>big advantage</strong> over the plain R code - it doesn&#x27;t load the whole dataframe into R,
but just the result of the query.</p><p>The same advantage can be had by using <a href="/whg-training-resources/programming/analysing_genes_across_species/Counting_genes_1/dbplyr.tidyverse.org]/">dbplyr</a> as well - see the next tab.</p></div><div role="tabpanel" class="tabItem_OmH5" hidden=""><p>Another way to do this uses <a href="https://dbplyr.tidyverse.org" target="_blank" rel="noopener noreferrer">dbplyr</a>.  dbplyr works just like dplyr, but transforms
its work into database queries under the hood.  </p><p>To try this, make sure you have a connection <code>db</code> to the database (as on the &#x27;R&#x27; tab).  You then use the dbplyr <code>tbl()</code>
function to represent a database table of data which you can then operate on.  LIie this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  db</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  %&gt;% tbl( &quot;gff&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  %&gt;% filter( type == &#x27;gene&#x27; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  %&gt;% group_by( dataset )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  %&gt;% summarise (count = n() )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  %&gt;% collect()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Interestingly, although this looks like R code, it is actually translating to sql code under the hood.  To see this, try
re-running it but changing the <code>collect()</code> call to <code>show_query()</code>.  You should see:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">SELECT `dataset`, COUNT(*) AS `count`</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">FROM `gff`</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">WHERE (`type` = &#x27;gene&#x27;)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GROUP BY `dataset`</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>In other words, dbplyr has worked out for itself how to crate the right query above from the R code.  LIke the combined
R-and-sql code, this has only loaded the bits of data it needs, not the whole dataframe, into memory.</p></div><div role="tabpanel" class="tabItem_OmH5" hidden=""><p>This can of course also be done in python / pandas.  First connect to the database:</p><div class="language-python codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-python codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token keyword" style="color:#00009f">import</span><span class="token plain"> pandas</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> sqlite3</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">db </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> sqlite3</span><span class="token punctuation" style="color:#393A34">.</span><span class="token plain">connect</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;genes.sqlite&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>And then read from it using <a href="https://pandas.pydata.org/docs/reference/api/pandas.read_sql.html" target="_blank" rel="noopener noreferrer">read_sql</a>:</p><div class="language-python codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-python codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">genes </span><span class="token operator" style="color:#393A34">=</span><span class="token plain"> pandas</span><span class="token punctuation" style="color:#393A34">.</span><span class="token plain">read_sql</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;SELECT * FROM gff WHERE type == &#x27;gene&#x27;&quot;</span><span class="token punctuation" style="color:#393A34">,</span><span class="token plain"> db </span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now let&#x27;s group by datasets, and summarise just as we did in sql:</p><div class="language-python codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-python codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    genes</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token punctuation" style="color:#393A34">.</span><span class="token plain">groupby</span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;dataset&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    </span><span class="token punctuation" style="color:#393A34">.</span><span class="token plain">size</span><span class="token punctuation" style="color:#393A34">(</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token punctuation" style="color:#393A34">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This produces:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">dataset</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Acanthochromis_polyacanthus.ASM210954v1.104    24027</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Camelus_dromedarius.CamDro2.104.chr.gff3       18919</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Gallus_gallus.GRCg6a.104                       16666</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Homo_sapiens.GRCh38.104                        21451</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Mus_musculus.GRCm39.104                        25655</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Pan_troglodytes.Pan_tro_3.0.104.chr            22056</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">PlasmoDB-54_Pfalciparum3D7                      5318</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">dtype: int64</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div></div></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/programming/analysing_genes_across_species/002_Counting_genes_1.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/programming/analysing_genes_across_species/getting_setup/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Getting setup</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/programming/analysing_genes_across_species/Counting_genes_2/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">What&#x27;s not a gene?</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#getting-data" class="table-of-contents__link toc-highlight">Getting data</a></li><li><a href="#counting-genes-in-sqlite" class="table-of-contents__link toc-highlight">Counting genes in sqlite</a></li><li><a href="#counting-genes-in-r-or-python" class="table-of-contents__link toc-highlight">Counting genes in R or python</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>