<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/sanger_sequence_data/introduction">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Viewing Sanger trace files | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Viewing Sanger trace files | The WHG training resources"><meta data-rh="true" name="description" content="If you got here, you should have downloaded a folder of Sanger sequence data.  (If not please [go and do that"><meta data-rh="true" property="og:description" content="If you got here, you should have downloaded a folder of Sanger sequence data.  (If not please [go and do that"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/introduction/">Viewing Sanger trace files</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/"><span itemprop="name">Analysing Sanger sequence data</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Viewing Sanger trace files</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Viewing Sanger trace files</h1><p>If you got here, you should have downloaded a folder of Sanger sequence data.  (If not please <a href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/#getting-the-data">go and do that
now</a>.)</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="whats-in-the-folder">What&#x27;s in the folder<a class="hash-link" href="#whats-in-the-folder" title="Direct link to heading">​</a></h2><p>Now open your data folder - you should see a list of files, something like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/file_listing-3090fc813aec966d5d3779dc348afafe.png" width="1448" height="572" class="img_E7b_"></p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>I&#x27;m viewing these on a mac - yours will look slightly different if you are working on Windows.</p></div></div><p>The files consist of Sanger Sequencing &#x27;DNA Electropherogram&#x27; results encoded in the <code>.ab1</code> format. To view one of the <code>.ab1</code>
files - double-click on it (or right-click and choose <code>Open with -&gt; FinchTV</code>). </p><p><strong>Note</strong> Please start with one of the <code>_GATA1</code> files.</p><p>You should see something like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/finchtv-415d1dc072575fa4f51ef11ed9b7e715.png" width="2478" height="1072" class="img_E7b_"></p><p>The files are also knows as <em>trace files</em> because they contain all the data that came out of the sequencing run. This includes
the chromatogram itself i.e. the measured flourescence intensities of the four bases incorporated at each point in the run
(presented as coloured lines), the inferred DNA sequence (presented at the top of the window if you turn on the right option),
and quality information about the bases (grey bars at the top of the window). </p><p>Try the icons along the top to see what they do (skip the load/save ones for now). You should be able to look at the information
in the file, alter how the chromatogram is viewed, and also see it in reverse orientation. This icon is particularly usefule:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/finchtv_wrap_button-e376fa422f009e8c13a0fb5cc2514b3e.png" width="80" height="76" class="img_E7b_"></p><p>as it lets you see the whole sequence at once. (You can also use the sliders at the left and right to zoom in and out of the
trace.)</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Questions</h5></div><div class="admonition-content"><p>Explore the data now and try to answer the following questions:</p><ul><li>How long is the sequence?</li><li>How long did it take to run?</li><li>What do you notice about the signal at different positions along the sequence?</li><li>Are there and &#x27;N&#x27; base calls?  Why?</li><li>Are there any non-N base calls you wouldn&#x27;t be confident in?  Why?  What do the quality scores for these bases look like?</li></ul></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="three-experiments-six-files">Three experiments, six files<a class="hash-link" href="#three-experiments-six-files" title="Direct link to heading">​</a></h3><p>You should have six files in your folder. These correspond to the three experiments you ran, which were something like this:</p><table><thead><tr><th>samples / assays</th><th>GATA1 site assay</th><th>SEA variant assay</th></tr></thead><tbody><tr><td>SEA deletion variant sample (S1)</td><td></td><td>Y</td></tr><tr><td>GATA1 control sample (GC)</td><td>Y</td><td></td></tr><tr><td>GATA1 &#x27;Vanuatu&#x27; mutation sample (G1)</td><td>Y</td><td></td></tr></tbody></table><p>The reason there are <em>six</em> files (not three) is that each product was sequenced twice - once on forward and once on reverse
strand.  This is typical of Sanger sequencing providers and provides an extra view of each sequence.</p><p>The filenames are of the form:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">[run_id]_[group]_[sample]_[assay]_[fwd or reverse]_[well id].ab1</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>which consists of:</p><ul><li>an identifier for the sequencing run (<code>471994201</code>), which the sequencing service has created to track this run;</li><li>an identifier (<code>Gr1</code> - <code>Gr11</code>) for the group;</li><li>the sample identifier (<code>G1</code>, <code>GC</code>, or <code>S1</code>);</li><li>the PCR assay used (<code>GATA</code> or <code>SEA</code>);</li><li>whether the file corresponds to forward strand (<code>F</code>) or reverse strand (<code>R</code>) sequence;</li><li>and an identifier for the well the sample was submitted in.</li></ul><p>Of these, the group id, sample id and assay name were of course generated by our experiment, while the sequencing provider
generated the other parts to track its process.</p><p>The primer pairs for the PCR reaction were:</p><table><thead><tr><th>assay</th><th>primer sequence</th></tr></thead><tbody><tr><td>GATA1 assay forward primer</td><td><code>CATTGTTGAGATGATTTGCTGGAGACACACAGATG</code></td></tr><tr><td>GATA1 assay reverse primer</td><td><code>GAAATACCAAGAGCTTCCTTGCACCAACAGCT</code></td></tr><tr><td>SEA variant assay forward primer</td><td><code>CTCTGTGTTCTCAGTATTGGAGGGAAGGAG</code></td></tr><tr><td>SEA variant assay reverse primer</td><td><code>ATATATGGGTCTGGAAGTGTATCCCTCCCA</code></td></tr></tbody></table><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Question</h5></div><div class="admonition-content"><p>Does the primer sequence appear in the resulting sequence file? Is this what you expected from the experiment?</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="aligning-the-sequence">Aligning the sequence<a class="hash-link" href="#aligning-the-sequence" title="Direct link to heading">​</a></h3><p>To go further than this, let&#x27;s use the <a href="https://genome.ucsc.edu/cgi-bin/hgBlat" target="_blank" rel="noopener noreferrer">UCSC Genome Browser BLAT tool</a> to align the
sequences to the human reference assembly and see where they lie.</p><p>There are a couple of ways to do this - the simplest one is to click the information icon and copy the sequence onto your
clipboard. Now visit the <a href="https://genome.ucsc.edu/cgi-bin/hgBlat" target="_blank" rel="noopener noreferrer">BLAT page</a> and paste in the sequence. It should look something
like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/BLAT-77b48a849f1228a023a362d714a82fb1.png" width="2058" height="662" class="img_E7b_"></p><p>For our purposes we will use the &quot;Feb 2009 (GRCh37/hg19)&quot; version of the genome reference assembly - make sure you have selected
this at the top.  (The reason is that the <a href="https://canvas.ox.ac.uk/courses/158645/modules#module_340266" target="_blank" rel="noopener noreferrer">resources used to design the
primers</a> are all from a few years ago - they report build 37
coordinates, rather than the more recent build 38, so it&#x27;s most convenient to use that here.)</p><p>Once you have pasted the sequence in and selected the assembly version - you&#x27;re ready to BLAT! Click &#x27;Submit&#x27; and wait a few
moments for the results to appear.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>Another related tool is <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank" rel="noopener noreferrer">NCBI BLAST</a>, which will align the sequence to a huge database
of known sequences (including from other organisms). You can try that directly from FinchTV by right-clicking on the sequence
and choosing &#x27;BLAST Sequence -&gt; Nucleotide BLAST&#x27;. Feel free to try this as well. However, UCSC Genome Browser provides probably
the best visualisation so we&#x27;ll use that here.</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="understanding-alignment-outputs">Understanding alignment outputs<a class="hash-link" href="#understanding-alignment-outputs" title="Direct link to heading">​</a></h3><p>You will hopefully see something like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/BLAT_results-3d224a2d37981755e1bba94774da2f4b.png" width="1536" height="554" class="img_E7b_"></p><p>You&#x27;ll probably notice a few things about the results - first, it will probably have several rows representing several different
alignments! For each of these, the output shows information about the part of the query sequence aligned (<code>QUERY</code>, <code>SCORE</code>,
<code>START</code>, <code>END</code>, and <code>QSIZE</code> columns) and information about where it aligned in the reference genome (<code>CHROM</code>, <code>STRAND</code>, <code>START</code>,
<code>END</code> and <code>SPAN</code>).</p><p>Look at the results and try to answer these questions:</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Questions</h5></div><div class="admonition-content"><ul><li><p>How many alignments did BLAT report?</p></li><li><p>What part of the query sequence was aligned in each result? How many results represent alignments of the whole query (or most of it?)</p></li><li><p>What level of identity did each alignment have?</p></li><li><p>Make sure you understand what the last five columns are showing. The <em>HBA1</em>/<em>HBA2</em> genes are on chromosome 16 - is that where
the alignments are?  Which strand are they on and how long did they span on the reference assembly?</p></li></ul></div></div><p>BLAT is a very &#x27;sensitive&#x27; alignment tool - it tries to find <em>all</em> reasonable alignments, even those between substrings.
However, we are mainly interested in full-length or nearly full-length alignments here. Before going on make sure you know which
row corresponds to the &#x27;best&#x27; alignment here.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="inspecting-an-alignment">Inspecting an alignment<a class="hash-link" href="#inspecting-an-alignment" title="Direct link to heading">​</a></h3><p>Let&#x27;s have a look at the best alignment in more detail. First, try clicking the &#x27;details&#x27; button. This page gives you a complete
description how the alignment to the reference assembly looks. Possibly the best thing to look at it the side-by-side alignment
at the bottom of the page.  It might look something like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">000001 tacntccctggatcaggccctggggcctgagtccggaagagaggtctgta 000050</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;&gt;&gt;&gt;&gt;&gt; ||| |||||||||||||||||||||||||||||||||||||||||||||| &gt;&gt;&gt;&gt;&gt;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">209565 tacatccctggatcaggccctggggcctgagtccggaagagaggtctgta 209614</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">000051 tggacacacccatcaatgggagcaccaggacacagatggaggctaatgtc 000100</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;&gt;&gt;&gt;&gt;&gt; |||||||||||||||||||||||||||||||||||||||||||||||||| &gt;&gt;&gt;&gt;&gt;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">209615 tggacacacccatcaatgggagcaccaggacacagatggaggctaatgtc 209664</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">000101 atgttgtagacaggatgggtgctgagctgccacacccacattattagaaa 000150</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;&gt;&gt;&gt;&gt;&gt; |||||||||||||||||||||||||||||||||||||||||||||||||| &gt;&gt;&gt;&gt;&gt;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">209665 atgttgtagacaggatgggtgctgagctgccacacccacattattagaaa 209714</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">000151 ataacagcacaggcttggggtggaggcgggacacaagactagccagaagg 000200</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;&gt;&gt;&gt;&gt;&gt; |||||||||||||||||||||||||||||||||||||||||||||||||| &gt;&gt;&gt;&gt;&gt;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">209715 ataacagcacaggcttggggtggaggcgggacacaagactagccagaagg 209764</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">000201 agaaagaaaggtgaaaagctgttggtgcaaggaagctctttgggtatttc 000250</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;&gt;&gt;&gt;&gt;&gt; ||||||||||||||||||||||||||||||||||||||||  |||||||| &gt;&gt;&gt;&gt;&gt;&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">209765 agaaagaaaggtgaaaagctgttggtgcaaggaagctctt..ggtatttc 209812</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You get: the query sequence (top rows), and the reference sequence (bottom rows). Every vertical line is a &#x27;match&#x27; and missing
lines are mismatches, insertions or deletions between the sequences.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Question</h5></div><div class="admonition-content"><p>The original plan was to sequence samples carrying the gain-of-function mutation from <a href="https://doi.org/10.1038/s41467-021-23980-6" target="_blank" rel="noopener noreferrer">Bozhilov et al Nature Communications
2021</a>. As that paper says:</p><blockquote><p>&quot;The candidate SNV (a T to C transition at coordinate hg19 chr16:209,709)
<!-- -->[...]<!-- --> is thought to down-regulate α-globin expression <!-- -->[and]<!-- --> is found within an
unremarkable non-coding region of the α-globin locus.</p></blockquote><p>Does your sequence have the &#x27;T&#x27; allele or the &#x27;C&#x27; allele at position 209,709? Does it have a GATA motif there? What about the
refrence sequence? Is this what you expect given which sample it is?</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="visualising-alignments-on-the-genome-browser">Visualising alignments on the genome browser<a class="hash-link" href="#visualising-alignments-on-the-genome-browser" title="Direct link to heading">​</a></h3><p>Finally let&#x27;s look at all our alignments on the genome browser.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Challenge</h5></div><div class="admonition-content"><p>Create a FASTA file with all of your sequences in.  There are a few ways to do this, but here is one:</p><ul><li><p>For each .ab1 file, load the file into FinchTV and inspect it to check it looks sensible. You should see a nice, clean trace,
with clear signal for bases - except maybe at the start and end.</p></li><li><p>If the trace looks good, go ahead and export it as a FASTA file. You can do this by clicking on the <code>File -&gt; Export -&gt; DNA
Sequence:FASTA</code> option. Choose the default options should create a new file with the same name but a <code>.seq</code> ending in the same
directory. (This is a FASTA file that just contains the sequence.)</p></li><li><p>Now using a text editor (such as <code>Notepad</code>) or your UNIX command-line skills, concatenate all these FASTA files together into
a single file.</p></li></ul><p>For example, if you are on Windows 10 you could do this by opening the Ubuntu terminal and doing:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">cd /mnt/c/Users/[your usename]/Desktop/HBA_practical_group_[n]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">cat *.seq &gt; combined.fasta</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>(If on a Mac or linux, you could try <code>cd ~/Desktop/HBA_practial_group_[n]</code> instead.)</p><p>Finally - open the file in a text editor (like <code>Notepad</code>) and let&#x27;s add in the original primer sequences. In FASTA format they
should look something like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">&gt;GATA1_fwd</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">CATTGTTGAGATGATTTGCTGGAGACACACAGATG</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;GATA1_rev</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">GAAATACCAAGAGCTTCCTTGCACCAACAGCT</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;SEA_fwd</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">CTCTGTGTTCTCAGTATTGGAGGGAAGGAG</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">&gt;SEA_rev</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">ATATATGGGTCTGGAAGTGTATCCCTCCCA</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Add these lines to the end of <code>combined.fasta</code> and save it.</p></div></div><p>You should now have a file <code>combined.fasta</code> with all your sequences in - at least, the ones that looked like they sequenced
properly.  </p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>The SEA sequences look a bit different to the GATA1 sequences. What&#x27;s the difference? Is there anything wrong with these traces?</p></div></div><p>Now let&#x27;s BLAT these all at once. Open a new window with the <a href="https://genome.ucsc.edu/cgi-bin/hgBlat" target="_blank" rel="noopener noreferrer">UCSC Genome Browser BLAT
tool</a> and paste in your FASTA file. Make sure you have selected the <code>Feb. 2009
(GRCh37/hg19)</code> reference assembly and click &#x27;Submit&#x27;. You should see <em>lots</em> of results - all the alignments of each of the
sequences. This time, let&#x27;s visualise these on the genome by clicking on &#x27;Build a custom track with these results&#x27; - you should
see something like:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/BLAT_custom_track-797f12e013e7f5b23e85c1759b02b7ef.png" width="2112" height="428" class="img_E7b_"></p><p>To visualise them, find the first alignment on chromosome 16 (probably the top row) and click the first &#x27;browser&#x27; link.</p><p>Congratulations!  You can now see how the alignments are arranged along the genome.</p><p>The <strong>Challenge questions</strong> below are based on using the genome browser to explore your alignments.</p><p>Here are some suggestions on how to do this. First, it&#x27;s good to get a bird&#x27;s eye view by zooming out. You will probably want to
zoom out about 300x to see all the alignments, so click the &#x27;zoom out 10x&#x27; button a couple of times and &#x27;zoom out 3x&#x27;.  </p><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>Before going on, it&#x27;s worth making a note of the region shown - then if you get &#x27;lost&#x27; you can easily get back here. The region
is shown in the box at the top left - it will look something like <code>chr16:187,324-260,34</code>. Make a note of this now before going
on.</p></div></div><p>To move around, you can click and drag the canvas area (try to grab the &#x27;white&#x27; bit as opposed to any of the objects), or use
the zoom buttons to zoom in and out. Another really useful way to zoom around is to select a region to zoom to, by dragging
along the ruler at the top. Try this now - if you can get lost you can always zoom out again or reset to your coordinates noted
above.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Challenge questions</h5></div><div class="admonition-content"><p>Use the genome browser to explore the alignments now.  Here are some questions to consider.</p><ul><li><p>Do the PCR primer sequences align on the forward or reverse strand - is this what you expect? (Note: you can click on the
alignment to get more information on it. You can always get back again by clicking the back button in your browser, or the
&#x27;Genome Browser&#x27; button at the top of the page, or else the &#x27;browser&#x27; button in the alignments.)</p></li><li><p>Where do the sequenced segments lie compared to the PCR primers? Are they directly on top or not?  Is this what you expect?</p></li><li><p>Zoom into the alignments near the GATA1 site. Which of your sequences carry the mutant &#x27;C&#x27; base at chr16:709,209? Is this what
you expect?</p></li><li><p>The SEA variant sequence seems to be split into two segments. Zoom into these to take a look at them. One is around
chr16:215,000, which is upstream of <em>HBM</em>, and the other around <code>chr16:235,000</code>, downstream of <em>HBQ1</em>. Why do your sequences
split into two alignments like this?  Are they in the expected place?</p></li></ul></div></div><p><strong>Note.</strong> You can compare your results to the informatino <a href="https://canvas.ox.ac.uk/courses/158645/modules#module_340266" target="_blank" rel="noopener noreferrer">on
canvas</a> - in particular this figure:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/deletion_variant_diagram-a6f5fb2a1fe81d70a530ef6f7cc0f3f1.png" width="1365" height="803" class="img_E7b_"></p><p>which comes from <a href="https://ezproxy-prd.bodleian.ox.ac.uk:2102/10.1182/blood.V95.1.360" target="_blank" rel="noopener noreferrer">Chong et. al <em>Blood</em>  (2000) 95 (1): 360–362</a></p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Extra challenge</h5></div><div class="admonition-content"><p>Look back at the site chr16:709,209 and look at the DNA sequence there. Does it look like a T-&gt;C transition at that site creates
a GATA-1 site, as the <a href="https://doi.org/10.1038/s41467-021-23980-6" target="_blank" rel="noopener noreferrer">authors of this paper</a> claim? Why?</p><p><strong>Hint.</strong> GATA transcription factors are named after the motif they bind. But they might bind in either orientation - forward
sequence, or <em>reverse complement</em>. (To reverse-complement the sequence, you have to first <strong>reverse</strong> the sequence, then swap
all the bases to their complementary ones <code>C &lt;-&gt; G</code> and <code>A &lt;-&gt; T</code>.</p><p><strong>Hint 2</strong>. If you want to see empirical evidence about what GATA1 really binds, you can look at
<a href="https://factorbook.org" target="_blank" rel="noopener noreferrer">Factorbook</a>. Searching for GATA1 produces <a href="https://www.factorbook.org/tf/human/GATA1/motif/ENCSR000EFT" target="_blank" rel="noopener noreferrer">this
result</a>. GATA1 really does bind a GATA sequence! (How closely do
the other bases in the motif match this particular sequence?)</p></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="next-steps">Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">​</a></h2><p>The UCSC genome browser has an incredible wealth of information on it about many of the thousands of experiments conducted on
the human genome (as well as mouse and other genomes). We will explore this a bit more in the session later today. Feel free to
click around and explore the data - but be warned the amount and breadth of data can be pretty overwhelming at first!</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/sanger_sequence_data/introduction.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Analysing Sanger sequence data</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Analysing paired-end sequencing data</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#whats-in-the-folder" class="table-of-contents__link toc-highlight">What&#39;s in the folder</a><ul><li><a href="#three-experiments-six-files" class="table-of-contents__link toc-highlight">Three experiments, six files</a></li><li><a href="#aligning-the-sequence" class="table-of-contents__link toc-highlight">Aligning the sequence</a></li><li><a href="#understanding-alignment-outputs" class="table-of-contents__link toc-highlight">Understanding alignment outputs</a></li><li><a href="#inspecting-an-alignment" class="table-of-contents__link toc-highlight">Inspecting an alignment</a></li><li><a href="#visualising-alignments-on-the-genome-browser" class="table-of-contents__link toc-highlight">Visualising alignments on the genome browser</a></li></ul></li><li><a href="#next-steps" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>