<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-population_genetics/simulation/popgen_simulation">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Genetic drift | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/population_genetics/simulation/popgen_simulation/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Genetic drift | The WHG training resources"><meta data-rh="true" name="description" content="The Wright-Fisher model is about the simplest population-genetic model there is.  It consists of:"><meta data-rh="true" property="og:description" content="The Wright-Fisher model is about the simplest population-genetic model there is.  It consists of:"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/population_genetics/simulation/popgen_simulation/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/population_genetics/simulation/popgen_simulation/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/population_genetics/simulation/popgen_simulation/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/population_genetics/">Population genetics</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/population_genetics/plotting_haplotypes/">Visualising haplotypes in R</a><button aria-label="Toggle the collapsible sidebar category &#x27;Visualising haplotypes in R&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/population_genetics/simulation/">Simulating populations</a><button aria-label="Toggle the collapsible sidebar category &#x27;Simulating populations&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/population_genetics/simulation/popgen_simulation/">Genetic drift</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/population_genetics/simulation/mutations/">Adding mutations</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/population_genetics/simulation/statistics/">Some population-genetic statistics</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/population_genetics/principal_components_analysis/">Principal components analysis</a><button aria-label="Toggle the collapsible sidebar category &#x27;Principal components analysis&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/population_genetics/admixture_and_biobanks/">Population structure and admixture</a><button aria-label="Toggle the collapsible sidebar category &#x27;Population structure and admixture&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/population_genetics/simulation/"><span itemprop="name">Simulating populations</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Genetic drift</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Genetic drift</h1><p>The Wright-Fisher model is about the simplest population-genetic model there is.  It consists of:</p><ul><li>A population of <em>N</em> individuals (which stays fixed over time).</li><li>non-overlapping generations</li><li>in each generation, each individual &#x27;samples&#x27; its parent uniformly from individuals in the parent generation.</li></ul><p>It&#x27;s pretty simple to implement, but even so it turns out to have many of the key features of real populations. We are going
to implement it.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="getting-started">Getting started<a class="hash-link" href="#getting-started" title="Direct link to heading">​</a></h2><p>As stated above there isnt really anything about <em>genetics</em> in the model. So to give ourselves some genetic data to work with,
we&#x27;ll also generate a set of haplotypes and have them evolve through the model.</p><p>Let&#x27;s get started.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="generating-a-starting-set-of-haplotypes">Generating a starting set of haplotypes<a class="hash-link" href="#generating-a-starting-set-of-haplotypes" title="Direct link to heading">​</a></h3><p>Let&#x27;s imagine we are looking at set of <em>L = 20</em> SNPs to start with and that there are (say) up to 10 different haplotypes in our population at the start of the simulation:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">L = 20</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">H = 10</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">haplotypes = matrix( NA, nrow = H, ncol = L )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>To generate some simple haplotypes to look at, let&#x27;s take a simple situation in which the variants are all biallelic (we&#x27;ll encode the alleles as <code>0</code> or <code>1</code>), they all start off independent of each other, and they all have 20% frequency.  We can simulate using <code>rbinom()</code> like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">haplotypes[,] = rbinom( n = H*L, size = 1, prob = 0.2 )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Here&#x27;s function to make a quick plot of our starting haplotypes:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.haplotypes &lt;- function( haplotypes ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  image(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    t(haplotypes),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    x = 1:ncol(haplotypes),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    y = 1:nrow(haplotypes),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlab = &quot;variant&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylab = &quot;haplotype&quot;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>If you call it</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.haplotypes( haplotypes )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You should see something like this:</p><p><img loading="lazy" alt="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASwAAADICAYAAABS39xVAAAEGWlDQ1BrQ0dDb2xvclNwYWNlR2VuZXJpY1JHQgAAOI2NVV1oHFUUPrtzZyMkzlNsNIV0qD8NJQ2TVjShtLp/3d02bpZJNtoi6GT27s6Yyc44M7v9oU9FUHwx6psUxL+3gCAo9Q/bPrQvlQol2tQgKD60+INQ6Ium65k7M5lpurHeZe58853vnnvuuWfvBei5qliWkRQBFpquLRcy4nOHj4g9K5CEh6AXBqFXUR0rXalMAjZPC3e1W99Dwntf2dXd/p+tt0YdFSBxH2Kz5qgLiI8B8KdVy3YBevqRHz/qWh72Yui3MUDEL3q44WPXw3M+fo1pZuQs4tOIBVVTaoiXEI/MxfhGDPsxsNZfoE1q66ro5aJim3XdoLFw72H+n23BaIXzbcOnz5mfPoTvYVz7KzUl5+FRxEuqkp9G/Ajia219thzg25abkRE/BpDc3pqvphHvRFys2weqvp+krbWKIX7nhDbzLOItiM8358pTwdirqpPFnMF2xLc1WvLyOwTAibpbmvHHcvttU57y5+XqNZrLe3lE/Pq8eUj2fXKfOe3pfOjzhJYtB/yll5SDFcSDiH+hRkH25+L+sdxKEAMZahrlSX8ukqMOWy/jXW2m6M9LDBc31B9LFuv6gVKg/0Szi3KAr1kGq1GMjU/aLbnq6/lRxc4XfJ98hTargX++DbMJBSiYMIe9Ck1YAxFkKEAG3xbYaKmDDgYyFK0UGYpfoWYXG+fAPPI6tJnNwb7ClP7IyF+D+bjOtCpkhz6CFrIa/I6sFtNl8auFXGMTP34sNwI/JhkgEtmDz14ySfaRcTIBInmKPE32kxyyE2Tv+thKbEVePDfW/byMM1Kmm0XdObS7oGD/MypMXFPXrCwOtoYjyyn7BV29/MZfsVzpLDdRtuIZnbpXzvlf+ev8MvYr/Gqk4H/kV/G3csdazLuyTMPsbFhzd1UabQbjFvDRmcWJxR3zcfHkVw9GfpbJmeev9F08WW8uDkaslwX6avlWGU6NRKz0g/SHtCy9J30o/ca9zX3Kfc19zn3BXQKRO8ud477hLnAfc1/G9mrzGlrfexZ5GLdn6ZZrrEohI2wVHhZywjbhUWEy8icMCGNCUdiBlq3r+xafL549HQ5jH+an+1y+LlYBifuxAvRN/lVVVOlwlCkdVm9NOL5BE4wkQ2SMlDZU97hX86EilU/lUmkQUztTE6mx1EEPh7OmdqBtAvv8HdWpbrJS6tJj3n0CWdM6busNzRV3S9KTYhqvNiqWmuroiKgYhshMjmhTh9ptWhsF7970j/SbMrsPE1suR5z7DMC+P/Hs+y7ijrQAlhyAgccjbhjPygfeBTjzhNqy28EdkUh8C+DU9+z2v/oyeH791OncxHOs5y2AtTc7nb/f73TWPkD/qwBnjX8BoJ98VQNcC+8AABOXSURBVHgB7d0LbBRV38fxf7kJFstdQQsoReUiKIIiEbnEQCMgwVe56MtNIAJGvFSiEeMl+MRKvCAieWwUqwQ0gAoCNqFV8VEEgyUoYBFeKAReFQoiFF4opWXfPfM8bahddGfmzMyene8ka3dnzplzzucsP2dvMymR6CIsCCCAgAECdQzoI11EAAEELAECiycCAggYI0BgGTNVdBQBBAgsngMIIGCMAIFlzFTRUQQQILB4DiCAgDECBJYxU0VHEUCAwOI5gAACxggQWMZMFR1FAAECi+cAAggYI0BgGTNVdBQBBAgsngMIIGCMAIFlzFTRUQQQqGcKwd69e2XFirnSrdvVpnQ5kH4umfqerXb/O2eirfKJWNjumL0eQzKYemn0r3/9KP/4xzuOmtAeWBUVFXLixAlp1qyZow5dqNLBgwdl9OiBcsUVaRcqwvqowCd7S2w5DBrUxVb5RCxsd8xejyEZTL02crp/LS8Jy8vLZdasWdK2bVtp0KCBNG/eXFJTU+W6666T3Nxcp32jHgIIIFBDQMsR1owZM0QdAX322WfSoUMHK6xKS0ulqKhIHn30USkrK5Pp06fXaJgHCCCAgF0BLUdY+fn5kpOTI927d5fGjRtLSkqKNGnSRPr06SPz5s2TlStX2u0X5RFAAIFaAloCS730W7duXa2dqxVr1qyRVq1axdzGSgQQQMCOgJaXhLNnz5b77rtP5s6dKxkZGZKWlibHjx+XHTt2iHoTPi8vz06fKIsAAgjEFNASWD169JAtW7bIxo0bZd++fdb7WeqoSr1v1a9fP+slYszW/7Ry7dq1kp2d/ae1/35YUlIiS5fO5FPCmDqsRCAcAloCS1E1bNhQBg4c6EotMzNT1C3WsnjxYqlXrzLWJtYhgEBIBLS8hxUSK4aJAAIBC2g5wnr11Vfl7NmzFxxKp06dZMSIERfczgYEEEAgHgEtgaXet3rzzTdlwoQJ1new/twwnxL+WYTHCCDgREBLYM2fP1/OnTtn3RYsWOCkH9RBAAEE/lZA23tYc+bMEfXt9pMnT/5toxRAAAEEnAhoOcJSDatvuC9ZssRJH6iDAAIIxCWgLbDiai3BC01Pmeh5D/8Zec9WG3b7ZHf/tjoTLZxo/bHbf1XeayMnffKyTjLMWZWPtpeEVTvkLwIIIOCVAIHllSz7RQAB7QIElnZSdogAAl4JEFheybJfBBDQLkBgaSdlhwgg4JUAgeWVLPtFAAHtAgSWdlJ2iAACXgkQWF7Jsl8EENAuQGBpJ2WHCCDglQCB5ZUs+0UAAe0CBJZ2UnaIAAJeCRBYXsmyXwQQ0C5AYGknZYcIIOCVgFFna8ju9byknqqI28Lur/LtllcdsftL+Lg7/5+CTvpktw3Ty9s1sjtndvefaJ6m9/98T46wztfgPgIIJLQAgZXQ00PnEEDgfAEC63wN7iOAQEILeBZY6qIUnN89oeeeziFgnICWwFIXn3jllVfkzjvvlHXr1smqVavksssuk8svv1weeOABgsu4pwUdRiAxBbR8SvjSSy/J7t27ZciQIfLII49IRUWFFVrXXnutZGVlyfLly+X+++//W4E9e/bI+vXrY5bbsGGDdRmxmBtZiQACoRDQEliffvqpbNq0ybqI6qFDh+TIkSPSp08fC/Cpp56yQiuewAqFOINEAAHHAloCq3PnzlJQUCADBw6Ur7/+Wk6fPl3doa1bt8qNN95Y/fiv7mRkZIi6xVrq1q0r+Yvzo5vOxdrMOgQQCIGAlsB6/PHHZdKkSVJcXCwPP/ywnDhxQlSIXX/99dZLvK+++ioElAwRAQS8FtASWOrlX1FRkRw9elRatGghZ86ckbVr18qxY8ckNzdXGjVq5PU42D8CCIRAQEtgKaeUlBQrrNT9iy66SIYPH67usiCAAALaBLR8rUFbb9gRAggg8BcCBNZf4LAJAQQSSyAlEl0Sq0uxe7N48WLp2bMy+mZ+euwCGtba/RW/ajKZfgmvgbDWLuyaOvH0o41aA0viFXY97VKcvupSea94s91qVnmOsByxUQkBBIIQILCCUKdNBBBwJEBgOWKjEgIIBCFAYAWhTpsIIOBIgMByxEYlBBAIQoDACkKdNhFAwJEAgeWIjUoIIBCEAIEVhDptIoCAIwECyxEblRBAIAgBAisIddpEAAFHAgSWIzYqIYBAEAIEVhDqtIkAAo4ECCxHbFRCAIEgBLSdwC+Izutu08mZAnT3gf3ZFzB93uyeHcHr8Xq9/4KCIvuT/J8aHGE5pqMiAgj4LUBg+S1Oewgg4FiAwHJMR0UEEPBbwLPAKisrk7Nnz/o9HtpDAIEkFtASWPv375fx48dLYWGhHD58WCZPniytW7eWpk2bWtcrLC8vT2JChoYAAn4JaAmsZ599Vtq1ayddu3aV+fPnS0VFhWzfvl3UVZ/VRVVfeOEFv8ZDOwggkMQCWr7WoC5P//PPP0uDBg1kxYoVsnLlSklP//fFIlRYTZs2LS5CdfHV7OzsmGUPHTokS5fOjLmNlQggEA4BLYF1zTXXyKJFi2TKlCkyYMAAycvLkxkzZliCa9askauvvjouzczMTFG3WIu6ak79+pWxNrEOAQRCIqAlsBYsWCDDhg2ThQsXSseOHWXmzJny7rvvSp06daS0tFTUERgLAggg4FZAS2BlZGRIUVGRFBQUyM6dO633s5o1a2YdWQ0dOlTq1dPSjNuxUh8BBAwX0JYkKSkpMnjwYOtmuAndRwCBBBXQ8ilhgo6NbiGAQJIJEFhJNqEMB4FkFiCwknl2GRsCSSag7T2sZHCxe5oPNWavT8VhuqtdnzDOgV0j058TbvrPEZYbPeoigICvAgSWr9w0hgACbgQILDd61EUAAV8FCCxfuWkMAQTcCBBYbvSoiwACvgoQWL5y0xgCCLgRILDc6FEXAQR8FSCwfOWmMQQQcCNAYLnRoy4CCPgqQGD5yk1jCCDgRoDAcqNHXQQQ8FWAwPKVm8YQQMCNAIHlRo+6CCDgq4BRZ2vI7vW8pJ6qiBvI7q/g7ZaPuyM+FrR7toNkGLNd3kQz8ro/Xu/frr+b8hxhudGjLgII+CpAYPnKTWMIIOBGgMByo0ddBBDwVcCzwCorK7OuSejraGgMAQSSWsCzwPr4448lKysrqfEYHAII+Cug5VNCdSn6I0eO1Oh5eXm5VFRUiAquESNGSG5ubo3tsR6oi7B+8cUXsTbJpk2bpLLyXMxtrEQAgXAIaAksFUaTJk2SsWPHyoQJEyy5lStXysaNG2XOnDmSmpoal2bDhg2lVatWMctecskl8kdKzE2sRACBkAhoCay+fftKYWGhPPTQQ9bLwJycHGnZsqU0btxY2rdvHzelKnuh8mfOnJH8d9dE98VRVtygFEQgyQS0BJYySUtLk0WLFsmyZcukX79+0rt3b6lbt26ScTEcBBAIUkD7m+6jRo2S/Px86z2t1q1bBzk22kYAgSQT0HaEdb5Lenq6rF69+vxV3EcAAQRcC2g/wnLdI3aAAAIIXECAwLoADKsRQCDxBDx5SejVMJ8qfF46d073avdJsd8wnn3B7sQlmpHX/fF6/3b93ZTnCMuNHnURQMBXAQLLV24aQwABNwIElhs96iKAgK8CBJav3DSGAAJuBAgsN3rURQABXwUILF+5aQwBBNwIEFhu9KiLAAK+ChBYvnLTGAIIuBEgsNzoURcBBHwVILB85aYxBBBwI0BgudGjLgII+CpAYPnKTWMIIOBGgMByo0ddBBDwVcCoszVk93peUk9VxA2UTL9Sj3vQNgtOT5los4a94nbnwG55e71xVtqukd0xeL1/Z6NOzFocYSXmvNArBBCIIUBgxUBhFQIIJKYAgZWY80KvEEAghoDngVVZWSnqmoIsCCCAgFsBLYF14MABGT9+vHXh1EGDBsnu3bur+7V8+XIZN25c9WPuIIAAAk4FtATW3LlzpU2bNtbVn/v06WNdSHXXrl1O+0Q9BBBAIKaAlq815OXlyZYtW6RRo0Yye/Zs6dKli2RmZsr69etjNnqhlWvXrpXs7OyYm0tKSqRH9OUlCwIIhFdAS2CpgCosLJTbbrvNkhwzZoz8+uuvcscdd8jUqVPj1lUhp26xlo0bN8qS/5omlQePxtrMOgQQCIGAlpeE06ZNk5EjR8qcOXOqybKysuTuu++Wxx57rHoddxBAAAE3AlqOsAYPHix79uyR4uLiGn157rnnpH///ta2Ght4gAACCDgQ0BJYqt3U1FTp1q1brS4MGDBA1I0FAQQQcCug5SWh205QHwEEEIhHgMCKR4kyCCCQEAIEVkJMA51AAIF4BFIi0SWegkGXUV9raNfuoFxxRVrQXaF9BLQK2D29jN3G7Z7uxu7+7ZYvKCiSQYNm2K1mlecIyxEblRBAIAgBAisIddpEAAFHAgSWIzYqIYBAEAIEVhDqtIkAAo4ECCxHbFRCAIEgBAisINRpEwEEHAkQWI7YqIQAAkEIEFhBqNMmAgg4EiCwHLFRCQEEghAgsIJQp00EEHAkQGA5YqMSAggEIUBgBaFOmwgg4EhA2wn8HLVuo1Ljxo1l8+afozU62ahlRtHjx0ujHY1ISkqKGR122Uv1c/uKigqpX9+Yp5/LEYtUVp6Lzq9InTq1jxH+72JvHXbs+F/X/XeyAzXHzZs3r1V127b/if74udbquFYYc7YGNZpt27bJyZMn4xqYSYVmzZold911V8zJNWkc8fb17Nmzoi4N98QTT8Rbxfhy6opQLVq0kF69ehk/lngH8OKLL8rChQtrFW/durVcddVVtdbHs8LbaI+nBzbKxDoFs43qCVu0bdu2VmCpv2FYysrKZNmyZTJ27NgwDNca4++//y7t27eXESNGhGbM77zzjqjrlOpcah+f6tw7+0IAAQQ0ChBYGjHZFQIIeCtAYHnry94RQECjAIGlEZNdIYCAtwIElre+7B0BBDQK1H0+umjcH7tyINCqVSvJyMiIfi+pvoPa5lVR30Vq06aNdOjQwbzOO+xxkyZN5MorrxT1NyyLmuOOHTtqHa5R38PSOnJ2hgACxgnwktC4KaPDCIRXgMAK79wzcgSMEyCwjJsyOoxAeAUIrPDOPSNHwDgBAsu4KaPDCIRXgMAK79wzcgSMEyCwjJsyOoxAeAUIrPDOve8jVyd0i6iz94VkUWOtrKwMyWj9GSaB5Y9zzFYKCwulXbt2NW6//PJLzLKmrzx06JD1zfbi4uLqofzxxx8yatQoSU9PF3Wus82bN1dvS4Y748aNk5dffrnGUG6++eYa852Tk1Nju4kP1P+InnzySenevbt1e+aZZ0SdpFEtuueYwArwGaICq2/fvvLNN99U39TZGJNtWbJkidx+++1y8ODBGkObPn269dONffv2SXZ2tgwfPlzUyf1MX7Zu3SpDhw6V1atX1xiKOonfzp07q+dazfu9995bo4yJD95//33Zvn27bNiwwbp9//33smjRImsouueYwArwGfLjjz9Kv379rHO5q9/XqTNS1q1bN8Ae6W+6vLxccnNz5aOPPpKmTZvWaCAvL0+ysrKkXr16MmzYMFFhrf4Rm768/fbbMn78+FpnVFXz3bNnT2nZsqWcOHHCOrJMS0szfbjWUdW8efNEXXdB3Xr06CHr16+3xqV7jgmsAJ8uP/zwg7z++uvSv39/62WCOrd7si0NGjSQzz//XDp1qnnxkGPHjsmZM2esf7xVY1aBVVJSUvXQ2L/z58+X0aNH1+q/mu9du3bJDTfcYJ3b/ZZbbpHjx4/XKmfaiptuuqn6R86nTp2SDz/8UIYMGSJezDGBFeCzo2vXrrJgwQLZu3evfPfdd9aFGdTLhjAsR44ckdTU1BpDVY+T8SIjVYNUV5CZOnWqFVr79++3jizVkWeyLOq9LHWefvVe1siRI8WLOTbqIhTJMrFV41An6a9aevfubZ2wf8WKFTJlypSq1Un7V11BRr0sOn8pLS21Tjtz/rpkuj9x4sTq4Vx66aUyYcIEWbp0qUyePLl6val3VFiNGTPGOqpSLwPV4sUcc4QV0DNEvbn82muviXqPp2pRH4GH5XxJzZo1E/Vy8bfffqsavqijjmS+ctAnn3wi6g35qiVZ5vvcuXOiwvjw4cPWBw0NGza0hujFHBNYVc8en/+qSVVPYPV6Xy0//fSTqE9XMjMzfe5JcM3dc8898sYbb4h6wufn51sfhXfp0iW4Dnnc8oEDB+Tpp5+2voumji4/+OAD65NRj5v1fPdvvfWWFcSrVq0S9bxWQazmVC3a5zj65TaWgAS+/fbbyK233hqJfgcpEv0ELRL9NC2gnvjTbPTMqpHdu3dXN7Znz55I586dI9EzcUaiZ1yNfPnll9XbkuHOgw8+GIl+XaN6KNGQikTfjI9EQzkSfbkUiX5PK3L69Onq7abeUfMXzaYat+ib7tZwdM8xZxy1/j8Q7H+OHj1qfeQf6zLmwfbMn9bVSwl1muiwLOqTNLVcfPHFYRmy9XJRxxwTWKF5yjBQBMwX4D0s8+eQESAQGgECKzRTzUARMF+AwDJ/DhkBAqERILBCM9UMFAHzBQgs8+eQESAQGgECKzRTzUARMF+AwDJ/DhkBAqERILBCM9UMFAHzBQgs8+eQESAQGgECKzRTzUARMF+AwDJ/DhkBAqERILBCM9UMFAHzBQgs8+eQESAQGgECKzRTzUARMF+AwDJ/DhkBAqER+H92MMy/w+sICgAAAABJRU5ErkJggg==" width="300" height="200" class="img_E7b_"></p><h3 class="anchor anchorWithStickyNavbar_mojV" id="generating-a-population">Generating a population<a class="hash-link" href="#generating-a-population" title="Direct link to heading">​</a></h3><p>We will now generate a set of <em>N</em> (haploid) individuals each of whom has one of the starting haplotypes. To keep track of the
evolution over <em>G</em> generations, let&#x27;s record these haplotypes in the first row of a matrix:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">N = 100 # number of individuals</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">G = 100 # number of generations</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">population = matrix(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  NA, nrow = G, ncol = N,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  dimnames = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    sprintf( &quot;generation=%d&quot;, 1:G ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    sprintf( &quot;individual=%d&quot;, 1:N )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>To generate our starting population let&#x27;s randomly sample our generated haplotypes for the first generation:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">population[1,] = sample( 1:H, size = N, replace = TRUE )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now we can plot what our population looks like:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.haplotypes( haplotypes[population[1,],] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Which should show something like this:</p><p>![img]<!-- -->(/Population genetics/solutions/initial_population.png)</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="evolving-the-population">Evolving the population<a class="hash-link" href="#evolving-the-population" title="Direct link to heading">​</a></h2><p>In the Wright-Fisher model, in each generation, each individual samples its haplotype from a randomly chosen parent in the generation before.  Let&#x27;s write a function to generate a new generation given the current one (a row of <code>population</code>):</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">evolve &lt;- function( currentGeneration ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  parents = sample( 1:N, size = N, replace = TRUE )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  return( currentGeneration[parents] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>That was pretty easy!  Now we can evolve the population:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">for( generation in 2:G ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  population[generation,] = evolve( population[generation-1,] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Look at the result before and after evolution:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">layout( matrix( 1:2, nrow = 2 ) )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">par( mar = c( 3, 3, 1, 1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.haplotypes( haplotypes[population[1,],])</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.haplotypes( haplotypes[population[G,],])</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Question.</strong> What features of this do you notice?</p><p><strong>Note.</strong> It might be worth emphasising here that there is nothing going on in our model except
inheritance - i.e. genetic drift.  Any structure that appears is purely due to random inheritance.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="how-frequencies-evolve">How frequencies evolve<a class="hash-link" href="#how-frequencies-evolve" title="Direct link to heading">​</a></h3><p>If you look at the &#x27;before&#x27; and &#x27;after&#x27; haplotypes, you&#x27;ll probably see some features:</p><ul><li>many variants are no longer polymorphic - the other alleles have been &#x27;lost&#x27; or have &#x27;fixed&#x27;</li><li>many of the remaining alleles are either rare or common.</li></ul><p>But how do these evolve over time? To see that let&#x27;s write some quick functions to compute the haplotype and variant frequencies across time:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">compute.haplotype.frequencies &lt;- function( population, haplotypes ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # We compute the frequency of each variant at each generation</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  result = matrix(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    nrow = G,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ncol = H,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    dimnames = list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      sprintf( &quot;generation=%d&quot;, 1:G ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      sprintf( &quot;haplotype=%d&quot;, 1:H )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  for( i in 1:G ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    currentGeneration = population[i,]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    result[i,] = sapply( 1:H, function(h) { length( which( currentGeneration == h )) } ) / N</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  return( result )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>And plot them:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">blank.plot &lt;- function( xlim = c( 0, 1 ), ylim = c( 0, 1 ), xlab = &quot;&quot;, ylab = &quot;&quot; ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # this function plots a blank canvas</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  plot(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    0, 0,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = &#x27;white&#x27;, # draw points white</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    bty = &#x27;n&#x27;,     # no border</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xaxt = &#x27;n&#x27;,    # no x axis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    yaxt = &#x27;n&#x27;,    # no y axis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlab = xlab,   # no x axis label</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylab = ylab,   # no x axis label</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlim = xlim,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylim = ylim</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">frequencies = compute.haplotype.frequencies( population, haplotypes )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">blank.plot( xlim = c( 1, G ), ylim = c( 0, 1 ), xlab = &quot;Generations&quot;, ylab = &quot;Haplotype frequencies&quot;)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">palette = rainbow( H )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">for( i in 1:H ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  points( 1:G, frequencies[,i], col = palette[i], type = &#x27;l&#x27; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">axis(1)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">axis(2)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Question.</strong> What happens to the haplotype frequencies over time?  What is causing this?</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="how-linkage-disequilibrium-evolves">How linkage disequilibrium evolves<a class="hash-link" href="#how-linkage-disequilibrium-evolves" title="Direct link to heading">​</a></h3><p>Because all our SNPs were independent of each other, and all the samples were unrelated, we
wouldn&#x27;t expect to see any strong linkage disequilibrium (correlation between genotypes at
different variants) at the start of our simulation. Or do we? Let&#x27;s compute the LD now and see:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">compute.ld &lt;- function( haplotypes ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  cor( haplotypes )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">compute.ld( haplotypes[population[1,],] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You should get a big 20x20 matrix of correlation values, with ones on the diagonal and lower values between SNPs.  The <code>image()</code> function can be used to visualise this.  By default <code>image()</code> chooses its own colour set, but we&#x27;ll specify a fixed set using the <code>breaks</code> and <code>col</code> arguments:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.ld &lt;- function( haplotypes ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  breaks = c( -0.01, seq( from = 0.1, to = 1, by = 0.1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  ld.colours = c( &#x27;grey95&#x27;, rev( heat.colors( length(breaks)-2 )))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  r = compute.ld( haplotypes )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  image(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    r^2,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    breaks = breaks,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = ld.colours</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.ld( haplotypes[population[1,],])</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Here we&#x27;ve plotted <em>r<sup>2</sup></em>; the colour scale is set up so <em>red points</em> are variants in high LD (also seen on the diagonal); yellow &amp; orange points have intermediate LD; and grey points have <em>r<sup>2</sup> &lt; 0.1</em>.  And white points have missing values (which occurs if one or other of the variants was actually monomorphic.)  It should look something like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/ld-02d64324ec78a2a413ba4954d6cd4dbb.png" width="400" height="300" class="img_E7b_"></p><p><strong>Note.</strong> As it turns out, even at this starting point in our simulation, there can be a bit of LD.
It arises due to our finite set of starting haplotypes.</p><p>Now plot the LD for the final generation and compare to the original:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">layout( matrix( 1:2, ncol = 1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.ld( haplotypes[population[1,],] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.ld( haplotypes[population[G,],] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Question.</strong> What do you see?</p><p><strong>Note.</strong> Depending on your simulation, you <em>might</em> see patterns emerging.  On the other hand, you might just see lots of white (missing) LD - it depends what happened in the simulation.)</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="exploring-the-simulations">Exploring the simulations.<a class="hash-link" href="#exploring-the-simulations" title="Direct link to heading">​</a></h3><p>In the file <a target="_blank" href="/whg-training-resources/assets/files/simulation_and_visualisation-60b89c36cb32eb5f3d0a7a2a013d9006.R/"><code>simulation_and_visualisation.R</code></a>
(also available <a href="https://github.com/whg-training/whg-training-resources/blob/main/docs/population_genetics/simulation_and_visualisation.R" target="_blank" rel="noopener noreferrer">on github</a>)
I&#x27;ve wrapped up the above code into two key functions:</p><ul><li><code>simulate.population()</code> which simulates from the model as above.</li><li><code>plot.simulation()</code> which makes a multi-panel plot of the result.</li></ul><p>To get this code, <code>source()</code> it into your session:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">source( &quot;simulation_and_visualisation.R&quot; )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>You can run it like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">sim = simulate.population( L = 20, N = 100, G = 100 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">plot.simulation(sim)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>or on one line:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.simulation( simulate.population( L = 20, N = 100, G = 100 ) )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>It produces a plot like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/simulation-6476da8b1dc80b7e7ea849d374bdd628.png" width="500" height="600" class="img_E7b_"></p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Questions</h5></div><div class="admonition-content"><ul><li>Explore the simulations and try varying the parameters. In particular, how does varying the population size <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em"></span><span class="mord mathnormal" style="margin-right:0.10903em">N</span></span></span></span></span> affect the
simulations?</li></ul><p><strong>Note.</strong> Remember that <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>L</mi></mrow><annotation encoding="application/x-tex">L</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em"></span><span class="mord mathnormal">L</span></span></span></span></span> is the number of variants, <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em"></span><span class="mord mathnormal" style="margin-right:0.10903em">N</span></span></span></span></span> is the population size, and <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>G</mi></mrow><annotation encoding="application/x-tex">G</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em"></span><span class="mord mathnormal">G</span></span></span></span></span> is the number of generations. (You
can also specify <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>H</mi></mrow><annotation encoding="application/x-tex">H</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em"></span><span class="mord mathnormal" style="margin-right:0.08125em">H</span></span></span></span></span>, the number of different haplotypes we created at the simulation start, if you want to.)</p><ul><li><p>Describe in general what happens to haplotypes and variants as the simulations proceed.</p></li><li><p>What does genetic drift do to LD?</p></li></ul></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="next-steps">Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">​</a></h2><p>Next let&#x27;s look at some <a href="/whg-training-resources/population_genetics/simulation/statistics/">key population genetic statistics</a>.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/population_genetics/simulation/popgen_simulation.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/population_genetics/simulation/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Simulating populations</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/population_genetics/simulation/mutations/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Adding mutations</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#getting-started" class="table-of-contents__link toc-highlight">Getting started</a><ul><li><a href="#generating-a-starting-set-of-haplotypes" class="table-of-contents__link toc-highlight">Generating a starting set of haplotypes</a></li><li><a href="#generating-a-population" class="table-of-contents__link toc-highlight">Generating a population</a></li></ul></li><li><a href="#evolving-the-population" class="table-of-contents__link toc-highlight">Evolving the population</a><ul><li><a href="#how-frequencies-evolve" class="table-of-contents__link toc-highlight">How frequencies evolve</a></li><li><a href="#how-linkage-disequilibrium-evolves" class="table-of-contents__link toc-highlight">How linkage disequilibrium evolves</a></li><li><a href="#exploring-the-simulations" class="table-of-contents__link toc-highlight">Exploring the simulations.</a></li></ul></li><li><a href="#next-steps" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>