<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Meta-analysing two studies. | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Meta-analysing two studies. | The WHG training resources"><meta data-rh="true" name="description" content="Up to the table of contents / Back to the Introduction / Forward to the fine-mapping section"><meta data-rh="true" property="og:description" content="Up to the table of contents / Back to the Introduction / Forward to the fine-mapping section"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/">Genome-wide association study practical</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genome-wide association study practical&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/genome_wide_association_studies/">Genome-wide association analysis</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/">Meta-analysis and fine-mapping tutorial</a><button aria-label="Toggle the collapsible sidebar category &#x27;Meta-analysis and fine-mapping tutorial&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction/">Meta-analysis and fine-mapping practical</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/">Meta-analysing two studies.</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/">Using FINEMAP to fine-map associations</a></li></ul></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"><span itemprop="name">Meta-analysis and fine-mapping tutorial</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Meta-analysing two studies.</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Meta-analysing two studies.</h1><p><a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/">Up to the table of contents</a> / <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction/">Back to the Introduction</a> / <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/">Forward to the fine-mapping section</a></p><h3 class="anchor anchorWithStickyNavbar_mojV" id="conducting-fixed-effect-meta-analysis">Conducting fixed-effect meta-analysis<a class="hash-link" href="#conducting-fixed-effect-meta-analysis" title="Direct link to heading">​</a></h3><p>The simplest way to perform meta-analysis is as follows.  We assume:</p><ul><li>that the <em>true effect</em> is the same in both studies.</li><li>that the <em>observed effect</em> in each study is equal to the <em>true effect</em> plus noise.  (The noise is given by the study standard error, and is assumed to be gaussian in the usual way.)</li></ul><p>The full name of this is <em>inverse variance weighted fixed-effect meta-analysis</em>.  </p><p>Here&#x27;s how it works: form a weighted average of the effect estimates, weighted by the inverse of the variances:</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mi>b</mi><mo>=</mo><munder><mo>∑</mo><mi>i</mi></munder><mfrac><msub><mi>β</mi><mi>i</mi></msub><msubsup><mtext>se</mtext><mi>i</mi><mn>2</mn></msubsup></mfrac></mrow><annotation encoding="application/x-tex">b = \sum_i \frac{\beta_i}{\text{se}_i^2}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6944em"></span><span class="mord mathnormal">b</span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:2.6491em;vertical-align:-1.2777em"></span><span class="mop op-limits"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.05em"><span style="top:-1.8723em;margin-left:0em"><span class="pstrut" style="height:3.05em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span><span style="top:-3.05em"><span class="pstrut" style="height:3.05em"></span><span><span class="mop op-symbol large-op">∑</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.2777em"><span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.3714em"><span style="top:-2.314em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord"><span class="mord text"><span class="mord">se</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7959em"><span style="top:-2.4231em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span><span style="top:-3.0448em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2769em"><span></span></span></span></span></span></span></span></span><span style="top:-3.23em"><span class="pstrut" style="height:3em"></span><span class="frac-line" style="border-bottom-width:0.04em"></span></span><span style="top:-3.677em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord"><span class="mord mathnormal" style="margin-right:0.05278em">β</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3117em"><span style="top:-2.55em;margin-left:-0.0528em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9629em"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></div><p>Compute also the sum of weights:</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><mi>w</mi><mo>=</mo><munder><mo>∑</mo><mi>i</mi></munder><mfrac><mn>1</mn><msubsup><mtext>se</mtext><mi>i</mi><mn>2</mn></msubsup></mfrac></mrow><annotation encoding="application/x-tex">w = \sum_i \frac{1}{\text{se}_i^2}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.4306em"></span><span class="mord mathnormal" style="margin-right:0.02691em">w</span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:2.5991em;vertical-align:-1.2777em"></span><span class="mop op-limits"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.05em"><span style="top:-1.8723em;margin-left:0em"><span class="pstrut" style="height:3.05em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span><span style="top:-3.05em"><span class="pstrut" style="height:3.05em"></span><span><span class="mop op-symbol large-op">∑</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:1.2777em"><span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.3214em"><span style="top:-2.314em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord"><span class="mord text"><span class="mord">se</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.7959em"><span style="top:-2.4231em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span><span style="top:-3.0448em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">2</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2769em"><span></span></span></span></span></span></span></span></span><span style="top:-3.23em"><span class="pstrut" style="height:3em"></span><span class="frac-line" style="border-bottom-width:0.04em"></span></span><span style="top:-3.677em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.9629em"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></div><p>Then the <em>meta-analysis estimate</em> is</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><msub><mi>β</mi><mtext>meta</mtext></msub><mo>=</mo><mi>b</mi><mo>∗</mo><mi>w</mi></mrow><annotation encoding="application/x-tex">\beta_{\text{meta}} = b * w</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8889em;vertical-align:-0.1944em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05278em">β</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.2806em"><span style="top:-2.55em;margin-left:-0.0528em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord text mtight"><span class="mord mtight">meta</span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:0.6944em"></span><span class="mord mathnormal">b</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mbin">∗</span><span class="mspace" style="margin-right:0.2222em"></span></span><span class="base"><span class="strut" style="height:0.4306em"></span><span class="mord mathnormal" style="margin-right:0.02691em">w</span></span></span></span></span></div><p>and the <em>meta-analysis standard error</em> is</p><div class="math math-display"><span class="katex-display"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><msub><mi>β</mi><mtext>meta</mtext></msub><mo>=</mo><mfrac><mn>1</mn><msqrt><mi>w</mi></msqrt></mfrac></mrow><annotation encoding="application/x-tex">\beta_{\text{meta}} = \frac{1}{\sqrt{w}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8889em;vertical-align:-0.1944em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05278em">β</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.2806em"><span style="top:-2.55em;margin-left:-0.0528em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord text mtight"><span class="mord mtight">meta</span></span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.2778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:0.2778em"></span></span><span class="base"><span class="strut" style="height:2.2514em;vertical-align:-0.93em"></span><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:1.3214em"><span style="top:-2.3097em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord sqrt"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.8003em"><span class="svg-align" style="top:-3em"><span class="pstrut" style="height:3em"></span><span class="mord" style="padding-left:0.833em"><span class="mord mathnormal" style="margin-right:0.02691em">w</span></span></span><span style="top:-2.7603em"><span class="pstrut" style="height:3em"></span><span class="hide-tail" style="min-width:0.853em;height:1.08em"><svg xmlns="http://www.w3.org/2000/svg" width="400em" height="1.08em" viewBox="0 0 400000 1080" preserveAspectRatio="xMinYMin slice"><path d="M95,702
c-2.7,0,-7.17,-2.7,-13.5,-8c-5.8,-5.3,-9.5,-10,-9.5,-14
c0,-2,0.3,-3.3,1,-4c1.3,-2.7,23.83,-20.7,67.5,-54
c44.2,-33.3,65.8,-50.3,66.5,-51c1.3,-1.3,3,-2,5,-2c4.7,0,8.7,3.3,12,10
s173,378,173,378c0.7,0,35.3,-71,104,-213c68.7,-142,137.5,-285,206.5,-429
c69,-144,104.5,-217.7,106.5,-221
l0 -0
c5.3,-9.3,12,-14,20,-14
H400000v40H845.2724
s-225.272,467,-225.272,467s-235,486,-235,486c-2.7,4.7,-9,7,-19,7
c-6,0,-10,-1,-12,-3s-194,-422,-194,-422s-65,47,-65,47z
M834 80h400000v40h-400000z"></path></svg></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2397em"><span></span></span></span></span></span></span></span><span style="top:-3.23em"><span class="pstrut" style="height:3em"></span><span class="frac-line" style="border-bottom-width:0.04em"></span></span><span style="top:-3.677em"><span class="pstrut" style="height:3em"></span><span class="mord"><span class="mord">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.93em"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>There are two good reasons to compute the estimate this way. Firstly, it makes sense: the meta-analysis estimate is a
weighted average of the per-study estimates, and they are weighted by the variance. So studies with lots of uncertainty (large
variance) get weighted down while studies with little uncertainty (small variance) get higher weight.</p><p>The second reason is that, as it turns out, this meta-analysis estimate is precisely <strong>what you would have computed</strong> if you
had estimated using all the data together** (at least approximately). The maths behind this is based on the remarkable fact
that products of normal distributions are also normal distributions. If we approximate the likelihood function from each study
by a normal - that is, by its estimate <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\beta_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.8889em;vertical-align:-0.1944em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:0.05278em">β</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3117em"><span style="top:-2.55em;margin-left:-0.0528em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span></span></span></span></span> and standard deviation <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext>se</mtext><mi>i</mi></msub></mrow><annotation encoding="application/x-tex">\text{se}_i</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.5806em;vertical-align:-0.15em"></span><span class="mord"><span class="mord text"><span class="mord">se</span></span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.3117em"><span style="top:-2.55em;margin-right:0.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.15em"><span></span></span></span></span></span></span></span></span></span></span> - then it turns out that this product is
a new normal density, and it has exactly the meta-analysis estimates as mean and standard deviation.</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="running-the-meta-analsis">Running the meta-analsis<a class="hash-link" href="#running-the-meta-analsis" title="Direct link to heading">​</a></h3><p>Here is a function to run the meta-analysis from our two studies:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">meta.analyse &lt;- function( beta1, se1, beta2, se2 ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  inverse.variances = c( 1/se1^2, 1/se2^2 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  b = c( beta1, beta2 ) * inverse.variances</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  meta.inverse.variance = sum( inverse.variances )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  meta.beta = sum( b ) / meta.inverse.variance</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  meta.se = sqrt( 1 / meta.inverse.variance )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # This formulation computes a two-tailed P-value</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # as discussed in lectures</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  meta.P = pnorm( -abs( meta.beta ), sd = meta.se ) * 2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # This formulation computes a Bayes factor under a N(0,0.2^2) prior</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  meta.BF = (</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    pnorm( meta.beta, mean = 0, sd = sqrt( meta.se^2 + 0.2 ) ) /</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    pnorm( meta.beta, mean = 0, sd = meta.se )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  return( list(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    meta.beta = meta.beta,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    meta.se = meta.se,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    meta.P = meta.P,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    log10_meta.BF = log10( meta.BF )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Try it on some fake data, for example:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">&gt; meta.analyse( beta1 = 1, se1 = 0.1, beta2 = 1, se2 = 0.1 )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Try a few different input values.  You should see:</p><ul><li>The <em>meta-analysis estimate</em> should be somewhere between the two input estimates.</li><li>The <em>meta-analysis standard error</em> should be smaller than either of the two study standard errors.</li><li>The <em>meta-analysis P-value</em> depends on how well the two estimates &#x27;stack up&#x27; on each other.  If they are both in the same direction and of roughly comparable size, the P-value will be lower than either of the study P-values.</li></ul><p>Here is a way to run the above function across all the data in our study.</p><p><strong>Note.</strong> We will use the <a href="https://purrr.tidyverse.org" target="_blank" rel="noopener noreferrer"><code>purrr</code></a> function <code>map_dfr()</code> to run the
above function across all rows and return a data frame. If you don&#x27;t have <code>purrr</code>, you could use a
base R function like <code>lapply()</code> instead.</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">meta_analysis = map_dfr(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  1:nrow(study1),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  function( i ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    c(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      rsid = study1$rsid[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      chromosome = study1$chromosome[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      position = study1$position[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      allele1 = study1$allele1[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      allele2 = study1$allele2[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study1.beta = study1$beta[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study1.se = study1$se[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study1.P = study1$P[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      log10_study1.BF = study1$log10_BF[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study2.beta = study2$beta[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study2.se = study2$se[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      study2.P = study2$P[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      log10_study2.BF = study2$log10_BF[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      meta.analyse(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        study1$beta[i], study1$se[i],</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        study2$beta[i], study2$se[i]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">      )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Question</h5></div><div class="admonition-content"><p>What does the meta-analysis look like for the &#x27;top&#x27; SNPs (e.g. those with the lowest
P-values or highest Bayes factors) in study1 or 2? Is the evidence stronger or weaker across
studies? Has the meta-analysis changed the SNP or SNPs with the strongest evidence? Why?</p><p><strong>Hint.</strong> One way to pull out the &#x27;top&#x27; SNps is to use the <code>order()</code> function. For example to pull
out the results for the ten SNPs with the lowest P-values in study 1, in order:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">meta_analysis[ order( meta_analysis$study1.P )[1:10], ]</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="making-a-forest-plot">Making a forest plot<a class="hash-link" href="#making-a-forest-plot" title="Direct link to heading">​</a></h3><p>To make best sense of this the tool for the job is a <a href="https://en.wikipedia.org/wiki/Forest_plot" target="_blank" rel="noopener noreferrer"><em>forest plot</em></a>.  We plot the study estimates and the meta-analysis estimate, along with their confidence/credible intervals, on seperate lines.  Let&#x27;s do that now.  First here&#x27;s our utility function to make a blank canvas:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">blank.plot &lt;- function( xlim = c( 0, 1 ), ylim = c( 0, 1 ), xlab = &quot;&quot;, ylab = &quot;&quot; ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # this function plots a blank canvas</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  plot(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    0, 0,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = &#x27;white&#x27;, # draw points white</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    bty = &#x27;n&#x27;,     # no border</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xaxt = &#x27;n&#x27;,    # no x axis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    yaxt = &#x27;n&#x27;,    # no y axis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlab = xlab,   # no x axis label</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylab = ylab,   # no x axis label</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlim = xlim,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylim = ylim</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now let&#x27;s make a general forest plot function:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">draw.forest.plot &lt;- function(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  betas,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  ses,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  names</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # y axis locations for the lines.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # We assume the meta-analysis will go on the last line, so we separate it slightly by putting it at 1/2</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  y = c( length(betas):2, 0.5 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # learn a good x axis range by going out 3 standard errors from each estimate:</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xlim = c( min( betas - 3 * ses ), max( betas + 3 * ses ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Also let&#x27;s make sure to include zero in our range</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xlim[1] = min( xlim[1], 0 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xlim[2] = max( xlim[2], 0 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # expand the range slightly</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xcentre = mean(xlim)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xlim[1] = xcentre + (xlim[1] - xcentre) * 1.1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  xlim[2] = xcentre + (xlim[2] - xcentre) * 1.1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Give ourselves a big left margin for the row labels</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  par( mar = c( 4.1, 6.1, 2.1, 2.1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  blank.plot(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlim = xlim,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylim = c( range(y) + c( -0.5, 0.5 )),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlab = &quot;Estimate and 95% CI&quot;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Draw the intervals first so they don&#x27;t draw over the points</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  segments(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    x0 = betas - 1.96 * ses, x1 = betas + 1.96 * ses,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    y0 = y, y1 = y,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = &#x27;grey&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Now plot the estimates</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  points(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    x = betas,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    y = y,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = &#x27;black&#x27;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    pch = 19</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # ... and add labels.  We put them 10% further left than the leftmost point</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # and we right-align them</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  text.x = xcentre + (xlim[1] - xcentre) * 1.1</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  text(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    x = text.x,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    y = y,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    labels = names,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    adj = 1, # right-adjust</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xpd = NA # this means &quot;Allow text outside the plot area&quot;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Add an x axis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  axis( 1 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # add grid lines</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # add a solid line at 0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  abline( v = 0, col = rgb( 0, 0, 0, 0.2 ), lwd = 2 )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For example, we could now make a forest plot for the first SNP:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">  forest.plot &lt;- function( row ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    betas = c( row$study1.beta, row$study2.beta, row$meta.beta )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ses = c( row$study1.se, row$study2.se, row$meta.se )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    names = c( &quot;Study 1&quot;, &quot;Study 2&quot;, &quot;Meta-analysis&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    draw.forest.plot( betas, ses, names )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  forest.plot( meta_analysis[1,] )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/forest_plot_1st_SNP-cac3aaf427b2a427fb43946ff64bb08b.png" width="400" height="300" class="img_E7b_"></p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Question</h5></div><div class="admonition-content"><p>What does the forest plot look like for the &#x27;top&#x27; SNPs, i.e. for those with the lowest P-values (or highest Bayes factors)?  Plot a few of them and look at them.  Make sure you understand how the data in the meta-analysis file corresponds to the data on the plot.</p></div></div><p><strong>Note.</strong> Forest plots are deceptively simple - they don&#x27;t look like there&#x27;s much going on, but whenever I try to draw one I
find it takes quite a bit of code. (Like the <code>draw.forest.plot()</code> function above, which already has ~40 lines of code). This
is typical of visualisation in general - the code always gets lengthy - and I think it is not really surprising. It is because
in a visualistion you are trying to convey a great deal of information clearly in a small space: this often takes great deal
of careful tweaking to get right. Don&#x27;t skimp on this!</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Challenge</h5></div><div class="admonition-content"><p> For a &quot;working&quot; plot, one thing it might be good to add to the plot would be more text listing the numerical
values (i.e. the estimate and 95% confidence intervals) E.g. this code:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">labels = sprintf(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  &quot;%.2f (%.2f - %.2f)&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  betas,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  betas - 1.96 * ses,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  betas + 1.96 * ses</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>produces the right kind of text.  Can you add this to the plot in the right place?</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="other-meta-analysis-models">Other meta-analysis models<a class="hash-link" href="#other-meta-analysis-models" title="Direct link to heading">​</a></h3><p>The above is a simple meta-analysis that assumes that the true effect is the same in both studies. However, as we saw in the
lecture, more complex models are also possible, including a &#x27;random&#x27; effects analysis (that assumes that true effects are
normally distributed around some mean) and include a heterogeneity test. Here is a way to do this.</p><p>First, you will need to install the <code>meta</code> library: run</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">install.packages(&quot;meta&quot;)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>or use the <code>Tools</code> -&gt; <code>Install Packages</code> menu in Rstudio.</p><p>We can then run a random-effect meta-analysis like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">library(meta)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">forest.meta(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  metagen(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    c( study1$beta[1], study2$beta[1] ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    c( study1$se[1],study2$se[1] )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Question</h5></div><div class="admonition-content"><p>What does this look like for the most-associated SNPs? Is there any evidence of differences in effect between studies?</p></div></div><h3 class="anchor anchorWithStickyNavbar_mojV" id="fine-mapping-the-association">Fine-mapping the association<a class="hash-link" href="#fine-mapping-the-association" title="Direct link to heading">​</a></h3><p>By now you should have a good sense of which SNPs have lots of evidence for association in the two studies - and maybe these
are the &#x27;causal&#x27; SNPs. However, an obvious possibility (if the gene is relevant for the disease) is that there could be
multiple causal SNPs. In the <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/">next section</a> we will see one way to try to discover how many causal SNPs there
are - and what they are.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Meta-analysis and fine-mapping practical</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Using FINEMAP to fine-map associations</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#conducting-fixed-effect-meta-analysis" class="table-of-contents__link toc-highlight">Conducting fixed-effect meta-analysis</a></li><li><a href="#running-the-meta-analsis" class="table-of-contents__link toc-highlight">Running the meta-analsis</a></li><li><a href="#making-a-forest-plot" class="table-of-contents__link toc-highlight">Making a forest plot</a></li><li><a href="#other-meta-analysis-models" class="table-of-contents__link toc-highlight">Other meta-analysis models</a></li><li><a href="#fine-mapping-the-association" class="table-of-contents__link toc-highlight">Fine-mapping the association</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>