<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Step 2: Inspecting sequence data quality | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Step 2: Inspecting sequence data quality | The WHG training resources"><meta data-rh="true" name="description" content="There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality"><meta data-rh="true" property="og:description" content="There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites/">Getting the prerequisites</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/">Practical outline</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs/">Step 1: What&#x27;s in the fastq files?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">Step 2: Inspecting sequence data quality</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">Step 3: Aligning reads</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/">Step 4: Viewing alignments</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/">Challenge questions</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/">Bonus tutorial: how accurate is Illumina sequencing?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">Appendix 1: Paired-end sequencing in theory and practice</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/">Appendix 2: understanding sequence duplication levels</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">Appendix 3: read trimming</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq/">Appendix 4: RNA-seq links</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links/">Useful links</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"><span itemprop="name">Analysing paired-end sequencing data</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Step 2: Inspecting sequence data quality</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Step 2: Inspecting sequence data quality</h1><p>There are a set of analyses that can be carried out directly on a FASTQ file to assess the quality
of sequencing. These include looking for poor quality data, looking for read duplication, assessing
GC content, looking for over-represented sequences and the presence of sequence adapters in reads.
Luckily there are also well-established tools that make running these analyses easy. The one we&#x27;ll
try is called <code>fastqc</code> and it can be run as follows.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="running-fastqc">Running <code>fastqc</code><a class="hash-link" href="#running-fastqc" title="Direct link to heading">​</a></h2><p>First make a directory for the output and run fastqc:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">mkdir fastqc_output</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">fastqc -o fastqc_output malaria/*.fastq.gz</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This will process both reads and will take a couple of minutes to run.</p><p>If you look in the folder now you&#x27;ll see there are some new files:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">$ ls fastq_output</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">QG0033-C_Illumina-HiSeq_read1.fastq.zip</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">QG0033-C_Illumina-HiSeq_read1.fastq.html</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">QG0033-C_Illumina-HiSeq_read2.fastq.zip</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">QG0033-C_Illumina-HiSeq_read2.fastq.html</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">...</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><code>fastqc</code> has created some <code>.html</code> files, as well as some <code>.zip</code> files. Look at these by opening the
<code>.html</code> files in a web browser.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="a-survey-of-sequencing-performance">A survey of sequencing performance<a class="hash-link" href="#a-survey-of-sequencing-performance" title="Direct link to heading">​</a></h2><p>In the rest of the page below I&#x27;m going to look at the <code>QG0033-C</code> dataset, which you will have
generated if you followed the instructions above. However, it&#x27;s useful to see what the output looks
like across datasets - so I collected some.  See:</p><ul><li><p><strong>a malaria sample</strong> sequenced on a HiSeq with an unusual GC bump.  (That&#x27;s this dataset, described further below). <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read1_fastqc.html#M5" target="_blank" rel="noopener noreferrer">-&gt; link</a> </p></li><li><p>a <strong>second malaria sample</strong> sequenced on the Illumina XTEN for which the reads have PCR primer contamination from the library prep step. <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read1_fastqc.html#M9" target="_blank" rel="noopener noreferrer">-&gt; read 1 link</a>, <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read2_fastqc.html#M9" target="_blank" rel="noopener noreferrer">read 2 link</a>.</p></li><li><p><strong>a human sample</strong> sequenced on a Novaseq at the <a href="https://www.well.ox.ac.uk" target="_blank" rel="noopener noreferrer">WHG</a>, for which read 2 is enriched for poly-G sequences. <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-illumina_novaseq_2_fastqc.html#M9" target="_blank" rel="noopener noreferrer">-&gt; link</a>.</p></li><li><p><strong>another human sample</strong>, sequenced during a fragmentation testing process, that has the same problem but much worse (nearly 3% of reads are like this.) <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/NA12878-illumina_novaseq-NE755566_R2_fastqc.html#M5" target="_blank" rel="noopener noreferrer">-&gt; link</a>.  </p></li><li><p>The <strong>same human sample</strong> sequenced <a href="https://www.ebi.ac.uk/ena/browser/view/ERR3239334?show=reads" target="_blank" rel="noopener noreferrer">at the NYGC</a>, where
the base qualities have been compressed to a different set of levels.  <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/ERR3239334-Illumina_NovaSeq_6000_read2_fastqc.html#M1" target="_blank" rel="noopener noreferrer">-&gt;
link</a>.</p></li></ul><p>Have a look at these. All except the second-last dataset would probably be considered reasonable quality data, but they
all have at least some level of artifacts.</p><p>(More examples can be found <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreferrer">on the fastq documentation
page</a>.)</p><p>Finally here is an example of <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-PacBio_CCS_fastqc.html" target="_blank" rel="noopener noreferrer">a long-read human dataset</a> (PacBio Hifi) - this looks quite different.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="next-steps">Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">​</a></h3><p>The rest of this page contains information on what&#x27;s in this fastqc output. There&#x27;s lots in here - feel free to read.
When you&#x27;re ready move on to <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">align the reads to the reference genome</a>.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="interpreting-fastqc-output">Interpreting fastqc output<a class="hash-link" href="#interpreting-fastqc-output" title="Direct link to heading">​</a></h2><p>For <code>QG0033-C</code> fastqc has provided us a detailed overview of information it can gain purely from
the fastq files. Let&#x27;s take a look.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="basic-statistics">Basic statistics<a class="hash-link" href="#basic-statistics" title="Direct link to heading">​</a></h3><p>The &#x27;Basic statistics&#x27; section tells you some stuff you already know - the number of reads, read
length and so on: </p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_basic_stats-936918dadde9e9f6d2ab81a3e230f17b.png" width="675" height="307" class="img_E7b_"></p><p>It also tells you the average GC content (the proportion of read bases that are
Gs or Cs). This is around 23%, which is about right because the <em>P.falciparum</em> genome <a href="https://doi.org/10.1038/nature01097" target="_blank" rel="noopener noreferrer">has such
high A/T content</a>.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="per-base-sequence-quality">Per base sequence quality<a class="hash-link" href="#per-base-sequence-quality" title="Direct link to heading">​</a></h3><p>The &#x27;Per base sequence quality&#x27; section is interesting: it suggests that the <strong>quality of
sequencing is lower at the ends of the read</strong>.</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_per_base_quality-4afa967904b1a3a534da7489ed88eb0e.png" width="910" height="676" class="img_E7b_"></p><p>Here, this is particularly seen at the start of the read, which is slightly unusual. If you look at
e.g. the results from the data run on XTEN you&#x27;ll see that it really tails off toward the end of
the read - but that&#x27;s not so noticeable here.</p><p><strong>Note.</strong> It&#x27;s important to realise this is the sequencer&#x27;s estimate of base quality - it&#x27;s generally
thought pretty good, but is not necessarily fully calibrated.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="per-tile-sequence-quality">Per tile sequence quality<a class="hash-link" href="#per-tile-sequence-quality" title="Direct link to heading">​</a></h3><p>This is a map of the &#x27;tiles&#x27; on the flowcell with the average base quality for each tile on the
Illumina sequencer. (Here, a &#x27;tile&#x27; is a physical region of a sequencer&#x27;s flowcell lane that is
imaged to detect bases. Inside each tile will be multiple &#x27;clusters&#x27; which, if the chemistry works
out, are each amplified from a single DNA fragment and correspond to one read pair. See the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">short
read theory page</a> for more on this).</p><p><strong>Question.</strong> How does fastqc know which tile the reads are from?</p><p>In this output you can see some tiles that have performed badly (red indicating below average base
quality) but most look pretty good.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="per-base-sequence-content">Per-base sequence content<a class="hash-link" href="#per-base-sequence-content" title="Direct link to heading">​</a></h3><p>This shows the average proportion of each base in each location in the reads:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_per_base_content-55a12a5f28059ec9bdcaf46c3ca33bd6.png" width="924" height="676" class="img_E7b_"></p><p>Here you see a fairly typical picture, as follows:</p><ul><li>In general <em>there should be about the same number of G as C bases, and the same number of A as T
bases</em>. This is because either strand of DNA ought to be equally likely to be sequenced. In the
above just under 40% of bases are G or C, and just over 10% are A or T, and it is pretty flat
across the read. So this looks about right for a malaria genome with ~23% A/T content.</li><li>There is nevertheless a skew of base content toward the start of the read. This seems to be
typical for Illumina sequencing and it could be caused by a number of things. One possibility is
that there are <a href="https://doi.org/10.1186/s12859-016-0976-y" target="_blank" rel="noopener noreferrer">biases in the sequence content due to DNA
fragmentation</a>; it&#x27;s also true that base quality was
lower at these bases so maybe there is another aspect of the chemistry causing this.</li></ul><h3 class="anchor anchorWithStickyNavbar_mojV" id="per-sequence-gc-content">Per-sequence GC content<a class="hash-link" href="#per-sequence-gc-content" title="Direct link to heading">​</a></h3><p>This plot measures the distribution of G and C bases (&#x27;GC&#x27; content) in all reads - and for this
sample it looks kind of interesting:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_gc_content-484e6b7afd5a3c86f25b9ebbcc832ed3.png" width="902" height="656" class="img_E7b_"></p><p>The bump on the left is expected, since <em>P.falciparum</em> has around 23% GC content on average. This
varies a bit around the genome so we should expect to see some spread around 23%.</p><p>But this sample also has a second bump of reads at around 60% GC. What could this be? Here are some
data points:</p><ul><li><p>If you look at the <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read2_fastqc.html#M5" target="_blank" rel="noopener noreferrer">read 2 fastqc output</a> file you&#x27;ll see it there as well.</p></li><li><p>It&#x27;s not seen in other samples (such as those linked above) - so it&#x27;s something about this sample.</p></li><li><p>Human DNA has about 41% GC content, but this is up at 60%... so it&#x27;s not obviously whole-genome human DNA contamination.</p></li></ul><p>Something fishy is going on... what could it be?</p><p><strong>Note.</strong> This will actually be one of our <strong><a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/challenges.md/">challenge questions</a></strong>. Leave it for now as
we&#x27;ll need to align reads to make progress on it.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="sequence-duplication-levels-adapter-content-and-over-represented-sequences">Sequence duplication levels, adapter content and over-represented sequences<a class="hash-link" href="#sequence-duplication-levels-adapter-content-and-over-represented-sequences" title="Direct link to heading">​</a></h3><p>Near the end are a set of plots that can provide evidence of contamination or chemistry issues -
such as high levels of duplicaets which could indicate a &#x27;low-diversity&#x27; (low DNA quantity input)
library, presence of adapter sequences which could be due to too short fragments, or other
artifacts.</p><p><strong>Note.</strong> The list of possible artifacts is long - see the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">paired-end sequencing
theory</a> for a description of some of this. (The
<a href="https://sequencing.qcfail.com" target="_blank" rel="noopener noreferrer">sequencing.qcfail.com</a> site is also a good read if you want a
more in-depth look at problems.) First of these analyses is the sequence duplication plot:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_duplication-57a97a49454b52b7b91dd6f76ae0941c.png" width="1754" height="1314" class="img_E7b_"></p><p>For this sample about 3% of sequences seem to be duplicated.  To interpret this, you should note that</p><ul><li><p>fastqc is only looking at one of the reads here (not both pairs, which would be a better analysis
to look for artifacts, because most duplication arising from amplification will be of the whole
fragment not just one of the ends).  So it might actually be over-estimating.</p></li><li><p>on the other hand sequencing errors might make duplicate fragments appear not duplicated, leading to underestimation
of the number of duplicates. For this reason fastqc focusses on the first 50bp of each read, but <a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/QG0033-C_Illumina-HiSeq_read1_fastqc.html#M1" target="_blank" rel="noopener noreferrer">as we saw above
there can be low-quality bases at the start of the read as
well</a>.</p></li><li><p>Some level of &#x27;genuine&#x27; duplication is expected in high-coverage data , of course. This is both
because we are sequencing randomly-placed fragments around the genome that occasionally overlap
by chance, and because genomes contain repetitive sequence. For more on this see the
<a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/">understanding sequence duplication levels</a> page.</p></li></ul><p>The upshot of this particular plot is that there aren&#x27;t worrying levels of duplicate reads in this
dataset, although because of the low qualities at the start of the read, you might guess this is an
underestimate.</p><p><strong>Note.</strong> It is common to see much higher levels of duplication - for example this
<a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/ERR3239334-Illumina_NovaSeq_6000_read2_fastqc.html#M8" target="_blank" rel="noopener noreferrer">sequencing of human sample NA12878</a>
shows around a 20% duplication rate even though it used a PCR-free library preparation.
PCR-based library preparation typically needs to still higher duplication rates.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="adapter-content-and-over-represented-sequences">Adapter content and over-represented sequences<a class="hash-link" href="#adapter-content-and-over-represented-sequences" title="Direct link to heading">​</a></h3><p>Finally fastqc has a couple of analyses that search reads for the presence of known adapter and
primer sequences, or other sequences that are unusually over-represented and suggest contamination
or chemistry issues. In brief, adapter sequences can appear in the reads <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">if the fragments are too
short</a> so that the sequencer &#x27;reads through&#x27; into the adapters; PCR primers
could appear for example due to primer self-amplification during PCR; other sequences (such as
poly-G sequences) can arise when chemistry for one of the primers fails.</p><p>This sample doesn&#x27;t seem to have any problems here.  But see:</p><ul><li><a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/malaria/ERR6227518_Illumina-XTEN_read2_fastqc.html#M9" target="_blank" rel="noopener noreferrer">this malaria dataset</a> with PCR primer contamination;</li><li><a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-illumina_novaseq_2_fastqc.html#M9" target="_blank" rel="noopener noreferrer">this human dataset</a> which has poly-G sequences in the second read.</li></ul><p>If you see high levels of artifacts of this type, you may want to try <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">read trimming</a> before analysis.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="next-steps-1">Next steps<a class="hash-link" href="#next-steps-1" title="Direct link to heading">​</a></h3><p>When you&#x27;ve read enough, move on to the section on <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">aligning reads</a>.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Step 1: What&#x27;s in the fastq files?</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Step 3: Aligning reads</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#running-fastqc" class="table-of-contents__link toc-highlight">Running <code>fastqc</code></a></li><li><a href="#a-survey-of-sequencing-performance" class="table-of-contents__link toc-highlight">A survey of sequencing performance</a><ul><li><a href="#next-steps" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></li><li><a href="#interpreting-fastqc-output" class="table-of-contents__link toc-highlight">Interpreting fastqc output</a><ul><li><a href="#basic-statistics" class="table-of-contents__link toc-highlight">Basic statistics</a></li><li><a href="#per-base-sequence-quality" class="table-of-contents__link toc-highlight">Per base sequence quality</a></li><li><a href="#per-tile-sequence-quality" class="table-of-contents__link toc-highlight">Per tile sequence quality</a></li><li><a href="#per-base-sequence-content" class="table-of-contents__link toc-highlight">Per-base sequence content</a></li><li><a href="#per-sequence-gc-content" class="table-of-contents__link toc-highlight">Per-sequence GC content</a></li><li><a href="#sequence-duplication-levels-adapter-content-and-over-represented-sequences" class="table-of-contents__link toc-highlight">Sequence duplication levels, adapter content and over-represented sequences</a></li><li><a href="#adapter-content-and-over-represented-sequences" class="table-of-contents__link toc-highlight">Adapter content and over-represented sequences</a></li><li><a href="#next-steps-1" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>