<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Challenge questions | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Challenge questions | The WHG training resources"><meta data-rh="true" name="description" content="We have three challenges.:"><meta data-rh="true" property="og:description" content="We have three challenges.:"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites/">Getting the prerequisites</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/">Practical outline</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs/">Step 1: What&#x27;s in the fastq files?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">Step 2: Inspecting sequence data quality</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">Step 3: Aligning reads</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/">Step 4: Viewing alignments</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/">Challenge questions</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/">Bonus tutorial: how accurate is Illumina sequencing?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">Appendix 1: Paired-end sequencing in theory and practice</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/">Appendix 2: understanding sequence duplication levels</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">Appendix 3: read trimming</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq/">Appendix 4: RNA-seq links</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links/">Useful links</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"><span itemprop="name">Analysing paired-end sequencing data</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Challenge questions</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Challenge questions</h1><p>We have three challenges.:</p><ul><li><p>If you want more practice running the pipelines outlined in the practical, try Challenge 1.</p></li><li><p>If you have <code>IGV</code> installed on your laptop, and want experience visualising reads in the desktop application (IGV),
try Challenge 2.</p></li><li><p>If you want a real challenge understanding the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/#per-sequence-gc-content">GC bump</a>, try Challenge 3.</p></li></ul><h2 class="anchor anchorWithStickyNavbar_mojV" id="challenge-1-analyse-some-human-data">Challenge 1: analyse some human data<a class="hash-link" href="#challenge-1-analyse-some-human-data" title="Direct link to heading">​</a></h2><p>In the folder <code>human/</code> you&#x27;ll find some data from a human sample - NA12878:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">ls ~sequence_data_analysis/human/</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Challenge.</strong> QC and align this data using the methods described in the practical. Then look at the result
using <code>samtools tview</code>.</p><p><strong>Note.</strong> This data is subsampled to chromosome 19. It is a bit bigger than the malaria data; to
speed this up you can probably get away with using up to 4 threads in the <code>bwa mem</code> step - but no
more than 4 please!</p><p>A good region to view is around the gene <em>FUT2</em>, which is at <code>chr19:48645971-48755951</code>.</p><p><strong>Question.</strong> The SNP <a href="https://www.ncbi.nlm.nih.gov/snp/rs601338" target="_blank" rel="noopener noreferrer">rs601338</a> in the <em>FUT2</em>
determines <em>secretor status</em>. Being a <em>non-secretor</em> means that blood group antigens (encoded by
ABO and other loci) are not secreted or expressed into bodily fluids in the normal way.
Non-secretors are <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6171556/" target="_blank" rel="noopener noreferrer">protected against some pathogens including
<em>norovirus</em></a>, but are also more susceptible
to others - so it can be medically important what status you have.</p><p>Secretor status is a recessive trait, so the table of genotypes and secretor status at rs601338 is:</p><table><thead><tr><th>genotype</th><th>secretor status</th></tr></thead><tbody><tr><td>GG</td><td>secretor</td></tr><tr><td>GA</td><td>secretor</td></tr><tr><td>AA</td><td>non-secretor</td></tr></tbody></table><p>What genotype does NA12878 have at rs601338?  Are they a secretor or non-secretor?</p><p><strong>Note.</strong> the SNP is at <code>chr19:48703417</code> as you can see <a href="https://www.ncbi.nlm.nih.gov/snp/rs601338" target="_blank" rel="noopener noreferrer">on dbSNP</a>.</p><p>Look for other variation.  Can you see an indel?</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="challenge-2-explore-reads-using-igv">Challenge 2: explore reads using <code>IGV</code><a class="hash-link" href="#challenge-2-explore-reads-using-igv" title="Direct link to heading">​</a></h2><p>This challenge is an alternative to Challenge 1 and will <strong>only work if you have IGV installed on your laptop</strong>.
To get set up:</p><ul><li>Start IGV on your laptop (not JupyterHub) and make sure you have selected human build 38 from the &#x27;genomes&#x27; drop down.</li><li>Load the track for NA12878 data from the following urls:</li></ul><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">  BAM URL: https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/solutions/human/</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">INDEX URL: https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/solutions/human/</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now browse the genome in IGV.</p><p><strong>Challenge.</strong> Can you find:</p><ol><li>a heterozygous SNP?</li><li>a homozygous SNP?</li><li>an insertion or deletion?</li><li>a larger structural variant?</li><li>a region where the coverage is unusually high (or low)?</li><li>With reference to <strong>challenge 1</strong>, what is the secretor status of this sample?</li><li>Can you find a read with clipped bases?  (Hint: to show clipped bases, you need to go to <code>View-&gt;Preferences-&gt;Alignments</code> and turn &#x27;show soft-clipped bases&#x27; on.)</li></ol><h3 class="anchor anchorWithStickyNavbar_mojV" id="challenge-3-figure-out-the-gc-bump">Challenge 3: figure out the GC bump.<a class="hash-link" href="#challenge-3-figure-out-the-gc-bump" title="Direct link to heading">​</a></h3><p><strong>Note.</strong> This is a real challenge.  I don&#x27;t fully know the answer right now.</p><p>In the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">QC section</a> we saw this weird pattern of GC content for the malaria sample <code>QG0033-C</code>:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/fastqc_gc_content-484e6b7afd5a3c86f25b9ebbcc832ed3.png" width="902" height="656" class="img_E7b_"></p><p><strong>Challenge</strong> figure out what the extra bump is.</p><p><strong>Hints.</strong> </p><ul><li><p>Maybe this is some kind of contamination? Since we have aligned the reads to the <em>P.falciparum</em>
genome, we might be able to find the contamination by searching for <em>unmapped reads</em>.</p></li><li><p>unmapped reads are identified by one of the <a href="https://broadinstitute.github.io/picard/explain-flags.html" target="_blank" rel="noopener noreferrer">SAM
flags</a>.  To pull them out you could do something like this:</p></li></ul><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">samtools view -f 4 -o QG0033-C-unaligned.bam QG0033-C.bam</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">samtools view -F 4 -o QG0033-C-onlyaligned.bam QG0033-C.bam</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>(The <code>-f</code> and <code>-F</code> option means: exclude or include reads with these flags).</p><ul><li>Now you could run both these bam files through fastqc to estimate the GC content seperately.  Does this explain the bump?</li></ul><h4 class="anchor anchorWithStickyNavbar_mojV" id="using-ncbi-blast">Using NCBI BLAST<a class="hash-link" href="#using-ncbi-blast" title="Direct link to heading">​</a></h4><p>One way to go further now is to take some reads from the unaligned BAM file and try BLASTing them
to lots of other sequences. You can do that as follows:</p><ul><li>Visit <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank" rel="noopener noreferrer">NCBI blast</a> and choose &#x27;Nucleotide BLAST&#x27;</li><li>Pick a read from the unaligned BAM and paste its sequence into the &#x27;FASTA sequence&#x27; box</li><li>To get this to work, for most reads I found I had to choose the <code>&#x27;Somewhat similar sequences (blastn)</code> option.</li></ul><p>Do the reads align?  To what organism?  What&#x27;s going on?</p><p><strong>Note</strong>. For example this was one result I got:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/BLAST_result-075c479ff7d62c3d9b9bdd63976907ad.png" width="1824" height="682" class="img_E7b_"></p><p>You can try looking at where these reads align to (here <code>Pf3D7_08_v3:1525593-1525692</code>) e.g. in <a href="https://plasmodb.org/plasmo/app/jbrowse?data=/a/service/jbrowse/tracks/default&amp;tracks=gene" target="_blank" rel="noopener noreferrer">PlasmoDB
browser</a> or the <a href="https://www.malariagen.net/apps/pf3k/release_3/index.html#genomebrowser" target="_blank" rel="noopener noreferrer">Pf3k
browser</a>. What could be going on?</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Step 4: Viewing alignments</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Bonus tutorial: how accurate is Illumina sequencing?</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#challenge-1-analyse-some-human-data" class="table-of-contents__link toc-highlight">Challenge 1: analyse some human data</a></li><li><a href="#challenge-2-explore-reads-using-igv" class="table-of-contents__link toc-highlight">Challenge 2: explore reads using <code>IGV</code></a><ul><li><a href="#challenge-3-figure-out-the-gc-bump" class="table-of-contents__link toc-highlight">Challenge 3: figure out the GC bump.</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>