<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Appendix 1: Paired-end sequencing in theory and practice | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Appendix 1: Paired-end sequencing in theory and practice | The WHG training resources"><meta data-rh="true" name="description" content="What paired-end sequence data should look like"><meta data-rh="true" property="og:description" content="What paired-end sequence data should look like"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/sequence_data_analysis/">Next-generation sequencing</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/sanger_sequence_data/">Analysing Sanger sequence data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing Sanger sequence data&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/">Analysing paired-end sequencing data</a><button aria-label="Toggle the collapsible sidebar category &#x27;Analysing paired-end sequencing data&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Prerequisites/">Getting the prerequisites</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/">Practical outline</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Inspecting_the_fastqs/">Step 1: What&#x27;s in the fastq files?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">Step 2: Inspecting sequence data quality</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Aligning_reads/">Step 3: Aligning reads</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Viewing_alignments/">Step 4: Viewing alignments</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Challenge_questions/">Challenge questions</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/">Bonus tutorial: how accurate is Illumina sequencing?</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory/">Appendix 1: Paired-end sequencing in theory and practice</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/">Appendix 2: understanding sequence duplication levels</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Read_trimming/">Appendix 3: read trimming</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/rna_seq/">Appendix 4: RNA-seq links</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links/">Useful links</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/IGV/">Exploring Reads in IGV</a><button aria-label="Toggle the collapsible sidebar category &#x27;Exploring Reads in IGV&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/building_an_ngs_pipeline/">A sequence data pipelining challenge!</a><button aria-label="Toggle the collapsible sidebar category &#x27;A sequence data pipelining challenge!&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/sequence_data_analysis/variant_calling_and_imputation/">Variant calling, phasing and imputation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Variant calling, phasing and imputation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/"><span itemprop="name">Analysing paired-end sequencing data</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Appendix 1: Paired-end sequencing in theory and practice</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Appendix 1: Paired-end sequencing in theory and practice</h1><h2 class="anchor anchorWithStickyNavbar_mojV" id="what-paired-end-sequence-data-should-look-like">What paired-end sequence data should look like<a class="hash-link" href="#what-paired-end-sequence-data-should-look-like" title="Direct link to heading">​</a></h2><p>You&#x27;ll know that typical short-read datasets are made up of pairs of reads representing sequencing
of both ends of short DNA fragments. The basic model is that we should be generating short DNA
fragments that get read from either end like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">        READ 1:    --------&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  DNA FRAGMENT: 5&#x27; ------------------------------------ 3&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">        READ 2:                               &lt;--------</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Read 1 and 2 form what is called a <strong>read pair</strong> (and this is <strong>paired-end sequencing</strong>); this is
important for analysis because we know that they came from the same fragment.</p><p>Typical data contains paired-end reads from millions of number of fragments that are of reasonable
size (often around 400-500bp). In an ideal world these would be randomly sampled from around the
genome of interest, so generating approximately uniform coverage.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="what-paired-end-data-actually-looks-like">What paired-end data actually looks like<a class="hash-link" href="#what-paired-end-data-actually-looks-like" title="Direct link to heading">​</a></h2><p>In real data things are not quite so simple. To understand this you have to know a bit about the
chemistry. It works like this (I&#x27;m describing Illumina sequencing here, other technologies may be
subtly different):</p><p>First, the fragment gets extra <em>adapter sequences</em> ligated onto the end:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27; ========-------------======= 3&#x27;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The fragment is often called the <strong>insert</strong>, since it is &quot;inserted&quot; between the adapter sequences.</p><p>The adapters are themselves structured: moving outward from the fragment, they have a <em>primer
sequence</em> (that tells the polymerase where to start sequencing), an <em>index</em> that identifies the
sample, and then a specific sequence whose job is to anneal to the oligos on the flowcell. So they
look something like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27; OOIIIPPPP--------------PPPPIIIOO 3&#x27;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Where <code>O</code> = oligo-binding sequence, <code>I</code> = the sample index, and <code>P</code> = the sequencing primer that
indicates the sequencing start point. (The sample index is used to de-multiplex reads from
different samples run on the same lane at the same time - it is read in seperate index cycles.)</p><p>If all goes well, the sequencing starts from just after the read 1 primer and ends after a fixed
number of cycles (often 100-150 for high-throughput sequencers). Then the fragment is reversed (by
a slightly complicated process) and read 2 is read in the same way. So we should get this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 1:             -----&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27; OOIIIPPPP------------------PPPPIIIOO 3&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 2:                         &lt;-----</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Here&#x27;s another picture of the same thing from <a href="https://support.illumina.com/bulletins/2016/04/adapter-trimming-why-are-adapter-sequences-trimmed-from-only-the--ends-of-reads.html" target="_blank" rel="noopener noreferrer">Illumina&#x27;s page on adapter
trimming</a>:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/PEcell2-4d18c1f63495bb083fb5971c0d0e9fab.png" width="896" height="410" class="img_E7b_"></p><p><strong>Note.</strong> What actually <em>are</em> the adapter sequences? This depends on the kit, but you can find out
if you want to - for Illumina has a
<a href="https://support-docs.illumina.com/SHARE/AdapterSeq/Content/SHARE/AdapterSeq/AdapterSequencesIntro.htm" target="_blank" rel="noopener noreferrer">list of adapter and index barcodes</a>.
Tools like <code>fastqc</code> use these known sequences to look for adapter contamination.</p><p>You can also watch <a href="https://youtu.be/fCd6B5HRaZ8" target="_blank" rel="noopener noreferrer">Illumina&#x27;s short video on the sequencing process</a>.</p><p>To generate enough intensity for imaging, the ligated fragments above undergo
<em>amplification on the flowcell</em> that generates multiple copies of each fragment - called a
<strong>cluster</strong>. This process keeps the fragments local (they stay within the imaging &#x27;tile&#x27;) so that
they are imaged together, generating the output sequence data.</p><p>Finally, the sequencer proceeds to sequence each read in <strong>cycles</strong>. In each cycle, travelling in
from the primer, the polymerase incorporates a single flourescently-labelled base complementary to
the insert, and an image is taken to capture the colour of the incorporated base. Then, in the next
cycle the flourescent label is removed which allows the subsequent base to be incorporated, and so
on until the read is complete.</p><p><strong>Note.</strong> See the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Useful_links/">useful links page</a> for links to other resources on sequencing.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="so-what-goes-wrong">So what goes wrong?<a class="hash-link" href="#so-what-goes-wrong" title="Direct link to heading">​</a></h2><p>Well here are a few common issues:</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-fragments-are-too-short">The fragments are too short<a class="hash-link" href="#the-fragments-are-too-short" title="Direct link to heading">​</a></h3><p>If the fragments are too short you&#x27;ll start to see <strong>overlapping reads</strong>, <strong>adapter contamination</strong>,
or maybe <strong>failed sequencing bases</strong>.</p><p>This is because the reads may start to look like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 1:             --------&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27; OOIIIPPPP--------------PPPPIIIOO 3&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 2:                  &lt;--------</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>or this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 1:             --------&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27; OOIIIPPPP---PPPPIIIOO 3&#x27;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 2:       &lt;--------</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Or maybe even this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 1:                --------------&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  FRAGMENT WITH ADAPTERS: 5&#x27;    OOIIIPPPP--PPPPIIIOO</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">                  READ 2:    &lt;-------------</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The reads end up with bits of overlapping sequence, read-through into the adapters, or even read
right off the end of the sequence.</p><p>(What happens in the last case depends a bit on which sequencer is being used. Recent machines such
as the Novaseq use a &#x27;2-color&#x27; imaging process, which does not distinguish between a G base and a
lack of intensity, so this type of read-through leads to long sequences of apparent &#x27;G&#x27; bases in
the reads.  See the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">QC page</a> for some examples.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-fragments-are-too-long">The fragments are too long<a class="hash-link" href="#the-fragments-are-too-long" title="Direct link to heading">​</a></h3><p>Having too-long fragments doesn&#x27;t cause the <em>same</em> problems as too-short fragments, but it does turn out to hurt read 2.
In particular the efficiency of the turnover step (where the fragments get reversed before read 2 is sequenced) seems to
depend heavily on having fragments that are around 500bp in length or less.</p><p>As fragments get longer, it&#x27;s typical to see <a href="https://www.nature.com/articles/s41598-019-39076-7#:~:text=Abstract,error%20rate%20below%201%2F1000" target="_blank" rel="noopener noreferrer">elevated error rates in read 2</a>.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-adapter-chemistry-goes-wrong">The adapter chemistry goes wrong<a class="hash-link" href="#the-adapter-chemistry-goes-wrong" title="Direct link to heading">​</a></h3><p>There&#x27;s a bit of asymmetry in the process between the two reads. This is because if the read 1
chemistry doesn&#x27;t work out, then the fragment may not anneal to the flowcell so won&#x27;t start a cluster at all. But if read 2
chemistry doesn&#x27;t work out we&#x27;ll get artifacts in the second read. Some examples of that are <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/">on
the QC page</a> where a fraction of read pairs read 2 entirely composed of <code>G</code>
bases. (As in
<a href="https://www.well.ox.ac.uk/~gav/projects/oxford_statgen_summer_school/sequence_data_analysis/fastqc_examples/human/HV31-illumina_novaseq_2_fastqc.html#M9" target="_blank" rel="noopener noreferrer">this sample</a>,
sequenced during workflow testing, for which I presume the read 2 primers are not properly ligated leading to low intensities for
the second read.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-library-has-too-little-input-dna">The library has too little input DNA<a class="hash-link" href="#the-library-has-too-little-input-dna" title="Direct link to heading">​</a></h3><p>If the library has too little input DNA then we will end up sequencing the same fragments over and
over again - generating <strong>duplicate reads</strong>.</p><p>This occurs because of duplication of the same fragments by the PCR process, and it means we will
get high levels of <strong>duplicate reads</strong>. (This is often referred to as a &#x27;low diversity&#x27; library,
but note this is not referring to <em>genetic diversity</em> or the actual genetic makeup of the sample -
it is just about about how well the genome is represented by random fragments.)</p><p><strong>Note.</strong> Some level of duplication is expected anyway, both by random chance of where fragments
are sequenced, and because genomes contain repetitive sequence. Howeer, an important feature of
artifactual duplicates above is that they sequence the <strong>exact same fragment</strong>, so both read 1 and
read 2 will appear to be duplicated. Because there is generally a spread of fragment sizes, this is
very unlikely even for high-coverage sequencing.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="there-is-spillover-between-clusters">There is spillover between clusters<a class="hash-link" href="#there-is-spillover-between-clusters" title="Direct link to heading">​</a></h3><p>Even with PCR duplicates, some clusters on the flowcell may end up large enough that they get
imaged as two or more clusters. (Another version of this is that the original amplified fragment
moves off and forms another nearby cluster.) This has the effect of also generating duplicate reads,
known as <strong>optical duplicates</strong>. </p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>The way these are detected is as identical or very similar reads that lie in nearby physical locations on the flowcell. </p></div></div><p>These duplicates can occur even if the library preparation steps do not use PCR amplification. See
<a href="https://sequencing.qcfail.com/articles/illumina-patterned-flow-cells-generate-duplicated-sequences/" target="_blank" rel="noopener noreferrer">sequencing.qcfail.com</a>
for an analysis of this.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="incorporation-of-bases-gets-out-of-sync">Incorporation of bases gets out of sync<a class="hash-link" href="#incorporation-of-bases-gets-out-of-sync" title="Direct link to heading">​</a></h3><p>The sequencing process relies on generating large clusters of identical DNA molecules that are all
imaged together. However, as the process proceeds along the read, it can happen that the different
copies get out of &#x27;sync&#x27;. As the read proceeds this affects a larger and larger subset of the
copies leading to <strong>worse sequencing toward the end of the read</strong>. This is often reflected in the
lowered base qualitites.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-fragmentation--chemistry-introduces-biases">The fragmentation / chemistry introduces biases<a class="hash-link" href="#the-fragmentation--chemistry-introduces-biases" title="Direct link to heading">​</a></h3><p>It is thought that common fragmentation methods themselves introduce bias in the DNA content at the
start of the fragments. It is generally true that there is some bias in the GC content of the first
few bases in the reads, as seen in this file.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="next-steps">Next steps<a class="hash-link" href="#next-steps" title="Direct link to heading">​</a></h2><p>Now get back <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Pipeline_outline/#the-practical-in-a-nutshell">back to the practical</a>, or
see the <a href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Quality_control/#a-survey-of-sequencing-performance">examples on the QC page</a>.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/Short_read_theory.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/De_novo_error_rate_estimation/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Bonus tutorial: how accurate is Illumina sequencing?</div></a><a class="pagination-nav__link pagination-nav__link--next" href="/whg-training-resources/sequence_data_analysis/introduction_to_next_generation_sequencing_data_analysis/duplicates/"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Appendix 2: understanding sequence duplication levels</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#what-paired-end-sequence-data-should-look-like" class="table-of-contents__link toc-highlight">What paired-end sequence data should look like</a></li><li><a href="#what-paired-end-data-actually-looks-like" class="table-of-contents__link toc-highlight">What paired-end data actually looks like</a></li><li><a href="#so-what-goes-wrong" class="table-of-contents__link toc-highlight">So what goes wrong?</a><ul><li><a href="#the-fragments-are-too-short" class="table-of-contents__link toc-highlight">The fragments are too short</a></li><li><a href="#the-fragments-are-too-long" class="table-of-contents__link toc-highlight">The fragments are too long</a></li><li><a href="#the-adapter-chemistry-goes-wrong" class="table-of-contents__link toc-highlight">The adapter chemistry goes wrong</a></li><li><a href="#the-library-has-too-little-input-dna" class="table-of-contents__link toc-highlight">The library has too little input DNA</a></li><li><a href="#there-is-spillover-between-clusters" class="table-of-contents__link toc-highlight">There is spillover between clusters</a></li><li><a href="#incorporation-of-bases-gets-out-of-sync" class="table-of-contents__link toc-highlight">Incorporation of bases gets out of sync</a></li><li><a href="#the-fragmentation--chemistry-introduces-biases" class="table-of-contents__link toc-highlight">The fragmentation / chemistry introduces biases</a></li></ul></li><li><a href="#next-steps" class="table-of-contents__link toc-highlight">Next steps</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>