<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.20">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css" integrity="sha384-odtC+0UGzzFL/6PNoE8rX/SPcQDXBJ+uRepguP4QkPCm2LBxH3FA3y+fKSiJ+AmM" crossorigin="anonymous"><title data-rh="true">Using FINEMAP to fine-map associations | The WHG training resources</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Using FINEMAP to fine-map associations | The WHG training resources"><meta data-rh="true" name="description" content="Up to the table of contents / Back to the meta-analysis section"><meta data-rh="true" property="og:description" content="Up to the table of contents / Back to the meta-analysis section"><link data-rh="true" rel="icon" href="/whg-training-resources/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/" hreflang="en"><link data-rh="true" rel="alternate" href="https://well.ox.ac.uk//whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/" hreflang="x-default"><link rel="stylesheet" href="/whg-training-resources/assets/css/styles.f24e0b7d.css">
<link rel="preload" href="/whg-training-resources/assets/js/runtime~main.8fe67747.js" as="script">
<link rel="preload" href="/whg-training-resources/assets/js/main.91c67f1f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/whg-training-resources/"><div class="navbar__logo"><img src="/whg-training-resources/img/wchg.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--light_TfLj"><img src="/whg-training-resources/img/wchg-white.png" alt="My Site Logo" class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title text--truncate">Training Resources</b></a><a class="navbar__item navbar__link" href="/whg-training-resources/overview/">Tutorials home</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div><div class="searchBox_dLyj"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docsWrapper_mKqt"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><div class="docPage_ualW"><aside class="theme-doc-sidebar-container docSidebarContainer_UQUJ"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/whg-training-resources/genome_wide_association_studies/genome_wide_association_analysis/">Genome-wide association study practical</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genome-wide association study practical&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/whg-training-resources/genome_wide_association_studies/">Genome-wide association analysis</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/">Meta-analysis and fine-mapping tutorial</a><button aria-label="Toggle the collapsible sidebar category &#x27;Meta-analysis and fine-mapping tutorial&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction/">Meta-analysis and fine-mapping practical</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/">Meta-analysing two studies.</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/">Using FINEMAP to fine-map associations</a></li></ul></li></ul></nav></div></aside><main class="docMainContainer_uL0j"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="Breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a aria-label="Home page" class="breadcrumbs__link" href="/whg-training-resources/"><svg viewBox="0 0 24 24" class="breadcrumbHomeIcon_kU5B"><path d="M10 19v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-7h1.7c.46 0 .68-.57.33-.87L12.67 3.6c-.38-.34-.96-.34-1.34 0l-8.36 7.53c-.34.3-.13.87.33.87H5v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1z" fill="currentColor"></path></svg></a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/"><span itemprop="name">Meta-analysis and fine-mapping tutorial</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="name">Using FINEMAP to fine-map associations</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_bZGK theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_l22C">On this page</button></div><div class="theme-doc-markdown markdown"><h1>Using FINEMAP to fine-map associations</h1><p><a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/">Up to the table of contents</a> / <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/">Back to the meta-analysis section</a></p><p>If you followed the <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/">meta-analysis section</a> you should now have a dataframe with meta-analysis results.  It looks something like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">&gt; meta_analysis</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># A tibble: 730 x 16</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   rsid             chromosome position allele1 allele2 study1.beta study1.se   study1.P log10_study1.BF study2.beta study2.se study2.P meta.beta meta.se   meta.P log10_meta.BF</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   &lt;chr&gt;                 &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;         &lt;dbl&gt;     &lt;dbl&gt;      &lt;dbl&gt;           &lt;dbl&gt;       &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 1 19:49149607:G:A          19 49149607 G       A            0.202     0.228  0.376              -0.0925     -0.146      0.295  0.310      0.0176  0.0894 8.44e- 1       -0.0497</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 2 19:49149711:G:A          19 49149711 G       A           -0.0183    0.202  0.928               0.0195      0.239      0.261  0.180      0.0245  0.0894 7.84e- 1       -0.0668</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 3 19:49149937:T:G          19 49149937 T       G           -0.357     0.0785 0.00000540          4.90       -0.264      0.101  0.00463   -0.669   0.0894 7.45e-14       12.3   </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 4 19:49150048:G:T          19 49150048 G       T           -0.332     0.0788 0.0000257           4.26       -0.271      0.102  0.00387   -0.636   0.0894 1.12e-12       11.2   </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 5 19:49150130:G:T          19 49150130 G       T           -0.196     0.110  0.0756              0.947      -0.204      0.143  0.0767    -0.209   0.0894 1.96e- 2        1.52  </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 6 19:49150233:G:A          19 49150233 G       A            0.133     0.134  0.321              -0.137      -0.0794     0.173  0.323      0.0381  0.0894 6.70e- 1       -0.0958</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 7 19:49151312:C:T          19 49151312 C       T            0.238     0.145  0.101              -0.137       0.101      0.187  0.295      0.114   0.0894 2.03e- 1       -0.176 </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 8 19:49151323:T:G          19 49151323 T       G            0.202     0.228  0.376              -0.0925     -0.146      0.295  0.310      0.0176  0.0894 8.44e- 1       -0.0497</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"> 9 19:49151460:C:CA         19 49151460 C       CA          -0.283     0.153  0.0636              0.936      -0.218      0.197  0.134     -0.142   0.0894 1.12e- 1        0.830 </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">10 19:49152118:C:T          19 49152118 C       T           -0.609     0.328  0.0632              0.634      -0.604      0.423  0.0767    -0.0723  0.0894 4.19e- 1        0.319 </span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>We can look at the strongest signals in the data (either by P-value or Bayes factor) and compare across the studies and the meta-analysis.</p><p>However, <strong>what if there are multiple causal variants in the region</strong>?</p><p>If there are, then the single-SNP estimate at each SNP we see above would be a combination of several things:</p><ul><li>A contribution from the true, causal effect of the SNP.  (This will be zero if the SNP doesn&#x27;t have a causal effect)</li><li>A contribution from sampling noise (represented in the standard error)</li><li>A contribution from other causal SNPs that happen to be in linkage disequilibrium (i.e. correlated) with this SNP.</li></ul><p>Moreover it is possible that looking at the &#x27;top&#x27; SNPs as above could be misleading in this case
(for example, if there were a SNP that is in strong LD with two causal SNPs, it might have an
apparent strong effect.)</p><p>In this part of the practical we will use the <a href="https:/www.finemap.me" target="_blank" rel="noopener noreferrer"><code>FINEMAP</code></a> tool to fit a model that allows for multiple
possbile causal SNPs.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="preparing-finemap-input-files">Preparing FINEMAP input files<a class="hash-link" href="#preparing-finemap-input-files" title="Direct link to heading">​</a></h2><p>To run finemap, we need to set up some input files in <a href="https:/www.finemap.me" target="_blank" rel="noopener noreferrer">the particular format that FINEMAP needs</a>.
<code>FINEMAP</code> takes:</p><ul><li><p>A set of per-SNP summary statistics (betas and standard errors) from the study. To maximise
power, we will use your meta-analysed results here.</p></li><li><p>A matrix of between-SNP linkage disquilibrium (correlation) values.</p></li><li><p>It also requires us to specify the sample size - in our study the sample sizes were 12,500 (study
1) and 7,500 (study 2), so the total sample size is 20,000.</p></li></ul><p>We have precomputed the LD values for you in the file <a target="_blank" href="/whg-training-resources/assets/files/study_LD-7240737f948f325c4f01f2f2cc916f38.ld/"><code>study_LD.ld</code></a>. (It is a giant
<span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>730</mn><mo>×</mo><mn>730</mn></mrow><annotation encoding="application/x-tex">730 \times 730</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.7278em;vertical-align:-0.0833em"></span><span class="mord">730</span><span class="mspace" style="margin-right:0.2222em"></span><span class="mbin">×</span><span class="mspace" style="margin-right:0.2222em"></span></span><span class="base"><span class="strut" style="height:0.6444em"></span><span class="mord">730</span></span></span></span></span> matrix represented in a text file). But we need to make a file of meta-analysis
results.</p><p>If you look at <a href="https:/www.finemap.me" target="_blank" rel="noopener noreferrer">the documentation</a> you&#x27;ll see that FINEMAP needs a
space-separated file with these columns: rsid, chromosome, position, allele1, allele2, maf, beta,
se. The only thing we don&#x27;t have that we need right now is the <em>minor allele frequency</em>. So let&#x27;s
compute this first using the minor allele frequencies from the two studies and their sample sizes
(which in this data are study1: 12,500 and study2: 7,500):</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">meta_analysis$maf = ((study1$maf * 12500) + (study2$maf * 7500)) / (20000)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note.</strong> The above calculation is <strong>dangerous</strong>: it would be wrong if the minor allele is not the
same in the two studies. For this data we happen to know that is not the case, but in real data you
should always aim to <strong>compute the allele frequency of a specific allele</strong> instead.</p><p>Now let&#x27;s write a FINEMAP input file:</p><p><strong>Still in your R / RStudio session</strong>:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain"># format the data...</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">finemap.data = meta_analysis[,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    c( &quot;rsid&quot;, &quot;chromosome&quot;, &quot;position&quot;, &quot;allele1&quot;, &quot;allele2&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    &quot;maf&quot;, &quot;meta.beta&quot;, &quot;meta.se&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">colnames( finemap.data )[7:8] = c( &quot;beta&quot;, &quot;se&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"># ...and write it to a file</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">write_delim( finemap.data, &quot;combined_study.z&quot;, delim = &quot; &quot;)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>To get <code>FINEMAP</code> to run we also have to give it another input file telling it what all the input
and output files are. This one is a bit weird: it is a <em>semi-colon delimited</em> text file. Create a
new file called <code>finemap.master</code> and put this in:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">z;ld;snp;config;cred;log;n_samples</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">combined_study.z;study_LD.ld;finemap_meta.snp;finemap_meta.config;finemap_meta.cred;finemap_meta.log;20000</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Note</h5></div><div class="admonition-content"><p>The format of the file is <a href="http://www.christianbenner.com/#input" target="_blank" rel="noopener noreferrer">described here</a>. What input files are we using?
What output files have we specified? What&#x27;s the number <code>20000</code>?</p></div></div><p>Now you are ready to run <code>FINEMAP</code>...</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="running-finemap">Running FINEMAP<a class="hash-link" href="#running-finemap" title="Direct link to heading">​</a></h2><p>I&#x27;m going to assume you have downloaded and been able to run <code>FINEMAP</code> as described in the
<a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Introduction/">Introduction</a>. (If not please go and download it now.) Hopefully you know where
it is and can run it with a command of the following form. (<strong>Note.</strong> in this section we are back
to working on the command-line - these are not R commands!)  Try this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">/path/to/finemap_v[version]/finemap_v[version]</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For example, on my older Mac OS system I had to use finemap v1.3, so running it looks like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">$ /path/to/finemap_v1.3.1_MacOSX/finemap_v1.3.1_MacOSX </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--------------------------------------|</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Welcome to FINEMAP v1.3.1            |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| (c) 2015-2018 University of Helsinki |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Help :                               |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - ./finemap --help                   |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - www.finemap.me                     |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - www.christianbenner.com            |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Contact :                            |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - christian.benner@helsinki.fi       |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - matti.pirinen@helsinki.fi          |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--------------------------------------|</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">Error : No flags detected!</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>For ease of use, I suggest to copy the finemap executable into the current directory:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">cp /path/to/finemap_v[version]/finemap_v[version] ./finemap</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Running FINEMAP is hopefully now easy:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">./finemap --sss --in-files finemap.master --log</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Be prepared for a bit of a wait as FINEMAP searches the space of possible causal configurations.</p><p>So what <em>is</em> FINEMAP doing? Well, you can read the <a href="https://doi.org/10.1093/bioinformatics/btw018" target="_blank" rel="noopener noreferrer">FINEMAP paper</a>, but in
short: finemap conducts a <em>shotgun stochastic search</em> around the space of possible causal configurations (groups of SNPs that
might be causal). It starts off with all single-SNP models, then it randomly tries a bunch of ways of adding a second SNP to
the first (picking the best model). Then it keeps going randomly trying to add or remove SNPs so that it explores the space of
configurations that have high probability.</p><p>In general it would need the raw genotype data to do this. However, because it assumes an additive
model (of genotypes on phenotype) it turns out that an approximation based on the SNP effects and
ld is sufficient, which is how FINEMAP can run so fast.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="interpreting-finemap-output">Interpreting FINEMAP output<a class="hash-link" href="#interpreting-finemap-output" title="Direct link to heading">​</a></h3><p>Your run of FINEMAP should have produced several output files (the ones you named in your
<code>finemap.master</code> file):</p><ul><li>A &#x27;configurations&#x27; file (<code>finemap_meta.config</code>), listing the causal configurations that FINEMAP thinks are plausible.</li><li>A &#x27;credible intervals&#x27; file, (<code>finemap_meta.cred</code>) listing variants wiht high probabilty of being the causal SNP for each signal in the top configuration.</li><li>A &#x27;SNP&#x27; file (<code>finemap_meta.snp</code>), listing the evidence that each SNP is one of the causal ones across configurations</li><li>And a log file (<code>finemap_meta.log</code>) telling you what it has done.</li></ul><p>Let&#x27;s look at these files now.</p><p><strong>Note.</strong> If for whatever reason you couldn&#x27;t get FINEMAP to run, you can use my output files which
are in the <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping/solutions/"><code>solutions</code></a> folder.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-finemap-log-file">The FINEMAP log file.<a class="hash-link" href="#the-finemap-log-file" title="Direct link to heading">​</a></h3><p>Before you look at the output files, look at the results FINEMAP prints to stdout when you run it
(and saves to the .log_sss file). Mine looks like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">|--------------------------------------|</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Welcome to FINEMAP v1.3.1            |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| (c) 2015-2018 University of Helsinki |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Help :                               |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - ./finemap --help                   |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - www.finemap.me                     |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - www.christianbenner.com            |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|                                      |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| Contact:                             |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - christian.benner@helsinki.fi       |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">| - matti.pirinen@helsinki.fi          |</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">|--------------------------------------|</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">--------</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">SETTINGS</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">--------</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- dataset         : all</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- corr-config     : 0.95</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- n-causal-snps   : 5</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- n-configs-top   : 50000</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- n-convergence   : 1000</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- n-iterations    : 100000</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- prior-k0        : 0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- prior-std       : 0.05 </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- prob-tol        : 0.001</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">------------</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">FINE-MAPPING (1/1)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">------------</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- GWAS summary stats               : combined_study.z</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- SNP correlations                 : study_LD.ld</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Causal SNP stats                 : finemap_meta.snp</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Causal configurations            : finemap_meta.config</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Credible sets                    : finemap_meta.cred</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Log file                         : finemap_meta.log_sss</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Number of GWAS samples           : 20000</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Number of SNPs                   : 730</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Prior-Pr(# of causal SNPs is k)  : </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  (0 -&gt; 0)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   1 -&gt; 0.583</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   2 -&gt; 0.291</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   3 -&gt; 0.097</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   4 -&gt; 0.0242</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   5 -&gt; 0.00482</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- 757559 configurations evaluated (1.152/100%) : converged after 1152 iterations</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Regional SNP heritability        : 0.00486 (SD: 0.000961 ; 95% CI: [0.00316,0.00689])</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Log10-BF of &gt;= one causal SNP    : 14.3</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Post-Pr(# of causal SNPs is k)   : </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  (0 -&gt; 0)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   1 -&gt; 0.00308</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   2 -&gt; 0.923</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   3 -&gt; 0.074</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   4 -&gt; 0.000159</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   5 -&gt; 0</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">- Run time                         : 0 hours, 0 minutes, 18 seconds</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>There are two bits of information here that allow you to interpret the fine-mapping: the overall
Bayes factor for the region, and the posterior distribution on the number of variants.</p><p>First, look at the line starting &quot;Log10-BF of &gt;= one causal SNP&quot;. This tells us the overall Bayes
factor (i.e. the Bayes factor that there is at least one causal variant in the region).</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">Log10-BF of &gt;= one causal SNP : 14.3</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>This Bayes factor is very large (over 10<sup>14</sup>), and shows that FINEMAP sees clear evidence that there is signal in
the region. (This is not surprising, of course, because we started with a region of strong
association.)</p><p>Second, let&#x27;s look at the posterior distribution on the number of causal variants. These occur at
the end of the log file on these lines:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">- Post-Pr(# of causal SNPs is k)   : </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  (0 -&gt; 0)</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   1 -&gt; 0.00308</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   2 -&gt; 0.923</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   3 -&gt; 0.074</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   4 -&gt; 0.000159</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">   5 -&gt; 0</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>FINEMAP is certain that there are at least two causal variants (as the posterior of k=1, i.e. there
being one variant is very small). And it quite strongly believes (92% posterior) that k = 2, i.e.
that there are exactly two causal variants.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-causal-configurations-output-file">The causal configurations output file<a class="hash-link" href="#the-causal-configurations-output-file" title="Direct link to heading">​</a></h3><p>Let&#x27;s look next at the <code>finemap_meta.config</code> file, which list gives the best 50,000 causal
configurations that FINEMAP found in its stochastic search. Load this file into your R session now:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">configurations = read_delim( &quot;finemap_meta.config&quot;, delim = &quot; &quot; , comment = &quot;#&#x27;)</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>We will look at the ones that have at least a 10% posterior:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">configurations[ configurations$prob &gt; 0.1, ]</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note.</strong> To print the full details here without truncating columns, I found I had to use <code>print()</code>
like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">print( configurations[ configurations$prob &gt; 0.1, ], width = 100 )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>As we would expect from the posterior distribution on <code>k</code>, all of the higher posterior
configurations include just two variants. In fact all three of them include the variant at position
<code>49208368</code>, but they vary a bit by swapping out the second variant in the pair. Note also that the
sum of the posterior for three causal configurations is only 64%, so while these are the most
likely it is also perfectly plausible that the true configuration lies somewhere further down the
list.</p><p>The configuration files can be a bit hard to interpret (and are mostly useful as input for
downstream analyses), so we will move on to some slightly easier files to read.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-snp-file">The SNP file<a class="hash-link" href="#the-snp-file" title="Direct link to heading">​</a></h3><p>FINEMAP also outputs the evidence for each individual SNP being causal (regardless of the
configuration).  Let&#x27;s see how many SNPs have an posterior greater than 10%:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">snps = read_delim( &quot;finemap_meta.snp&quot;, delim = &quot; &quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">print( snps[ snps$prob &gt; 0.1, ], width = 100 )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>The SNP at position <code>49208368</code> has a high posterior (85%), so is quite likely to be causal and you
would be quite confident taking this forward for follow-up experiments or further analyses. The
other variants have posteriors between 19% and 40%, so are reasonable candidates but you wouldn&#x27;t
bet the house on any one individually. Instead, if you were designing follow-up experiments based
on this, you would need to consider these variants as a group.</p><h3 class="anchor anchorWithStickyNavbar_mojV" id="the-credible-sets-file">The credible sets file<a class="hash-link" href="#the-credible-sets-file" title="Direct link to heading">​</a></h3><p>To group variants, FINEMAP also provides credible sets for each of the two signals:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">credible = read_delim( &quot;finemap_meta.cred2&quot;, delim = &quot; &quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">print( credible, width = 100 )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Also make sure look at the column names:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">&gt; </span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">colnames( credible )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">[1] &quot;index&quot; &quot;cred1&quot; &quot;prob1 &quot;cred2&quot; &quot;prob2&quot;</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p><strong>Note.</strong> My version of the file has trailing spaces at the end of each line - so R warns about
missing column names when I load it.  You can safely ignore that.</p><p>The data in this file shows:</p><ul><li><p>that the signal that contains the variant at <code>49208368</code> has a credible set size of 12 variants
(the possible variants are listed in the 2nd column).</p></li><li><p>and also that the signal that includes the variant at 49208380 has a credible set size of 4
variants (listed in the 4th column).</p></li></ul><p>This illustrates a counter-intuitive point: even though 49208368 itself has a high posterior, it is
in LD with a &quot;long tail&quot; of 11 other variants that are in tight LD with each other. The signal that
includes 49208380 is not well fine-mapped to a single causal variant, but it does localize well to
a small number of variants (4). So, if you are looking for a single variant that is reasonably
likely to be causal, you&#x27;d pick 49208368. But if you were looking for a small set of variants that,
overall, cover most of the possible causal posterior space, you would pick the second signal.</p><p><em>Question</em> Using genome browsers, investigate the location and function of these variants. Based on
the genetic and the functional information, which variant do you think is most likely to be cauasal
in each group?</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="plotting-finemap-output">Plotting FINEMAP output.<a class="hash-link" href="#plotting-finemap-output" title="Direct link to heading">​</a></h2><p>In statistics we are never done until the ink is dry on our plot. Let&#x27;s make a hitplot that (like
our GWAS practical hitplot) shows the marginal evidence at each SNP, but colours the SNPs according
to what credible set they fall in.  </p><p>First we have to reconstruct the marginal SNP P-value from the effect size and standard error (or z-score),
since FINEMAP doesn&#x27;t output the P-value:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">snps$pvalue = 2 * pnorm( -abs(snps$z ))</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Next let&#x27;s define the colours for our credible sets:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">colours = c( &quot;chocolate1&quot;, &quot;deepskyblue4&quot; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">snps$colour &lt;- &quot;grey&quot;</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">snps$colour[ snps$rsid %in% credible[[&#x27;cred1&#x27;]] ] &lt;- colours[1]</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">snps$colour[ snps$rsid %in% credible[[&#x27;cred2&#x27;]] ] &lt;- colours[2]</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Also let&#x27;s load genes using the same data / code from the <a href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/GWAS_analysis_practical/">GWAS
practical</a>:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">genes &lt;- read.table( &quot;resources/refGene_chr19.txt.gz&quot;, header=T, as.is=T )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Now here&#x27;s a function that makes the plot:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.finemap.output &lt;- function( snps, credible, xrange = NULL ) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Two big panels and a small one for genes</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  layout( matrix( 1:3, ncol = 1 ), heights = c( 1, 0.75, 0.75 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  if( is.null( xrange )) {</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xrange = range( snps$position )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  }</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Plot the per-SNP evidence</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  par( mar = c( 0.1, 4, 1, 1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  plot(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    snps$position,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    -log10( snps$pvalue ),</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlab = &quot;Position&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylab = &quot;-log10(P-value)&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = snps$colour,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    pch = 19,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xaxt = &#x27;n&#x27;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlim = xrange,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    bty = &#x27;n&#x27;            # no box, because I think they&#x27;re ugly!</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  legend( &quot;topleft&quot;, col = colours, pch = 19, legend = sprintf( &quot;Credible set %d&quot;, 1:2 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  # Plot the posteriors</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  par( mar = c( 0.1, 4, 0.1, 1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  plot(</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    snps$position,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    snps$prob,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    ylab = &quot;Posterior&quot;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    col = snps$colour,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    type = &quot;h&quot;,          # to plot vertical lines.  See `?plot`.</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xlim = xrange,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    xaxt = &#x27;n&#x27;,</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">    bty = &#x27;n&#x27;            # no box</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  par( mar = c( 4, 4, 0.1, 1 ))</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  plot.genes( genes, xrange, bty = &#x27;n&#x27; )</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  grid()</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">}</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>Run it like this:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.finemap.output( snps, credible )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>and inspect it.  You can also zoom in to see how the evidence stacks up over the genes:</p><div class="codeBlockContainer_MPoW theme-code-block" style="--prism-color:#393A34;--prism-background-color:#f6f8fa"><div class="codeBlockContent_B9tL"><pre tabindex="0" class="prism-code language-text codeBlock__0OG thin-scrollbar"><code class="codeBlockLines_gEuF"><span class="token-line" style="color:#393A34"><span class="token plain">plot.finemap.output( snps, credible, xrange = c( 49190000, 49229999 ) )</span><br></span></code></pre><div class="buttonGroup_hRr1"><button type="button" aria-label="Copy code to clipboard" title="Copy" class="clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div></div><p>It should look something like this:</p><p><img loading="lazy" alt="img" src="/whg-training-resources/assets/images/finemap_plot_zoom-41d6bda3fe3270436c7241d8c705ca5f.png" width="1800" height="1200" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="conclusion">Conclusion<a class="hash-link" href="#conclusion" title="Direct link to heading">​</a></h2><p>Congratulations!  You have fine-mapped your GWAS signal!</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/whg-training/whg-training-resources/edit/main/docs/genome_wide_association_studies/meta-analysis_and_fine-mapping/Fine-mapping.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><a class="pagination-nav__link pagination-nav__link--prev" href="/whg-training-resources/genome_wide_association_studies/meta-analysis_and_fine-mapping/Meta-analysis/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Meta-analysing two studies.</div></a></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#preparing-finemap-input-files" class="table-of-contents__link toc-highlight">Preparing FINEMAP input files</a></li><li><a href="#running-finemap" class="table-of-contents__link toc-highlight">Running FINEMAP</a><ul><li><a href="#interpreting-finemap-output" class="table-of-contents__link toc-highlight">Interpreting FINEMAP output</a></li><li><a href="#the-finemap-log-file" class="table-of-contents__link toc-highlight">The FINEMAP log file.</a></li><li><a href="#the-causal-configurations-output-file" class="table-of-contents__link toc-highlight">The causal configurations output file</a></li><li><a href="#the-snp-file" class="table-of-contents__link toc-highlight">The SNP file</a></li><li><a href="#the-credible-sets-file" class="table-of-contents__link toc-highlight">The credible sets file</a></li></ul></li><li><a href="#plotting-finemap-output" class="table-of-contents__link toc-highlight">Plotting FINEMAP output.</a></li><li><a href="#conclusion" class="table-of-contents__link toc-highlight">Conclusion</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">License</div><ul class="footer__items clean-list"><li class="footer__item"><a class="footer__link-item" href="/whg-training-resources/LICENSE/">License</a></li></ul></div><div class="col footer__col"><div class="footer__title">More</div><ul class="footer__items clean-list"><li class="footer__item"><a href="https://github.com/whg-training/whg-training-resources" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 University of Oxford.  Built with Docusaurus.</div></div></div></footer></div>
<script src="/whg-training-resources/assets/js/runtime~main.8fe67747.js"></script>
<script src="/whg-training-resources/assets/js/main.91c67f1f.js"></script>
</body>
</html>